---
title: "BeatAML - Analysis of Splicing Variants"
author: "Interactive Report"
output:
  html_document:
    toc: yes
    toc_float: yes
    toc_collapsed: yes
    toc_depth: 3
    number_sections: yes
  pdf_document:
    toc: yes
    toc_depth: '3'
---

```{r setup, include=FALSE} 
knitr::opts_chunk$set(warning = FALSE, message = FALSE) 
```

```{r echo = F, results = 'hide'}
library(ggplot2)
library(dplyr)
library(readxl)
library(openxlsx)
library(stringr)
library(sqldf)
library(tidyr)
library(DT)
library(plotly)
library(knitr)
library(kableExtra)
library(ggsignif)
library(stats)
library(rstatix)
```

```{r echo=FALSE}
extractedSJ_dir_in <- "../Results/BeatAML/SpliceJunction/ExtractedSJ-Annotated/"
normalizedEx_dir_out <- "../Results/BeatAML/SpliceJunction/NormalizedExpression/"
summary_dir_out <- "../Results/BeatAML/"
```


In the BeatAML cohort we found 28 variants of interest, of which 25 variants can be validated by RNASeq data. 

The samples of interest are those where the variant has been called either in DNA or RNA, and have RNASeq data available for validation. The splice junction search is made based on the STAR-SJCounts 1-based intronic positions.

```{r echo=FALSE}
found_variants <- read_excel("../Results/BeatAML/BeatAML.Variant_summary.xlsx", sheet = "Sheet 1")
validable_variants <- unique(found_variants$MutationKey_Hg38[found_variants$Validable == "Validable"])
```

```{r echo=FALSE}
to_be_validated = read.delim("../Data/Potential_SAVs.tsv", sep ="\t")
columns_to_select <- c("ID","SYMBOL" ,"MutationKey_Hg38","hg19","hg38","HGVSc","HGVSp","Consequence","Prediction_merged", "Predictors" ,"MINIGENE" , "SpliceAI_pred_DP_AG","SpliceAI_pred_DP_AL","SpliceAI_pred_DP_DG","SpliceAI_pred_DP_DL" ,
"SpliceAI_pred_DS_AG","SpliceAI_pred_DS_AL","SpliceAI_pred_DS_DG","SpliceAI_pred_DS_DL","SpliceAI_pred_SYMBOL", "MaxEntScan_alt","MaxEntScan_diff","MaxEntScan_ref","Motif","New.splice.site","Wild.Type","Mutant",   "Variation....","Motif.1","New.splice.site.1","Wild.Type.1","Mutant.1","Variation.....1" )
to_be_validated_2 <- to_be_validated[to_be_validated$MutationKey_Hg38 %in% validable_variants, columns_to_select]
```

Splicing alterations to be evaluated:
```{r  echo=FALSE}
datatable(
  to_be_validated_2,
  rownames = FALSE,
  extensions = 'Buttons',
  options = list(paging = TRUE,
              pageLength = 4,
              autoWidth = TRUE,
              dom = 'lifrtBp',
              buttons = c( 'excel'), 
              scrollX = TRUE,  
              scrollY = TRUE
              )
)
```

```{r echo=FALSE}
summary_res <- data.frame(matrix(ncol = 17, nrow = 0))

colnames(summary_res) <- c("Gene","MutationKey_Hg38","Prediction","splice_junction_status","splice_junction_position", "sample_id","case_id",
"RNA_Sample","DNA_Sample","NormalizedExpression","ECDF","Pvalue","VAF_RNA","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf")

```



# NRAS chr1,114716123,C,T {.tabset}
Variant found in **19 patients** of the BeatAML (**20 samples**).

* Patients with NRAS chr1,114716123,C,T variant: **19 patients (20 samples)**
* Patients with the variant and RNASeq for validation: **19 patients (17 samples)**

The splicing alterations being assessed are: 

* **Donor Gain**: chr1:114716126, found in the mutated samples.
* **Exon 2 Skipping**: chr1:114713979-114716657, found in the mutated samples.
* **Donor Gain**: chr1:114713979-114716076, found in the splice junction collection, but not found in the mutated samples.

Variant information:
```{r echo=FALSE}
datatable(to_be_validated[to_be_validated$MutationKey_Hg38 == "chr1,114716123,C,T",],
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp', 
              scrollX = TRUE,  
              scrollY = TRUE,
              columnDefs = list(list(targets = '_all', className = 'dt-center')))
)
```

Load the extracted splice junctions of the gene harboring the mutation.
```{r}
extractedSJ_path <- paste0(extractedSJ_dir_in,"NRAS_UM_annotSJ.tsv")
GeneSJ <- read.delim(extractedSJ_path, sep ="\t")
```

Set the sample's group: Mutated (MUT) or No Mutated (WT)
```{r}
samples_df <- found_variants[found_variants$Gene=="NRAS" & found_variants$MutationKey_Hg38 == "chr1,114716123,C,T",]

cases <- samples_df$RNA_Sample[samples_df$Validable == "Validable"]
GeneSJ$GROUP <- ifelse(GeneSJ$sample_id %in% cases , "MUT", "WT")
```


## SJ Lookup {.tabset}

Search for the splice junctions of interest in the extracted splice junctions of the gene by position (chr_SJstart_SJend).


### Donor Gain
Search: chr1:114716126 

Show all the splice junctions containing the position 114716126
```{r}
colnames(GeneSJ)[grep("11471612",colnames(GeneSJ))]
```

Found: chr1:114713979-114716126

Reads of all the AML samples (mutated and no mutated) for the splice junction:
```{r}
GeneSJ$chr1_114713979_114716126
```
Samples with the SJ of interest:
```{r}
table(GeneSJ$chr1_114713979_114716126>0) 
```
Groups of the samples having the alternative splice junction:
```{r}
table(GeneSJ$GROUP[GeneSJ$chr1_114713979_114716126 > 0])
```

Alternative SJ found in the mutated samples. 

### Exon Skipping 2
Search: chr1:114713979-114716657

Show all the splice junctions containing the position 114713979
```{r}
colnames(GeneSJ)[grep("114713979",colnames(GeneSJ))]
```
Found: chr1:114713979-114716657

Reads of all the AML samples (mutated and no mutated) for the splice junction:
```{r}
GeneSJ$chr1_114713979_114716657
```

Samples with the SJ of interest:
```{r}
table(GeneSJ$chr1_114713979_114716657>0) 
```

Groups of the samples having the alternative splice junction:
```{r}
table(GeneSJ$GROUP[GeneSJ$chr1_114713979_114716657 > 0])
```

Alternative SJ found in the mutated samples.

### Donor Gain 2
Search: chr1:114713979-114716076 

Show all the splice junctions containing the positions between 114716070 - 114716079
```{r}
colnames(GeneSJ)[grep("11471607",colnames(GeneSJ))]
```

Reads of all the AML samples (mutated and no mutated) for the splice junction:
```{r}
GeneSJ$chr1_114713979_114716076
```

Samples with the SJ of interest:
```{r}
table(GeneSJ$chr1_114713979_114716076 >0) 
```

Groups of the samples having the alternative splice junction:
```{r}
table(GeneSJ$GROUP[GeneSJ$chr1_114713979_114716076 > 0])
```

Alternative SJ not found in the mutated samples of the splice junction collection.

### Canonical SJ
Exon1-2: chr1:114716178-114716657

Exon2-3: chr1:114713979-114716049

Reads of all the AML samples (mutated and no mutated) for the splice junction:
```{r}
GeneSJ$chr1_114716178_114716657
```

Reads of all the AML samples (mutated and no mutated) for the splice junction:
```{r}
GeneSJ$chr1_114713979_114716049
```


## Normalization
Count the reads of all the splice junctions of the gene harboring the variant:
```{r}
GeneSJ$rowSum_SJtotal <- rowSums(GeneSJ[,grep("chr", names(GeneSJ))])
```

Normalization of the expression by the total read counts of all the splice junctions of the gene:
```{r}
GeneSJ$Normalized_CanonEx1_2 <- (GeneSJ$chr1_114716178_114716657)/GeneSJ$rowSum_SJtotal*100
GeneSJ$Normalized_CanonEx2_3 <- (GeneSJ$chr1_114713979_114716049)/GeneSJ$rowSum_SJtotal*100
GeneSJ$Normalized_DG <- (GeneSJ$chr1_114713979_114716126)/GeneSJ$rowSum_SJtotal*100
GeneSJ$Normalized_ES2 <- (GeneSJ$chr1_114713979_114716657)/GeneSJ$rowSum_SJtotal*100
```

Download the normalized values for the assessed splice junctions of all the AML samples:
```{r echo=FALSE}
normalization_df <- GeneSJ[,c("sample_id","case_id", "GROUP","Normalized_CanonEx1_2", "Normalized_CanonEx2_3", "Normalized_DG", "Normalized_ES2","chr1_114716178_114716657", "chr1_114713979_114716049","chr1_114713979_114716126", "chr1_114713979_114716657", "rowSum_SJtotal","file_id.BAM","file_name.BAM","file_id.STAR.SJCounts","file_name.STAR.SJCounts" )]

colnames(normalization_df)[grep("chr1_114716178_114716657", names(normalization_df))] <- "chr1_114716178_114716657_CanonEx1_2"
colnames(normalization_df)[grep("chr1_114713979_114716049", names(normalization_df))] <- "chr1_114713979_114716049_CanonEx2_3"
colnames(normalization_df)[grep("chr1_114713979_114716126", names(normalization_df))] <- "chr1_114713979_114716126_DG"
colnames(normalization_df)[grep("chr1_114713979_114716657", names(normalization_df))] <- "chr1_114713979_114716657_ES2"

write.xlsx(normalization_df, paste0(normalizedEx_dir_out, "BeatAML.NRAS.chr1-114716123-C-T.xlsx"), rowNames = FALSE)
```

```{r echo=FALSE}
datatable(normalization_df,
extensions = 'Buttons',
rownames = FALSE, 
filter = 'top',
options = list(paging = TRUE,
              pageLength = 4,
              autoWidth = FALSE,
              dom = 'lifrtBp',
              buttons = c('excel'), 
              scrollX = TRUE,  
              scrollY = TRUE,
              columnDefs = list(list(targets = '_all', className = 'dt-center'))
))
```

## VAF
```{r echo=FALSE}
GeneSJ_subset <- GeneSJ[GeneSJ$sample_id %in% cases,]
GeneSJ_subset$Gene <- "NRAS"
GeneSJ_subset$MutationKey_Hg38 <- "chr1,114716123,C,T"

annot_vaf <- merge(found_variants[,c("MutationKey_Hg38","Gene","sample_id","case_id",                                               "RNA_Sample","DNA_Sample","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf",                         "MuTect2_filter","VarScan2_filter","SomaticSniper_filter","MuSE_filter","MUTReads","WTReads","VAF")], 
                   GeneSJ_subset[,c("Gene","MutationKey_Hg38","sample_id", "case_id", "Normalized_CanonEx1_2", "Normalized_CanonEx2_3","Normalized_DG", "Normalized_ES2")], 
                   by.x=c("RNA_Sample", "case_id","Gene", "MutationKey_Hg38"), 
                   by.y=c("sample_id", "case_id","Gene", "MutationKey_Hg38"))


annot_vaf <- annot_vaf %>% select(Gene,MutationKey_Hg38,
                                  sample_id,
                                  case_id, RNA_Sample,DNA_Sample,
                                  MuTect2_vaf,VarScan2_vaf,SomaticSniper_vaf,MuSE_vaf,
                                  VAF,
                                  Normalized_CanonEx1_2, Normalized_CanonEx2_3, Normalized_DG,Normalized_ES2,
                                  everything()
                                  )
colnames(annot_vaf)[grep("VAF", names(annot_vaf))] <- "VAF_RNA"
colnames(annot_vaf)[grep("MUTReads", names(annot_vaf))] <- "MUTReads_RNA"
colnames(annot_vaf)[grep("WTReads", names(annot_vaf))] <- "WTReads_RNA"

write.xlsx(annot_vaf, paste0(normalizedEx_dir_out, "BeatAML.NRAS.chr1-114716123-C-T.MUT.xlsx"), rowNames = FALSE)
```

Mutated samples vaf:
```{r echo=FALSE}
datatable(annot_vaf,
extensions = 'Buttons',
rownames = FALSE, 
filter = 'top',
options = list(paging = TRUE,
              pageLength = 4,
              autoWidth = TRUE,
              dom = 'lifrtBp',
              buttons = c('excel'), 
              scrollX = TRUE,  
              scrollY = TRUE
              )
)
```

## Plots {.tabset}

```{r echo=FALSE}
GeneSJ$GROUP <- ifelse(GeneSJ$GROUP == "MUT", "MUT", "WT")
GeneSJ$GROUP <- as.factor(GeneSJ$GROUP)
```

### Static Dot Plots 
Canonical SJ:

```{r echo=FALSE}
ggplot(GeneSJ, aes(sample_id,Normalized_CanonEx1_2, color=GROUP)) + 
  geom_point(size=.8)  +
  ggtitle("Normalized expression of NRAS Exon 1-2 Splice Junction") +
  labs(color='GROUP') 

ggplot(GeneSJ, aes(sample_id,Normalized_CanonEx2_3,color=GROUP)) + 
  geom_point(size=.8) +
  ggtitle("Normalized expression of NRAS Exon 2-3 Splice Junction") +
  labs(color='GROUP') 
```

Splicing alterations:

```{r echo=FALSE}
ggplot(GeneSJ, aes(sample_id,Normalized_DG,color=GROUP)) +
  geom_point(size=.8) +
  ggtitle("Normalized expression of NRAS Donor Gain") +
  labs(color='GROUP') 
```

```{r echo=FALSE}
ggplot(GeneSJ, aes(sample_id,Normalized_ES2,color=GROUP)) +
  geom_point(size=.8) +
  ggtitle("Normalized expression of NRAS Exon Skipping 2 Splice Junction") +
  labs(color='GROUP') 
```

### Interactive Dot Plots

```{r echo=FALSE}
plot_df <- merge(GeneSJ,annot_vaf[c("RNA_Sample", "VAF_RNA", 
                                    "MuTect2_vaf", "VarScan2_vaf", "SomaticSniper_vaf", "MuSE_vaf")],
                 by.x="sample_id", by.y="RNA_Sample", all.x = TRUE)
```

Canonical SJ:

```{r echo=FALSE}
ggplotly(ggplot(plot_df, aes(sample_id,Normalized_CanonEx1_2, color=GROUP)) + 
  geom_point(aes(text=paste0("VAF RNA: ",VAF_RNA,"\nVAF MuTect: ",MuTect2_vaf,
                             "\nVAF VarScan2: ",VarScan2_vaf,
                             "\nVAF SomaticSniper: ",SomaticSniper_vaf,
                             "\nVAF MuSE: ",MuSE_vaf)),size= 0.8)  +
  ggtitle("Normalized expression of NRAS Exon 1-2 Splice Junction") +
  labs(color='GROUP') 
)
```


```{r echo=FALSE}
ggplotly(ggplot(plot_df, aes(sample_id,Normalized_CanonEx2_3,color=GROUP)) + 
  geom_point(aes(text=paste0("VAF RNA: ",VAF_RNA,"\nVAF MuTect: ",MuTect2_vaf,
                             "\nVAF VarScan2: ",VarScan2_vaf,
                             "\nVAF SomaticSniper: ",SomaticSniper_vaf,
                             "\nVAF MuSE: ",MuSE_vaf)),size= 0.8) +
  ggtitle("Normalized expression of NRAS Exon 2-3 Splice Junction") +
  labs(color='GROUP') 
)
```

Splicing alteration:

```{r echo=FALSE}
ggplotly(ggplot(plot_df, aes(sample_id,Normalized_DG,color=GROUP)) +
  geom_point(aes(text=paste0("VAF RNA: ",VAF_RNA,"\nVAF MuTect: ",MuTect2_vaf,
                             "\nVAF VarScan2: ",VarScan2_vaf,
                             "\nVAF SomaticSniper: ",SomaticSniper_vaf,
                             "\nVAF MuSE: ",MuSE_vaf)),size= 0.8)  +
  ggtitle("Normalized expression of NRAS Donor Gain") +
  labs(color='GROUP') 
)
```

```{r echo=FALSE}
ggplotly(ggplot(plot_df, aes(sample_id,Normalized_ES2,color=GROUP)) +
  geom_point(aes(text=paste0("VAF RNA: ",VAF_RNA,"\nVAF MuTect: ",MuTect2_vaf,
                             "\nVAF VarScan2: ",VarScan2_vaf,
                             "\nVAF SomaticSniper: ",SomaticSniper_vaf,
                             "\nVAF MuSE: ",MuSE_vaf)),size= 0.8)  +
  ggtitle("Normalized expression of NRAS Exon Skipping 2 Splice Junction") +
  labs(color='GROUP') 
)
```

### Violin Plots
Violin Plots for the alternative splice junctions interrogated:
```{r echo=FALSE}
plot_boxplot <- plot_df[plot_df$GROUP %in% c("WT"),]
plot_dotplot <- plot_df[plot_df$GROUP %in% c("MUT"),]

GEN = "NRAS"
SJ = "chr1:114713979-114716126"

ggplot(plot_df, aes(y=Normalized_DG, x=factor(GROUP, level=c("WT", "MUT")))) + 
       geom_violin(scale="area") +
       geom_point(data = plot_dotplot, 
                  aes(y=Normalized_DG, x=GROUP),
                  color="black",
                  size=2
                  ) +
        
       ylab("Normalized SJ expression") +
       xlab(paste(GEN, SJ))
```


```{r echo=FALSE}
plot_boxplot <- plot_df[plot_df$GROUP %in% c("WT"),]
plot_dotplot <- plot_df[plot_df$GROUP %in% c("MUT"),]

GEN = "NRAS"
SJ = "chr1:114713979-114716657"

ggplot(plot_df, aes(y=Normalized_ES2, x=factor(GROUP, level=c("WT", "MUT")))) + 
       geom_violin(scale="area") +
       geom_point(data = plot_dotplot, 
                  aes(y=Normalized_ES2, x=GROUP),
                  color="black",
                  size=2
                  ) +
        
       ylab("Normalized SJ expression") +
       xlab(paste(GEN, SJ))
```

## Statistical Analysis {.tabset}

```{r}
SJCounts <- GeneSJ #BeatAML.NRAS.chr1-114716123-C-T.xlsx
```

### Donor Gain {.tabset}

#### Expression Difference 

Normality Test:
```{r}
shapiro.test(SJCounts$Normalized_DG[SJCounts$GROUP == "WT"])
```

Value of Mean Normalized Expression of the Alternative SJ in WT samples:
```{r}
mean_WT_SJi <- mean(SJCounts$Normalized_DG[SJCounts$GROUP == "WT"], na.rm=TRUE)
mean_WT_SJi
```

Normalized Expression Value of the Alternative SJ in the MUT sample:
```{r}
MUT_SJi <- SJCounts$Normalized_DG[SJCounts$GROUP == "MUT"]
MUT_SJi
```
Deviation from the mean normalized expression:
```{r}
SJCounts$Difference <-  SJCounts$Normalized_DG - mean_WT_SJi
```

```{r echo=FALSE}
table <- SJCounts[SJCounts$GROUP == "WT", c("sample_id", "Normalized_DG", "Difference")]
table <- table[order(table$Difference, decreasing = TRUE),]
```

```{r echo=FALSE}
datatable(table,
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp'
))
```

Difference in the MUT sample: deviation of the Normalized expression of the MUT patient from the mean normalized WT expression
```{r}
SJCounts[SJCounts$GROUP == "MUT", c("sample_id", "Difference")]
```

```{r echo=FALSE}
ggplot(SJCounts, aes(sample_id,Difference)) + geom_point(aes(colour=factor(GROUP))) +
  ylab("Differential Expression") +
  labs(color = "GROUP",
       title = "Differential Expression Distribution")
```


```{r echo=FALSE}
ggplot(SJCounts, aes(y=Difference, x=factor(GROUP, level=c("WT", "MUT")))) + 
  geom_boxplot() +
  ylab("Differential Expression") +
  xlab("GROUP")
```

#### ECDF

```{r}
v_ecdf <- ecdf(SJCounts$Difference[SJCounts$GROUP == "WT"])
v_ecdf
```

```{r}
plot(ecdf(SJCounts$Difference[SJCounts$GROUP == "WT"]))
```

```{r}
v_ecdf(SJCounts$Difference[SJCounts$GROUP == "MUT"])
```


```{r}
MUT_df <- SJCounts[SJCounts$GROUP == "MUT",c("sample_id","case_id", "Normalized_DG")]
colnames(MUT_df) <- c("sample_id", "case_id", "NormalizedExpression")

MUT_df$Prediction <- "Donor Gain"
MUT_df$splice_junction_status <- "AlternativeSJ found in MUT samples"
MUT_df$splice_junction_position <- "chr1:114713979-114716126"
MUT_df$ECDF <- v_ecdf(SJCounts$Difference[SJCounts$GROUP == "MUT"])
MUT_df$Pvalue <- NA

MUT_df$ECDF <- ifelse(MUT_df$NormalizedExpression == 0, NA,MUT_df$ECDF)

```

```{r echo=FALSE}
vaf_select <- annot_vaf[,c("Gene","MutationKey_Hg38","sample_id","RNA_Sample","DNA_Sample","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf", "VAF_RNA")]
MUT_vaf <- merge(MUT_df, vaf_select, by.x="sample_id", by.y = "RNA_Sample")
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id"] <- "RNA_Sample"
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id.y"] <- "sample_id"

col_order <- c("Gene", "MutationKey_Hg38", "Prediction","splice_junction_status", "splice_junction_position","sample_id", "case_id","RNA_Sample","DNA_Sample","NormalizedExpression","ECDF", "Pvalue", "VAF_RNA","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf")
MUT_vaf <- MUT_vaf[, col_order]
```

Download the vaf, inferred percentiles and pvalues of the mutated samples:
```{r echo=FALSE}
datatable(MUT_vaf,
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp',
              scrollX = TRUE
))
```


```{r echo=FALSE}
stat_df <- data.frame(matrix(ncol = 17, nrow = 1))
colnames(stat_df) <- c("Gene","MutationKey_Hg38","Prediction","splice_junction_status","splice_junction_position", "sample_id","case_id","RNA_Sample","DNA_Sample","NormalizedExpression","ECDF","Pvalue","VAF_RNA","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf")

stat_df$Gene <- unique(MUT_vaf$Gene)
stat_df$MutationKey_Hg38 <- unique(MUT_vaf$MutationKey_Hg38)
stat_df$sample_id <- paste(MUT_vaf$sample_id, collapse = ';\n')
stat_df$case_id <- paste(MUT_vaf$case_id, collapse = ';\n')
stat_df$RNA_Sample <- paste(MUT_vaf$RNA_Sample, collapse = ';\n')
stat_df$DNA_Sample <- paste(MUT_vaf$DNA_Sample, collapse = ';\n')
stat_df$NormalizedExpression <- paste0(MUT_vaf$NormalizedExpression, collapse= ";\n")
stat_df$ECDF <- paste0(MUT_vaf$ECDF, collapse = ";\n")
stat_df$VAF_RNA <-  paste(MUT_vaf$VAF_RNA, collapse = ';\n')
stat_df$MuTect2_vaf <-  paste(MUT_vaf$MuTect2_vaf, collapse = ';\n')
stat_df$VarScan2_vaf <-  paste(MUT_vaf$VarScan2_vaf, collapse = ';\n')
stat_df$SomaticSniper_vaf <-  paste(MUT_vaf$SomaticSniper_vaf, collapse = ';\n')
stat_df$MuSE_vaf <-  paste(MUT_vaf$MuSE_vaf, collapse = ';\n')
stat_df$Prediction <- "Donor Gain"
stat_df$splice_junction_status <- "AlternativeSJ found in MUT samples"
stat_df$splice_junction_position <- "chr1:114713979-114716126"
stat_df$Pvalue <- NA
```

```{r echo=FALSE}
summary_res<- rbind(summary_res,stat_df)
```


### Exon Skipping {.tabset}
#### Expression Difference 
Normality Test:

```{r}
shapiro.test(SJCounts$Normalized_ES2[SJCounts$GROUP == "WT"])
```

Value of Mean Normalized Expression of the Alternative SJ in WT samples:
```{r}
mean_WT_SJi <- mean(SJCounts$Normalized_ES2[SJCounts$GROUP == "WT"], na.rm=TRUE)
mean_WT_SJi
```

Normalized Expression Value of the Alternative SJ in the MUT sample:
```{r}
MUT_SJi <- SJCounts$Normalized_ES2[SJCounts$GROUP == "MUT"]
MUT_SJi
```

Deviation from the mean normalized expression:
```{r}
SJCounts$Difference <-  SJCounts$Normalized_ES2 - mean_WT_SJi
```

```{r echo=FALSE}
table <- SJCounts[SJCounts$GROUP == "WT", c("sample_id", "Normalized_ES2", "Difference")]
table <- table[order(table$Difference, decreasing = TRUE),]
```


```{r echo=FALSE}
datatable(table,
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp'
))
```

Difference in the MUT sample: deviation of the Normalized expression of the MUT patient from the mean normalized WT expression
```{r}
SJCounts[SJCounts$GROUP == "MUT", c("sample_id", "Difference")]
```

```{r echo=FALSE}
ggplot(SJCounts, aes(sample_id,Difference)) + geom_point(aes(colour=factor(GROUP))) +
  ylab("Differential Expression") +
  labs(color = "GROUP",
       title = "Differential Expression Distribution")
```


```{r echo=FALSE}
ggplot(SJCounts, aes(y=Difference, x=factor(GROUP, level=c("WT", "MUT")))) + 
  geom_boxplot() +
  ylab("Differential Expression") +
  xlab("GROUP")
```

#### ECDF

```{r}
v_ecdf <- ecdf(SJCounts$Difference[SJCounts$GROUP == "WT"])
v_ecdf
```

```{r}
plot(ecdf(SJCounts$Difference[SJCounts$GROUP == "WT"]))
```

```{r}
v_ecdf(SJCounts$Difference[SJCounts$GROUP == "MUT"])
```


```{r}
MUT_df <- SJCounts[SJCounts$GROUP == "MUT",c("sample_id","case_id", "Normalized_ES2")]
colnames(MUT_df) <- c("sample_id", "case_id", "NormalizedExpression")

MUT_df$ECDF <- v_ecdf(SJCounts$Difference[SJCounts$GROUP == "MUT"])
MUT_df$Pvalue <- NA
MUT_df$Prediction <- "Exon Skipping"
MUT_df$splice_junction_status <- "AlternativeSJ found in MUT samples"
MUT_df$splice_junction_position <- "chr1:114713979-114716657"

MUT_df$ECDF <- ifelse(MUT_df$NormalizedExpression == 0, NA,MUT_df$ECDF)
```


```{r echo=FALSE}
vaf_select <- annot_vaf[,c("Gene","MutationKey_Hg38","sample_id","RNA_Sample","DNA_Sample","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf", "VAF_RNA")]
MUT_vaf <- merge(MUT_df, vaf_select, by.x="sample_id", by.y = "RNA_Sample")
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id"] <- "RNA_Sample"
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id.y"] <- "sample_id"

col_order <- c("Gene", "MutationKey_Hg38", "Prediction","splice_junction_status","splice_junction_position","sample_id", "case_id","RNA_Sample","DNA_Sample","NormalizedExpression","ECDF", "Pvalue", "VAF_RNA","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf")
MUT_vaf <- MUT_vaf[, col_order]
```

Download the vaf, inferred percentiles and pvalues of the mutated samples:
```{r echo=FALSE}
datatable(MUT_vaf,
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp',
              scrollX=TRUE
))
```


#### Mann Whitney 
Normality:
```{r}
shapiro.test(SJCounts$Normalized_ES2[SJCounts$GROUP== "WT"])
```

```{r}
shapiro.test(SJCounts$Normalized_ES2[SJCounts$GROUP== "MUT"])
```

Mann-Whitney:
```{r}
wt <- wilcox.test(x=SJCounts$Normalized_ES2[SJCounts$GROUP== "MUT"], 
                  y=SJCounts$Normalized_ES2[SJCounts$GROUP== "WT"],
                  alternative = "two.sided", 
                  paired = FALSE, 
                  conf.int = 0.95)
wt
```


```{r echo=FALSE}
stat_df <- data.frame(matrix(ncol = 17, nrow = 1))
colnames(stat_df) <- c("Gene","MutationKey_Hg38","Prediction","splice_junction_status","splice_junction_position", "sample_id","case_id","RNA_Sample","DNA_Sample","NormalizedExpression","ECDF","Pvalue","VAF_RNA","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf")

stat_df$Gene <- unique(MUT_vaf$Gene)
stat_df$MutationKey_Hg38 <- unique(MUT_vaf$MutationKey_Hg38)
stat_df$sample_id <- paste(MUT_vaf$sample_id, collapse = ';\n') 
stat_df$case_id <- paste(MUT_vaf$case_id, collapse = ';\n') 
stat_df$RNA_Sample <- paste(MUT_vaf$RNA_Sample, collapse = ';\n') 
stat_df$DNA_Sample <- paste(MUT_vaf$DNA_Sample, collapse = ';\n')
stat_df$NormalizedExpression <- paste0(MUT_vaf$NormalizedExpression, collapse= ";\n")
stat_df$ECDF <- paste0(MUT_vaf$ECDF, collapse = ";\n")
stat_df$VAF_RNA <-  paste(MUT_vaf$VAF_RNA, collapse = ';\n')
stat_df$MuTect2_vaf <-  paste(MUT_vaf$MuTect2_vaf, collapse = ';\n')
stat_df$VarScan2_vaf <-  paste(MUT_vaf$VarScan2_vaf, collapse = ';\n')
stat_df$SomaticSniper_vaf <-  paste(MUT_vaf$SomaticSniper_vaf, collapse = ';\n')
stat_df$MuSE_vaf <-  paste(MUT_vaf$MuSE_vaf, collapse = ';\n')
stat_df$Prediction <- "Exon Skipping"
stat_df$splice_junction_status <- "AlternativeSJ found in MUT samples"
stat_df$splice_junction_position <- "chr1:114713979-114716657"

stat_df$Pvalue <- wt$p.value
```

```{r echo=FALSE}
summary_res<- rbind(summary_res,stat_df)
```



```{r echo=FALSE}
## Add not found effects information:
MUT_df <- MUT_df[,c("sample_id", "case_id")]
MUT_df$NormalizedExpression <- NA
MUT_df$Prediction <- "Donor Gain"
MUT_df$splice_junction_status <- "AlternativeSJ not found in MUT samples"
MUT_df$splice_junction_position <- "chr1:114713979-114716076"
MUT_df$Pvalue <- NA

vaf_select <- annot_vaf[,c("Gene","MutationKey_Hg38","sample_id","RNA_Sample","DNA_Sample","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf", "VAF_RNA")]
MUT_vaf <- merge(MUT_df, vaf_select, by.x="sample_id", by.y = "RNA_Sample")
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id"] <- "RNA_Sample"
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id.y"] <- "sample_id"

col_order <- c("Gene", "MutationKey_Hg38", "Prediction","splice_junction_status","splice_junction_position","sample_id", "case_id","RNA_Sample","DNA_Sample","NormalizedExpression", "Pvalue", "VAF_RNA","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf")
MUT_vaf <- MUT_vaf[, col_order]


stat_df <- data.frame(matrix(ncol = 17, nrow = 1))
colnames(stat_df) <- c("Gene","MutationKey_Hg38","Prediction","splice_junction_status","splice_junction_position", "sample_id","case_id","RNA_Sample","DNA_Sample","NormalizedExpression","ECDF","Pvalue","VAF_RNA","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf")

stat_df$Gene <- unique(MUT_vaf$Gene)
stat_df$MutationKey_Hg38 <- unique(MUT_vaf$MutationKey_Hg38)
stat_df$sample_id <- paste(MUT_vaf$sample_id, collapse = ';\n') 
stat_df$case_id <- paste(MUT_vaf$case_id, collapse = ';\n') 
stat_df$RNA_Sample <- paste(MUT_vaf$RNA_Sample, collapse = ';\n') 
stat_df$DNA_Sample <- paste(MUT_vaf$DNA_Sample, collapse = ';\n')
stat_df$NormalizedExpression <- NA
stat_df$ECDF <- NA
stat_df$VAF_RNA <-  paste(MUT_vaf$VAF_RNA, collapse = ';\n')
stat_df$MuTect2_vaf <-  paste(MUT_vaf$MuTect2_vaf, collapse = ';\n')
stat_df$VarScan2_vaf <-  paste(MUT_vaf$VarScan2_vaf, collapse = ';\n')
stat_df$SomaticSniper_vaf <-  paste(MUT_vaf$SomaticSniper_vaf, collapse = ';\n')
stat_df$MuSE_vaf <-  paste(MUT_vaf$MuSE_vaf, collapse = ';\n')
stat_df$Prediction <- unique(MUT_vaf$Prediction)
stat_df$splice_junction_status <- unique(MUT_vaf$splice_junction_status)
stat_df$splice_junction_position <- unique(MUT_vaf$splice_junction_position)
stat_df$Pvalue <- NA

## Save
summary_res <- rbind(summary_res, stat_df)

```


```{r echo=FALSE}
rm(GeneSJ, normalization_df, GeneSJ_subset, annot_vaf, plot_df, samples_df,table, plot_boxplot, plot_dotplot, vaf_select, SJCounts, MUT_df, MUT_vaf)
```


# KRAS chr12,25245347,C,T {.tabset}
Variant found in **8 patients** of the BeatAML (**8 samples**)

* Patients with KRAS chr12,25245347,C,T variant: **8 patients (8 samples)**
* Patients with the variant and RNASeq for validation: **7 patients (7 samples)**

The splicing alterations being assessed are: 

* **Donor Gain**: predicted at 1bp from the variant, chr12:25245345, not found in the splice junction collection.
* **Donor Gain**: chr12:25245281-25245348, not found in the splice junction collection.

Variant information:
```{r echo=FALSE}
datatable(to_be_validated[to_be_validated$MutationKey_Hg38 == "chr12,25245347,C,T",],
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp', 
              scrollX = TRUE,  
              scrollY = TRUE,
              columnDefs = list(list(targets = '_all', className = 'dt-center')))
)
```

Load the extracted splice junctions of the gene harboring the mutation.
```{r}
extractedSJ_path <- paste0(extractedSJ_dir_in,"KRAS_UM_annotSJ.tsv")
GeneSJ <- read.delim(extractedSJ_path, sep ="\t")
```

Set the sample's group: Mutated (MUT) or No Mutated (WT)

```{r}
samples_df <- found_variants[found_variants$Gene=="KRAS" & found_variants$MutationKey_Hg38 == "chr12,25245347,C,T",]

cases <- samples_df$RNA_Sample[samples_df$Validable == "Validable"]
GeneSJ$GROUP <- ifelse(GeneSJ$sample_id %in% cases , "MUT", "WT")
```


## SJ Lookup{.tabset}

Search for the splice junctions of interest in the extracted splice junctions of the gene by position (chr_SJstart_SJend).

### Donor Gain 
Search: predicted at 1bp from the variant 25245347, chr12:25245345

Show all the splice junctions containing the positions from 25245340 to 25245349
```{r}
colnames(GeneSJ)[grepl("2524534", colnames(GeneSJ))] 
```
Alternative SJ not found in the splice junction collection.

### Donor Gain 
Search: chr12:25245281-25245348

Show all the splice junctions containing the positions from 25245340 to 25245349
```{r}
colnames(GeneSJ)[grepl("2524534", colnames(GeneSJ))] 
```

Show all the splice junctions containing the positions from 25245280 to 25245289
```{r}
colnames(GeneSJ)[grepl("2524528", colnames(GeneSJ))] 
```


Alternative SJ not found in the splice junction collection.

## VAF
```{r echo=FALSE}
GeneSJ_subset <- GeneSJ[GeneSJ$sample_id %in% cases,]
GeneSJ_subset$Gene <- "KRAS"
GeneSJ_subset$MutationKey_Hg38 <- "chr12,25245347,C,T"

annot_vaf <- merge(found_variants[,c("MutationKey_Hg38","Gene","sample_id","case_id",                                               "RNA_Sample","DNA_Sample","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf",                         "MuTect2_filter","VarScan2_filter","SomaticSniper_filter","MuSE_filter","MUTReads","WTReads","VAF")], 
                   GeneSJ_subset[,c("Gene","MutationKey_Hg38","sample_id", "case_id")], 
                   by.x=c("RNA_Sample", "case_id","Gene", "MutationKey_Hg38"), 
                   by.y=c("sample_id", "case_id","Gene", "MutationKey_Hg38"))


annot_vaf <- annot_vaf %>% select(Gene,MutationKey_Hg38,
                                  sample_id,
                                  case_id, RNA_Sample,DNA_Sample,
                                  MuTect2_vaf,VarScan2_vaf,SomaticSniper_vaf,MuSE_vaf,
                                  VAF,
                                  everything()
                                  )
colnames(annot_vaf)[grep("VAF", names(annot_vaf))] <- "VAF_RNA"
colnames(annot_vaf)[grep("MUTReads", names(annot_vaf))] <- "MUTReads_RNA"
colnames(annot_vaf)[grep("WTReads", names(annot_vaf))] <- "WTReads_RNA"

write.xlsx(annot_vaf, paste0(normalizedEx_dir_out, "BeatAML.KRAS.chr12-25245347-C-T.MUT.xlsx"), rowNames = FALSE)
```

Mutated samples vaf:
```{r echo=FALSE}
datatable(annot_vaf,
extensions = 'Buttons',
rownames = FALSE, 
filter = 'top',
options = list(paging = TRUE,
              pageLength = 4,
              autoWidth = TRUE,
              dom = 'lifrtBp',
              buttons = c('excel'), 
              scrollX = TRUE,  
              scrollY = TRUE
              )
)
```


```{r echo=FALSE}
## Generate Mut_df
MUT_df <- GeneSJ[GeneSJ$GROUP == "MUT",c("sample_id","case_id")]
```


```{r echo=FALSE}
## Add not found effects information: Donor Gain 1
MUT_df <- MUT_df[,c("sample_id", "case_id")]
MUT_df$NormalizedExpression <- NA
MUT_df$Prediction <- "Donor Gain"
MUT_df$splice_junction_status <- "AlternativeSJ not found in SJ collection"
MUT_df$splice_junction_position <- "chr12:25245345"
MUT_df$Pvalue <- NA

vaf_select <- annot_vaf[,c("Gene","MutationKey_Hg38","sample_id","RNA_Sample","DNA_Sample","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf", "VAF_RNA")]
MUT_vaf <- merge(MUT_df, vaf_select, by.x="sample_id", by.y = "RNA_Sample")
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id"] <- "RNA_Sample"
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id.y"] <- "sample_id"

col_order <- c("Gene", "MutationKey_Hg38", "Prediction","splice_junction_status","splice_junction_position","sample_id", "case_id","RNA_Sample","DNA_Sample","NormalizedExpression", "Pvalue", "VAF_RNA","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf")
MUT_vaf <- MUT_vaf[, col_order]

stat_df <- data.frame(matrix(ncol = 17, nrow = 1))
colnames(stat_df) <- c("Gene","MutationKey_Hg38","Prediction","splice_junction_status","splice_junction_position", "sample_id","case_id","RNA_Sample","DNA_Sample","NormalizedExpression","ECDF","Pvalue","VAF_RNA","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf")

stat_df$Gene <- unique(MUT_vaf$Gene)
stat_df$MutationKey_Hg38 <- unique(MUT_vaf$MutationKey_Hg38)
stat_df$sample_id <- paste(MUT_vaf$sample_id, collapse = ';\n') 
stat_df$case_id <- paste(MUT_vaf$case_id, collapse = ';\n') 
stat_df$RNA_Sample <- paste(MUT_vaf$RNA_Sample, collapse = ';\n') 
stat_df$DNA_Sample <- paste(MUT_vaf$DNA_Sample, collapse = ';\n')
stat_df$NormalizedExpression <- NA
stat_df$ECDF <- NA
stat_df$VAF_RNA <-  paste(MUT_vaf$VAF_RNA, collapse = ';\n')
stat_df$MuTect2_vaf <-  paste(MUT_vaf$MuTect2_vaf, collapse = ';\n')
stat_df$VarScan2_vaf <-  paste(MUT_vaf$VarScan2_vaf, collapse = ';\n')
stat_df$SomaticSniper_vaf <-  paste(MUT_vaf$SomaticSniper_vaf, collapse = ';\n')
stat_df$MuSE_vaf <-  paste(MUT_vaf$MuSE_vaf, collapse = ';\n')
stat_df$Prediction <- unique(MUT_vaf$Prediction)
stat_df$splice_junction_status <- unique(MUT_vaf$splice_junction_status)
stat_df$splice_junction_position <- unique(MUT_vaf$splice_junction_position)
stat_df$Pvalue <- NA

## Save
summary_res <- rbind(summary_res, stat_df)
```

```{r echo=FALSE}
## Add not found effects information: Donor Gain 2
MUT_df <- MUT_df[,c("sample_id", "case_id")]
MUT_df$NormalizedExpression <- NA
MUT_df$Prediction <- "Donor Gain"
MUT_df$splice_junction_status <- "AlternativeSJ not found in SJ collection"
MUT_df$splice_junction_position <- "chr12:25245281-25245348"
MUT_df$Pvalue <- NA

vaf_select <- annot_vaf[,c("Gene","MutationKey_Hg38","sample_id","RNA_Sample","DNA_Sample","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf", "VAF_RNA")]
MUT_vaf <- merge(MUT_df, vaf_select, by.x="sample_id", by.y = "RNA_Sample")
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id"] <- "RNA_Sample"
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id.y"] <- "sample_id"

col_order <- c("Gene", "MutationKey_Hg38", "Prediction","splice_junction_status","splice_junction_position","sample_id", "case_id","RNA_Sample","DNA_Sample","NormalizedExpression", "Pvalue", "VAF_RNA","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf")
MUT_vaf <- MUT_vaf[, col_order]

stat_df <- data.frame(matrix(ncol = 17, nrow = 1))
colnames(stat_df) <- c("Gene","MutationKey_Hg38","Prediction","splice_junction_status","splice_junction_position", "sample_id","case_id","RNA_Sample","DNA_Sample","NormalizedExpression","ECDF","Pvalue","VAF_RNA","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf")

stat_df$Gene <- unique(MUT_vaf$Gene)
stat_df$MutationKey_Hg38 <- unique(MUT_vaf$MutationKey_Hg38)
stat_df$sample_id <- paste(MUT_vaf$sample_id, collapse = ';\n') 
stat_df$case_id <- paste(MUT_vaf$case_id, collapse = ';\n') 
stat_df$RNA_Sample <- paste(MUT_vaf$RNA_Sample, collapse = ';\n') 
stat_df$DNA_Sample <- paste(MUT_vaf$DNA_Sample, collapse = ';\n')
stat_df$NormalizedExpression <- NA
stat_df$ECDF <- NA
stat_df$VAF_RNA <-  paste(MUT_vaf$VAF_RNA, collapse = ';\n')
stat_df$MuTect2_vaf <-  paste(MUT_vaf$MuTect2_vaf, collapse = ';\n')
stat_df$VarScan2_vaf <-  paste(MUT_vaf$VarScan2_vaf, collapse = ';\n')
stat_df$SomaticSniper_vaf <-  paste(MUT_vaf$SomaticSniper_vaf, collapse = ';\n')
stat_df$MuSE_vaf <-  paste(MUT_vaf$MuSE_vaf, collapse = ';\n')
stat_df$Prediction <- unique(MUT_vaf$Prediction)
stat_df$splice_junction_status <- unique(MUT_vaf$splice_junction_status)
stat_df$splice_junction_position <- unique(MUT_vaf$splice_junction_position)
stat_df$Pvalue <- NA

## Save
summary_res <- rbind(summary_res, stat_df)
```

```{r echo=FALSE}
rm(GeneSJ, normalization_df, GeneSJ_subset, annot_vaf, plot_df, samples_df,table, plot_boxplot, plot_dotplot, vaf_select, SJCounts, MUT_df, MUT_vaf)
```


# KMT2D chr12,49022063,G,A {.tabset}
Variant found in **4 patients** of the BeatAML (**4 samples**)

* Patients with KMT2D chr12,49022063,G,A variant: **4 patients (4 samples)**
* Patients with the variant and RNASeq for validation: **4 patients (4 samples)**

The splicing alterations being assessed are: 

* **Donor Gain**: predicted at 2bp from the variant, chr12:49022064, not found in the splice junction collection.
* **Donor Gain**: chr12:49022070, not found in the splice junction collection.

Variant information:
```{r echo=FALSE}
datatable(to_be_validated[to_be_validated$MutationKey_Hg38 == "chr12,49022063,G,A",],
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp', 
              scrollX = TRUE,  
              scrollY = TRUE,
              columnDefs = list(list(targets = '_all', className = 'dt-center')))
)
```

Load the extracted splice junctions of the gene harboring the mutation.
```{r}
extractedSJ_path <- paste0(extractedSJ_dir_in,"KMT2D_UM_annotSJ.tsv")
GeneSJ <- read.delim(extractedSJ_path, sep ="\t")
```

Set the sample's group: Mutated (MUT) or No Mutated (WT)

```{r}
samples_df <- found_variants[found_variants$Gene=="KMT2D" & found_variants$MutationKey_Hg38 == "chr12,49022063,G,A",]

cases <- samples_df$RNA_Sample[samples_df$Validable == "Validable"]
GeneSJ$GROUP <- ifelse(GeneSJ$sample_id %in% cases , "MUT", "WT")
```


## SJ Lookup{.tabset}

Search for the splice junctions of interest in the extracted splice junctions of the gene by position (chr_SJstart_SJend).

### Donor Gain 
Search: chr12:49022064

Show all the splice junctions containing the positions from 49022060 to 49022069
```{r}
colnames(GeneSJ)[grepl("4902206", colnames(GeneSJ))] 
```
Alternative SJ not found in the splice junction collection.

### Donor Gain 
Search: chr12:49022070
Show all the splice junctions containing the positions from 49022070 to 49022079
```{r}
colnames(GeneSJ)[grepl("4902207", colnames(GeneSJ))] 
```

Alternative SJ not found in the splice junction collection.

## VAF
```{r echo=FALSE}
GeneSJ_subset <- GeneSJ[GeneSJ$sample_id %in% cases,]
GeneSJ_subset$Gene <- "KMT2D"
GeneSJ_subset$MutationKey_Hg38 <- "chr12,49022063,G,A"

annot_vaf <- merge(found_variants[,c("MutationKey_Hg38","Gene","sample_id","case_id",                                               "RNA_Sample","DNA_Sample","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf",                         "MuTect2_filter","VarScan2_filter","SomaticSniper_filter","MuSE_filter","MUTReads","WTReads","VAF")], 
                   GeneSJ_subset[,c("Gene","MutationKey_Hg38","sample_id", "case_id")], 
                   by.x=c("RNA_Sample", "case_id","Gene", "MutationKey_Hg38"), 
                   by.y=c("sample_id", "case_id","Gene", "MutationKey_Hg38"))


annot_vaf <- annot_vaf %>% select(Gene,MutationKey_Hg38,
                                  sample_id,
                                  case_id, RNA_Sample,DNA_Sample,
                                  MuTect2_vaf,VarScan2_vaf,SomaticSniper_vaf,MuSE_vaf,
                                  VAF,
                                  everything()
                                  )
colnames(annot_vaf)[grep("VAF", names(annot_vaf))] <- "VAF_RNA"
colnames(annot_vaf)[grep("MUTReads", names(annot_vaf))] <- "MUTReads_RNA"
colnames(annot_vaf)[grep("WTReads", names(annot_vaf))] <- "WTReads_RNA"

write.xlsx(annot_vaf, paste0(normalizedEx_dir_out, "BeatAML.KMT2D.chr12-49022063-G-A.MUT.xlsx"), rowNames = FALSE)
```

Mutated samples vaf:
```{r echo=FALSE}
datatable(annot_vaf,
extensions = 'Buttons',
rownames = FALSE, 
filter = 'top',
options = list(paging = TRUE,
              pageLength = 4,
              autoWidth = TRUE,
              dom = 'lifrtBp',
              buttons = c('excel'), 
              scrollX = TRUE,  
              scrollY = TRUE
              )
)
```


```{r echo=FALSE}
## Generate Mut_df
MUT_df <- GeneSJ[GeneSJ$GROUP == "MUT",c("sample_id","case_id")]
```


```{r echo=FALSE}
## Add not found effects information: Donor Gain 1
MUT_df <- MUT_df[,c("sample_id", "case_id")]
MUT_df$NormalizedExpression <- NA
MUT_df$Prediction <- "Donor Gain"
MUT_df$splice_junction_status <- "AlternativeSJ not found in SJ collection"
MUT_df$splice_junction_position <- "chr12:49022064"
MUT_df$Pvalue <- NA

vaf_select <- annot_vaf[,c("Gene","MutationKey_Hg38","sample_id","RNA_Sample","DNA_Sample","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf", "VAF_RNA")]
MUT_vaf <- merge(MUT_df, vaf_select, by.x="sample_id", by.y = "RNA_Sample")
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id"] <- "RNA_Sample"
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id.y"] <- "sample_id"

col_order <- c("Gene", "MutationKey_Hg38", "Prediction","splice_junction_status","splice_junction_position","sample_id", "case_id","RNA_Sample","DNA_Sample","NormalizedExpression", "Pvalue", "VAF_RNA","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf")
MUT_vaf <- MUT_vaf[, col_order]

stat_df <- data.frame(matrix(ncol = 17, nrow = 1))
colnames(stat_df) <- c("Gene","MutationKey_Hg38","Prediction","splice_junction_status","splice_junction_position", "sample_id","case_id","RNA_Sample","DNA_Sample","NormalizedExpression","ECDF","Pvalue","VAF_RNA","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf")

stat_df$Gene <- unique(MUT_vaf$Gene)
stat_df$MutationKey_Hg38 <- unique(MUT_vaf$MutationKey_Hg38)
stat_df$sample_id <- paste(MUT_vaf$sample_id, collapse = ';\n') 
stat_df$case_id <- paste(MUT_vaf$case_id, collapse = ';\n') 
stat_df$RNA_Sample <- paste(MUT_vaf$RNA_Sample, collapse = ';\n') 
stat_df$DNA_Sample <- paste(MUT_vaf$DNA_Sample, collapse = ';\n')
stat_df$NormalizedExpression <- NA
stat_df$ECDF <- NA
stat_df$VAF_RNA <-  paste(MUT_vaf$VAF_RNA, collapse = ';\n')
stat_df$MuTect2_vaf <-  paste(MUT_vaf$MuTect2_vaf, collapse = ';\n')
stat_df$VarScan2_vaf <-  paste(MUT_vaf$VarScan2_vaf, collapse = ';\n')
stat_df$SomaticSniper_vaf <-  paste(MUT_vaf$SomaticSniper_vaf, collapse = ';\n')
stat_df$MuSE_vaf <-  paste(MUT_vaf$MuSE_vaf, collapse = ';\n')
stat_df$Prediction <- unique(MUT_vaf$Prediction)
stat_df$splice_junction_status <- unique(MUT_vaf$splice_junction_status)
stat_df$splice_junction_position <- unique(MUT_vaf$splice_junction_position)
## Save
summary_res <- rbind(summary_res, stat_df)
```

```{r echo=FALSE}
## Add not found effects information: Donor Gain 2
MUT_df <- MUT_df[,c("sample_id", "case_id")]
MUT_df$NormalizedExpression <- NA
MUT_df$Prediction <- "Donor Gain"
MUT_df$splice_junction_status <- "AlternativeSJ not found in SJ collection"
MUT_df$splice_junction_position <- "chr12:49022070"
MUT_df$Pvalue <- NA

vaf_select <- annot_vaf[,c("Gene","MutationKey_Hg38","sample_id","RNA_Sample","DNA_Sample","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf", "VAF_RNA")]
MUT_vaf <- merge(MUT_df, vaf_select, by.x="sample_id", by.y = "RNA_Sample")
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id"] <- "RNA_Sample"
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id.y"] <- "sample_id"

col_order <- c("Gene", "MutationKey_Hg38", "Prediction","splice_junction_status","splice_junction_position","sample_id", "case_id","RNA_Sample","DNA_Sample","NormalizedExpression", "Pvalue", "VAF_RNA","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf")
MUT_vaf <- MUT_vaf[, col_order]

stat_df <- data.frame(matrix(ncol = 17, nrow = 1))
colnames(stat_df) <- c("Gene","MutationKey_Hg38","Prediction","splice_junction_status","splice_junction_position", "sample_id","case_id","RNA_Sample","DNA_Sample","NormalizedExpression","ECDF","Pvalue","VAF_RNA","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf")

stat_df$Gene <- unique(MUT_vaf$Gene)
stat_df$MutationKey_Hg38 <- unique(MUT_vaf$MutationKey_Hg38)
stat_df$sample_id <- paste(MUT_vaf$sample_id, collapse = ';\n') 
stat_df$case_id <- paste(MUT_vaf$case_id, collapse = ';\n') 
stat_df$RNA_Sample <- paste(MUT_vaf$RNA_Sample, collapse = ';\n') 
stat_df$DNA_Sample <- paste(MUT_vaf$DNA_Sample, collapse = ';\n')
stat_df$NormalizedExpression <- NA
stat_df$ECDF <- NA
stat_df$VAF_RNA <-  paste(MUT_vaf$VAF_RNA, collapse = ';\n')
stat_df$MuTect2_vaf <-  paste(MUT_vaf$MuTect2_vaf, collapse = ';\n')
stat_df$VarScan2_vaf <-  paste(MUT_vaf$VarScan2_vaf, collapse = ';\n')
stat_df$SomaticSniper_vaf <-  paste(MUT_vaf$SomaticSniper_vaf, collapse = ';\n')
stat_df$MuSE_vaf <-  paste(MUT_vaf$MuSE_vaf, collapse = ';\n')
stat_df$Prediction <- unique(MUT_vaf$Prediction)
stat_df$splice_junction_status <- unique(MUT_vaf$splice_junction_status)
stat_df$splice_junction_position <- unique(MUT_vaf$splice_junction_position)

## Save
summary_res <- rbind(summary_res, stat_df)
```


```{r echo=FALSE}
rm(GeneSJ, normalization_df, GeneSJ_subset, annot_vaf, plot_df, samples_df,table, plot_boxplot, plot_dotplot, vaf_select, SJCounts, MUT_df, MUT_vaf)
```


# FLT3 chr13,28018485,G,T {.tabset}
Variant found in **3 patients** of the BeatAML (**3 samples**)

* Patients with FLT3 chr13,28018485,G,T variant: **3 patients (3 samples)**
* Patients with the variant and RNASeq for validation: **1 patients (1 samples)**


The splicing alterations being assessed are: 

* **Donor Loss**: chr13:28015702-28018466; donor splice site chr13:28018466.
* **Exon Skipping 20**: chr13:28015702-28023349, found in mutated samples.
* **Exon Skipping 19**: chr13:28018590-28024860, found in mutated samples.
* **Exon Skipping 19 + 20**: chr13:28015702-28024860, found in mutated samples.
 
Variant information:
```{r echo=FALSE}
datatable(to_be_validated[to_be_validated$MutationKey_Hg38 == "chr13,28018485,G,T",],
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp', 
              scrollX = TRUE,  
              scrollY = TRUE,
              columnDefs = list(list(targets = '_all', className = 'dt-center')))
)
```

Load the extracted splice junctions of the gene harboring the mutation.
```{r}
extractedSJ_path <- paste0(extractedSJ_dir_in,"FLT3_UM_annotSJ.tsv")
GeneSJ <- read.delim(extractedSJ_path, sep ="\t")
```

Set the sample's group: Mutated (MUT) or No Mutated (WT)

```{r}
samples_df <- found_variants[found_variants$Gene=="FLT3" & found_variants$MutationKey_Hg38 == "chr13,28018485,G,T",]

cases <- samples_df$RNA_Sample[samples_df$Validable == "Validable"]
GeneSJ$GROUP <- ifelse(GeneSJ$sample_id %in% cases , "MUT", "WT")
```


## SJ Lookup{.tabset}

Search for the splice junctions of interest in the extracted splice junctions of the gene by position (chr_SJstart_SJend).

### Exon Skipping 20
Search: chr13:28015702-28023349 

Show all the splice junctions containing the position 28015702_28023349
```{r}
colnames(GeneSJ)[grep("28015702_28023349",colnames(GeneSJ))]
```

Found: chr13:28015702-28023349

Reads of all the AML samples (mutated and no mutated) for the splice junction:
```{r}
GeneSJ$chr13_28015702_28023349
```
Samples with the SJ of interest:
```{r}
table(GeneSJ$chr13_28015702_28023349>0) 
```
Groups of the samples having the alternative splice junction:
```{r}
table(GeneSJ$GROUP[GeneSJ$chr13_28015702_28023349 > 0])
```

Alternative SJ found in the mutated samples.

### Exon Skipping 19
Search: chr13:28018590-28024860

Show all the splice junctions containing the positions 28018590-28024860
```{r}
colnames(GeneSJ)[grep("28018590_28024860",colnames(GeneSJ))]
```

Found: chr13:28018590-28024860

Reads of all the AML samples (mutated and no mutated) for the splice junction:
```{r}
GeneSJ$chr13_28018590_28024860
```

Samples with the SJ of interest:
```{r}
table(GeneSJ$chr13_28018590_28024860>0) 
```

Groups of the samples having the alternative splice junction:
```{r}
table(GeneSJ$GROUP[GeneSJ$chr13_28018590_28024860 > 0])
```
Alternative SJ found in the mutated samples.

### Exon Skipping 19+20
Search: chr13:28015702-28024860

Show all the splice junctions containing the positions 28015702-28024860
```{r}
colnames(GeneSJ)[grep("28015702_28024860",colnames(GeneSJ))]
```

Reads of all the AML samples (mutated and no mutated) for the splice junction:
```{r}
GeneSJ$chr13_28015702_28024860
```

Samples with the SJ of interest:
```{r}
table(GeneSJ$chr13_28015702_28024860 >0) 
```

Groups of the samples having the alternative splice junction:
```{r}
table(GeneSJ$GROUP[GeneSJ$chr13_28015702_28024860 > 0])
```
Alternative SJ found in the mutated samples.

### Canonical SJ
Exon 19-20	chr13:28018590-28023349

Exon 20-21	chr13:28015702-28018466; splice site chr13:28018466

Reads of all the AML samples (mutated and no mutated) for the splice junction:
```{r}
GeneSJ$chr13_28018590_28023349
```

Reads of all the AML samples (mutated and no mutated) for the splice junction:
```{r}
GeneSJ$chr13_28015702_28018466
```


## Normalization
Count the reads of all the splice junctions of the gene harboring the variant:
```{r}
GeneSJ$rowSum_SJtotal <- rowSums(GeneSJ[,grep("chr", names(GeneSJ))])
```

Normalization of the expression by the total read counts of all the splice junctions of the gene:
```{r}
GeneSJ$Normalized_CanonEx19_20 <- (GeneSJ$chr13_28018590_28023349)/GeneSJ$rowSum_SJtotal*100
GeneSJ$Normalized_CanonEx20_21 <- (GeneSJ$chr13_28015702_28018466)/GeneSJ$rowSum_SJtotal*100

GeneSJ$Normalized_SE20 <- (GeneSJ$chr13_28015702_28023349)/GeneSJ$rowSum_SJtotal*100
GeneSJ$INCLUSION_Ex20 <- GeneSJ$chr13_28018590_28023349 + GeneSJ$chr13_28015702_28018466
GeneSJ$PSI_SE20 <- (GeneSJ$INCLUSION_Ex20)/(GeneSJ$chr13_28015702_28023349+GeneSJ$INCLUSION_Ex20)


GeneSJ$Normalized_SE19 <- (GeneSJ$chr13_28018590_28024860)/GeneSJ$rowSum_SJtotal*100
GeneSJ$INCLUSION_Ex19 <- GeneSJ$chr13_28018590_28023349 + GeneSJ$chr13_28023478_28024860
GeneSJ$PSI_SE19 <- (GeneSJ$INCLUSION_Ex19)/(GeneSJ$chr13_28018590_28024860+GeneSJ$INCLUSION_Ex19)

GeneSJ$Normalized_SE1920 <- (GeneSJ$chr13_28015702_28024860)/GeneSJ$rowSum_SJtotal*100
```

Download the normalized values for the assessed splice junctions of all the AML samples:
```{r echo=FALSE}
normalization_df <- GeneSJ[,c("sample_id","case_id", "GROUP",
                              "Normalized_CanonEx19_20", "Normalized_CanonEx20_21", "Normalized_SE20", "INCLUSION_Ex20","PSI_SE20",
                              "Normalized_SE19", "INCLUSION_Ex19","PSI_SE19",
                              "Normalized_SE1920", 
                              "chr13_28018590_28023349", "chr13_28015702_28018466", "chr13_28015702_28023349", 
                              "chr13_28018590_28024860", "chr13_28015702_28024860", "rowSum_SJtotal","file_id.BAM","file_name.BAM","file_id.STAR.SJCounts","file_name.STAR.SJCounts" )]

colnames(normalization_df)[grep("chr13_28018590_28023349", names(normalization_df))] <- "chr13_28018590_28023349_CanonEx19_20"
colnames(normalization_df)[grep("chr13_28015702_28018466", names(normalization_df))] <- "chr13_28015702_28018466_CanonEx20_21"
colnames(normalization_df)[grep("chr13_28015702_28023349", names(normalization_df))] <- "chr13_28015702_28023349_SE20"
colnames(normalization_df)[grep("chr13_28018590_28024860", names(normalization_df))] <- "chr13_28018590_28024860_SE19"
colnames(normalization_df)[grep("chr13_28015702_28024860", names(normalization_df))] <- "chr13_28015702_28024860_SE1920"

write.xlsx(normalization_df, paste0(normalizedEx_dir_out, "BeatAML.FLT3.chr13-28018485-G-T.xlsx"), rowNames = FALSE)
```

```{r echo=FALSE}
datatable(normalization_df,
extensions = 'Buttons',
rownames = FALSE, 
filter = 'top',
options = list(paging = TRUE,
              pageLength = 4,
              autoWidth = FALSE,
              dom = 'lifrtBp',
              buttons = c('excel'), 
              scrollX = TRUE,  
              scrollY = TRUE,
              columnDefs = list(list(targets = '_all', className = 'dt-center'))
))
```

## VAF
```{r echo=FALSE}
GeneSJ_subset <- GeneSJ[GeneSJ$sample_id %in% cases,]
GeneSJ_subset$Gene <- "FLT3"
GeneSJ_subset$MutationKey_Hg38 <- "chr13,28018485,G,T"

annot_vaf <- merge(found_variants[,c("MutationKey_Hg38","Gene","sample_id","case_id",                                               "RNA_Sample","DNA_Sample","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf",                         "MuTect2_filter","VarScan2_filter","SomaticSniper_filter","MuSE_filter","MUTReads","WTReads","VAF")], 
                   GeneSJ_subset[,c("Gene","MutationKey_Hg38","sample_id", "case_id", "Normalized_CanonEx19_20", "Normalized_CanonEx20_21", "Normalized_SE20", "PSI_SE20", "Normalized_SE19", "Normalized_SE1920")], 
                   by.x=c("RNA_Sample", "case_id","Gene", "MutationKey_Hg38"), 
                   by.y=c("sample_id", "case_id","Gene", "MutationKey_Hg38"))


annot_vaf <- annot_vaf %>% select(Gene,MutationKey_Hg38,
                                  sample_id,
                                  case_id, RNA_Sample,DNA_Sample,
                                  MuTect2_vaf,VarScan2_vaf,SomaticSniper_vaf,MuSE_vaf,
                                  VAF,
                                  Normalized_CanonEx19_20, Normalized_CanonEx20_21, Normalized_SE20, PSI_SE20,
                                  Normalized_SE19, Normalized_SE1920,
                                  everything()
                                  )
colnames(annot_vaf)[grep("VAF", names(annot_vaf))] <- "VAF_RNA"
colnames(annot_vaf)[grep("MUTReads", names(annot_vaf))] <- "MUTReads_RNA"
colnames(annot_vaf)[grep("WTReads", names(annot_vaf))] <- "WTReads_RNA"

write.xlsx(annot_vaf, paste0(normalizedEx_dir_out, "BeatAML.FLT3.chr13-28018485-G-T.MUT.xlsx"), rowNames = FALSE)
```

Mutated samples vaf:
```{r echo=FALSE}
datatable(annot_vaf,
extensions = 'Buttons',
rownames = FALSE, 
filter = 'top',
options = list(paging = TRUE,
              pageLength = 4,
              autoWidth = TRUE,
              dom = 'lifrtBp',
              buttons = c('excel'), 
              scrollX = TRUE,  
              scrollY = TRUE
              )
)
```

## Plots {.tabset}

```{r echo=FALSE}
GeneSJ$GROUP <- ifelse(GeneSJ$GROUP == "MUT", "MUT", "WT")
GeneSJ$GROUP <- as.factor(GeneSJ$GROUP)
```

### Static Dot Plots 
Canonical splice junction: Exon 20-21	chr13:28015702-28018466; donor splice site chr13:28018466

```{r echo=FALSE}
ggplot(GeneSJ, aes(sample_id,Normalized_CanonEx20_21,color=GROUP)) + 
  geom_point(size=.8) +
  labs(color='GROUP', 
       title = "Normalized Expression of FLT3 chr13:28015702-28018466", 
       subtitle="Potential Donor Loss Effect",
       y = "Normalized Expression")
```


Splicing alterations:

```{r echo=FALSE}
ggplot(GeneSJ, aes(sample_id,Normalized_SE20,color=GROUP)) + 
  geom_point(size=.8) +
  labs(color='GROUP', 
       title = "Normalized Expression of FLT3 chr13:28015702-28023349", 
       subtitle="Potential Exon 20 Skipping",
       y = "Normalized Expression")
```


```{r echo=FALSE}
ggplot(GeneSJ, aes(sample_id,Normalized_SE19,color=GROUP)) + 
  geom_point(size=.8) +
  labs(color='GROUP', 
       title = "Normalized Expression of FLT3 chr13:28018590-28024860", 
       subtitle="Potential Exon 19 Skipping",
       y = "Normalized Expression")
```


```{r echo=FALSE}
ggplot(GeneSJ, aes(sample_id,Normalized_SE1920,color=GROUP)) + 
  geom_point(size=.8) +
  labs(color='GROUP', 
       title = "Normalized Expression of FLT3 chr13:28015702-28024860", 
       subtitle="Potential Exon 19+20 Skipping",
       y = "Normalized Expression")
```

### Interactive Dot Plots

```{r echo=FALSE}
plot_df <- merge(GeneSJ,annot_vaf[c("RNA_Sample", "VAF_RNA", 
                                    "MuTect2_vaf", "VarScan2_vaf", "SomaticSniper_vaf", "MuSE_vaf")],
                 by.x="sample_id", by.y="RNA_Sample", all.x = TRUE)
```

Canonical splice junction: Exon 20-21	chr13:28015702-28018466; donor splice site chr13:28018466

```{r echo=FALSE}
ggplotly(ggplot(plot_df, aes(sample_id,Normalized_CanonEx20_21,color=GROUP)) + 
  geom_point(aes(text=paste0("VAF RNA: ",VAF_RNA,"\nVAF MuTect: ",MuTect2_vaf,
                             "\nVAF VarScan2: ",VarScan2_vaf,
                             "\nVAF SomaticSniper: ",SomaticSniper_vaf,
                             "\nVAF MuSE: ",MuSE_vaf)),size= 0.8) +
  ggtitle("Normalized expression of FLT3 Exon 20-21 Splice Junction") +
  labs(color='GROUP')
)
```

Splicing alterations:

```{r echo=FALSE}
ggplotly(ggplot(plot_df, aes(sample_id,Normalized_SE20,color=GROUP)) + 
  geom_point(aes(text=paste0("VAF RNA: ",VAF_RNA,"\nVAF MuTect: ",MuTect2_vaf,
                             "\nVAF VarScan2: ",VarScan2_vaf,
                             "\nVAF SomaticSniper: ",SomaticSniper_vaf,
                             "\nVAF MuSE: ",MuSE_vaf)),size= 0.8)  +
  ggtitle("Normalized expression of FLT3 Exon Skipping 20 Splice Junction") +
  labs(color='GROUP')
)
  
ggplotly(ggplot(plot_df, aes(sample_id,Normalized_SE19,color=GROUP)) + 
  geom_point(aes(text=paste0("VAF RNA: ",VAF_RNA,"\nVAF MuTect: ",MuTect2_vaf,
                             "\nVAF VarScan2: ",VarScan2_vaf,
                             "\nVAF SomaticSniper: ",SomaticSniper_vaf,
                             "\nVAF MuSE: ",MuSE_vaf)),size= 0.8)  +
  ggtitle("Normalized expression of FLT3 Exon Skipping 19 Splice Junction") +
  labs(color='GROUP')
)

ggplotly(ggplot(plot_df, aes(sample_id,Normalized_SE1920,color=GROUP)) + 
  geom_point(aes(text=paste0("VAF RNA: ",VAF_RNA,"\nVAF MuTect: ",MuTect2_vaf,
                             "\nVAF VarScan2: ",VarScan2_vaf,
                             "\nVAF SomaticSniper: ",SomaticSniper_vaf,
                             "\nVAF MuSE: ",MuSE_vaf)),size= 0.8)  +
  ggtitle("Normalized expression of FLT3 Exon Skipping 19+20 Splice Junction") +
  labs(color='GROUP')
)
```


### Violin Plots
Violin Plots for the alternative splice junctions interrogated:

```{r echo=FALSE}
plot_boxplot <- plot_df[plot_df$GROUP %in% c("WT"),]
plot_dotplot <- plot_df[plot_df$GROUP %in% c("MUT"),]

GEN = "FLT3"
SJ = "chr13:28015702-28023349"

ggplot(plot_df, aes(y=Normalized_SE20, x=factor(GROUP, level=c("WT", "MUT")))) + 
       geom_violin(scale="area") +
       geom_point(data = plot_dotplot, 
                  aes(y=Normalized_SE20, x=GROUP),
                  color="black",
                  size=2
                  ) +
        
       ylab("Normalized Expression of ES20 (%)") +
       xlab(paste(GEN, SJ))
```


## Statistical Analysis {.tabset}
```{r}
SJCounts <- GeneSJ
```

### Donor Loss {.tabset}
#### Expression Difference 
Normality Test:
```{r}
shapiro.test(SJCounts$Normalized_CanonEx20_21[SJCounts$GROUP == "WT"])
```

Value of Mean Normalized Expression of the Alternative SJ in WT samples:
```{r}
mean_WT_SJi <- mean(SJCounts$Normalized_CanonEx20_21[SJCounts$GROUP == "WT"], na.rm=TRUE)
mean_WT_SJi
```

Normalized Expression Value of the Alternative SJ in the MUT sample:
```{r}
MUT_SJi <- SJCounts$Normalized_CanonEx20_21[SJCounts$GROUP == "MUT"]
MUT_SJi
```

Deviation from the mean normalized expression:
```{r}
SJCounts$Difference <-  SJCounts$Normalized_CanonEx20_21 - mean_WT_SJi
```

```{r echo=FALSE}
table <- SJCounts[SJCounts$GROUP == "WT", c("sample_id", "Normalized_CanonEx20_21", "Difference")]
table <- table[order(table$Difference, decreasing = TRUE),]
```

```{r echo=FALSE}
datatable(table,
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp'
))
```

Difference in the MUT sample: deviation of the Normalized expression of the MUT patient from the mean normalized WT expression
```{r}
SJCounts$Difference[SJCounts$GROUP == "MUT"]
```

```{r echo=FALSE}
ggplot(SJCounts, aes(sample_id,Difference)) + geom_point(aes(colour=factor(GROUP))) +
  ylab("Differential Expression") +
  labs(color = "GROUP",
       title = "Differential Expression Distribution")
```


```{r echo=FALSE}
ggplot(SJCounts, aes(y=Difference, x=factor(GROUP, level=c("WT", "MUT")))) + 
  geom_boxplot() +
  ylab("Differential Expression") +
  xlab("GROUP")
```

#### ECDF - Pvalue Inference

```{r}
v_ecdf <- ecdf(SJCounts$Difference[SJCounts$GROUP == "WT"])
v_ecdf
```

```{r}
plot(ecdf(SJCounts$Difference[SJCounts$GROUP == "WT"]))
```

```{r}
v_ecdf(SJCounts$Difference[SJCounts$GROUP == "MUT"])
print(paste0("MUT Percentile: ", v_ecdf(SJCounts$Difference[SJCounts$GROUP == "MUT"])))
```

```{r}
print(paste0("Inferred Pvalue: ", v_ecdf(SJCounts$Difference[SJCounts$GROUP == "MUT"])))
```


```{r}
MUT_df <- SJCounts[SJCounts$GROUP == "MUT",c("sample_id","case_id", "Normalized_CanonEx20_21")]
colnames(MUT_df) <- c("sample_id", "case_id", "NormalizedExpression")

MUT_df$ECDF <- v_ecdf(SJCounts$Difference[SJCounts$GROUP == "MUT"])
MUT_df$Pvalue <-  MUT_df$ECDF

MUT_df$Prediction <- "Donor Loss"
MUT_df$splice_junction_status <- "CanonicalSJ"
MUT_df$splice_junction_position <- "chr13:28015702-28018466"

MUT_df$ECDF <- ifelse(MUT_df$NormalizedExpression == 0, NA,MUT_df$ECDF)
MUT_df$Pvalue <- ifelse(MUT_df$NormalizedExpression == 0, NA,MUT_df$Pvalue)
```

```{r echo=FALSE}
vaf_select <- annot_vaf[,c("Gene","MutationKey_Hg38","sample_id","RNA_Sample","DNA_Sample","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf", "VAF_RNA")]
MUT_vaf <- merge(MUT_df, vaf_select, by.x="sample_id", by.y = "RNA_Sample")
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id"] <- "RNA_Sample"
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id.y"] <- "sample_id"

col_order <- c("Gene", "MutationKey_Hg38", "Prediction","splice_junction_status","splice_junction_position","sample_id", "case_id","RNA_Sample","DNA_Sample","NormalizedExpression","ECDF", "Pvalue", "VAF_RNA","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf")
MUT_vaf <- MUT_vaf[, col_order]
```

Download the vaf, inferred percentiles and pvalues of the mutated samples:
```{r echo=FALSE}
datatable(MUT_vaf,
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp',
              scrollX=TRUE
))
```

```{r echo=FALSE}
summary_res <- rbind(summary_res, MUT_vaf)
```

### Exon Skipping 20 {.tabset}
#### Expression Difference 
Normality Test:
```{r}
shapiro.test(SJCounts$Normalized_SE20[SJCounts$GROUP == "WT"])
```

Value of Mean Normalized Expression of the SJ in WT samples:
```{r}
mean_WT_SJi <- mean(SJCounts$Normalized_SE20[SJCounts$GROUP == "WT"], na.rm=TRUE)
mean_WT_SJi
```

Normalized Expression Value of the  SJ in the MUT sample:
```{r}
MUT_SJi <- SJCounts$Normalized_SE20[SJCounts$GROUP == "MUT"]
MUT_SJi
```

Deviation from the mean normalized expression:
```{r}
SJCounts$Difference <-  SJCounts$Normalized_SE20 - mean_WT_SJi
```

```{r echo=FALSE}
table <- SJCounts[SJCounts$GROUP == "WT", c("sample_id", "Normalized_SE20", "Difference")]
table <- table[order(table$Difference, decreasing = TRUE),]
```

```{r echo=FALSE}
datatable(table,
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp', 
              scrollX = TRUE,  
              scrollY = TRUE
))
```
Difference in the MUT sample: deviation of the Normalized expression of the MUT patient from the mean normalized WT expression
```{r}
SJCounts$Difference[SJCounts$GROUP == "MUT"]
```

```{r echo=FALSE}
ggplot(SJCounts, aes(sample_id,Difference)) + geom_point(aes(colour=factor(GROUP))) +
  ylab("Differential Expression") +
  labs(color = "GROUP",
       title = "Differential Expression Distribution")
```


```{r echo=FALSE}
ggplot(SJCounts, aes(y=Difference, x=factor(GROUP, level=c("WT", "MUT")))) + 
  geom_boxplot() +
  ylab("Differential Expression") +
  xlab("GROUP")
```

#### ECDF - Pvalue Inference

```{r}
v_ecdf <- ecdf(SJCounts$Difference[SJCounts$GROUP == "WT"])
v_ecdf
```

```{r}
plot(ecdf(SJCounts$Difference[SJCounts$GROUP == "WT"]))
```

```{r}
v_ecdf(SJCounts$Difference[SJCounts$GROUP == "MUT"])
print(paste0("MUT Percentile: ", v_ecdf(SJCounts$Difference[SJCounts$GROUP == "MUT"])))
```

```{r}
1-v_ecdf(SJCounts$Difference[SJCounts$GROUP == "MUT"])
print(paste0("Inferred Pvalue: ", 1-v_ecdf(SJCounts$Difference[SJCounts$GROUP == "MUT"])))
```


```{r echo=FALSE}
MUT_df <- SJCounts[SJCounts$GROUP == "MUT",c("sample_id","case_id", "Normalized_SE20")]
colnames(MUT_df) <- c("sample_id", "case_id", "NormalizedExpression")

MUT_df$ECDF <- v_ecdf(SJCounts$Difference[SJCounts$GROUP == "MUT"])
MUT_df$Pvalue <- 1- MUT_df$ECDF

MUT_df$Prediction <- "Exon Skipping"
MUT_df$splice_junction_status <- "AlternativeSJ found in MUT samples"
MUT_df$splice_junction_position <- "chr13:28015702-28023349"

MUT_df$ECDF <- ifelse(MUT_df$NormalizedExpression == 0, NA,MUT_df$ECDF)
MUT_df$Pvalue <- ifelse(MUT_df$NormalizedExpression == 0, NA,MUT_df$Pvalue)
```


```{r echo=FALSE}
vaf_select <- annot_vaf[,c("Gene","MutationKey_Hg38","sample_id","RNA_Sample","DNA_Sample","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf", "VAF_RNA")]
MUT_vaf <- merge(MUT_df, vaf_select, by.x="sample_id", by.y = "RNA_Sample")
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id"] <- "RNA_Sample"
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id.y"] <- "sample_id"

col_order <- c("Gene", "MutationKey_Hg38", "Prediction","splice_junction_status","splice_junction_position","sample_id", "case_id","RNA_Sample","DNA_Sample","NormalizedExpression","ECDF", "Pvalue", "VAF_RNA","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf")
MUT_vaf <- MUT_vaf[, col_order]
```

Download the vaf, inferred percentiles and pvalues of the mutated samples:
```{r echo=FALSE}
datatable(MUT_vaf,
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp',
              scrollX=TRUE
))
```

```{r echo=FALSE}
summary_res <- rbind(summary_res, MUT_vaf)
```

### Exon Skipping 19 {.tabset}
#### Expression Difference 
Normality Test:
```{r}
shapiro.test(SJCounts$Normalized_SE19[SJCounts$GROUP == "WT"])
```

Value of Mean Normalized Expression of the SJ in WT samples:
```{r}
mean_WT_SJi <- mean(SJCounts$Normalized_SE19[SJCounts$GROUP == "WT"], na.rm=TRUE)
mean_WT_SJi
```

Normalized Expression Value of the SJ in the MUT sample:
```{r}
MUT_SJi <- SJCounts$Normalized_SE19[SJCounts$GROUP == "MUT"]
MUT_SJi
```



Deviation from the mean normalized expression:
```{r}
SJCounts$Difference <-  SJCounts$Normalized_SE19 - mean_WT_SJi
```

```{r echo=FALSE}
table <- SJCounts[SJCounts$GROUP == "WT", c("sample_id", "Normalized_SE19", "Difference")]
table <- table[order(table$Difference, decreasing = TRUE),]
```

```{r echo=FALSE}
datatable(table,
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp'
))
```
Difference in the MUT sample: deviation of the Normalized expression of the MUT patient from the mean normalized WT expression
```{r}
SJCounts$Difference[SJCounts$GROUP == "MUT"]
```

```{r echo=FALSE}
ggplot(SJCounts, aes(sample_id,Difference)) + geom_point(aes(colour=factor(GROUP))) +
  ylab("Differential Expression") +
  labs(color = "GROUP",
       title = "Differential Expression Distribution")
```


```{r echo=FALSE}
ggplot(SJCounts, aes(y=Difference, x=factor(GROUP, level=c("WT", "MUT")))) + 
  geom_boxplot() +
  ylab("Differential Expression") +
  xlab("GROUP")
```

#### ECDF - Pvalue Inference {.tabset}

```{r}
v_ecdf <- ecdf(SJCounts$Difference[SJCounts$GROUP == "WT"])
v_ecdf
```

```{r}
plot(ecdf(SJCounts$Difference[SJCounts$GROUP == "WT"]))
```

```{r}
v_ecdf(SJCounts$Difference[SJCounts$GROUP == "MUT"])
print(paste0("MUT Percentile: ", v_ecdf(SJCounts$Difference[SJCounts$GROUP == "MUT"])))
```

```{r}
1-v_ecdf(SJCounts$Difference[SJCounts$GROUP == "MUT"])
print(paste0("Inferred Pvalue: ", 1-v_ecdf(SJCounts$Difference[SJCounts$GROUP == "MUT"])))
```

```{r}
v_ecdf(SJCounts$Difference[SJCounts$GROUP == "MUT"])
```

```{r}
MUT_df <- SJCounts[SJCounts$GROUP == "MUT",c("sample_id","case_id", "Normalized_SE19")]
colnames(MUT_df) <- c("sample_id", "case_id", "NormalizedExpression")

MUT_df$ECDF <- v_ecdf(SJCounts$Difference[SJCounts$GROUP == "MUT"])
MUT_df$Pvalue <- 1- MUT_df$ECDF

MUT_df$Prediction <- "Exon Skipping"
MUT_df$splice_junction_status <- "AlternativeSJ found in MUT samples"
MUT_df$splice_junction_position <- "chr13:28018590-28024860"

MUT_df$ECDF <- ifelse(MUT_df$NormalizedExpression == 0, NA,MUT_df$ECDF)
MUT_df$Pvalue <- ifelse(MUT_df$NormalizedExpression == 0, NA,MUT_df$Pvalue)
```


```{r echo=FALSE}
vaf_select <- annot_vaf[,c("Gene","MutationKey_Hg38","sample_id","RNA_Sample","DNA_Sample","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf", "VAF_RNA")]
MUT_vaf <- merge(MUT_df, vaf_select, by.x="sample_id", by.y = "RNA_Sample")
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id"] <- "RNA_Sample"
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id.y"] <- "sample_id"

col_order <- c("Gene", "MutationKey_Hg38", "Prediction","splice_junction_status","splice_junction_position","sample_id", "case_id","RNA_Sample","DNA_Sample","NormalizedExpression","ECDF", "Pvalue", "VAF_RNA","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf")
MUT_vaf <- MUT_vaf[, col_order]
```

Download the vaf, inferred percentiles and pvalues of the mutated samples:
```{r echo=FALSE}
datatable(MUT_vaf,
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp',
              scrollX=TRUE
))
```

```{r echo=FALSE}
summary_res <- rbind(summary_res, MUT_vaf)
```

### Exon Skipping 19 + 20 {.tabset}
#### Expression Difference 
Normality Test:
```{r}
shapiro.test(SJCounts$Normalized_SE1920[SJCounts$GROUP == "WT"])
```


Value of Mean Normalized Expression of the SJ in WT samples:
```{r}
mean_WT_SJi <- mean(SJCounts$Normalized_SE1920[SJCounts$GROUP == "WT"], na.rm=TRUE)
mean_WT_SJi
```

Value of Mean Normalized Expression of the SJ in WT samples:
```{r}
MUT_SJi <- SJCounts$Normalized_SE1920[SJCounts$GROUP == "MUT"]
MUT_SJi
```



Deviation from the mean normalized expression:

```{r}
SJCounts$Difference <-  SJCounts$Normalized_SE1920 - mean_WT_SJi
```

```{r echo=FALSE}
table <- SJCounts[SJCounts$GROUP == "WT", c("sample_id", "Normalized_SE1920", "Difference")]
table <- table[order(table$Difference, decreasing = TRUE),]
```

```{r echo=FALSE}
datatable(table,
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp'
))
```
Difference in the MUT sample: deviation of the Normalized expression of the MUT patient from the mean normalized WT expression
```{r}
SJCounts$Difference[SJCounts$GROUP == "MUT"]
```

```{r echo=FALSE}
ggplot(SJCounts, aes(sample_id,Difference)) + geom_point(aes(colour=factor(GROUP))) +
  ylab("Differential Expression") +
  labs(color = "GROUP",
       title = "Differential Expression Distribution")
```

```{r echo=FALSE}
ggplot(SJCounts, aes(y=Difference, x=factor(GROUP, level=c("WT", "MUT")))) + 
  geom_boxplot() +
  ylab("Differential Expression") +
  xlab("GROUP")
```

#### ECDF - Pvalue Inference {.tabset}

```{r}
v_ecdf <- ecdf(SJCounts$Difference[SJCounts$GROUP == "WT"])
v_ecdf
```

```{r}
plot(ecdf(SJCounts$Difference[SJCounts$GROUP == "WT"]))
```

```{r}
v_ecdf(SJCounts$Difference[SJCounts$GROUP == "MUT"])
print(paste0("MUT Percentile: ", v_ecdf(SJCounts$Difference[SJCounts$GROUP == "MUT"])))
```

```{r}
1-v_ecdf(SJCounts$Difference[SJCounts$GROUP == "MUT"])
print(paste0("Inferred Pvalue: ", 1-v_ecdf(SJCounts$Difference[SJCounts$GROUP == "MUT"])))
```

```{r}
v_ecdf(SJCounts$Difference[SJCounts$GROUP == "MUT"])
```

```{r}
MUT_df <- SJCounts[SJCounts$GROUP == "MUT",c("sample_id","case_id", "Normalized_SE1920")]
colnames(MUT_df) <- c("sample_id", "case_id", "NormalizedExpression")

MUT_df$ECDF <- v_ecdf(SJCounts$Difference[SJCounts$GROUP == "MUT"])
MUT_df$Pvalue <- 1- MUT_df$ECDF

MUT_df$Prediction <- "Exon Skipping"
MUT_df$splice_junction_status <- "AlternativeSJ found in MUT samples"
MUT_df$splice_junction_position <- "chr13:28015702-28024860"

MUT_df$ECDF <- ifelse(MUT_df$NormalizedExpression == 0, NA,MUT_df$ECDF)
MUT_df$Pvalue <- ifelse(MUT_df$NormalizedExpression == 0, NA,MUT_df$Pvalue)

```

```{r echo=FALSE}
vaf_select <- annot_vaf[,c("Gene","MutationKey_Hg38","sample_id","RNA_Sample","DNA_Sample","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf", "VAF_RNA")]
MUT_vaf <- merge(MUT_df, vaf_select, by.x="sample_id", by.y = "RNA_Sample")
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id"] <- "RNA_Sample"
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id.y"] <- "sample_id"

col_order <- c("Gene", "MutationKey_Hg38", "Prediction","splice_junction_status","splice_junction_position","sample_id", "case_id","RNA_Sample","DNA_Sample","NormalizedExpression","ECDF", "Pvalue", "VAF_RNA","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf")
MUT_vaf <- MUT_vaf[, col_order]
```

Download the vaf, inferred percentiles and pvalues of the mutated samples:
```{r echo=FALSE}
datatable(MUT_vaf,
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp',
              scrollX=TRUE
)
)
```

```{r echo=FALSE}
summary_res <- rbind(summary_res, MUT_vaf)
```


```{r echo=FALSE}
rm(GeneSJ, normalization_df, GeneSJ_subset, annot_vaf, plot_df, samples_df,table, plot_boxplot, plot_dotplot, vaf_select, SJCounts, MUT_df, MUT_vaf)
```


# IDH1 {.tabset}
Three (3) variants: chr2,208248389,G,A, chr2,208248389,G,T & chr2,208248389,G,C

Variants found in **30 patients** of the BeatAML (**35 samples**)

* Patients with IDH1 chr2,208248389,G,A variant: **23 patients (27 samples)**
* Patients with IDH1 chr2,208248389,G,T variant: **5 patients (5 samples)**
* Patients with IDH1 chr2,208248389,G,C variant: **2 patients (3 samples)**

Patients with the variant and RNASeq for validation: 

* Patients with IDH1 chr2,208248389,G,A variant and RNASeq for validation: **14 patients (15 samples)**
* Patients with IDH1 chr2,208248389,G,T variant and RNASeq for validation: **4 patients (4 samples)**
* Patients with IDH1 chr2,208248389,G,C variant and RNASeq for validation: **2 patients (3 samples)**


The splicing alterations being assessed are: 

* **Donor Gain**: chr2:208245425-208248422, found in the mutated samples.
* **Exon Skipping 4**: chr2:208245425-208251429, found in the splice junction collection, not found in the mutated samples.
* **Donor Loss**: chr2:208245425-208248368; donor splice site chr2:208248368.

Variant information:
```{r echo=FALSE}
datatable(to_be_validated[to_be_validated$MutationKey_Hg38 %in% c( "chr2,208248389,G,A","chr2,208248389,G,T" ,"chr2,208248389,G,C"),],
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp', 
              scrollX = TRUE,  
              scrollY = TRUE,
              columnDefs = list(list(targets = '_all', className = 'dt-center')))
)
```

Load the extracted splice junctions of the gene harboring the mutation.
```{r}
extractedSJ_path <- paste0(extractedSJ_dir_in,"IDH1_UM_annotSJ.tsv")
GeneSJ <- read.delim(extractedSJ_path, sep ="\t")
```

Set the sample's group: Mutated (MUT) or No Mutated (WT)
```{r}
samples_df <- found_variants[found_variants$Gene=="IDH1" & found_variants$MutationKey_Hg38 %in% c( "chr2,208248389,G,A","chr2,208248389,G,T" ,"chr2,208248389,G,C"),]
```

```{r}
R132C <- samples_df$RNA_Sample[samples_df$MutationKey_Hg38 == "chr2,208248389,G,A" & samples_df$Validable =="Validable"] #n=15 G>A
R132G <- samples_df$RNA_Sample[samples_df$MutationKey_Hg38 == "chr2,208248389,G,C" & samples_df$Validable =="Validable"] #n=3 G>C
R132S  <- samples_df$RNA_Sample[samples_df$MutationKey_Hg38 == "chr2,208248389,G,T" & samples_df$Validable =="Validable"] #n=4 G>T

cases <- append(R132C, R132G)
cases <- append(cases, R132S)

GeneSJ$GROUP <- ifelse(GeneSJ$sample_id %in% R132C, "MUT", 
                                ifelse(GeneSJ$sample_id %in% R132G, "MUT",
                                      ifelse(GeneSJ$sample_id %in% R132S, "MUT", 
                                                     "WT")))


GeneSJ$Variant_status <- ifelse(GeneSJ$sample_id %in% R132C, "R132C", 
                                ifelse(GeneSJ$sample_id %in% R132G, "R132G",
                                              ifelse(GeneSJ$sample_id %in% R132S, "R132S", 
                                                     "WT")))
```


## SJ Lookup{.tabset}

Search for the splice junctions of interest in the extracted splice junctions of the gene by position (chr_SJstart_SJend).

### Donor Gain
Search: chr2:208245425-208248422


Show all the splice junctions containing the position 208245425-208248422
```{r}
colnames(GeneSJ)[grep("208245425_208248422",colnames(GeneSJ))]
```

Found: chr2_208245425_208248422

Reads of all the AML samples (mutated and no mutated) for the splice junction:
```{r}
GeneSJ$chr2_208245425_208248422
```
Samples with the SJ of interest:
```{r}
table(GeneSJ$chr2_208245425_208248422>0) 
```
Groups of the samples having the alternative splice junction:
```{r}
table(GeneSJ$GROUP[GeneSJ$chr2_208245425_208248422 > 0])
```
Alternative SJ found in the mutated samples.

### Exon Skipping 4
Search: chr2:208245425-208251429

Show all the splice junctions containing the positions 208245425-208251429
```{r}
colnames(GeneSJ)[grep("208245425_208251429",colnames(GeneSJ))]
```

Found: chr2_208245425_208251429

Reads of all the AML samples (mutated and no mutated) for the splice junction:
```{r}
GeneSJ$chr2_208245425_208251429
```

Samples with the SJ of interest:
```{r}
table(GeneSJ$chr2_208245425_208251429>0) 
```

Groups of the samples having the alternative splice junction:
```{r}
table(GeneSJ$GROUP[GeneSJ$chr2_208245425_208251429 > 0])
```
Alternative SJ not found in the mutated samples of the splice junction collection.

### Canonical SJ
Exon 3-4: chr2:208248661-208251429

Exon 4-5: chr2:208245425-208248368; splice site chr2:208248368


Reads of all the AML samples (mutated and no mutated) for the splice junction:
```{r}
GeneSJ$chr2_208248661_208251429
```

Reads of all the AML samples (mutated and no mutated) for the splice junction:
```{r}
GeneSJ$chr2_208245425_208248368
```


## Normalization
Count the reads of all the splice junctions of the gene harboring the variant:
```{r}
GeneSJ$rowSum_SJtotal <- rowSums(GeneSJ[,grep("chr", names(GeneSJ))])
```

Normalization of the expression by the total read counts of all the splice junctions of the gene:
```{r}
GeneSJ$Normalized_CanonEx3_4 <- (GeneSJ$chr2_208248661_208251429)/GeneSJ$rowSum_SJtotal*100
GeneSJ$Normalized_CanonEx4_5 <- (GeneSJ$chr2_208245425_208248368)/GeneSJ$rowSum_SJtotal*100

GeneSJ$Normalized_DGEx4 <- (GeneSJ$chr2_208245425_208248422)/GeneSJ$rowSum_SJtotal*100
```

Download the normalized values for the assessed splice junctions of all the AML samples:
```{r echo=FALSE}
normalization_df <- GeneSJ[,c("sample_id","case_id", "GROUP","Variant_status","Normalized_CanonEx3_4", "Normalized_CanonEx4_5", "Normalized_DGEx4","chr2_208248661_208251429", "chr2_208245425_208248368", "chr2_208245425_208248422", "rowSum_SJtotal","file_id.BAM","file_name.BAM","file_id.STAR.SJCounts","file_name.STAR.SJCounts" )]

colnames(normalization_df)[grep("chr2_208248661_208251429", names(normalization_df))] <- "chr2_208248661_208251429_CanonEx3_4"
colnames(normalization_df)[grep("chr2_208245425_208248368", names(normalization_df))] <- "chr2_208245425_208248368_CanonEx4_5"
colnames(normalization_df)[grep("chr2_208245425_208248422", names(normalization_df))] <- "chr2_208245425_208248422_DGEx4"

write.xlsx(normalization_df, paste0(normalizedEx_dir_out, "BeatAML.IDH1.xlsx"), rowNames = FALSE)
```

```{r echo=FALSE}
datatable(normalization_df,
extensions = 'Buttons',
rownames = FALSE, 
filter = 'top',
options = list(paging = TRUE,
              pageLength = 4,
              autoWidth = FALSE,
              dom = 'lifrtBp',
              buttons = c('excel'), 
              scrollX = TRUE,  
              scrollY = TRUE,
              columnDefs = list(list(targets = '_all', className = 'dt-center'))
))
```

## VAF
```{r echo=FALSE}
GeneSJ_subset <- GeneSJ[GeneSJ$sample_id %in% cases,]
GeneSJ_subset$Gene <- "IDH1"

annot_vaf <- merge(found_variants[,c("MutationKey_Hg38","Gene","sample_id","case_id",                                               "RNA_Sample","DNA_Sample","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf",                         "MuTect2_filter","VarScan2_filter","SomaticSniper_filter","MuSE_filter","MUTReads","WTReads","VAF")], 
                   GeneSJ_subset[,c("Gene","sample_id", "case_id","Normalized_CanonEx3_4", "Normalized_CanonEx4_5", "Normalized_DGEx4")], 
                   by.x=c("RNA_Sample", "case_id","Gene"), 
                   by.y=c("sample_id", "case_id","Gene"))


annot_vaf <- annot_vaf %>% select(Gene,MutationKey_Hg38,
                                  sample_id,
                                  case_id, RNA_Sample,DNA_Sample,
                                  MuTect2_vaf,VarScan2_vaf,SomaticSniper_vaf,MuSE_vaf,
                                  VAF,
                                  Normalized_CanonEx3_4, Normalized_CanonEx4_5, Normalized_DGEx4,
                                  everything()
                                  )
colnames(annot_vaf)[grep("VAF", names(annot_vaf))] <- "VAF_RNA"
colnames(annot_vaf)[grep("MUTReads", names(annot_vaf))] <- "MUTReads_RNA"
colnames(annot_vaf)[grep("WTReads", names(annot_vaf))] <- "WTReads_RNA"

write.xlsx(annot_vaf, paste0(normalizedEx_dir_out, "BeatAML.IDH1.MUT.xlsx"), rowNames = FALSE)
```

Mutated samples vaf:
```{r echo=FALSE}
datatable(annot_vaf,
extensions = 'Buttons',
rownames = FALSE, 
filter = 'top',
options = list(paging = TRUE,
              pageLength = 4,
              autoWidth = TRUE,
              dom = 'lifrtBp',
              buttons = c('excel'), 
              scrollX = TRUE,  
              scrollY = TRUE
              )
)
```

## Plots {.tabset}

```{r echo=FALSE}
GeneSJ$GROUP <- ifelse(GeneSJ$GROUP == "MUT", "MUT", "WT")
GeneSJ$GROUP <- as.factor(GeneSJ$GROUP)
```

### Static Dot Plots 
Canonical splice junction: Exon 4-5: chr2:208245425-208248368; donor splice site chr2:208248368

```{r echo=FALSE}
ggplot(GeneSJ, aes(sample_id,Normalized_CanonEx4_5,color=GROUP)) + 
  geom_point(size=.8) +
  labs(color='GROUP', 
       title = "Normalized expression of IDH1 Exon 4-5 Splice Junction", 
       subtitle="Potential Donor Loss Effect",
       y = "Normalized Expression")
```


Splicing alterations:

```{r echo=FALSE}
ggplot(GeneSJ, aes(sample_id,Normalized_DGEx4,color=GROUP)) + 
  geom_point(size=.8) +
  labs(color='GROUP', 
       title = "Normalized expression of IDH1 Exon 4 Donor Gain Splice Junction", 
       subtitle="Potential Exon 4 Donor Gain",
       y = "Normalized Expression")
```


```{r echo=FALSE}
ggplot(GeneSJ, aes(sample_id,Normalized_DGEx4,color=Variant_status)) + 
  geom_point(size=.8) +
  labs(color='variant_status', 
       title = "Normalized expression of IDH1 Exon 4 Donor Gain Splice Junction", 
       subtitle="Potential Exon 4 Donor Gain",
       y = "Normalized Expression",
       color="Variant status")
```


### Interactive Dot Plots

```{r echo=FALSE}
plot_df <- merge(GeneSJ,annot_vaf[c("RNA_Sample", "VAF_RNA", 
                                    "MuTect2_vaf", "VarScan2_vaf", "SomaticSniper_vaf", "MuSE_vaf")],
                 by.x="sample_id", by.y="RNA_Sample", all.x = TRUE)
```

Canonical splice junction: Exon 4-5: chr2:208245425-208248368; donor splice site chr2:208248368

```{r echo=FALSE}
ggplotly(ggplot(plot_df, aes(sample_id,Normalized_CanonEx4_5,color=GROUP)) + 
  geom_point(aes(text=paste0("Variant: ",Variant_status ,"\nVAF RNA: ",VAF_RNA,"\nVAF MuTect: ",MuTect2_vaf,
                             "\nVAF VarScan2: ",VarScan2_vaf,
                             "\nVAF SomaticSniper: ",SomaticSniper_vaf,
                             "\nVAF MuSE: ",MuSE_vaf)),size= 0.8) +
  ggtitle("Normalized expression of IDH1 Exon 4-5 Splice Junction") +
  labs(color='GROUP')
)
```

Splicing alteration:

```{r echo=FALSE}
ggplotly(ggplot(plot_df, aes(sample_id,Normalized_DGEx4,color=GROUP)) + 
  geom_point(aes(text=paste0("Variant: ",Variant_status ,"\nVAF RNA: ",VAF_RNA,"\nVAF MuTect: ",MuTect2_vaf,
                             "\nVAF VarScan2: ",VarScan2_vaf,
                             "\nVAF SomaticSniper: ",SomaticSniper_vaf,
                             "\nVAF MuSE: ",MuSE_vaf)),size= 0.8) +
  ggtitle("Normalized expression of IDH1 Exon 4 Donor Gain Splice Junction") +
  labs(color='GROUP')
)

```


```{r echo=FALSE}
ggplotly(ggplot(plot_df, aes(sample_id,Normalized_DGEx4,color=Variant_status)) + 
  geom_point(aes(text=paste0("VAF RNA: ",VAF_RNA,"\nVAF MuTect: ",MuTect2_vaf,
                             "\nVAF VarScan2: ",VarScan2_vaf,
                             "\nVAF SomaticSniper: ",SomaticSniper_vaf,
                             "\nVAF MuSE: ",MuSE_vaf)),size= 0.8) +
  ggtitle("Normalized expression of IDH1 Exon 4 Donor Gain Splice Junction") +
  labs(color='Variant status') 
)

```


### Violin Plots
Violin Plots for the alternative splice junctions interrogated:

```{r echo=FALSE}
plot_boxplot <- plot_df[plot_df$GROUP %in% c("WT"),]
plot_dotplot <- plot_df[plot_df$GROUP %in% c("MUT"),]

GEN = "IDH1"
SJ = "chr2:208245425-208248422"

ggplot(plot_df, aes(y=Normalized_DGEx4, x=factor(GROUP, level=c("WT", "MUT")))) + 
       geom_violin(scale="area") +
       geom_point(data = plot_dotplot, 
                  aes(y=Normalized_DGEx4, x=GROUP),
                  color="black",
                  size=2
                  ) +
        
       ylab("Normalized Expression of potential Donor Gain") +
       xlab(paste(GEN, SJ))
```


## Statistical Analysis {.tabset}

```{r}
SJCounts <- GeneSJ
```
 
### Donor Gain {.tabset}
#### Expression Difference 
Normality Test:
```{r}
shapiro.test(SJCounts$Normalized_DGEx4[SJCounts$GROUP == "WT"])
```

Value of Mean Normalized Expression of the Alternative SJ in WT samples:
```{r}
mean_WT_SJi <- mean(SJCounts$Normalized_DGEx4[SJCounts$GROUP == "WT"], na.rm=TRUE)
mean_WT_SJi
```

Normalized Expression Value of the Alternative SJ in the MUT sample:
```{r}
MUT_SJi <- SJCounts$Normalized_DGEx4[SJCounts$GROUP == "MUT"]
MUT_SJi
```



Deviation from the mean normalized expression:
```{r}
SJCounts$Difference <-  SJCounts$Normalized_DGEx4 - mean_WT_SJi
```

```{r echo=FALSE}
table <- SJCounts[SJCounts$GROUP == "WT", c("sample_id", "Normalized_DGEx4", "Difference")]
table <- table[order(table$Difference, decreasing = TRUE),]
```

```{r echo=FALSE}
datatable(table,
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp'
))
```

Difference in the MUT sample: deviation of the Normalized expression of the MUT patient from the mean normalized WT expression
```{r}
SJCounts[SJCounts$GROUP == "MUT", c("sample_id", "Difference")]
```

```{r echo=FALSE}
ggplot(SJCounts, aes(sample_id,Difference)) + geom_point(aes(colour=factor(GROUP))) +
  ylab("Differential Expression") +
  labs(color = "GROUP",
       title = "Differential Expression Distribution")
```


```{r echo=FALSE}
ggplot(SJCounts, aes(y=Difference, x=factor(GROUP, level=c("WT", "MUT")))) + 
  geom_boxplot() +
  ylab("Differential Expression") +
  xlab("GROUP")
```

#### ECDF

```{r}
v_ecdf <- ecdf(SJCounts$Difference[SJCounts$GROUP == "WT"])
v_ecdf
```

```{r}
plot(ecdf(SJCounts$Difference[SJCounts$GROUP == "WT"]))
```

```{r}
v_ecdf(SJCounts$Difference[SJCounts$GROUP == "MUT"])
```

```{r}
MUT_df <- SJCounts[SJCounts$GROUP == "MUT",c("sample_id","case_id", "Normalized_DGEx4")]
colnames(MUT_df) <- c("sample_id", "case_id", "NormalizedExpression")

MUT_df$ECDF <- v_ecdf(SJCounts$Difference[SJCounts$GROUP == "MUT"])
MUT_df$Pvalue <- NA

MUT_df$Prediction <- "Donor Gain"
MUT_df$splice_junction_status <- "AlternativeSJ found in MUT samples"
MUT_df$splice_junction_position <- "chr2:208245425-208248422"

MUT_df$ECDF <- ifelse(MUT_df$NormalizedExpression == 0, NA,MUT_df$ECDF)
```

```{r echo=FALSE}
vaf_select <- annot_vaf[,c("Gene","MutationKey_Hg38","sample_id","RNA_Sample","DNA_Sample","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf", "VAF_RNA")]
MUT_vaf <- merge(MUT_df, vaf_select, by.x="sample_id", by.y = "RNA_Sample")
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id"] <- "RNA_Sample"
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id.y"] <- "sample_id"

col_order <- c("Gene", "MutationKey_Hg38", "Prediction","splice_junction_status","splice_junction_position","sample_id", "case_id","RNA_Sample","DNA_Sample","NormalizedExpression","ECDF", "Pvalue", "VAF_RNA","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf")
MUT_vaf <- MUT_vaf[, col_order]

```

Download the vaf, inferred percentiles and pvalues of the mutated samples:
```{r echo=FALSE}
datatable(MUT_vaf,
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp',
              scrollX=TRUE
))
```

#### Kruskal Wallis 

Kruskal Wallis:
```{r}
kruskal.test(Normalized_DGEx4 ~ Variant_status, data = SJCounts)
```
Pairwise comparisons:
```{r}
pairwise.wilcox.test(SJCounts$Normalized_DGEx4, SJCounts$Variant_status)
```

Detailed:
```{r}
pkw <- pairwise_wilcox_test(SJCounts,Normalized_DGEx4 ~ Variant_status)
pkw
```


```{r echo=FALSE}
MUT_vaf_2 <- MUT_vaf[MUT_vaf$MutationKey_Hg38 == "chr2,208248389,G,A",]

stat_df <- data.frame(matrix(ncol = 17, nrow = 1))
colnames(stat_df) <- c("Gene","MutationKey_Hg38","Prediction","splice_junction_status","splice_junction_position", "sample_id","case_id","RNA_Sample","DNA_Sample","NormalizedExpression","ECDF","Pvalue","VAF_RNA","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf")

stat_df$Gene <- unique(MUT_vaf_2$Gene)
stat_df$MutationKey_Hg38 <- unique(MUT_vaf_2$MutationKey_Hg38)
stat_df$sample_id <- paste(MUT_vaf_2$sample_id, collapse = ';\n') 
stat_df$case_id <- paste(MUT_vaf_2$case_id, collapse = ';\n') 
stat_df$RNA_Sample <- paste(MUT_vaf_2$RNA_Sample, collapse = ';\n') 
stat_df$DNA_Sample <- paste(MUT_vaf_2$DNA_Sample, collapse = ';\n')
stat_df$NormalizedExpression <- paste0(MUT_vaf_2$NormalizedExpression, collapse= ";\n")
stat_df$ECDF <- paste0(MUT_vaf_2$ECDF, collapse = ";\n")
stat_df$VAF_RNA <-  paste(MUT_vaf_2$VAF_RNA, collapse = ';\n')
stat_df$MuTect2_vaf <-  paste(MUT_vaf_2$MuTect2_vaf, collapse = ';\n')
stat_df$VarScan2_vaf <-  paste(MUT_vaf_2$VarScan2_vaf, collapse = ';\n')
stat_df$SomaticSniper_vaf <-  paste(MUT_vaf_2$SomaticSniper_vaf, collapse = ';\n')
stat_df$MuSE_vaf <-  paste(MUT_vaf_2$MuSE_vaf, collapse = ';\n')
stat_df$Prediction <- "Donor Gain"
stat_df$splice_junction_status <- "AlternativeSJ found in MUT samples"
stat_df$splice_junction_position <- "chr2:208245425-208248422"

stat_df$Pvalue <- pkw$p.adj[3]
```

```{r echo=FALSE}
summary_res<- rbind(summary_res,stat_df)
```


```{r echo=FALSE}
MUT_vaf_2 <- MUT_vaf[MUT_vaf$MutationKey_Hg38 == "chr2,208248389,G,C",]

stat_df <- data.frame(matrix(ncol = 17, nrow = 1))
colnames(stat_df) <- c("Gene","MutationKey_Hg38","Prediction","splice_junction_status","splice_junction_position", "sample_id","case_id","RNA_Sample","DNA_Sample","NormalizedExpression","ECDF","Pvalue","VAF_RNA","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf")
stat_df$Gene <- unique(MUT_vaf_2$Gene)
stat_df$MutationKey_Hg38 <- unique(MUT_vaf_2$MutationKey_Hg38)
stat_df$sample_id <- paste(MUT_vaf_2$sample_id, collapse = ';\n') 
stat_df$case_id <- paste(MUT_vaf_2$case_id, collapse = ';\n') 
stat_df$RNA_Sample <- paste(MUT_vaf_2$RNA_Sample, collapse = ';\n') 
stat_df$DNA_Sample <- paste(MUT_vaf_2$DNA_Sample, collapse = ';\n')
stat_df$NormalizedExpression <- paste0(MUT_vaf_2$NormalizedExpression, collapse= ";\n")
stat_df$ECDF <- paste0(MUT_vaf_2$ECDF, collapse = ";\n")
stat_df$VAF_RNA <-  paste(MUT_vaf_2$VAF_RNA, collapse = ';\n')
stat_df$MuTect2_vaf <-  paste(MUT_vaf_2$MuTect2_vaf, collapse = ';\n')
stat_df$VarScan2_vaf <-  paste(MUT_vaf_2$VarScan2_vaf, collapse = ';\n')
stat_df$SomaticSniper_vaf <-  paste(MUT_vaf_2$SomaticSniper_vaf, collapse = ';\n')
stat_df$MuSE_vaf <-  paste(MUT_vaf_2$MuSE_vaf, collapse = ';\n')
stat_df$Prediction <- "Donor Gain"
stat_df$splice_junction_status <- "AlternativeSJ found in MUT samples"
stat_df$splice_junction_position <- "chr2:208245425-208248422"

stat_df$Pvalue <- pkw$p.adj[5]
```

```{r echo=FALSE}
summary_res<- rbind(summary_res,stat_df)
```


```{r echo=FALSE}
MUT_vaf_2 <- MUT_vaf[MUT_vaf$MutationKey_Hg38 == "chr2,208248389,G,T",]

stat_df <- data.frame(matrix(ncol = 17, nrow = 1))
colnames(stat_df) <- c("Gene","MutationKey_Hg38","Prediction","splice_junction_status","splice_junction_position", "sample_id","case_id","RNA_Sample","DNA_Sample","NormalizedExpression","ECDF","Pvalue","VAF_RNA","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf")
stat_df$Gene <- unique(MUT_vaf_2$Gene)
stat_df$MutationKey_Hg38 <- unique(MUT_vaf_2$MutationKey_Hg38)
stat_df$sample_id <- paste(MUT_vaf_2$sample_id, collapse = ';\n') 
stat_df$case_id <- paste(MUT_vaf_2$case_id, collapse = ';\n') 
stat_df$RNA_Sample <- paste(MUT_vaf_2$RNA_Sample, collapse = ';\n') 
stat_df$DNA_Sample <- paste(MUT_vaf_2$DNA_Sample, collapse = ';\n')
stat_df$NormalizedExpression <- paste0(MUT_vaf_2$NormalizedExpression, collapse= ";\n")
stat_df$ECDF <- paste0(MUT_vaf_2$ECDF, collapse = ";\n")
stat_df$VAF_RNA <-  paste(MUT_vaf_2$VAF_RNA, collapse = ';\n')
stat_df$MuTect2_vaf <-  paste(MUT_vaf_2$MuTect2_vaf, collapse = ';\n')
stat_df$VarScan2_vaf <-  paste(MUT_vaf_2$VarScan2_vaf, collapse = ';\n')
stat_df$SomaticSniper_vaf <-  paste(MUT_vaf_2$SomaticSniper_vaf, collapse = ';\n')
stat_df$MuSE_vaf <-  paste(MUT_vaf_2$MuSE_vaf, collapse = ';\n')
stat_df$Prediction <- "Donor Gain"
stat_df$splice_junction_status <- "AlternativeSJ found in MUT samples"
stat_df$splice_junction_position <- "chr2:208245425-208248422"

stat_df$Pvalue <- pkw$p.adj[6]
```

```{r echo=FALSE}
summary_res<- rbind(summary_res,stat_df)
```


### Donor Loss {.tabset}
#### Expression Difference 
Normality Test:
```{r}
shapiro.test(SJCounts$Normalized_CanonEx4_5[SJCounts$GROUP == "WT"])
```

Value of Mean Normalized Expression of the Alternative SJ in WT samples:
```{r}
mean_WT_SJi <- mean(SJCounts$Normalized_CanonEx4_5[SJCounts$GROUP == "WT"], na.rm=TRUE)
mean_WT_SJi
```

Normalized Expression Value of the Alternative SJ in the MUT sample:
```{r}
MUT_SJi <- SJCounts$Normalized_CanonEx4_5[SJCounts$GROUP == "MUT"]
MUT_SJi
```



Deviation from the mean normalized expression:
```{r}
SJCounts$Difference <-  SJCounts$Normalized_CanonEx4_5 - mean_WT_SJi
```

```{r echo=FALSE}
table <- SJCounts[SJCounts$GROUP == "WT", c("sample_id", "Normalized_CanonEx4_5", "Difference")]
table <- table[order(table$Difference, decreasing = TRUE),]
```

```{r echo=FALSE}
datatable(table,
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp'
))
```

Difference in the MUT sample: deviation of the Normalized expression of the MUT patient from the mean normalized WT expression
```{r}
SJCounts$Difference[SJCounts$GROUP == "MUT"]
```

```{r echo=FALSE}
ggplot(SJCounts, aes(sample_id,Difference)) + geom_point(aes(colour=factor(GROUP))) +
  ylab("Differential Expression") +
  labs(color = "GROUP",
       title = "Differential Expression Distribution")
```


```{r echo=FALSE}
ggplot(SJCounts, aes(y=Difference, x=factor(GROUP, level=c("WT", "MUT")))) + 
  geom_boxplot() +
  ylab("Differential Expression") +
  xlab("GROUP")
```

#### ECDF

```{r}
v_ecdf <- ecdf(SJCounts$Difference[SJCounts$GROUP == "WT"])
v_ecdf
```

```{r}
plot(ecdf(SJCounts$Difference[SJCounts$GROUP == "WT"]))
```

```{r}
v_ecdf(SJCounts$Difference[SJCounts$GROUP == "MUT"])
```

```{r}
MUT_df <- SJCounts[SJCounts$GROUP == "MUT",c("sample_id","case_id", "Normalized_CanonEx4_5")]
colnames(MUT_df) <- c("sample_id", "case_id", "NormalizedExpression")

MUT_df$ECDF <- v_ecdf(SJCounts$Difference[SJCounts$GROUP == "MUT"])
MUT_df$Pvalue <- NA
MUT_df$Prediction  <- "Donor Loss"
MUT_df$splice_junction_status <- "CanonicalSJ"
MUT_df$splice_junction_position <- "chr2:208245425-208248368"

MUT_df$ECDF <- ifelse(MUT_df$NormalizedExpression == 0, NA,MUT_df$ECDF)
```

```{r echo=FALSE}
vaf_select <- annot_vaf[,c("Gene","MutationKey_Hg38","sample_id","RNA_Sample","DNA_Sample","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf", "VAF_RNA")]
MUT_vaf <- merge(MUT_df, vaf_select, by.x="sample_id", by.y = "RNA_Sample")
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id"] <- "RNA_Sample"
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id.y"] <- "sample_id"

col_order <- c("Gene", "MutationKey_Hg38", "Prediction","splice_junction_status","splice_junction_position","sample_id", "case_id","RNA_Sample","DNA_Sample","NormalizedExpression","ECDF", "Pvalue", "VAF_RNA","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf")
MUT_vaf <- MUT_vaf[, col_order]
```

Download the vaf, inferred percentiles and pvalues of the mutated samples:
```{r echo=FALSE}
datatable(MUT_vaf,
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp',
              scrollX=TRUE
))
```




#### Kruskal Wallis

Kruskal Wallis:
```{r}
kruskal.test(Normalized_CanonEx4_5 ~ Variant_status, data = SJCounts)
```

Pairwise comparisons:
```{r}
pairwise.wilcox.test(SJCounts$Normalized_CanonEx4_5, SJCounts$Variant_status)
```

Detailed:
```{r}
pkw <- pairwise_wilcox_test(SJCounts,Normalized_CanonEx4_5 ~ Variant_status)
pkw
```


```{r echo=FALSE}
MUT_vaf_2 <- MUT_vaf[MUT_vaf$MutationKey_Hg38 == "chr2,208248389,G,A",]

stat_df <- data.frame(matrix(ncol = 17, nrow = 1))
colnames(stat_df) <- c("Gene","MutationKey_Hg38","Prediction","splice_junction_status","splice_junction_position", "sample_id","case_id","RNA_Sample","DNA_Sample","NormalizedExpression","ECDF","Pvalue","VAF_RNA","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf")
stat_df$Gene <- unique(MUT_vaf_2$Gene)
stat_df$MutationKey_Hg38 <- unique(MUT_vaf_2$MutationKey_Hg38)
stat_df$sample_id <- paste(MUT_vaf_2$sample_id, collapse = ';\n') 
stat_df$case_id <- paste(MUT_vaf_2$case_id, collapse = ';\n') 
stat_df$RNA_Sample <- paste(MUT_vaf_2$RNA_Sample, collapse = ';\n') 
stat_df$DNA_Sample <- paste(MUT_vaf_2$DNA_Sample, collapse = ';\n')
stat_df$NormalizedExpression <- paste0(MUT_vaf_2$NormalizedExpression, collapse= ";\n")
stat_df$ECDF <- paste0(MUT_vaf_2$ECDF, collapse = ";\n")
stat_df$VAF_RNA <-  paste(MUT_vaf_2$VAF_RNA, collapse = ';\n')
stat_df$MuTect2_vaf <-  paste(MUT_vaf_2$MuTect2_vaf, collapse = ';\n')
stat_df$VarScan2_vaf <-  paste(MUT_vaf_2$VarScan2_vaf, collapse = ';\n')
stat_df$SomaticSniper_vaf <-  paste(MUT_vaf_2$SomaticSniper_vaf, collapse = ';\n')
stat_df$MuSE_vaf <-  paste(MUT_vaf_2$MuSE_vaf, collapse = ';\n')
stat_df$Prediction  <- "Donor Loss"
stat_df$splice_junction_status <- "CanonicalSJ"
stat_df$splice_junction_position <- "chr2:208245425-208248368"

stat_df$Pvalue <-pkw$p.adj[3]
```

```{r echo=FALSE}
summary_res<- rbind(summary_res,stat_df)
```



```{r echo=FALSE}
MUT_vaf_2 <- MUT_vaf[MUT_vaf$MutationKey_Hg38 == "chr2,208248389,G,C",]

stat_df <- data.frame(matrix(ncol = 17, nrow = 1))
colnames(stat_df) <- c("Gene","MutationKey_Hg38","Prediction","splice_junction_status","splice_junction_position", "sample_id","case_id","RNA_Sample","DNA_Sample","NormalizedExpression","ECDF","Pvalue","VAF_RNA","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf")
stat_df$Gene <- unique(MUT_vaf_2$Gene)
stat_df$MutationKey_Hg38 <- unique(MUT_vaf_2$MutationKey_Hg38)
stat_df$sample_id <- paste(MUT_vaf_2$sample_id, collapse = ';\n') 
stat_df$case_id <- paste(MUT_vaf_2$case_id, collapse = ';\n') 
stat_df$RNA_Sample <- paste(MUT_vaf_2$RNA_Sample, collapse = ';\n') 
stat_df$DNA_Sample <- paste(MUT_vaf_2$DNA_Sample, collapse = ';\n')
stat_df$NormalizedExpression <- paste0(MUT_vaf_2$NormalizedExpression, collapse= ";\n")
stat_df$ECDF <- paste0(MUT_vaf_2$ECDF, collapse = ";\n")
stat_df$VAF_RNA <-  paste(MUT_vaf_2$VAF_RNA, collapse = ';\n')
stat_df$MuTect2_vaf <-  paste(MUT_vaf_2$MuTect2_vaf, collapse = ';\n')
stat_df$VarScan2_vaf <-  paste(MUT_vaf_2$VarScan2_vaf, collapse = ';\n')
stat_df$SomaticSniper_vaf <-  paste(MUT_vaf_2$SomaticSniper_vaf, collapse = ';\n')
stat_df$MuSE_vaf <-  paste(MUT_vaf_2$MuSE_vaf, collapse = ';\n')
stat_df$Prediction  <- "Donor Loss"
stat_df$splice_junction_status <- "CanonicalSJ"
stat_df$splice_junction_position <- "chr2:208245425-208248368"

stat_df$Pvalue <- pkw$p.adj[5]
```

```{r echo=FALSE}
summary_res<- rbind(summary_res,stat_df)
```


```{r echo=FALSE}
MUT_vaf_2 <- MUT_vaf[MUT_vaf$MutationKey_Hg38 == "chr2,208248389,G,T",]

stat_df <- data.frame(matrix(ncol = 17, nrow = 1))
colnames(stat_df) <- c("Gene","MutationKey_Hg38","Prediction","splice_junction_status","splice_junction_position", "sample_id","case_id","RNA_Sample","DNA_Sample","NormalizedExpression","ECDF","Pvalue","VAF_RNA","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf")
stat_df$Gene <- unique(MUT_vaf_2$Gene)
stat_df$MutationKey_Hg38 <- unique(MUT_vaf_2$MutationKey_Hg38)
stat_df$sample_id <- paste(MUT_vaf_2$sample_id, collapse = ';\n') 
stat_df$case_id <- paste(MUT_vaf_2$case_id, collapse = ';\n') 
stat_df$RNA_Sample <- paste(MUT_vaf_2$RNA_Sample, collapse = ';\n') 
stat_df$DNA_Sample <- paste(MUT_vaf_2$DNA_Sample, collapse = ';\n')
stat_df$NormalizedExpression <- paste0(MUT_vaf_2$NormalizedExpression, collapse= ";\n")
stat_df$ECDF <- paste0(MUT_vaf_2$ECDF, collapse = ";\n")
stat_df$VAF_RNA <-  paste(MUT_vaf_2$VAF_RNA, collapse = ';\n')
stat_df$MuTect2_vaf <-  paste(MUT_vaf_2$MuTect2_vaf, collapse = ';\n')
stat_df$VarScan2_vaf <-  paste(MUT_vaf_2$VarScan2_vaf, collapse = ';\n')
stat_df$SomaticSniper_vaf <-  paste(MUT_vaf_2$SomaticSniper_vaf, collapse = ';\n')
stat_df$MuSE_vaf <-  paste(MUT_vaf_2$MuSE_vaf, collapse = ';\n')
stat_df$Prediction  <- "Donor Loss"
stat_df$splice_junction_status <- "CanonicalSJ"
stat_df$splice_junction_position <- "chr2:208245425-208248368"

stat_df$Pvalue <- pkw$p.adj[6]
```

```{r echo=FALSE}
summary_res<- rbind(summary_res,stat_df)
```


```{r echo=FALSE}
## Add not found effects information:
MUT_df <- SJCounts[SJCounts$Variant_status == "R132C",c("sample_id","case_id")]

MUT_df$NormalizedExpression <- NA
MUT_df$Prediction <- "Exon Skipping"
MUT_df$splice_junction_status <- "AlternativeSJ not found in MUT samples"
MUT_df$splice_junction_position <- "chr2:208245425-208251429"
MUT_df$Pvalue <- NA

vaf_select <- annot_vaf[,c("Gene","MutationKey_Hg38","sample_id","RNA_Sample","DNA_Sample","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf", "VAF_RNA")]
MUT_vaf <- merge(MUT_df, vaf_select, by.x="sample_id", by.y = "RNA_Sample")
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id"] <- "RNA_Sample"
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id.y"] <- "sample_id"

col_order <- c("Gene", "MutationKey_Hg38", "Prediction","splice_junction_status","splice_junction_position","sample_id", "case_id","RNA_Sample","DNA_Sample","NormalizedExpression", "Pvalue", "VAF_RNA","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf")
MUT_vaf <- MUT_vaf[, col_order]

stat_df <- data.frame(matrix(ncol = 17, nrow = 1))
colnames(stat_df) <- c("Gene","MutationKey_Hg38","Prediction","splice_junction_status","splice_junction_position", "sample_id","case_id","RNA_Sample","DNA_Sample","NormalizedExpression","ECDF","Pvalue","VAF_RNA","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf")

stat_df$Gene <- unique(MUT_vaf$Gene)
stat_df$MutationKey_Hg38 <- unique(MUT_vaf$MutationKey_Hg38)
stat_df$sample_id <- paste(MUT_vaf$sample_id, collapse = ';\n') 
stat_df$case_id <- paste(MUT_vaf$case_id, collapse = ';\n') 
stat_df$RNA_Sample <- paste(MUT_vaf$RNA_Sample, collapse = ';\n') 
stat_df$DNA_Sample <- paste(MUT_vaf$DNA_Sample, collapse = ';\n')
stat_df$NormalizedExpression <- NA
stat_df$ECDF <- paste0(MUT_vaf$ECDF, collapse = ";\n")
stat_df$VAF_RNA <-  paste(MUT_vaf$VAF_RNA, collapse = ';\n')
stat_df$MuTect2_vaf <-  paste(MUT_vaf$MuTect2_vaf, collapse = ';\n')
stat_df$VarScan2_vaf <-  paste(MUT_vaf$VarScan2_vaf, collapse = ';\n')
stat_df$SomaticSniper_vaf <-  paste(MUT_vaf$SomaticSniper_vaf, collapse = ';\n')
stat_df$MuSE_vaf <-  paste(MUT_vaf$MuSE_vaf, collapse = ';\n')
stat_df$Prediction <- unique(MUT_vaf$Prediction)
stat_df$splice_junction_status <- unique(MUT_vaf$splice_junction_status)
stat_df$splice_junction_position <- unique(MUT_vaf$splice_junction_position)

## Save
summary_res <- rbind(summary_res, stat_df)

## Add not found effects information:
MUT_df <- SJCounts[SJCounts$Variant_status == "R132G",c("sample_id","case_id")]

MUT_df$NormalizedExpression <- NA
MUT_df$Prediction <- "Exon Skipping"
MUT_df$splice_junction_status <- "AlternativeSJ not found in MUT samples"
MUT_df$splice_junction_position <- "chr2:208245425-208251429"
MUT_df$Pvalue <- NA

vaf_select <- annot_vaf[,c("Gene","MutationKey_Hg38","sample_id","RNA_Sample","DNA_Sample","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf", "VAF_RNA")]
MUT_vaf <- merge(MUT_df, vaf_select, by.x="sample_id", by.y = "RNA_Sample")
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id"] <- "RNA_Sample"
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id.y"] <- "sample_id"

col_order <- c("Gene", "MutationKey_Hg38", "Prediction","splice_junction_status","splice_junction_position","sample_id", "case_id","RNA_Sample","DNA_Sample","NormalizedExpression", "Pvalue", "VAF_RNA","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf")
MUT_vaf <- MUT_vaf[, col_order]

stat_df <- data.frame(matrix(ncol = 17, nrow = 1))
colnames(stat_df) <- c("Gene","MutationKey_Hg38","Prediction","splice_junction_status","splice_junction_position", "sample_id","case_id","RNA_Sample","DNA_Sample","NormalizedExpression","ECDF","Pvalue","VAF_RNA","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf")

stat_df$Gene <- unique(MUT_vaf$Gene)
stat_df$MutationKey_Hg38 <- unique(MUT_vaf$MutationKey_Hg38)
stat_df$sample_id <- paste(MUT_vaf$sample_id, collapse = ';\n') 
stat_df$case_id <- paste(MUT_vaf$case_id, collapse = ';\n') 
stat_df$RNA_Sample <- paste(MUT_vaf$RNA_Sample, collapse = ';\n') 
stat_df$DNA_Sample <- paste(MUT_vaf$DNA_Sample, collapse = ';\n')
stat_df$NormalizedExpression <- NA
stat_df$ECDF <- paste0(MUT_vaf$ECDF, collapse = ";\n")
stat_df$VAF_RNA <-  paste(MUT_vaf$VAF_RNA, collapse = ';\n')
stat_df$MuTect2_vaf <-  paste(MUT_vaf$MuTect2_vaf, collapse = ';\n')
stat_df$VarScan2_vaf <-  paste(MUT_vaf$VarScan2_vaf, collapse = ';\n')
stat_df$SomaticSniper_vaf <-  paste(MUT_vaf$SomaticSniper_vaf, collapse = ';\n')
stat_df$MuSE_vaf <-  paste(MUT_vaf$MuSE_vaf, collapse = ';\n')
stat_df$Prediction <- unique(MUT_vaf$Prediction)
stat_df$splice_junction_status <- unique(MUT_vaf$splice_junction_status)
stat_df$splice_junction_position <- unique(MUT_vaf$splice_junction_position)

## Save
summary_res <- rbind(summary_res, stat_df)

## Add not found effects information:
MUT_df <- SJCounts[SJCounts$Variant_status == "R132S",c("sample_id","case_id")]

MUT_df$NormalizedExpression <- NA
MUT_df$Prediction <- "Exon Skipping"
MUT_df$splice_junction_status <- "AlternativeSJ not found in MUT samples"
MUT_df$splice_junction_position <- "chr2:208245425-208251429"
MUT_df$Pvalue <- NA

vaf_select <- annot_vaf[,c("Gene","MutationKey_Hg38","sample_id","RNA_Sample","DNA_Sample","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf", "VAF_RNA")]
MUT_vaf <- merge(MUT_df, vaf_select, by.x="sample_id", by.y = "RNA_Sample")
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id"] <- "RNA_Sample"
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id.y"] <- "sample_id"

col_order <- c("Gene", "MutationKey_Hg38", "Prediction","splice_junction_status","splice_junction_position","sample_id", "case_id","RNA_Sample","DNA_Sample","NormalizedExpression", "Pvalue", "VAF_RNA","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf")
MUT_vaf <- MUT_vaf[, col_order]

stat_df <- data.frame(matrix(ncol = 17, nrow = 1))
colnames(stat_df) <- c("Gene","MutationKey_Hg38","Prediction","splice_junction_status","splice_junction_position", "sample_id","case_id","RNA_Sample","DNA_Sample","NormalizedExpression","ECDF","Pvalue","VAF_RNA","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf")

stat_df$Gene <- unique(MUT_vaf$Gene)
stat_df$MutationKey_Hg38 <- unique(MUT_vaf$MutationKey_Hg38)
stat_df$sample_id <- paste(MUT_vaf$sample_id, collapse = ';\n') 
stat_df$case_id <- paste(MUT_vaf$case_id, collapse = ';\n') 
stat_df$RNA_Sample <- paste(MUT_vaf$RNA_Sample, collapse = ';\n') 
stat_df$DNA_Sample <- paste(MUT_vaf$DNA_Sample, collapse = ';\n')
stat_df$NormalizedExpression <- NA
stat_df$ECDF <- paste0(MUT_vaf$ECDF, collapse = ";\n")
stat_df$VAF_RNA <-  paste(MUT_vaf$VAF_RNA, collapse = ';\n')
stat_df$MuTect2_vaf <-  paste(MUT_vaf$MuTect2_vaf, collapse = ';\n')
stat_df$VarScan2_vaf <-  paste(MUT_vaf$VarScan2_vaf, collapse = ';\n')
stat_df$SomaticSniper_vaf <-  paste(MUT_vaf$SomaticSniper_vaf, collapse = ';\n')
stat_df$MuSE_vaf <-  paste(MUT_vaf$MuSE_vaf, collapse = ';\n')
stat_df$Prediction <- unique(MUT_vaf$Prediction)
stat_df$splice_junction_status <- unique(MUT_vaf$splice_junction_status)
stat_df$splice_junction_position <- unique(MUT_vaf$splice_junction_position)

## Save
summary_res <- rbind(summary_res, stat_df)
```


```{r echo=FALSE}
rm(GeneSJ, normalization_df, GeneSJ_subset, annot_vaf, plot_df, samples_df,table, plot_boxplot, plot_dotplot, vaf_select, SJCounts, MUT_df, MUT_vaf)
```


# U2AF1 chr21,43094667,T,C {.tabset}
Variant found in **3 patients** of the BeatAML (**3 samples**)

* Patients with U2AF1 chr21,43094667,T,C variant: **3 patients (3 samples)**
* Patients with the variant and RNASeq for validation: **3 patients (3 samples)**

The splicing alterations being assessed are: 

* **Donor Gain** : chr21:43094564-43094666, found in the mutated samples .
* **Exon 6 Skipping**: chr21:43094564-43095437, not found in the splice junction collection.
* **Exon 5 Skipping**: chr21:43094789-43095693, found in the mutated samples. 
* **Donor Loss**: chr21:43094564-43094654; donor splice site chr21:43094654.
* **Canonical SJ Intron 4-5**: chr21:43095537-43095693.
* **Canonical SJ Intron 5-6**: chr21:43094789-43095437.

Variant information:
```{r echo=FALSE}
datatable(to_be_validated[to_be_validated$MutationKey_Hg38 == "chr21,43094667,T,C",],
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp', 
              scrollX = TRUE,  
              scrollY = TRUE,
              columnDefs = list(list(targets = '_all', className = 'dt-center')))
)
```

Load the extracted splice junctions of the gene harboring the mutation.
```{r}
extractedSJ_pathUM <- paste0(extractedSJ_dir_in,"U2AF1_UM_annotSJ.tsv")
extractedSJ_pathMM <- paste0(extractedSJ_dir_in,"U2AF1_MM_annotSJ.tsv")
GeneSJ_UM <- read.delim(extractedSJ_pathUM, sep ="\t")
GeneSJ_MM <- read.delim(extractedSJ_pathMM, sep ="\t")

GeneSJ <- GeneSJ_UM[,grep("chr", names(GeneSJ_UM))] + GeneSJ_MM[,grep("chr", names(GeneSJ_MM))]
GeneSJ$INDEX <- GeneSJ_UM$INDEX
GeneSJ$sample_id <- GeneSJ_UM$sample_id
GeneSJ$case_id <- GeneSJ_UM$case_id
GeneSJ$file_id.BAM <- GeneSJ_UM$file_id.BAM
GeneSJ$file_name.BAM <- GeneSJ_UM$file_name.BAM
GeneSJ$file_id.STAR.SJCounts <- GeneSJ_UM$file_id.STAR.SJCounts
GeneSJ$file_name.STAR.SJCounts <- GeneSJ_UM$file_name.STAR.SJCounts
```

Set the sample's group: Mutated (MUT) or No Mutated (WT)

```{r}
samples_df <- found_variants[found_variants$Gene=="U2AF1" & found_variants$MutationKey_Hg38 == "chr21,43094667,T,C",]

cases <- samples_df$RNA_Sample[samples_df$Validable == "Validable"]
GeneSJ$GROUP <- ifelse(GeneSJ$sample_id %in% cases , "MUT", "WT")
```


## SJ Lookup {.tabset}

Search for the splice junctions of interest in the extracted splice junctions of the gene by position (chr_SJstart_SJend).

### Exon 6 Donor Gain 
Search: chr21:43094564-43094666

Show all the splice junctions containing the position chr21:43094564-43094666
```{r}
colnames(GeneSJ)[grep("43094564_43094666",colnames(GeneSJ))]
```

Found: chr21:43094564-43094666

Reads of all the AML samples (mutated and no mutated) for the splice junction:
```{r}
GeneSJ$chr21_43094564_43094666
```
Samples with the SJ of interest:
```{r}
table(GeneSJ$chr21_43094564_43094666>0) 
```
Groups of the samples having the alternative splice junction:
```{r}
table(GeneSJ$GROUP[GeneSJ$chr21_43094564_43094666 > 0])
```
Alternative SJ found in the mutated samples.

### Exon 6 Skipping
Search: chr21:43094564-43095437

```{r}
colnames(GeneSJ)[grep("43094564",colnames(GeneSJ))]
```

```{r}
colnames(GeneSJ)[grep("43095437",colnames(GeneSJ))]
```

Show all the splice junctions containing the position 43094564_43095437
```{r}
colnames(GeneSJ)[grep("43094564_43095437",colnames(GeneSJ))]
```
Alternative SJ not found in the splice junction collection.

### Exon 5 Skipping
Search: chr21:43094789-43095693

Show all the splice junctions containing the position chr21:43094789-43095693
```{r}
colnames(GeneSJ)[grep("43094789_43095693",colnames(GeneSJ))]
```
Found: chr21:43094789-43095693

Reads of all the AML samples (mutated and no mutated) for the splice junction:
```{r}
GeneSJ$chr21_43094789_43095693
```
Samples with the SJ of interest:
```{r}
table(GeneSJ$chr21_43094789_43095693>0) 
```
Groups of the samples having the alternative splice junction:
```{r}
table(GeneSJ$GROUP[GeneSJ$chr21_43094789_43095693 > 0])
```
Alternative SJ found in the mutated samples.

### Canonical SJ
Exon 4-5: chr21:43095537-43095693

Exon 5-6: chr21:43064789-43095437

Exon 6-7: chr21:43094564-43094654; splice site chr21:43094554

Reads of all the AML samples (mutated and no mutated) for the splice junction:
```{r}
GeneSJ$chr21_43095537_43095693 #SJ Intron 4 to 5
```

Reads of all the AML samples (mutated and no mutated) for the splice junction:
```{r}
GeneSJ$chr21_43094789_43095437 #SJ Intron 5 to 6
```

Reads of all the AML samples (mutated and no mutated) for the splice junction:
```{r}
GeneSJ$chr21_43094564_43094654 #SJ Intron 6 to 7
```

## Normalization
Count the reads of all the splice junctions of the gene harboring the variant:
```{r}
GeneSJ$rowSum_SJtotal <- rowSums(GeneSJ[,grep("chr", names(GeneSJ))])
```

Normalization of the expression by the total read counts of all the splice junctions of the gene:
```{r}
GeneSJ$Normalized_CanonEx4_5 <- (GeneSJ$chr21_43095537_43095693)/GeneSJ$rowSum_SJtotal*100
GeneSJ$Normalized_CanonEx5_6 <- (GeneSJ$chr21_43094789_43095437)/GeneSJ$rowSum_SJtotal*100
GeneSJ$Normalized_CanonEx6_7 <- (GeneSJ$chr21_43094564_43094654)/GeneSJ$rowSum_SJtotal*100
GeneSJ$Normalized_Ex6DG <- (GeneSJ$chr21_43094564_43094666)/GeneSJ$rowSum_SJtotal*100
GeneSJ$Normalized_ES5 <- (GeneSJ$chr21_43094789_43095693)/GeneSJ$rowSum_SJtotal*100
```

Download the normalized values for the assessed splice junctions of all the AML samples:
```{r echo=FALSE}
normalization_df <- GeneSJ[,c("sample_id","case_id", "GROUP","Normalized_CanonEx4_5","Normalized_CanonEx5_6", "Normalized_CanonEx6_7", "Normalized_Ex6DG" ,"Normalized_ES5", "chr21_43095537_43095693", "chr21_43094789_43095437", "chr21_43094564_43094654", "chr21_43094564_43094666","chr21_43094789_43095693", "rowSum_SJtotal","file_id.BAM","file_name.BAM","file_id.STAR.SJCounts","file_name.STAR.SJCounts" )]

colnames(normalization_df)[grep("chr21_43095537_43095693", names(normalization_df))] <- "chr21_43095537_43095693_CanonEx4_5"
colnames(normalization_df)[grep("chr21_43094789_43095437", names(normalization_df))] <- "chr21_43094789_43095437_CanonEx5_6"
colnames(normalization_df)[grep("chr21_43094564_43094654", names(normalization_df))] <- "chr21_43094564_43094654_CanonEx6_7"
colnames(normalization_df)[grep("chr21_43094564_43094666", names(normalization_df))] <- "chr21_43094564_43094666_Ex6DG"
colnames(normalization_df)[grep("chr21_43094789_43095693", names(normalization_df))] <- "chr21_43094789_43095693_ES5"

write.xlsx(normalization_df, paste0(normalizedEx_dir_out, "BeatAML.U2AF1.xlsx"), rowNames = FALSE)
```

```{r echo=FALSE}
datatable(normalization_df,
extensions = 'Buttons',
rownames = FALSE, 
filter = 'top',
options = list(paging = TRUE,
              pageLength = 4,
              autoWidth = FALSE,
              dom = 'lifrtBp',
              buttons = c('excel'), 
              scrollX = TRUE,  
              scrollY = TRUE,
              columnDefs = list(list(targets = '_all', className = 'dt-center'))
))
```

## VAF
```{r echo=FALSE}
GeneSJ_subset <- GeneSJ[GeneSJ$sample_id %in% cases,]
GeneSJ_subset$Gene <- "U2AF1"
GeneSJ_subset$MutationKey_Hg38 <- "chr21,43094667,T,C"

annot_vaf <- merge(found_variants[,c("MutationKey_Hg38","Gene","sample_id","case_id",                                               "RNA_Sample","DNA_Sample","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf",                         "MuTect2_filter","VarScan2_filter","SomaticSniper_filter","MuSE_filter","MUTReads","WTReads","VAF")], 
                   GeneSJ_subset[,c("Gene","MutationKey_Hg38","sample_id", "case_id",  "Normalized_CanonEx4_5","Normalized_CanonEx5_6", "Normalized_CanonEx6_7", "Normalized_Ex6DG" ,"Normalized_ES5")], 
                   by.x=c("RNA_Sample", "case_id","Gene", "MutationKey_Hg38"), 
                   by.y=c("sample_id", "case_id","Gene", "MutationKey_Hg38"))


annot_vaf <- annot_vaf %>% select(Gene,MutationKey_Hg38,
                                  sample_id,
                                  case_id, RNA_Sample,DNA_Sample,
                                  MuTect2_vaf,VarScan2_vaf,SomaticSniper_vaf,MuSE_vaf,
                                  VAF,
                                  Normalized_CanonEx4_5,Normalized_CanonEx5_6, Normalized_CanonEx6_7,Normalized_Ex6DG, Normalized_ES5,
                                  everything()
                                  )
colnames(annot_vaf)[grep("VAF", names(annot_vaf))] <- "VAF_RNA"
colnames(annot_vaf)[grep("MUTReads", names(annot_vaf))] <- "MUTReads_RNA"
colnames(annot_vaf)[grep("WTReads", names(annot_vaf))] <- "WTReads_RNA"

write.xlsx(annot_vaf, paste0(normalizedEx_dir_out, "BeatAML.U2AF1.MUT.xlsx"), rowNames = FALSE)
```

Mutated samples vaf:
```{r echo=FALSE}
datatable(annot_vaf,
extensions = 'Buttons',
rownames = FALSE, 
filter = 'top',
options = list(paging = TRUE,
              pageLength = 4,
              autoWidth = TRUE,
              dom = 'lifrtBp',
              buttons = c('excel'), 
              scrollX = TRUE,  
              scrollY = TRUE
              )
)
```

## Plots {.tabset}

```{r echo=FALSE}
GeneSJ$GROUP <- ifelse(GeneSJ$GROUP == "MUT", "MUT", "WT")
GeneSJ$GROUP <- as.factor(GeneSJ$GROUP)
```

### Static Dot Plots 
Canonical SJ:

```{r echo=FALSE}
ggplot(GeneSJ, aes(sample_id,Normalized_CanonEx6_7, color=GROUP)) + 
  geom_point(size=.8)  +
  labs(color='GROUP', 
       title = "Normalized Expression of U2AF1 chr21:43094564-43094654", 
       subtitle="Potential Donor Loss Effect",
       y = "Normalized Expression")
```

Splicing alterations:

```{r echo=FALSE}
ggplot(GeneSJ, aes(sample_id,Normalized_Ex6DG,color=GROUP)) +
  geom_point(size=.8) +
  labs(color='GROUP', 
       title = "Normalized Expression of U2AF1 chr21:43094564-43094666", 
       subtitle="Potential Donor Gain Effect",
       y = "Normalized Expression")
```

### Interactive Dot Plots

```{r echo=FALSE}
plot_df <- merge(GeneSJ,annot_vaf[c("RNA_Sample", "VAF_RNA", 
                                    "MuTect2_vaf", "VarScan2_vaf", "SomaticSniper_vaf", "MuSE_vaf")],
                 by.x="sample_id", by.y="RNA_Sample", all.x = TRUE)
```

Canonical SJ:

```{r echo=FALSE}
ggplotly(ggplot(plot_df, aes(sample_id,Normalized_CanonEx6_7, color=GROUP)) + 
  geom_point(aes(text=paste0("VAF RNA: ",VAF_RNA,"\nVAF MuTect: ",MuTect2_vaf,
                             "\nVAF VarScan2: ",VarScan2_vaf,
                             "\nVAF SomaticSniper: ",SomaticSniper_vaf,
                             "\nVAF MuSE: ",MuSE_vaf)),size= 0.8)  +
  ggtitle("Normalized expression of U2AF1 Canonical Donor Loss") +
  labs(color='GROUP') 
)
```


Splicing alteration:

```{r echo=FALSE}
ggplotly(ggplot(plot_df, aes(sample_id,Normalized_Ex6DG,color=GROUP)) +
  geom_point(aes(text=paste0("VAF RNA: ",VAF_RNA,"\nVAF MuTect: ",MuTect2_vaf,
                             "\nVAF VarScan2: ",VarScan2_vaf,
                             "\nVAF SomaticSniper: ",SomaticSniper_vaf,
                             "\nVAF MuSE: ",MuSE_vaf)),size= 0.8)  +
  ggtitle("Normalized expression of U2AF1 Donor Gain") +
  labs(color='GROUP') 
)
```

### Violin Plots
Violin Plots for the alternative splice junctions interrogated:
```{r echo=FALSE}
plot_boxplot <- plot_df[plot_df$GROUP %in% c("WT"),]
plot_dotplot <- plot_df[plot_df$GROUP %in% c("MUT"),]

GEN = "U2AF1"
SJ = "chr21:43094564-43094654"

ggplot(plot_df, aes(y=Normalized_CanonEx6_7, x=factor(GROUP, level=c("WT", "MUT")))) + 
       geom_violin(scale="area") +
       geom_point(data = plot_dotplot, 
                  aes(y=Normalized_CanonEx6_7, x=GROUP),
                  color="black",
                  size=2
                  ) +
        
       ylab("Normalized SJ expression") +
       xlab(paste(GEN, SJ))
```


```{r echo=FALSE}
plot_boxplot <- plot_df[plot_df$GROUP %in% c("WT"),]
plot_dotplot <- plot_df[plot_df$GROUP %in% c("MUT"),]

GEN = "U2AF1"
SJ = "chr21:43094564-43094666"

ggplot(plot_df, aes(y=Normalized_Ex6DG, x=factor(GROUP, level=c("WT", "MUT")))) + 
       geom_violin(scale="area") +
       geom_point(data = plot_dotplot, 
                  aes(y=Normalized_Ex6DG, x=GROUP),
                  color="black",
                  size=2
                  ) +
        
       ylab("Normalized SJ expression") +
       xlab(paste(GEN, SJ))
```
```{r echo=FALSE}
plot_boxplot <- plot_df[plot_df$GROUP %in% c("WT"),]
plot_dotplot <- plot_df[plot_df$GROUP %in% c("MUT"),]

GEN = "U2AF1"
SJ = "chr21:43094789-43095693"

ggplot(plot_df, aes(y=Normalized_ES5, x=factor(GROUP, level=c("WT", "MUT")))) + 
       geom_violin(scale="area") +
       geom_point(data = plot_dotplot, 
                  aes(y=Normalized_ES5, x=GROUP),
                  color="black",
                  size=2
                  ) +
        
       ylab("Normalized SJ expression") +
       xlab(paste(GEN, SJ))
```


## Statistical Analysis {.tabset}

```{r}
SJCounts <- GeneSJ 
```

### Donor Gain {.tabset}
#### Expression Difference 
Normality Test:
```{r}
shapiro.test(SJCounts$Normalized_Ex6DG[SJCounts$GROUP == "WT"])
```

Value of Mean Normalized Expression of the Alternative SJ in WT samples:
```{r}
mean_WT_SJi <- mean(SJCounts$Normalized_Ex6DG[SJCounts$GROUP == "WT"], na.rm=TRUE)
mean_WT_SJi
```

Normalized Expression Value of the Alternative SJ in the MUT sample:
```{r}
MUT_SJi <- SJCounts$Normalized_Ex6DG[SJCounts$GROUP == "MUT"]
MUT_SJi
```

Deviation from the mean normalized expression:
```{r}
SJCounts$Difference <-  SJCounts$Normalized_Ex6DG - mean_WT_SJi
```

```{r echo=FALSE}
table <- SJCounts[SJCounts$GROUP == "WT", c("sample_id", "Normalized_Ex6DG", "Difference")]
table <- table[order(table$Difference, decreasing = TRUE),]
```

```{r echo=FALSE}
datatable(table,
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp', 
              scrollX = TRUE,  
              scrollY = TRUE
))
```

Difference in the MUT sample: deviation of the Normalized expression of the MUT patient from the mean normalized WT expression
```{r}
SJCounts$Difference[SJCounts$GROUP == "MUT"]
```

```{r echo=FALSE}
ggplot(SJCounts, aes(sample_id,Difference)) + geom_point(aes(colour=factor(GROUP))) +
  ylab("Differential Expression") +
  labs(color = "GROUP",
       title = "Differential Expression Distribution")
```


```{r echo=FALSE}
ggplot(SJCounts, aes(y=Difference, x=factor(GROUP, level=c("WT", "MUT")))) + 
  geom_boxplot() +
  ylab("Differential Expression") +
  xlab("GROUP")
```

#### ECDF

```{r}
v_ecdf <- ecdf(SJCounts$Difference[SJCounts$GROUP == "WT"])
v_ecdf
```

```{r}
plot(ecdf(SJCounts$Difference[SJCounts$GROUP == "WT"]))
```

```{r}
v_ecdf(SJCounts$Difference[SJCounts$GROUP == "MUT"])
```

```{r}
MUT_df <- SJCounts[SJCounts$GROUP == "MUT",c("sample_id","case_id", "Normalized_Ex6DG")]
colnames(MUT_df) <- c("sample_id", "case_id", "NormalizedExpression")

MUT_df$ECDF <- v_ecdf(SJCounts$Difference[SJCounts$GROUP == "MUT"])
MUT_df$Pvalue <- 1- MUT_df$ECDF
MUT_df$Prediction <- "Donor Gain"
MUT_df$splice_junction_status <- "AlternativeSJ found in MUT samples"
MUT_df$splice_junction_position <- "chr21:43094564-43094666"

MUT_df$ECDF <- ifelse(MUT_df$NormalizedExpression == 0, NA,MUT_df$ECDF)
MUT_df$Pvalue <- ifelse(MUT_df$NormalizedExpression == 0, NA,MUT_df$Pvalue)
```

```{r echo=FALSE}
vaf_select <- annot_vaf[,c("Gene","MutationKey_Hg38","sample_id","RNA_Sample","DNA_Sample","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf", "VAF_RNA")]
MUT_vaf <- merge(MUT_df, vaf_select, by.x="sample_id", by.y = "RNA_Sample")
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id"] <- "RNA_Sample"
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id.y"] <- "sample_id"

col_order <- c("Gene", "MutationKey_Hg38", "Prediction","splice_junction_status","splice_junction_position","sample_id", "case_id","RNA_Sample","DNA_Sample","NormalizedExpression","ECDF", "Pvalue", "VAF_RNA","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf")
MUT_vaf <- MUT_vaf[, col_order]
```

Download the vaf, inferred percentiles and pvalues of the mutated samples:
```{r echo=FALSE}
datatable(MUT_vaf,
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp',
              scrollX=TRUE
))
```


#### T-test
Normality:
```{r}
shapiro.test(SJCounts$Normalized_Ex6DG[SJCounts$GROUP == "WT"])
```

```{r}
shapiro.test(SJCounts$Normalized_Ex6DG[SJCounts$GROUP == "MUT"])
```

Variances:
```{r}
res.ftest <- var.test(Normalized_Ex6DG ~ GROUP, SJCounts, alternative = "two.sided", conf.level=0.95)
res.ftest
```

Welch Two Sample t-test:
```{r}
t <- t.test(Normalized_Ex6DG ~ GROUP, data=SJCounts,
 alternative="two.sided")
t
```

```{r echo=FALSE}

stat_df <- data.frame(matrix(ncol = 17, nrow = 1))
colnames(stat_df) <- c("Gene","MutationKey_Hg38","Prediction","splice_junction_status","splice_junction_position", "sample_id","case_id","RNA_Sample","DNA_Sample","NormalizedExpression","ECDF","Pvalue","VAF_RNA","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf")
stat_df$Gene <- unique(MUT_vaf$Gene)
stat_df$MutationKey_Hg38 <- unique(MUT_vaf$MutationKey_Hg38)
stat_df$sample_id <- paste(MUT_vaf$sample_id, collapse = ';\n') 
stat_df$case_id <- paste(MUT_vaf$case_id, collapse = ';\n') 
stat_df$RNA_Sample <- paste(MUT_vaf$RNA_Sample, collapse = ';\n') 
stat_df$DNA_Sample <- paste(MUT_vaf$DNA_Sample, collapse = ';\n')
stat_df$NormalizedExpression <- paste0(MUT_vaf$NormalizedExpression, collapse= ";\n")
stat_df$ECDF <- paste0(MUT_vaf$ECDF, collapse = ";\n")
stat_df$VAF_RNA <-  paste(MUT_vaf$VAF_RNA, collapse = ';\n')
stat_df$MuTect2_vaf <-  paste(MUT_vaf$MuTect2_vaf, collapse = ';\n')
stat_df$VarScan2_vaf <-  paste(MUT_vaf$VarScan2_vaf, collapse = ';\n')
stat_df$SomaticSniper_vaf <-  paste(MUT_vaf$SomaticSniper_vaf, collapse = ';\n')
stat_df$MuSE_vaf <-  paste(MUT_vaf$MuSE_vaf, collapse = ';\n')
stat_df$Prediction <- "Donor Gain"
stat_df$splice_junction_status <- "AlternativeSJ found in MUT samples"
stat_df$splice_junction_position <- "chr21:43094564-43094666"

stat_df$Pvalue <- t$p.value
```

```{r echo=FALSE}
summary_res<- rbind(summary_res,stat_df)
```

### Canonical Donor Loss {.tabset}
#### Expression Difference 
Normality Test:
```{r}
shapiro.test(SJCounts$Normalized_CanonEx6_7[SJCounts$GROUP == "WT"])
```

Value of Mean Normalized Expression of the Alternative SJ in WT samples:
```{r}
mean_WT_SJi <- mean(SJCounts$Normalized_CanonEx6_7[SJCounts$GROUP == "WT"], na.rm=TRUE)
mean_WT_SJi
```

Normalized Expression Value of the Alternative SJ in the MUT sample:
```{r}
MUT_SJi <- SJCounts$Normalized_CanonEx6_7[SJCounts$GROUP == "MUT"]
MUT_SJi
```

Deviation from the mean normalized expression:
```{r}
SJCounts$Difference <-  SJCounts$Normalized_CanonEx6_7 - mean_WT_SJi
```

```{r echo=FALSE}
table <- SJCounts[SJCounts$GROUP == "WT", c("sample_id", "Normalized_CanonEx6_7", "Difference")]
table <- table[order(table$Difference, decreasing = TRUE),]
```

```{r echo=FALSE}
datatable(table,
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp'
))
```

Difference in the MUT sample: deviation of the Normalized expression of the MUT patient from the mean normalized WT expression
```{r}
SJCounts$Difference[SJCounts$GROUP == "MUT"]
```

```{r echo=FALSE}
ggplot(SJCounts, aes(sample_id,Difference)) + geom_point(aes(colour=factor(GROUP))) +
  ylab("Differential Expression") +
  labs(color = "GROUP",
       title = "Differential Expression Distribution")
```


```{r echo=FALSE}
ggplot(SJCounts, aes(y=Difference, x=factor(GROUP, level=c("WT", "MUT")))) + 
  geom_boxplot() +
  ylab("Differential Expression") +
  xlab("GROUP")
```

#### ECDF

```{r}
v_ecdf <- ecdf(SJCounts$Difference[SJCounts$GROUP == "WT"])
v_ecdf
```

```{r}
plot(ecdf(SJCounts$Difference[SJCounts$GROUP == "WT"]))
```

```{r}
v_ecdf(SJCounts$Difference[SJCounts$GROUP == "MUT"])
```

```{r}
MUT_df <- SJCounts[SJCounts$GROUP == "MUT",c("sample_id","case_id", "Normalized_CanonEx6_7")]
colnames(MUT_df) <- c("sample_id", "case_id", "NormalizedExpression")

MUT_df$ECDF <- v_ecdf(SJCounts$Difference[SJCounts$GROUP == "MUT"])
MUT_df$Pvalue <- NA
MUT_df$Prediction <- "Donor Loss"
MUT_df$splice_junction_status <- "CanonicalSJ"
MUT_df$splice_junction_position <- "chr21:43094564-43094654"

MUT_df$ECDF <- ifelse(MUT_df$NormalizedExpression == 0, NA,MUT_df$ECDF)

```


```{r echo=FALSE}
vaf_select <- annot_vaf[,c("Gene","MutationKey_Hg38","sample_id","RNA_Sample","DNA_Sample","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf", "VAF_RNA")]
MUT_vaf <- merge(MUT_df, vaf_select, by.x="sample_id", by.y = "RNA_Sample")
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id"] <- "RNA_Sample"
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id.y"] <- "sample_id"

col_order <- c("Gene", "MutationKey_Hg38", "Prediction","splice_junction_status","splice_junction_position","sample_id", "case_id","RNA_Sample","DNA_Sample","NormalizedExpression","ECDF", "Pvalue", "VAF_RNA","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf")
MUT_vaf <- MUT_vaf[, col_order]
```

Download the vaf, inferred percentiles and pvalues of the mutated samples:
```{r echo=FALSE}
datatable(MUT_vaf,
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp',
              scrollX=TRUE
))
```


#### T-test
Normality:
```{r}
shapiro.test(SJCounts$Normalized_CanonEx6_7[SJCounts$GROUP == "WT"])
```

```{r}
shapiro.test(SJCounts$Normalized_CanonEx6_7[SJCounts$GROUP == "MUT"])
```

Variances:
```{r}
res.ftest <- var.test(Normalized_CanonEx6_7 ~ GROUP, SJCounts, alternative = "two.sided", conf.level=0.95)
res.ftest
```

Two Sample t-test:
```{r}
t <- t.test(Normalized_CanonEx6_7 ~ GROUP, data=SJCounts,
 alternative="two.sided",
 var.equal=TRUE)
t
```

```{r echo=FALSE}

stat_df <- data.frame(matrix(ncol = 17, nrow = 1))
colnames(stat_df) <- c("Gene","MutationKey_Hg38","Prediction","splice_junction_status","splice_junction_position", "sample_id","case_id","RNA_Sample","DNA_Sample","NormalizedExpression","ECDF","Pvalue","VAF_RNA","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf")
stat_df$Gene <- unique(MUT_vaf$Gene)
stat_df$MutationKey_Hg38 <- unique(MUT_vaf$MutationKey_Hg38)
stat_df$sample_id <- paste(MUT_vaf$sample_id, collapse = ';\n') 
stat_df$case_id <- paste(MUT_vaf$case_id, collapse = ';\n') 
stat_df$RNA_Sample <- paste(MUT_vaf$RNA_Sample, collapse = ';\n') 
stat_df$DNA_Sample <- paste(MUT_vaf$DNA_Sample, collapse = ';\n')
stat_df$NormalizedExpression <- paste0(MUT_vaf$NormalizedExpression, collapse= ";\n")
stat_df$ECDF <- paste0(MUT_vaf$ECDF, collapse = ";\n")
stat_df$VAF_RNA <-  paste(MUT_vaf$VAF_RNA, collapse = ';\n')
stat_df$MuTect2_vaf <-  paste(MUT_vaf$MuTect2_vaf, collapse = ';\n')
stat_df$VarScan2_vaf <-  paste(MUT_vaf$VarScan2_vaf, collapse = ';\n')
stat_df$SomaticSniper_vaf <-  paste(MUT_vaf$SomaticSniper_vaf, collapse = ';\n')
stat_df$MuSE_vaf <-  paste(MUT_vaf$MuSE_vaf, collapse = ';\n')
stat_df$Prediction <- "Donor Loss"
stat_df$splice_junction_status <- "AlternativeSJ found in MUT samples"
stat_df$splice_junction_position <- "chr21:43094564-43094654"

stat_df$Pvalue <- t$p.value
```

```{r echo=FALSE}
summary_res<- rbind(summary_res,stat_df)
```

### Exon 5 Skipping {.tabset}
#### Expression Difference 
Normality Test:
```{r}
shapiro.test(SJCounts$Normalized_ES5[SJCounts$GROUP == "WT"])
```

Value of Mean Normalized Expression of the Alternative SJ in WT samples:
```{r}
mean_WT_SJi <- mean(SJCounts$Normalized_ES5[SJCounts$GROUP == "WT"], na.rm=TRUE)
mean_WT_SJi
```

Normalized Expression Value of the Alternative SJ in the MUT sample:
```{r}
MUT_SJi <- SJCounts$Normalized_ES5[SJCounts$GROUP == "MUT"]
MUT_SJi
```

Deviation from the mean normalized expression:
```{r}
SJCounts$Difference <-  SJCounts$Normalized_ES5 - mean_WT_SJi
```

```{r echo=FALSE}
table <- SJCounts[SJCounts$GROUP == "WT", c("sample_id", "Normalized_ES5", "Difference")]
table <- table[order(table$Difference, decreasing = TRUE),]
```

```{r echo=FALSE}
datatable(table,
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp', 
              scrollX = TRUE,  
              scrollY = TRUE
))
```

Difference in the MUT sample: deviation of the Normalized expression of the MUT patient from the mean normalized WT expression
```{r}
SJCounts$Difference[SJCounts$GROUP == "MUT"]
```

```{r echo=FALSE}
ggplot(SJCounts, aes(sample_id,Difference)) + geom_point(aes(colour=factor(GROUP))) +
  ylab("Differential Expression") +
  labs(color = "GROUP",
       title = "Differential Expression Distribution")
```


```{r echo=FALSE}
ggplot(SJCounts, aes(y=Difference, x=factor(GROUP, level=c("WT", "MUT")))) + 
  geom_boxplot() +
  ylab("Differential Expression") +
  xlab("GROUP")
```

#### ECDF

```{r}
v_ecdf <- ecdf(SJCounts$Difference[SJCounts$GROUP == "WT"])
v_ecdf
```

```{r}
plot(ecdf(SJCounts$Difference[SJCounts$GROUP == "WT"]))
```

```{r}
v_ecdf(SJCounts$Difference[SJCounts$GROUP == "MUT"])
```

```{r}
MUT_df <- SJCounts[SJCounts$GROUP == "MUT",c("sample_id","case_id", "Normalized_ES5")]
colnames(MUT_df) <- c("sample_id", "case_id", "NormalizedExpression")

MUT_df$ECDF <- v_ecdf(SJCounts$Difference[SJCounts$GROUP == "MUT"])
MUT_df$Pvalue <- NA

MUT_df$Prediction <- "Exon Skipping"
MUT_df$splice_junction_status <- "AlternativeSJ found in MUT samples"
MUT_df$splice_junction_position <- "chr21:43094789-43095693"

MUT_df$ECDF <- ifelse(MUT_df$NormalizedExpression == 0, NA,MUT_df$ECDF)
```

```{r echo=FALSE}
vaf_select <- annot_vaf[,c("Gene","MutationKey_Hg38","sample_id","RNA_Sample","DNA_Sample","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf", "VAF_RNA")]
MUT_vaf <- merge(MUT_df, vaf_select, by.x="sample_id", by.y = "RNA_Sample")
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id"] <- "RNA_Sample"
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id.y"] <- "sample_id"

col_order <- c("Gene", "MutationKey_Hg38", "Prediction","splice_junction_status","splice_junction_position","sample_id", "case_id","RNA_Sample","DNA_Sample","NormalizedExpression","ECDF", "Pvalue", "VAF_RNA","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf")
MUT_vaf <- MUT_vaf[, col_order]
```

Download the vaf, inferred percentiles and pvalues of the mutated samples:
```{r echo=FALSE}
datatable(MUT_vaf,
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp', 
              scrollX = TRUE
))
```


#### T-test
Normality:
```{r}
shapiro.test(SJCounts$Normalized_ES5[SJCounts$GROUP == "WT"])
```

```{r}
shapiro.test(SJCounts$Normalized_ES5[SJCounts$GROUP == "MUT"])
```

Variances:
```{r}
res.ftest <- var.test(Normalized_ES5 ~ GROUP, SJCounts, alternative = "two.sided", conf.level=0.95)
res.ftest
```

Welch Two Sample t-test:
```{r}
t <- t.test(Normalized_ES5 ~ GROUP, data=SJCounts,
 alternative="less",
 var.equal=FALSE)
t
```

```{r echo=FALSE}

stat_df <- data.frame(matrix(ncol = 17, nrow = 1))
colnames(stat_df) <- c("Gene","MutationKey_Hg38","Prediction","splice_junction_status","splice_junction_position", "sample_id","case_id","RNA_Sample","DNA_Sample","NormalizedExpression","ECDF","Pvalue","VAF_RNA","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf")
stat_df$Gene <- unique(MUT_vaf$Gene)
stat_df$MutationKey_Hg38 <- unique(MUT_vaf$MutationKey_Hg38)
stat_df$sample_id <- paste(MUT_vaf$sample_id, collapse = ';\n') 
stat_df$case_id <- paste(MUT_vaf$case_id, collapse = ';\n') 
stat_df$RNA_Sample <- paste(MUT_vaf$RNA_Sample, collapse = ';\n') 
stat_df$DNA_Sample <- paste(MUT_vaf$DNA_Sample, collapse = ';\n')
stat_df$NormalizedExpression <- paste0(MUT_vaf$NormalizedExpression, collapse= ";\n")
stat_df$ECDF <- paste0(MUT_vaf$ECDF, collapse = ";\n")
stat_df$VAF_RNA <-  paste(MUT_vaf$VAF_RNA, collapse = ';\n')
stat_df$MuTect2_vaf <-  paste(MUT_vaf$MuTect2_vaf, collapse = ';\n')
stat_df$VarScan2_vaf <-  paste(MUT_vaf$VarScan2_vaf, collapse = ';\n')
stat_df$SomaticSniper_vaf <-  paste(MUT_vaf$SomaticSniper_vaf, collapse = ';\n')
stat_df$MuSE_vaf <-  paste(MUT_vaf$MuSE_vaf, collapse = ';\n')
stat_df$Prediction <- "Exon Skipping"
stat_df$splice_junction_status <- "AlternativeSJ found in MUT samples"
stat_df$splice_junction_position <- "chr21:43094789-43095693"

stat_df$Pvalue <- t$p.value
```

```{r echo=FALSE}
summary_res<- rbind(summary_res,stat_df)
```

### Canonical SJ Intron 4-5 {.tabset}
#### Expression Difference 
Normality Test:
```{r}
shapiro.test(SJCounts$Normalized_CanonEx4_5[SJCounts$GROUP == "WT"])
```

Value of Mean Normalized Expression of the Alternative SJ in WT samples:
```{r}
mean_WT_SJi <- mean(SJCounts$Normalized_CanonEx4_5[SJCounts$GROUP == "WT"], na.rm=TRUE)
mean_WT_SJi
```

Normalized Expression Value of the Alternative SJ in the MUT sample:
```{r}
MUT_SJi <- SJCounts$Normalized_CanonEx4_5[SJCounts$GROUP == "MUT"]
MUT_SJi
```

Deviation from the mean normalized expression:
```{r}
SJCounts$Difference <-  SJCounts$Normalized_CanonEx4_5 - mean_WT_SJi
```

```{r echo=FALSE}
table <- SJCounts[SJCounts$GROUP == "WT", c("sample_id", "Normalized_CanonEx4_5", "Difference")]
table <- table[order(table$Difference, decreasing = TRUE),]
```

```{r echo=FALSE}
datatable(table,
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp', 
              scrollX = TRUE,  
              scrollY = TRUE
))
```

Difference in the MUT sample: deviation of the Normalized expression of the MUT patient from the mean normalized WT expression
```{r}
SJCounts$Difference[SJCounts$GROUP == "MUT"]
```

```{r echo=FALSE}
ggplot(SJCounts, aes(sample_id,Difference)) + geom_point(aes(colour=factor(GROUP))) +
  ylab("Differential Expression") +
  labs(color = "GROUP",
       title = "Differential Expression Distribution")
```


```{r echo=FALSE}
ggplot(SJCounts, aes(y=Difference, x=factor(GROUP, level=c("WT", "MUT")))) + 
  geom_boxplot() +
  ylab("Differential Expression") +
  xlab("GROUP")
```

#### ECDF

```{r}
v_ecdf <- ecdf(SJCounts$Difference[SJCounts$GROUP == "WT"])
v_ecdf
```

```{r}
plot(ecdf(SJCounts$Difference[SJCounts$GROUP == "WT"]))
```

```{r}
v_ecdf(SJCounts$Difference[SJCounts$GROUP == "MUT"])
```

```{r}
MUT_df <- SJCounts[SJCounts$GROUP == "MUT",c("sample_id","case_id", "Normalized_CanonEx4_5")]
colnames(MUT_df) <- c("sample_id", "case_id", "NormalizedExpression")

MUT_df$ECDF <- v_ecdf(SJCounts$Difference[SJCounts$GROUP == "MUT"])
MUT_df$Pvalue <- NA

MUT_df$Prediction <- "Canonical Loss"
MUT_df$splice_junction_status <- "CanonicalSJ"
MUT_df$splice_junction_position <- "chr21:43095537-43095693"

MUT_df$ECDF <- ifelse(MUT_df$NormalizedExpression == 0, NA,MUT_df$ECDF)


```

```{r echo=FALSE}
vaf_select <- annot_vaf[,c("Gene","MutationKey_Hg38","sample_id","RNA_Sample","DNA_Sample","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf", "VAF_RNA")]
MUT_vaf <- merge(MUT_df, vaf_select, by.x="sample_id", by.y = "RNA_Sample")
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id"] <- "RNA_Sample"
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id.y"] <- "sample_id"

col_order <- c("Gene", "MutationKey_Hg38", "Prediction","splice_junction_status","splice_junction_position","sample_id", "case_id","RNA_Sample","DNA_Sample","NormalizedExpression","ECDF", "Pvalue", "VAF_RNA","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf")
MUT_vaf <- MUT_vaf[, col_order]
```

Download the vaf, inferred percentiles and pvalues of the mutated samples:
```{r echo=FALSE}
datatable(MUT_vaf,
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp', 
              scrollX = TRUE
))
```
`

#### T-test
Normality:
```{r}
shapiro.test(SJCounts$Normalized_CanonEx4_5[SJCounts$GROUP == "WT"])
```

```{r}
shapiro.test(SJCounts$Normalized_CanonEx4_5[SJCounts$GROUP == "MUT"])
```

Variances:
```{r}
res.ftest <- var.test(Normalized_CanonEx4_5 ~ GROUP, SJCounts, alternative = "two.sided", conf.level=0.95)
res.ftest
```

Two Sample t-test:
```{r}
t <- t.test(Normalized_CanonEx4_5 ~ GROUP, data=SJCounts,
 alternative="two.sided",
 var.equal=TRUE)
t
```

```{r echo=FALSE}

stat_df <- data.frame(matrix(ncol = 17, nrow = 1))
colnames(stat_df) <- c("Gene","MutationKey_Hg38","Prediction","splice_junction_status","splice_junction_position", "sample_id","case_id","RNA_Sample","DNA_Sample","NormalizedExpression","ECDF","Pvalue","VAF_RNA","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf")
stat_df$Gene <- unique(MUT_vaf$Gene)
stat_df$MutationKey_Hg38 <- unique(MUT_vaf$MutationKey_Hg38)
stat_df$sample_id <- paste(MUT_vaf$sample_id, collapse = ';\n') 
stat_df$case_id <- paste(MUT_vaf$case_id, collapse = ';\n') 
stat_df$RNA_Sample <- paste(MUT_vaf$RNA_Sample, collapse = ';\n') 
stat_df$DNA_Sample <- paste(MUT_vaf$DNA_Sample, collapse = ';\n')
stat_df$NormalizedExpression <- paste0(MUT_vaf$NormalizedExpression, collapse= ";\n")
stat_df$ECDF <- paste0(MUT_vaf$ECDF, collapse = ";\n")
stat_df$VAF_RNA <-  paste(MUT_vaf$VAF_RNA, collapse = ';\n')
stat_df$MuTect2_vaf <-  paste(MUT_vaf$MuTect2_vaf, collapse = ';\n')
stat_df$VarScan2_vaf <-  paste(MUT_vaf$VarScan2_vaf, collapse = ';\n')
stat_df$SomaticSniper_vaf <-  paste(MUT_vaf$SomaticSniper_vaf, collapse = ';\n')
stat_df$MuSE_vaf <-  paste(MUT_vaf$MuSE_vaf, collapse = ';\n')
stat_df$Prediction <- "Canonical Loss"
stat_df$splice_junction_status <- "CanonicalSJ"
stat_df$splice_junction_position <- "chr21:43095537-43095693"

stat_df$Pvalue <- t$p.value
```

```{r echo=FALSE}
summary_res<- rbind(summary_res,stat_df)
```

### Canonical SJ Intron 5-6 {.tabset}
#### Expression Difference 
Normality Test:
```{r}
shapiro.test(SJCounts$Normalized_CanonEx5_6[SJCounts$GROUP == "WT"])
```

Value of Mean Normalized Expression of the Alternative SJ in WT samples:
```{r}
mean_WT_SJi <- mean(SJCounts$Normalized_CanonEx5_6[SJCounts$GROUP == "WT"], na.rm=TRUE)
mean_WT_SJi
```

Normalized Expression Value of the Alternative SJ in the MUT sample:
```{r}
MUT_SJi <- SJCounts$Normalized_CanonEx5_6[SJCounts$GROUP == "MUT"]
MUT_SJi
```

Deviation from the mean normalized expression:
```{r}
SJCounts$Difference <-  SJCounts$Normalized_CanonEx5_6 - mean_WT_SJi
```

```{r echo=FALSE}
table <- SJCounts[SJCounts$GROUP == "WT", c("sample_id", "Normalized_CanonEx5_6", "Difference")]
table <- table[order(table$Difference, decreasing = TRUE),]
```

```{r echo=FALSE}
datatable(table,
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp'
))
```

Difference in the MUT sample: deviation of the Normalized expression of the MUT patient from the mean normalized WT expression
```{r}
SJCounts$Difference[SJCounts$GROUP == "MUT"]
```

```{r echo=FALSE}
ggplot(SJCounts, aes(sample_id,Difference)) + geom_point(aes(colour=factor(GROUP))) +
  ylab("Differential Expression") +
  labs(color = "GROUP",
       title = "Differential Expression Distribution")
```


```{r echo=FALSE}
ggplot(SJCounts, aes(y=Difference, x=factor(GROUP, level=c("WT", "MUT")))) + 
  geom_boxplot() +
  ylab("Differential Expression") +
  xlab("GROUP")
```

#### ECDF

```{r}
v_ecdf <- ecdf(SJCounts$Difference[SJCounts$GROUP == "WT"])
v_ecdf
```

```{r}
plot(ecdf(SJCounts$Difference[SJCounts$GROUP == "WT"]))
```

```{r}
v_ecdf(SJCounts$Difference[SJCounts$GROUP == "MUT"])
```

```{r}
MUT_df <- SJCounts[SJCounts$GROUP == "MUT",c("sample_id","case_id", "Normalized_CanonEx5_6")]
colnames(MUT_df) <- c("sample_id", "case_id", "NormalizedExpression")

MUT_df$ECDF <- v_ecdf(SJCounts$Difference[SJCounts$GROUP == "MUT"])
MUT_df$Pvalue <- NA

MUT_df$Prediction <- "Canonical Loss"
MUT_df$splice_junction_status <- "CanonicalSJ"
MUT_df$splice_junction_position <- "chr21:43094789-43095437"

MUT_df$ECDF <- ifelse(MUT_df$NormalizedExpression == 0, NA,MUT_df$ECDF)
```

```{r echo=FALSE}
vaf_select <- annot_vaf[,c("Gene","MutationKey_Hg38","sample_id","RNA_Sample","DNA_Sample","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf", "VAF_RNA")]
MUT_vaf <- merge(MUT_df, vaf_select, by.x="sample_id", by.y = "RNA_Sample")
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id"] <- "RNA_Sample"
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id.y"] <- "sample_id"

col_order <- c("Gene", "MutationKey_Hg38", "Prediction","splice_junction_status","splice_junction_position","sample_id", "case_id","RNA_Sample","DNA_Sample","NormalizedExpression","ECDF", "Pvalue", "VAF_RNA","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf")
MUT_vaf <- MUT_vaf[, col_order]
```

Download the vaf, inferred percentiles and pvalues of the mutated samples:
```{r echo=FALSE}
datatable(MUT_vaf,
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp',
              scrollX=TRUE
))
```

#### T-test
Normality:
```{r}
shapiro.test(SJCounts$Normalized_CanonEx5_6[SJCounts$GROUP == "WT"])
```

```{r}
shapiro.test(SJCounts$Normalized_CanonEx5_6[SJCounts$GROUP == "MUT"])
```

Variances:
```{r}
res.ftest <- var.test(Normalized_CanonEx5_6 ~ GROUP, SJCounts, alternative = "two.sided", conf.level=0.95)
res.ftest
```

Two Sample t-test:
```{r}
t <- t.test(Normalized_CanonEx5_6 ~ GROUP, data=SJCounts,
 alternative="two.sided",
 var.equal=TRUE)
t
```

```{r echo=FALSE}

stat_df <- data.frame(matrix(ncol = 17, nrow = 1))
colnames(stat_df) <- c("Gene","MutationKey_Hg38","Prediction","splice_junction_status","splice_junction_position", "sample_id","case_id","RNA_Sample","DNA_Sample","NormalizedExpression","ECDF","Pvalue","VAF_RNA","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf")
stat_df$Gene <- unique(MUT_vaf$Gene)
stat_df$MutationKey_Hg38 <- unique(MUT_vaf$MutationKey_Hg38)
stat_df$sample_id <- paste(MUT_vaf$sample_id, collapse = ';\n') 
stat_df$case_id <- paste(MUT_vaf$case_id, collapse = ';\n') 
stat_df$RNA_Sample <- paste(MUT_vaf$RNA_Sample, collapse = ';\n') 
stat_df$DNA_Sample <- paste(MUT_vaf$DNA_Sample, collapse = ';\n')
stat_df$NormalizedExpression <- paste0(MUT_vaf$NormalizedExpression, collapse= ";\n")
stat_df$ECDF <- paste0(MUT_vaf$ECDF, collapse = ";\n")
stat_df$VAF_RNA <-  paste(MUT_vaf$VAF_RNA, collapse = ';\n')
stat_df$MuTect2_vaf <-  paste(MUT_vaf$MuTect2_vaf, collapse = ';\n')
stat_df$VarScan2_vaf <-  paste(MUT_vaf$VarScan2_vaf, collapse = ';\n')
stat_df$SomaticSniper_vaf <-  paste(MUT_vaf$SomaticSniper_vaf, collapse = ';\n')
stat_df$MuSE_vaf <-  paste(MUT_vaf$MuSE_vaf, collapse = ';\n')
stat_df$Prediction <- "Canonical Loss"
stat_df$splice_junction_status <- "CanonicalSJ"
stat_df$splice_junction_position <- "chr21:43064789-43095437"

stat_df$Pvalue <- t$p.value
```

```{r echo=FALSE}
summary_res<- rbind(summary_res,stat_df)
```



```{r echo=FALSE}
## Add not found effects information:
MUT_df <- MUT_df[,c("sample_id", "case_id")]
MUT_df$NormalizedExpression <- NA
MUT_df$Prediction <- "Exon Skipping"
MUT_df$splice_junction_status <- "AlternativeSJ not found in SJ collection"
MUT_df$splice_junction_position <- "chr21:43094564-43095437"
MUT_df$Pvalue <- NA

vaf_select <- annot_vaf[,c("Gene","MutationKey_Hg38","sample_id","RNA_Sample","DNA_Sample","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf", "VAF_RNA")]
MUT_vaf <- merge(MUT_df, vaf_select, by.x="sample_id", by.y = "RNA_Sample")
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id"] <- "RNA_Sample"
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id.y"] <- "sample_id"

col_order <- c("Gene", "MutationKey_Hg38", "Prediction","splice_junction_status","splice_junction_position","sample_id", "case_id","RNA_Sample","DNA_Sample","NormalizedExpression", "Pvalue", "VAF_RNA","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf")
MUT_vaf <- MUT_vaf[, col_order]

stat_df <- data.frame(matrix(ncol = 17, nrow = 1))
colnames(stat_df) <- c("Gene","MutationKey_Hg38","Prediction","splice_junction_status","splice_junction_position", "sample_id","case_id","RNA_Sample","DNA_Sample","NormalizedExpression","ECDF","Pvalue","VAF_RNA","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf")

stat_df$Gene <- unique(MUT_vaf$Gene)
stat_df$MutationKey_Hg38 <- unique(MUT_vaf$MutationKey_Hg38)
stat_df$sample_id <- paste(MUT_vaf$sample_id, collapse = ';\n') 
stat_df$case_id <- paste(MUT_vaf$case_id, collapse = ';\n') 
stat_df$RNA_Sample <- paste(MUT_vaf$RNA_Sample, collapse = ';\n') 
stat_df$DNA_Sample <- paste(MUT_vaf$DNA_Sample, collapse = ';\n')
stat_df$NormalizedExpression <- NA
stat_df$ECDF <- NA
stat_df$VAF_RNA <-  paste(MUT_vaf$VAF_RNA, collapse = ';\n')
stat_df$MuTect2_vaf <-  paste(MUT_vaf$MuTect2_vaf, collapse = ';\n')
stat_df$VarScan2_vaf <-  paste(MUT_vaf$VarScan2_vaf, collapse = ';\n')
stat_df$SomaticSniper_vaf <-  paste(MUT_vaf$SomaticSniper_vaf, collapse = ';\n')
stat_df$MuSE_vaf <-  paste(MUT_vaf$MuSE_vaf, collapse = ';\n')
stat_df$Prediction <- unique(MUT_vaf$Prediction)
stat_df$splice_junction_status <- unique(MUT_vaf$splice_junction_status)
stat_df$splice_junction_position <- unique(MUT_vaf$splice_junction_position)

## Save
summary_res <- rbind(summary_res, stat_df)
```

```{r echo=FALSE}
rm(GeneSJ, normalization_df, GeneSJ_subset, annot_vaf, plot_df, samples_df,table, plot_boxplot, plot_dotplot, vaf_select, SJCounts, MUT_df, MUT_vaf)
```


# WT1 chr11,32396364,G,T {.tabset}
Variant found in **3 patients** of the BeatAML (**3 samples**)

* Patients with WT1 chr11,32396364,G,T variant: **3 patients (3 samples)**
* Patients with the variant and RNASeq for validation: **2 patients (2 samples)**

The splicing alterations being assessed are: 

* **Aceptor Loss**: chr11:32396408-32399947; aceptor splice site chr11:32396408.
* **Donor Gain**: predicted at 3bp from the variant position, chr11,32396366, not found in the splice junction collection.
* **Exon Skipping**: chr11:32392756-32399947, found in the mutated samples.

Variant information:
```{r echo=FALSE}
datatable(to_be_validated[to_be_validated$MutationKey_Hg38 == "chr11,32396364,G,T",],
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp', 
              scrollX = TRUE,  
              scrollY = TRUE,
              columnDefs = list(list(targets = '_all', className = 'dt-center')))
)
```

Load the extracted splice junctions of the gene harboring the mutation.
```{r}
extractedSJ_path <- paste0(extractedSJ_dir_in,"WT1_UM_annotSJ.tsv")
GeneSJ <- read.delim(extractedSJ_path, sep ="\t")
```

Set the sample's group: Mutated (MUT) or No Mutated (WT)
```{r}
samples_df <- found_variants[found_variants$Gene=="WT1" & found_variants$MutationKey_Hg38 == "chr11,32396364,G,T",]

cases <- samples_df$RNA_Sample[samples_df$Validable == "Validable"]
GeneSJ$GROUP <- ifelse(GeneSJ$sample_id %in% cases , "MUT", "WT")
```


## SJ Lookup {.tabset}

Search for the splice junctions of interest in the extracted splice junctions of the gene by position (chr_SJstart_SJend).

### Donor Gain 
Search: predicted at 3bp from the variant position 32396364, chr11,32396366

Show all the splice junctions containing positions between 32396360 - 32396369
```{r}
colnames(GeneSJ)[grep("3239636",colnames(GeneSJ))]
```
Alternative SJ not found in the splice junction collection.

### Exon Skipping 
Search: chr11:32392756-32399947

Show all the splice junctions containing the position 32392756-32399947
```{r}
colnames(GeneSJ)[grep("32392756_32399947",colnames(GeneSJ))]
```
Found: chr11_32392756_32399947

Reads of all the AML samples (mutated and no mutated) for the splice junction:
```{r}
GeneSJ$chr11_32392756_32399947
```

Samples with the SJ of interest:
```{r}
table(GeneSJ$chr11_32392756_32399947>0) 
```

Groups of the samples having the alternative splice junction:
```{r}
table(GeneSJ$GROUP[GeneSJ$chr11_32392756_32399947 > 0])
```

Alternative SJ found in the mutated samples.

### Canonical SJ
Exon 7-6: chr11:32396408-32399947; aceptor splice site: chr11:32396408

Exon 8-7: chr11:32392756-32396256

Reads of all the AML samples (mutated and no mutated) for the splice junction:
```{r}
GeneSJ$chr11_32396408_32399947
```

Reads of all the AML samples (mutated and no mutated) for the splice junction:
```{r}
GeneSJ$chr11_32392756_32396256
```


## Normalization
Count the reads of all the splice junctions of the gene harboring the variant:
```{r}
GeneSJ$rowSum_SJtotal <- rowSums(GeneSJ[,grep("chr", names(GeneSJ))])
```

Normalization of the expression by the total read counts of all the splice junctions of the gene:
```{r}
GeneSJ$Normalized_CanonEx6_7 <- (GeneSJ$chr11_32396408_32399947)/GeneSJ$rowSum_SJtotal*100
GeneSJ$Normalized_CanonEx7_8 <- (GeneSJ$chr11_32392756_32396256)/GeneSJ$rowSum_SJtotal*100
GeneSJ$Normalized_ES7 <- (GeneSJ$chr11_32392756_32399947)/GeneSJ$rowSum_SJtotal*100
```

Download the normalized values for the assessed splice junctions of all the AML samples:
```{r echo=FALSE}
normalization_df <- GeneSJ[,c("sample_id","case_id", "GROUP","Normalized_CanonEx6_7", "Normalized_CanonEx7_8", "Normalized_ES7", "chr11_32396408_32399947", "chr11_32392756_32396256","chr11_32392756_32399947", "rowSum_SJtotal","file_id.BAM","file_name.BAM","file_id.STAR.SJCounts","file_name.STAR.SJCounts" )]

colnames(normalization_df)[grep("chr11_32396408_32399947", names(normalization_df))] <- "chr11_32396408_32399947_CanonEx6_7"
colnames(normalization_df)[grep("chr11_32392756_32396256", names(normalization_df))] <- "chr11_32392756_32396256_CanonEx7_8"
colnames(normalization_df)[grep("chr11_32392756_32399947", names(normalization_df))] <- "chr11_32392756_32399947_ES7"

write.xlsx(normalization_df, paste0(normalizedEx_dir_out, "BeatAML.WT1.chr11-32396364-G-T.xlsx"), rowNames = FALSE)
```

```{r echo=FALSE}
datatable(normalization_df,
extensions = 'Buttons',
rownames = FALSE, 
filter = 'top',
options = list(paging = TRUE,
              pageLength = 4,
              autoWidth = FALSE,
              dom = 'lifrtBp',
              buttons = c('excel'), 
              scrollX = TRUE,  
              scrollY = TRUE,
              columnDefs = list(list(targets = '_all', className = 'dt-center'))
))
```

## VAF
```{r echo=FALSE}
GeneSJ_subset <- GeneSJ[GeneSJ$sample_id %in% cases,]
GeneSJ_subset$Gene <- "WT1"
GeneSJ_subset$MutationKey_Hg38 <- "chr11,32396364,G,T"

annot_vaf <- merge(found_variants[,c("MutationKey_Hg38","Gene","sample_id","case_id",                                               "RNA_Sample","DNA_Sample","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf",                         "MuTect2_filter","VarScan2_filter","SomaticSniper_filter","MuSE_filter","MUTReads","WTReads","VAF")], 
                   GeneSJ_subset[,c("Gene","sample_id",  "MutationKey_Hg38","case_id", "Normalized_CanonEx6_7", "Normalized_CanonEx7_8", "Normalized_ES7")], 
                   by.x=c("RNA_Sample", "case_id","Gene", "MutationKey_Hg38"),
                   by.y=c("sample_id", "case_id","Gene", "MutationKey_Hg38"))


annot_vaf <- annot_vaf %>% select(Gene,MutationKey_Hg38,
                                  sample_id,
                                  case_id, RNA_Sample,DNA_Sample,
                                  MuTect2_vaf,VarScan2_vaf,SomaticSniper_vaf,MuSE_vaf,
                                  VAF,
                                  Normalized_CanonEx6_7, Normalized_CanonEx7_8, Normalized_ES7,
                                  everything()
                                  )

colnames(annot_vaf)[grep("VAF", names(annot_vaf))] <- "VAF_RNA"
colnames(annot_vaf)[grep("MUTReads", names(annot_vaf))] <- "MUTReads_RNA"
colnames(annot_vaf)[grep("WTReads", names(annot_vaf))] <- "WTReads_RNA"

write.xlsx(annot_vaf, paste0(normalizedEx_dir_out, "BeatAML.WT1.chr11-32396364-G-T.MUT.xlsx"), rowNames = FALSE)
```

Mutated samples vaf:
```{r echo=FALSE}
datatable(annot_vaf,
extensions = 'Buttons',
rownames = FALSE, 
filter = 'top',
options = list(paging = TRUE,
              pageLength = 4,
              autoWidth = TRUE,
              dom = 'lifrtBp',
              buttons = c('excel'), 
              scrollX = TRUE,  
              scrollY = TRUE
              )
)
```

## Plots {.tabset}

```{r echo=FALSE}
GeneSJ$GROUP <- ifelse(GeneSJ$GROUP == "MUT", "MUT", "WT")
GeneSJ$GROUP <- as.factor(GeneSJ$GROUP)
```

### Static Dot Plots 
Canonical SJ:

```{r echo=FALSE}

ggplot(GeneSJ, aes(sample_id,Normalized_CanonEx6_7,color=GROUP)) + 
  geom_point(size=.8) +
  ggtitle("Normalized expression of WT1 Exon 6-7 Splice Junction") +
  labs(color='GROUP')
```

Splicing alterations:

```{r echo=FALSE}
ggplot(GeneSJ, aes(sample_id,Normalized_ES7,color=GROUP)) +
  geom_point(size=.8) +
  ggtitle("Normalized expression of WT1 Exon 7 Skippping") +
  labs(color='GROUP') 
```


### Interactive Dot Plots

```{r echo=FALSE}
plot_df <- merge(GeneSJ,annot_vaf[c("RNA_Sample", "VAF_RNA", 
                                    "MuTect2_vaf", "VarScan2_vaf", "SomaticSniper_vaf", "MuSE_vaf")],
                 by.x="sample_id", by.y="RNA_Sample", all.x = TRUE)
```


Canonical splice junction: Exon 6-7: chr11:32396408-32399947; aceptor splice site: chr11:32396408
```{r echo=FALSE}
ggplotly(ggplot(plot_df, aes(sample_id,Normalized_CanonEx6_7, color=GROUP)) + 
  geom_point(aes(text=paste0("VAF RNA: ",VAF_RNA,"\nVAF MuTect: ",MuTect2_vaf,
                             "\nVAF VarScan2: ",VarScan2_vaf,
                             "\nVAF SomaticSniper: ",SomaticSniper_vaf,
                             "\nVAF MuSE: ",MuSE_vaf)),size= 0.8)  +
  ggtitle("Normalized expression of WT1 Exon 6-7 Splice Junction") +
  labs(color='GROUP') 
)
```

Splicing alteration:
```{r echo=FALSE}
ggplotly(ggplot(plot_df, aes(sample_id,Normalized_ES7,color=GROUP)) + 
  geom_point(aes(text=paste0("VAF RNA: ",VAF_RNA,"\nVAF MuTect: ",MuTect2_vaf,
                             "\nVAF VarScan2: ",VarScan2_vaf,
                             "\nVAF SomaticSniper: ",SomaticSniper_vaf,
                             "\nVAF MuSE: ",MuSE_vaf)),size= 0.8) +
  ggtitle("Normalized expression of WT1 Exon 7 Skippping") +
  labs(color='GROUP') 
)
```


### Violin Plots
Violin Plots for the alternative splice junctions interrogated:
```{r echo=FALSE}
plot_boxplot <- plot_df[plot_df$GROUP %in% c("WT"),]
plot_dotplot <- plot_df[plot_df$GROUP %in% c("MUT"),]

GEN = "WT1"
SJ = "chr11:32396408-32399947"

ggplot(plot_df, aes(y=Normalized_CanonEx6_7, x=factor(GROUP, level=c("WT", "MUT")))) + 
       geom_violin(scale="area") +
       geom_point(data = plot_dotplot, 
                  aes(y=Normalized_CanonEx6_7, x=GROUP),
                  color="black",
                  size=2
                  ) +
        
       ylab("Normalized SJ expression") +
       xlab(paste(GEN, SJ))
```


```{r echo=FALSE}
plot_boxplot <- plot_df[plot_df$GROUP %in% c("WT"),]
plot_dotplot <- plot_df[plot_df$GROUP %in% c("MUT"),]

GEN = "WT1"
SJ = "chr11:32392756-32399947"

ggplot(plot_df, aes(y=Normalized_ES7, x=factor(GROUP, level=c("WT", "MUT")))) + 
       geom_violin(scale="area") +
       geom_point(data = plot_dotplot, 
                  aes(y=Normalized_ES7, x=GROUP),
                  color="black",
                  size=2
                  ) +
        
       ylab("Normalized SJ expression") +
       xlab(paste(GEN, SJ))
```

## Statistical Analysis {.tabset}

```{r}
SJCounts <- GeneSJ
```

### Aceptor Loss {.tabset}

#### Expression Difference 

Normality Test:
```{r}
shapiro.test(SJCounts$Normalized_CanonEx6_7[SJCounts$GROUP == "WT"])
```

Value of Mean Normalized Expression of the Alternative SJ in WT samples:
```{r}
mean_WT_SJi <- mean(SJCounts$Normalized_CanonEx6_7[SJCounts$GROUP == "WT"], na.rm=TRUE)
mean_WT_SJi
```

Normalized Expression Value of the Alternative SJ in the MUT sample:
```{r}
MUT_SJi <- SJCounts$Normalized_CanonEx6_7[SJCounts$GROUP == "MUT"]
MUT_SJi
```
Deviation from the mean normalized expression:
```{r}
SJCounts$Difference <-  SJCounts$Normalized_CanonEx6_7 - mean_WT_SJi
```

```{r echo=FALSE}
table <- SJCounts[SJCounts$GROUP == "WT", c("sample_id", "Normalized_CanonEx6_7", "Difference")]
table <- table[order(table$Difference, decreasing = TRUE),]
```

```{r echo=FALSE}
datatable(table,
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp'
))
```

Difference in the MUT sample: deviation of the Normalized expression of the MUT patient from the mean normalized WT expression
```{r}
SJCounts$Difference[SJCounts$GROUP == "MUT"]
```

```{r echo=FALSE}
ggplot(SJCounts, aes(sample_id,Difference)) + geom_point(aes(colour=factor(GROUP))) +
  ylab("Differential Expression") +
  labs(color = "GROUP",
       title = "Differential Expression Distribution")
```


```{r echo=FALSE}
ggplot(SJCounts, aes(y=Difference, x=factor(GROUP, level=c("WT", "MUT")))) + 
  geom_boxplot() +
  ylab("Differential Expression") +
  xlab("GROUP")
```

#### ECDF - Pvalue Inference 

```{r}
v_ecdf <- ecdf(SJCounts$Difference[SJCounts$GROUP == "WT"])
v_ecdf
```

```{r}
plot(ecdf(SJCounts$Difference[SJCounts$GROUP == "WT"]))
```

```{r}
v_ecdf(SJCounts$Difference[SJCounts$GROUP == "MUT"])
```


```{r}
MUT_df <- SJCounts[SJCounts$GROUP == "MUT",c("sample_id","case_id", "Normalized_CanonEx6_7")]
colnames(MUT_df) <- c("sample_id", "case_id", "NormalizedExpression")

MUT_df$ECDF <- v_ecdf(SJCounts$Difference[SJCounts$GROUP == "MUT"])
MUT_df$Pvalue <- MUT_df$ECDF

MUT_df$Prediction <- "Aceptor Loss"
MUT_df$splice_junction_status <- "CanonicalSJ"
MUT_df$splice_junction_position <- "chr11:32396408-32399947"

MUT_df$ECDF <- ifelse(MUT_df$NormalizedExpression == 0, NA,MUT_df$ECDF)
MUT_df$Pvalue <- ifelse(MUT_df$NormalizedExpression == 0, NA,MUT_df$Pvalue)
```

```{r echo=FALSE}
vaf_select <- annot_vaf[,c("Gene","MutationKey_Hg38","sample_id","RNA_Sample","DNA_Sample","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf", "VAF_RNA")]
MUT_vaf <- merge(MUT_df, vaf_select, by.x="sample_id", by.y = "RNA_Sample")
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id"] <- "RNA_Sample"
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id.y"] <- "sample_id"

col_order <- c("Gene", "MutationKey_Hg38", "Prediction","splice_junction_status","splice_junction_position","sample_id", "case_id","RNA_Sample","DNA_Sample","NormalizedExpression","ECDF", "Pvalue", "VAF_RNA","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf")
MUT_vaf <- MUT_vaf[, col_order]
```

Download the vaf, inferred percentiles and pvalues of the mutated samples:
```{r echo=FALSE}
datatable(MUT_vaf,
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp',
              scrollX=TRUE
))
```

```{r echo=FALSE}
summary_res <- rbind(summary_res,MUT_vaf)
```


### Exon Skipping {.tabset}

#### Expression Difference 

Normality Test:
```{r}
shapiro.test(SJCounts$Normalized_ES7[SJCounts$GROUP == "WT"])
```

Value of Mean Normalized Expression of the Alternative SJ in WT samples:
```{r}
mean_WT_SJi <- mean(SJCounts$Normalized_ES7[SJCounts$GROUP == "WT"], na.rm=TRUE)
mean_WT_SJi
```

Normalized Expression Value of the Alternative SJ in the MUT sample:
```{r}
MUT_SJi <- SJCounts$Normalized_ES7[SJCounts$GROUP == "MUT"]
MUT_SJi
```
Deviation from the mean normalized expression:
```{r}
SJCounts$Difference <-  SJCounts$Normalized_ES7 - mean_WT_SJi
```

```{r echo=FALSE}
table <- SJCounts[SJCounts$GROUP == "WT", c("sample_id", "Normalized_ES7", "Difference")]
table <- table[order(table$Difference, decreasing = TRUE),]
```

```{r echo=FALSE}
datatable(table,
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp'
))
```

Difference in the MUT sample: deviation of the Normalized expression of the MUT patient from the mean normalized WT expression
```{r}
SJCounts$Difference[SJCounts$GROUP == "MUT"]
```

```{r echo=FALSE}
ggplot(SJCounts, aes(sample_id,Difference)) + geom_point(aes(colour=factor(GROUP))) +
  ylab("Differential Expression") +
  labs(color = "GROUP",
       title = "Differential Expression Distribution")
```


```{r echo=FALSE}
ggplot(SJCounts, aes(y=Difference, x=factor(GROUP, level=c("WT", "MUT")))) + 
  geom_boxplot() +
  ylab("Differential Expression") +
  xlab("GROUP")
```

#### ECDF - Pvalue Inference 

```{r}
v_ecdf <- ecdf(SJCounts$Difference[SJCounts$GROUP == "WT"])
v_ecdf
```

```{r}
plot(ecdf(SJCounts$Difference[SJCounts$GROUP == "WT"]))
```

```{r}
v_ecdf(SJCounts$Difference[SJCounts$GROUP == "MUT"])
```


```{r}
MUT_df <- SJCounts[SJCounts$GROUP == "MUT",c("sample_id","case_id", "Normalized_ES7")]
colnames(MUT_df) <- c("sample_id", "case_id", "NormalizedExpression")

MUT_df$ECDF <- v_ecdf(SJCounts$Difference[SJCounts$GROUP == "MUT"])
MUT_df$Pvalue <- 1- MUT_df$ECDF

MUT_df$Prediction <- "Exon Skipping"
MUT_df$splice_junction_status <- "AlternativeSJ found in MUT samples"
MUT_df$splice_junction_position <- "chr11:32392756-32399947"

MUT_df$ECDF <- ifelse(MUT_df$NormalizedExpression == 0, NA,MUT_df$ECDF)
MUT_df$Pvalue <- ifelse(MUT_df$NormalizedExpression == 0, NA,MUT_df$Pvalue)
```

```{r echo=FALSE}
vaf_select <- annot_vaf[,c("Gene","MutationKey_Hg38","sample_id","RNA_Sample","DNA_Sample","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf", "VAF_RNA")]
MUT_vaf <- merge(MUT_df, vaf_select, by.x="sample_id", by.y = "RNA_Sample")
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id"] <- "RNA_Sample"
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id.y"] <- "sample_id"

col_order <- c("Gene", "MutationKey_Hg38", "Prediction","splice_junction_status","splice_junction_position","sample_id", "case_id","RNA_Sample","DNA_Sample","NormalizedExpression","ECDF", "Pvalue", "VAF_RNA","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf")
MUT_vaf <- MUT_vaf[, col_order]
```

Download the vaf, inferred percentiles and pvalues of the mutated samples:
```{r echo=FALSE}
datatable(MUT_vaf,
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp',
              scrollX = TRUE,
              scrollY =TRUE
))
```

```{r echo=FALSE}
summary_res <- rbind(summary_res,MUT_vaf)
```



```{r echo=FALSE}
## Add not found effects information:
MUT_df <- MUT_df[,c("sample_id", "case_id")]
MUT_df$NormalizedExpression <- NA
MUT_df$Prediction <- "Donor Gain"
MUT_df$splice_junction_status <- "AlternativeSJ not found in SJ collection"
MUT_df$splice_junction_position <- "chr11:32396366"
MUT_df$Pvalue <- NA
MUT_df$ECDF <- NA

vaf_select <- annot_vaf[,c("Gene","MutationKey_Hg38","sample_id","RNA_Sample","DNA_Sample","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf", "VAF_RNA")]
MUT_vaf <- merge(MUT_df, vaf_select, by.x="sample_id", by.y = "RNA_Sample")
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id"] <- "RNA_Sample"
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id.y"] <- "sample_id"

col_order <- c("Gene", "MutationKey_Hg38", "Prediction","splice_junction_status","splice_junction_position","sample_id", "case_id","RNA_Sample","DNA_Sample","NormalizedExpression","ECDF", "Pvalue", "VAF_RNA","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf")
MUT_vaf <- MUT_vaf[, col_order]

## Save
summary_res <- rbind(summary_res, MUT_vaf)
```

```{r echo=FALSE}
rm(GeneSJ, normalization_df, GeneSJ_subset, annot_vaf, plot_df, samples_df,table, plot_boxplot, plot_dotplot, vaf_select, SJCounts, MUT_df, MUT_vaf)
```


# WT1 chr11,32392696,G,A {.tabset}
Variant found in **1 patient** of the BeatAML (**1 sample**)

* Patients with WT1 chr11,32392696,G,A variant: **1 patient (1 sample)**
* Patients with the variant and RNASeq for validation: **1 patient (1 sample)**

The splicing alterations being assessed are: 

* **Donor Loss**: chr11:32392065-32392665; donor splice site chr11:32392665.
* **Exon Skipping**: chr11:32392065-32396256, found in the mutated samples.

Variant information:
```{r echo=FALSE}
datatable(to_be_validated[to_be_validated$MutationKey_Hg38 == "chr11,32392696,G,A",],
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp', 
              scrollX = TRUE,  
              scrollY = TRUE,
              columnDefs = list(list(targets = '_all', className = 'dt-center')))
)
```

Load the extracted splice junctions of the gene harboring the mutation.
```{r}
extractedSJ_path <- paste0(extractedSJ_dir_in,"WT1_UM_annotSJ.tsv")
GeneSJ <- read.delim(extractedSJ_path, sep ="\t")
```

Set the sample's group: Mutated (MUT) or No Mutated (WT)
```{r}
samples_df <- found_variants[found_variants$Gene=="WT1" & found_variants$MutationKey_Hg38 == "chr11,32392696,G,A",]

cases <- samples_df$RNA_Sample[samples_df$Validable == "Validable"]
GeneSJ$GROUP <- ifelse(GeneSJ$sample_id %in% cases , "MUT", "WT")
```


## SJ Lookup {.tabset}

Search for the splice junctions of interest in the extracted splice junctions of the gene by position (chr_SJstart_SJend).

### Exon Skipping 
Search: chr11:32392065-32396256

Show all the splice junctions containing the position chr11:32392065-32396256
```{r}
colnames(GeneSJ)[grep("32392065_32396256",colnames(GeneSJ))]
```
Found: chr11_32392065_32396256

Reads of all the AML samples (mutated and no mutated) for the splice junction:
```{r}
GeneSJ$chr11_32392065_32396256
```

Samples with the SJ of interest:
```{r}
table(GeneSJ$chr11_32392065_32396256>0) 
```

Groups of the samples having the alternative splice junction:
```{r}
table(GeneSJ$GROUP[GeneSJ$chr11_32392065_32396256 > 0])
```

Alternative SJ found in the mutated samples.

### Canonical SJ
Exon upstream (UE): chr11:32392756-32396256

Exon downstream (DE): chr11:32392065-32392665; donor splice site chr11:32392665

Reads of all the AML samples (mutated and no mutated) for the splice junction:
```{r}
GeneSJ$chr11_32392756_32396256
```

Reads of all the AML samples (mutated and no mutated) for the splice junction:
```{r}
GeneSJ$chr11_32392065_32392665
```


## Normalization
Count the reads of all the splice junctions of the gene harboring the variant:
```{r}
GeneSJ$rowSum_SJtotal <- rowSums(GeneSJ[,grep("chr", names(GeneSJ))])
```

Normalization of the expression by the total read counts of all the splice junctions of the gene:
```{r}
GeneSJ$Normalized_CanonDE <- (GeneSJ$chr11_32392065_32392665)/GeneSJ$rowSum_SJtotal*100
GeneSJ$Normalized_ES <- (GeneSJ$chr11_32392065_32396256)/GeneSJ$rowSum_SJtotal*100
```

Download the normalized values for the assessed splice junctions of all the AML samples:
```{r echo=FALSE}
normalization_df <- GeneSJ[,c("sample_id","case_id", "GROUP","Normalized_CanonDE", "Normalized_ES", "chr11_32392065_32392665",  "chr11_32392065_32396256", "rowSum_SJtotal","file_id.BAM","file_name.BAM","file_id.STAR.SJCounts","file_name.STAR.SJCounts" )]

colnames(normalization_df)[grep("chr11:32392065-32392665", names(normalization_df))] <- "chr11:32392065-32392665_CanonDE"
colnames(normalization_df)[grep("chr11:32392065-32396256", names(normalization_df))] <- "chr11:32392065-32396256_ES"

write.xlsx(normalization_df, paste0(normalizedEx_dir_out, "BeatAML.WT1.chr11-32392696-G-A.xlsx"), rowNames = FALSE)
```

```{r echo=FALSE}
datatable(normalization_df,
extensions = 'Buttons',
rownames = FALSE, 
filter = 'top',
options = list(paging = TRUE,
              pageLength = 4,
              autoWidth = FALSE,
              dom = 'lifrtBp',
              buttons = c('excel'), 
              scrollX = TRUE,  
              scrollY = TRUE,
              columnDefs = list(list(targets = '_all', className = 'dt-center'))
))
```

## VAF
```{r echo=FALSE}
GeneSJ_subset <- GeneSJ[GeneSJ$sample_id %in% cases,]
GeneSJ_subset$Gene <- "WT1"
GeneSJ_subset$MutationKey_Hg38 <- "chr11,32392696,G,A"

annot_vaf <- merge(found_variants[,c("MutationKey_Hg38","Gene","sample_id","case_id",                                               "RNA_Sample","DNA_Sample","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf",                         "MuTect2_filter","VarScan2_filter","SomaticSniper_filter","MuSE_filter","MUTReads","WTReads","VAF")], 
                   GeneSJ_subset[,c("Gene","sample_id", "MutationKey_Hg38","case_id", "Normalized_CanonDE", "Normalized_ES")], 
                   by.x=c("RNA_Sample", "case_id","Gene", "MutationKey_Hg38"), 
                   by.y=c("sample_id", "case_id","Gene", "MutationKey_Hg38"))


annot_vaf <- annot_vaf %>% select(Gene,MutationKey_Hg38,
                                  sample_id,
                                  case_id, RNA_Sample,DNA_Sample,
                                  MuTect2_vaf,VarScan2_vaf,SomaticSniper_vaf,MuSE_vaf,
                                  VAF,
                                  Normalized_CanonDE,Normalized_ES,
                                  everything()
                                  )

colnames(annot_vaf)[grep("VAF", names(annot_vaf))] <- "VAF_RNA"
colnames(annot_vaf)[grep("MUTReads", names(annot_vaf))] <- "MUTReads_RNA"
colnames(annot_vaf)[grep("WTReads", names(annot_vaf))] <- "WTReads_RNA"

write.xlsx(annot_vaf, paste0(normalizedEx_dir_out, "BeatAML.WT1.chr11-32392696-G-A.MUT.xlsx"), rowNames = FALSE)
```

Mutated samples vaf:
```{r echo=FALSE}
datatable(annot_vaf,
extensions = 'Buttons',
rownames = FALSE, 
filter = 'top',
options = list(paging = TRUE,
              pageLength = 4,
              autoWidth = TRUE,
              dom = 'lifrtBp',
              buttons = c('excel'), 
              scrollX = TRUE,  
              scrollY = TRUE
              )
)
```

## Plots {.tabset}

```{r echo=FALSE}
GeneSJ$GROUP <- ifelse(GeneSJ$GROUP == "MUT", "MUT", "WT")
GeneSJ$GROUP <- as.factor(GeneSJ$GROUP)
```

### Static Dot Plots 
Canonical splice junction Exon downstream (DE): chr11:32392065-32392665; donor splice site chr11:32392665
```{r echo=FALSE}

ggplot(GeneSJ, aes(sample_id,Normalized_CanonDE,color=GROUP)) + 
  geom_point(size=.8) +
  labs(color='GROUP', 
       title = "Normalized Expression of WT1 chr11:32392065-32392665", 
       subtitle="Potential Canonical Donor Loss",
       y = "Normalized Expression")
```

Splicing alterations:

```{r echo=FALSE}
ggplot(GeneSJ, aes(sample_id,Normalized_ES,color=GROUP)) +
  geom_point(size=.8) +
  labs(color='GROUP', 
       title = "Normalized Expression of WT1 chr11:32392065-32396256", 
       subtitle="Potential Exon Skipping",
       y = "Normalized Expression")
```


### Interactive Dot Plots

```{r echo=FALSE}
plot_df <- merge(GeneSJ,annot_vaf[c("RNA_Sample", "VAF_RNA", 
                                    "MuTect2_vaf", "VarScan2_vaf", "SomaticSniper_vaf", "MuSE_vaf")],
                 by.x="sample_id", by.y="RNA_Sample", all.x = TRUE)
```


Canonical splice junction Exon downstream (DE): chr11:32392065-32392665, donor splice site: chr11:32392665
```{r echo=FALSE}
ggplotly(ggplot(plot_df, aes(sample_id,Normalized_CanonDE, color=GROUP)) + 
  geom_point(aes(text=paste0("VAF RNA: ",VAF_RNA,"\nVAF MuTect: ",MuTect2_vaf,
                             "\nVAF VarScan2: ",VarScan2_vaf,
                             "\nVAF SomaticSniper: ",SomaticSniper_vaf,
                             "\nVAF MuSE: ",MuSE_vaf)),size= 0.8)  +
  ggtitle("Normalized expression of WT1 Canonical Donor") +
  labs(color='GROUP') 
)
```

Splicing alteration:
```{r echo=FALSE}
ggplotly(ggplot(plot_df, aes(sample_id,Normalized_ES,color=GROUP)) + 
  geom_point(aes(text=paste0("VAF RNA: ",VAF_RNA,"\nVAF MuTect: ",MuTect2_vaf,
                             "\nVAF VarScan2: ",VarScan2_vaf,
                             "\nVAF SomaticSniper: ",SomaticSniper_vaf,
                             "\nVAF MuSE: ",MuSE_vaf)),size= 0.8) +
  ggtitle("Normalized expression of WT1 Exon Skipping") +
  labs(color='GROUP') 
)
```


### Violin Plots
Violin Plots for the alternative splice junctions interrogated:
```{r echo=FALSE}
plot_boxplot <- plot_df[plot_df$GROUP %in% c("WT"),]
plot_dotplot <- plot_df[plot_df$GROUP %in% c("MUT"),]

GEN = "WT1"
SJ = "chr11:32392065-32392665"

ggplot(plot_df, aes(y=Normalized_CanonDE, x=factor(GROUP, level=c("WT", "MUT")))) + 
       geom_violin(scale="area") +
       geom_point(data = plot_dotplot, 
                  aes(y=Normalized_CanonDE, x=GROUP),
                  color="black",
                  size=2
                  ) +
        
       ylab("Normalized SJ expression") +
       xlab(paste(GEN, SJ))
```


```{r echo=FALSE}
plot_boxplot <- plot_df[plot_df$GROUP %in% c("WT"),]
plot_dotplot <- plot_df[plot_df$GROUP %in% c("MUT"),]

GEN = "WT1"
SJ = "chr11:32392065-32396256"

ggplot(plot_df, aes(y=Normalized_ES, x=factor(GROUP, level=c("WT", "MUT")))) + 
       geom_violin(scale="area") +
       geom_point(data = plot_dotplot, 
                  aes(y=Normalized_ES, x=GROUP),
                  color="black",
                  size=2
                  ) +
        
       ylab("Normalized SJ expression") +
       xlab(paste(GEN, SJ))
```

## Statistical Analysis {.tabset}

```{r}
SJCounts <- GeneSJ
```

### Donor Loss {.tabset}

#### Expression Difference 

Normality Test:
```{r}
shapiro.test(SJCounts$Normalized_CanonDE[SJCounts$GROUP == "WT"])
```

Value of Mean Normalized Expression of the Alternative SJ in WT samples:
```{r}
mean_WT_SJi <- mean(SJCounts$Normalized_CanonDE[SJCounts$GROUP == "WT"], na.rm=TRUE)
mean_WT_SJi
```

Normalized Expression Value of the Alternative SJ in the MUT sample:
```{r}
MUT_SJi <- SJCounts$Normalized_CanonDE[SJCounts$GROUP == "MUT"]
MUT_SJi
```
Deviation from the mean normalized expression:
```{r}
SJCounts$Difference <-  SJCounts$Normalized_CanonDE - mean_WT_SJi
```

```{r echo=FALSE}
table <- SJCounts[SJCounts$GROUP == "WT", c("sample_id", "Normalized_CanonDE", "Difference")]
table <- table[order(table$Difference, decreasing = TRUE),]
```

```{r echo=FALSE}
datatable(table,
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp'
))
```

Difference in the MUT sample: deviation of the Normalized expression of the MUT patient from the mean normalized WT expression
```{r}
SJCounts$Difference[SJCounts$GROUP == "MUT"]
```

```{r echo=FALSE}
ggplot(SJCounts, aes(sample_id,Difference)) + geom_point(aes(colour=factor(GROUP))) +
  ylab("Differential Expression") +
  labs(color = "GROUP",
       title = "Differential Expression Distribution")
```


```{r echo=FALSE}
ggplot(SJCounts, aes(y=Difference, x=factor(GROUP, level=c("WT", "MUT")))) + 
  geom_boxplot() +
  ylab("Differential Expression") +
  xlab("GROUP")
```

#### ECDF - Pvalue Inference 

```{r}
v_ecdf <- ecdf(SJCounts$Difference[SJCounts$GROUP == "WT"])
v_ecdf
```

```{r}
plot(ecdf(SJCounts$Difference[SJCounts$GROUP == "WT"]))
```

```{r}
v_ecdf(SJCounts$Difference[SJCounts$GROUP == "MUT"])
```


```{r}
MUT_df <- SJCounts[SJCounts$GROUP == "MUT",c("sample_id","case_id", "Normalized_CanonDE")]
colnames(MUT_df) <- c("sample_id", "case_id", "NormalizedExpression")

MUT_df$ECDF <- v_ecdf(SJCounts$Difference[SJCounts$GROUP == "MUT"])
MUT_df$Pvalue <- MUT_df$ECDF

MUT_df$Prediction <- "Donor Loss"
MUT_df$splice_junction_status <- "CanonicalSJ"
MUT_df$splice_junction_position <- "chr11:32392065-32392665"

MUT_df$ECDF <- ifelse(MUT_df$NormalizedExpression == 0, NA,MUT_df$ECDF)
MUT_df$Pvalue <- ifelse(MUT_df$NormalizedExpression == 0, NA,MUT_df$Pvalue)
```

```{r echo=FALSE}
vaf_select <- annot_vaf[,c("Gene","MutationKey_Hg38","sample_id","RNA_Sample","DNA_Sample","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf", "VAF_RNA")]
MUT_vaf <- merge(MUT_df, vaf_select, by.x="sample_id", by.y = "RNA_Sample")
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id"] <- "RNA_Sample"
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id.y"] <- "sample_id"

col_order <- c("Gene", "MutationKey_Hg38", "Prediction","splice_junction_status","splice_junction_position","sample_id", "case_id","RNA_Sample","DNA_Sample","NormalizedExpression","ECDF", "Pvalue", "VAF_RNA","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf")
MUT_vaf <- MUT_vaf[, col_order]
```

Download the vaf, inferred percentiles and pvalues of the mutated samples:
```{r echo=FALSE}
datatable(MUT_vaf,
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp',
              scrollX=TRUE
))
```

```{r echo=FALSE}
summary_res <- rbind(summary_res,MUT_vaf)
```


### Exon Skipping {.tabset}

#### Expression Difference 

Normality Test:
```{r}
shapiro.test(SJCounts$Normalized_ES[SJCounts$GROUP == "WT"])
```

Value of Mean Normalized Expression of the Alternative SJ in WT samples:
```{r}
mean_WT_SJi <- mean(SJCounts$Normalized_ES[SJCounts$GROUP == "WT"], na.rm=TRUE)
mean_WT_SJi
```

Normalized Expression Value of the Alternative SJ in the MUT sample:
```{r}
MUT_SJi <- SJCounts$Normalized_ES[SJCounts$GROUP == "MUT"]
MUT_SJi
```
Deviation from the mean normalized expression:
```{r}
SJCounts$Difference <-  SJCounts$Normalized_ES - mean_WT_SJi
```

```{r echo=FALSE}
table <- SJCounts[SJCounts$GROUP == "WT", c("sample_id", "Normalized_ES", "Difference")]
table <- table[order(table$Difference, decreasing = TRUE),]
```

```{r echo=FALSE}
datatable(table,
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp'
))
```

Difference in the MUT sample: deviation of the Normalized expression of the MUT patient from the mean normalized WT expression
```{r}
SJCounts$Difference[SJCounts$GROUP == "MUT"]
```

```{r echo=FALSE}
ggplot(SJCounts, aes(sample_id,Difference)) + geom_point(aes(colour=factor(GROUP))) +
  ylab("Differential Expression") +
  labs(color = "GROUP",
       title = "Differential Expression Distribution")
```


```{r echo=FALSE}
ggplot(SJCounts, aes(y=Difference, x=factor(GROUP, level=c("WT", "MUT")))) + 
  geom_boxplot() +
  ylab("Differential Expression") +
  xlab("GROUP")
```

#### ECDF - Pvalue Inference 

```{r}
v_ecdf <- ecdf(SJCounts$Difference[SJCounts$GROUP == "WT"])
v_ecdf
```

```{r}
plot(ecdf(SJCounts$Difference[SJCounts$GROUP == "WT"]))
```

```{r}
v_ecdf(SJCounts$Difference[SJCounts$GROUP == "MUT"])
```


```{r}
MUT_df <- SJCounts[SJCounts$GROUP == "MUT",c("sample_id","case_id", "Normalized_ES")] 
colnames(MUT_df) <- c("sample_id", "case_id", "NormalizedExpression")

MUT_df$ECDF <- v_ecdf(SJCounts$Difference[SJCounts$GROUP == "MUT"])
MUT_df$Pvalue <- 1- MUT_df$ECDF

MUT_df$Prediction <- "Exon Skipping"
MUT_df$splice_junction_status <- "AlternativeSJ found in MUT samples"
MUT_df$splice_junction_position <- "chr11:32392065-32396256"

MUT_df$ECDF <- ifelse(MUT_df$NormalizedExpression == 0, NA,MUT_df$ECDF)
MUT_df$Pvalue <- ifelse(MUT_df$NormalizedExpression == 0, NA,MUT_df$Pvalue)
```

```{r echo=FALSE}
vaf_select <- annot_vaf[,c("Gene","MutationKey_Hg38","sample_id","RNA_Sample","DNA_Sample","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf", "VAF_RNA")]
MUT_vaf <- merge(MUT_df, vaf_select, by.x="sample_id", by.y = "RNA_Sample")
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id"] <- "RNA_Sample"
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id.y"] <- "sample_id"

col_order <- c("Gene", "MutationKey_Hg38", "Prediction","splice_junction_status","splice_junction_position","sample_id", "case_id","RNA_Sample","DNA_Sample","NormalizedExpression","ECDF", "Pvalue", "VAF_RNA","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf")
MUT_vaf <- MUT_vaf[, col_order]
```

Download the vaf, inferred percentiles and pvalues of the mutated samples:
```{r echo=FALSE}
datatable(MUT_vaf,
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp',
              scrollX = TRUE,
              scrollY =TRUE
))
```


```{r echo=FALSE}
summary_res <- rbind(summary_res,MUT_vaf)
```

```{r echo=FALSE}
rm(GeneSJ, normalization_df, GeneSJ_subset, annot_vaf, plot_df, samples_df,table, plot_boxplot, plot_dotplot, vaf_select, SJCounts, MUT_df, MUT_vaf)
```


# TP53 chr17,7675238,T,C {.tabset}
Variant found in **1 patient** of the BeatAML (**1 sample**)

* Patients with TP53 chr17,7675238,T,C variant: **1 patient (1 sample)**
* Patients with the variant and RNASeq for validation: **1 patient (1 sample)**

The splicing alterations being assessed are: 

* **Aceptor Loss**: chr17:7675237-7675993; aceptor splice site chr17:7675237.
* **Aceptor Gain**: chr17:7675216-7675993, found in the mutated samples.
* **Aceptor Gain**: chr17:7675234:7676033, found in the splice junction collection, not found in the mutated samples. 


Variant information:
```{r echo=FALSE}
datatable(to_be_validated[to_be_validated$MutationKey_Hg38 == "chr17,7675238,T,C",],
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp', 
              scrollX = TRUE,  
              scrollY = TRUE,
              columnDefs = list(list(targets = '_all', className = 'dt-center')))
)
```

Load the extracted splice junctions of the gene harboring the mutation.
```{r}
extractedSJ_path <- paste0(extractedSJ_dir_in,"TP53_UM_annotSJ.tsv")
GeneSJ <- read.delim(extractedSJ_path, sep ="\t")
```

Set the sample's group: Mutated (MUT) or No Mutated (WT)
```{r}
samples_df <- found_variants[found_variants$Gene=="TP53" & found_variants$MutationKey_Hg38 == "chr17,7675238,T,C",]

cases <- samples_df$RNA_Sample[samples_df$Validable == "Validable"]
GeneSJ$GROUP <- ifelse(GeneSJ$sample_id %in% cases , "MUT", "WT")
```


## SJ Lookup {.tabset}

Search for the splice junctions of interest in the extracted splice junctions of the gene by position (chr_SJstart_SJend).

### Aceptor Gain
Search: chr17:7675216-7675993

Show all the splice junctions containing the position chr17:7675216-7675993
```{r}
colnames(GeneSJ)[grep("7675216_7675993",colnames(GeneSJ))]
```
Found: chr17_7675216_7675993

Reads of all the AML samples (mutated and no mutated) for the splice junction:
```{r}
GeneSJ$chr17_7675216_7675993
```

Samples with the SJ of interest:
```{r}
table(GeneSJ$chr17_7675216_7675993>0) 
```

Groups of the samples having the alternative splice junction:
```{r}
table(GeneSJ$GROUP[GeneSJ$chr17_7675216_7675993 > 0])
```
Alternative SJ found in the mutated samples.


### Aceptor Gain
Search: chr17:7675234-7676033

Show all the splice junctions containing the positions between 7675230 and 7675239
```{r}
colnames(GeneSJ)[grep("767523",colnames(GeneSJ))]
```
Found: chr17_7675234_7676033

Reads of all the AML samples (mutated and no mutated) for the splice junction:
```{r}
GeneSJ$chr17_7675234_7676033
```

Samples with the SJ of interest:
```{r}
table(GeneSJ$chr17_7675234_7676033>0) 
```

Groups of the samples having the alternative splice junction:
```{r}
table(GeneSJ$GROUP[GeneSJ$chr17_7675234_7676033 > 0])
```

Alternative SJ not found in the mutated samples of the splice junction collection.



### Canonical SJ
Canonical DE: chr17:7674972-7675052

Canonical UE: chr17:7675237-7675993; aceptor splice site chr17:7675237

Reads of all the AML samples (mutated and no mutated) for the splice junction:
```{r}
GeneSJ$chr17_7674972_7675052
```

Reads of all the AML samples (mutated and no mutated) for the splice junction:
```{r}
GeneSJ$chr17_7675237_7675993
```


## Normalization
Count the reads of all the splice junctions of the gene harboring the variant:
```{r}
GeneSJ$rowSum_SJtotal <- rowSums(GeneSJ[,grep("chr", names(GeneSJ))])
```

Normalization of the expression by the total read counts of all the splice junctions of the gene:
```{r}
GeneSJ$Normalized_CanonDE <- (GeneSJ$chr17_7674972_7675052)/GeneSJ$rowSum_SJtotal*100
GeneSJ$Normalized_CanonUE <- (GeneSJ$chr17_7675237_7675993)/GeneSJ$rowSum_SJtotal*100

GeneSJ$Normalized_AG <- (GeneSJ$chr17_7675216_7675993)/GeneSJ$rowSum_SJtotal*100
```

Download the normalized values for the assessed splice junctions of all the AML samples:
```{r echo=FALSE}
normalization_df <- GeneSJ[,c("sample_id","case_id", "GROUP","Normalized_CanonUE", "Normalized_CanonDE","Normalized_AG","chr17_7675237_7675993","chr17_7674972_7675052", "chr17_7675216_7675993", "rowSum_SJtotal","file_id.BAM","file_name.BAM","file_id.STAR.SJCounts","file_name.STAR.SJCounts" )]

colnames(normalization_df)[grep("chr17_7674972_7675052", names(normalization_df))] <- "chr17_7674972_7675052_CanonDE"
colnames(normalization_df)[grep("chr17_7675237_7675993", names(normalization_df))] <- "chr17_7675237_7675993_CanonUE"
colnames(normalization_df)[grep("chr17_7675216_7675993", names(normalization_df))] <- "chr17_7675216_7675993_ExAG"

write.xlsx(normalization_df, paste0(normalizedEx_dir_out, "BeatAML.TP53.chr17-7675238-T-C.xlsx"), rowNames = FALSE)
```

```{r echo=FALSE}
datatable(normalization_df,
extensions = 'Buttons',
rownames = FALSE, 
filter = 'top',
options = list(paging = TRUE,
              pageLength = 4,
              autoWidth = FALSE,
              dom = 'lifrtBp',
              buttons = c('excel'), 
              scrollX = TRUE,  
              scrollY = TRUE,
              columnDefs = list(list(targets = '_all', className = 'dt-center'))
))
```

## VAF
```{r echo=FALSE}
GeneSJ_subset <- GeneSJ[GeneSJ$sample_id %in% cases,]
GeneSJ_subset$Gene <- "TP53"
GeneSJ_subset$MutationKey_Hg38 <- "chr17,7675238,T,C"

annot_vaf <- merge(found_variants[,c("MutationKey_Hg38","Gene","sample_id","case_id",                                               "RNA_Sample","DNA_Sample","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf",                         "MuTect2_filter","VarScan2_filter","SomaticSniper_filter","MuSE_filter","MUTReads","WTReads","VAF")], 
                   GeneSJ_subset[,c("Gene","sample_id", "MutationKey_Hg38","case_id",  "Normalized_CanonUE", "Normalized_CanonDE", "Normalized_AG")], 
                   by.x=c("RNA_Sample", "case_id","Gene", "MutationKey_Hg38"), 
                   by.y=c("sample_id", "case_id","Gene", "MutationKey_Hg38"))


annot_vaf <- annot_vaf %>% select(Gene,MutationKey_Hg38,
                                  sample_id,
                                  case_id, RNA_Sample,DNA_Sample,
                                  MuTect2_vaf,VarScan2_vaf,SomaticSniper_vaf,MuSE_vaf,
                                  VAF,
                                  Normalized_CanonUE, Normalized_CanonDE,
                                  Normalized_AG,
                                  everything()
                                  )

colnames(annot_vaf)[grep("VAF", names(annot_vaf))] <- "VAF_RNA"
colnames(annot_vaf)[grep("MUTReads", names(annot_vaf))] <- "MUTReads_RNA"
colnames(annot_vaf)[grep("WTReads", names(annot_vaf))] <- "WTReads_RNA"

write.xlsx(annot_vaf, paste0(normalizedEx_dir_out, "BeatAML.TP53.chr17-7675238-T-C.MUT.xlsx"), rowNames = FALSE)
```

Mutated samples vaf:
```{r echo=FALSE}
datatable(annot_vaf,
extensions = 'Buttons',
rownames = FALSE, 
filter = 'top',
options = list(paging = TRUE,
              pageLength = 4,
              autoWidth = TRUE,
              dom = 'lifrtBp',
              buttons = c('excel'), 
              scrollX = TRUE,  
              scrollY = TRUE
              )
)
```

## Plots {.tabset}

```{r echo=FALSE}
GeneSJ$GROUP <- ifelse(GeneSJ$GROUP == "MUT", "MUT", "WT")
GeneSJ$GROUP <- as.factor(GeneSJ$GROUP)
```

### Static Dot Plots 
Canonical splice junction chr17:7675237-7675993, aceptor splice site chr17:7675237
```{r echo=FALSE}

ggplot(GeneSJ, aes(sample_id,Normalized_CanonUE,color=GROUP)) + 
  geom_point(size=.8) +
  ggtitle("Normalized expression of TP53 Upstream Exon Splice Junction") +
  labs(color='GROUP')
```

Splicing alterations:

```{r echo=FALSE}
ggplot(GeneSJ, aes(sample_id,Normalized_AG,color=GROUP)) +
  geom_point(size=.8) +
  ggtitle("Normalized expression of TP53 Aceptor Gain") +
  labs(color='GROUP') 
```


### Interactive Dot Plots

```{r echo=FALSE}
plot_df <- merge(GeneSJ,annot_vaf[c("RNA_Sample", "VAF_RNA", 
                                    "MuTect2_vaf", "VarScan2_vaf", "SomaticSniper_vaf", "MuSE_vaf")],
                 by.x="sample_id", by.y="RNA_Sample", all.x = TRUE)
```

Canonical splice junction chr17:7675237-7675993, aceptor splice site chr17:7675237
```{r echo=FALSE}
ggplotly(ggplot(plot_df, aes(sample_id,Normalized_CanonUE, color=GROUP)) + 
  geom_point(aes(text=paste0("VAF RNA: ",VAF_RNA,"\nVAF MuTect: ",MuTect2_vaf,
                             "\nVAF VarScan2: ",VarScan2_vaf,
                             "\nVAF SomaticSniper: ",SomaticSniper_vaf,
                             "\nVAF MuSE: ",MuSE_vaf)),size= 0.8)  +
  ggtitle("Normalized expression of TP53 Upstream Exon Splice Junction") +
  labs(color='GROUP') 
)
```

Splicing alteration:
```{r echo=FALSE}
ggplotly(ggplot(plot_df, aes(sample_id,Normalized_AG,color=GROUP)) + 
  geom_point(aes(text=paste0("VAF RNA: ",VAF_RNA,"\nVAF MuTect: ",MuTect2_vaf,
                             "\nVAF VarScan2: ",VarScan2_vaf,
                             "\nVAF SomaticSniper: ",SomaticSniper_vaf,
                             "\nVAF MuSE: ",MuSE_vaf)),size= 0.8) +
  ggtitle("Normalized expression of TP53 aceptor Gain") +
  labs(color='GROUP') 
)
```


### Violin Plots
Violin Plots for the alternative splice junctions interrogated:
```{r echo=FALSE}
plot_boxplot <- plot_df[plot_df$GROUP %in% c("WT"),]
plot_dotplot <- plot_df[plot_df$GROUP %in% c("MUT"),]

GEN = "TP53"
SJ = "chr17:7675237-7675993"

ggplot(plot_df, aes(y=Normalized_CanonUE, x=factor(GROUP, level=c("WT", "MUT")))) + 
       geom_violin(scale="area") +
       geom_point(data = plot_dotplot, 
                  aes(y=Normalized_CanonUE, x=GROUP),
                  color="black",
                  size=2
                  ) +
        
       ylab("Normalized SJ expression") +
       xlab(paste(GEN, SJ))
```


```{r echo=FALSE}
plot_boxplot <- plot_df[plot_df$GROUP %in% c("WT"),]
plot_dotplot <- plot_df[plot_df$GROUP %in% c("MUT"),]

GEN = "TP53"
SJ = "chr17:7675216-7675993"

ggplot(plot_df, aes(y=Normalized_AG, x=factor(GROUP, level=c("WT", "MUT")))) + 
       geom_violin(scale="area") +
       geom_point(data = plot_dotplot, 
                  aes(y=Normalized_AG, x=GROUP),
                  color="black",
                  size=2
                  ) +
        
       ylab("Normalized SJ expression") +
       xlab(paste(GEN, SJ))
```

## Statistical Analysis {.tabset}

```{r}
SJCounts <- GeneSJ
```

### Aceptor Loss {.tabset}

#### Expression Difference 

Normality Test:
```{r}
shapiro.test(SJCounts$Normalized_CanonUE[SJCounts$GROUP == "WT"])
```

Value of Mean Normalized Expression of the Alternative SJ in WT samples:
```{r}
mean_WT_SJi <- mean(SJCounts$Normalized_CanonUE[SJCounts$GROUP == "WT"], na.rm=TRUE)
mean_WT_SJi
```

Normalized Expression Value of the Alternative SJ in the MUT sample:
```{r}
MUT_SJi <- SJCounts$Normalized_CanonUE[SJCounts$GROUP == "MUT"]
MUT_SJi
```
Deviation from the mean normalized expression:
```{r}
SJCounts$Difference <-  SJCounts$Normalized_CanonUE - mean_WT_SJi
```

```{r echo=FALSE}
table <- SJCounts[SJCounts$GROUP == "WT", c("sample_id", "Normalized_CanonUE", "Difference")]
table <- table[order(table$Difference, decreasing = TRUE),]
```

```{r echo=FALSE}
datatable(table,
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp'
))
```

Difference in the MUT sample: deviation of the Normalized expression of the MUT patient from the mean normalized WT expression
```{r}
SJCounts$Difference[SJCounts$GROUP == "MUT"]
```

```{r echo=FALSE}
ggplot(SJCounts, aes(sample_id,Difference)) + geom_point(aes(colour=factor(GROUP))) +
  ylab("Differential Expression") +
  labs(color = "GROUP",
       title = "Differential Expression Distribution")
```


```{r echo=FALSE}
ggplot(SJCounts, aes(y=Difference, x=factor(GROUP, level=c("WT", "MUT")))) + 
  geom_boxplot() +
  ylab("Differential Expression") +
  xlab("GROUP")
```

#### ECDF - Pvalue Inference 

```{r}
v_ecdf <- ecdf(SJCounts$Difference[SJCounts$GROUP == "WT"])
v_ecdf
```

```{r}
plot(ecdf(SJCounts$Difference[SJCounts$GROUP == "WT"]))
```

```{r}
v_ecdf(SJCounts$Difference[SJCounts$GROUP == "MUT"])
```


```{r}
MUT_df <- SJCounts[SJCounts$GROUP == "MUT",c("sample_id","case_id", "Normalized_CanonUE")] 
colnames(MUT_df) <- c("sample_id", "case_id", "NormalizedExpression")

MUT_df$ECDF <- v_ecdf(SJCounts$Difference[SJCounts$GROUP == "MUT"])
MUT_df$Pvalue <- MUT_df$ECDF

MUT_df$Prediction <- "Aceptor Loss"
MUT_df$splice_junction_status <- "CanonicalSJ"
MUT_df$splice_junction_position <- "chr17:7675237-7675993"

MUT_df$ECDF <- ifelse(MUT_df$NormalizedExpression == 0, NA,MUT_df$ECDF)
MUT_df$Pvalue <- ifelse(MUT_df$NormalizedExpression == 0, NA,MUT_df$Pvalue)
```

```{r echo=FALSE}
vaf_select <- annot_vaf[,c("Gene","MutationKey_Hg38","sample_id","RNA_Sample","DNA_Sample","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf", "VAF_RNA")]
MUT_vaf <- merge(MUT_df, vaf_select, by.x="sample_id", by.y = "RNA_Sample")
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id"] <- "RNA_Sample"
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id.y"] <- "sample_id"

col_order <- c("Gene", "MutationKey_Hg38", "Prediction","splice_junction_status","splice_junction_position","sample_id", "case_id","RNA_Sample","DNA_Sample","NormalizedExpression","ECDF", "Pvalue", "VAF_RNA","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf")
MUT_vaf <- MUT_vaf[, col_order]
```

Download the vaf, inferred percentiles and pvalues of the mutated samples:
```{r echo=FALSE}
datatable(MUT_vaf,
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp',
              scrollX=TRUE
))
```

```{r echo=FALSE}
summary_res <- rbind(summary_res,MUT_vaf)
```


### Aceptor Gain {.tabset}

#### Expression Difference 

Normality Test:
```{r}
shapiro.test(SJCounts$Normalized_AG[SJCounts$GROUP == "WT"])
```

Value of Mean Normalized Expression of the Alternative SJ in WT samples:
```{r}
mean_WT_SJi <- mean(SJCounts$Normalized_AG[SJCounts$GROUP == "WT"], na.rm=TRUE)
mean_WT_SJi
```

Normalized Expression Value of the Alternative SJ in the MUT sample:
```{r}
MUT_SJi <- SJCounts$Normalized_AG[SJCounts$GROUP == "MUT"]
MUT_SJi
```
Deviation from the mean normalized expression:
```{r}
SJCounts$Difference <-  SJCounts$Normalized_AG - mean_WT_SJi
```

```{r echo=FALSE}
table <- SJCounts[SJCounts$GROUP == "WT", c("sample_id", "Normalized_AG", "Difference")]
table <- table[order(table$Difference, decreasing = TRUE),]
```

```{r echo=FALSE}
datatable(table,
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp'
))
```

Difference in the MUT sample: deviation of the Normalized expression of the MUT patient from the mean normalized WT expression
```{r}
SJCounts$Difference[SJCounts$GROUP == "MUT"]
```

```{r echo=FALSE}
ggplot(SJCounts, aes(sample_id,Difference)) + geom_point(aes(colour=factor(GROUP))) +
  ylab("Differential Expression") +
  labs(color = "GROUP",
       title = "Differential Expression Distribution")
```


```{r echo=FALSE}
ggplot(SJCounts, aes(y=Difference, x=factor(GROUP, level=c("WT", "MUT")))) + 
  geom_boxplot() +
  ylab("Differential Expression") +
  xlab("GROUP")
```

#### ECDF - Pvalue Inference 

```{r}
v_ecdf <- ecdf(SJCounts$Difference[SJCounts$GROUP == "WT"])
v_ecdf
```

```{r}
plot(ecdf(SJCounts$Difference[SJCounts$GROUP == "WT"]))
```

```{r}
v_ecdf(SJCounts$Difference[SJCounts$GROUP == "MUT"])
```


```{r}
MUT_df <- SJCounts[SJCounts$GROUP == "MUT",c("sample_id","case_id", "Normalized_AG")] 
colnames(MUT_df) <- c("sample_id", "case_id", "NormalizedExpression")
MUT_df$ECDF <- v_ecdf(SJCounts$Difference[SJCounts$GROUP == "MUT"])
MUT_df$Pvalue <- 1- MUT_df$ECDF

MUT_df$Prediction <- "Aceptor Gain"
MUT_df$splice_junction_status <- "AlternativeSJ found in MUT samples"
MUT_df$splice_junction_position <- "chr17:7675216-7675993"

MUT_df$ECDF <- ifelse(MUT_df$NormalizedExpression == 0, NA,MUT_df$ECDF)
MUT_df$Pvalue <- ifelse(MUT_df$NormalizedExpression == 0, NA,MUT_df$Pvalue)
```

```{r echo=FALSE}
vaf_select <- annot_vaf[,c("Gene","MutationKey_Hg38","sample_id","RNA_Sample","DNA_Sample","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf", "VAF_RNA")]
MUT_vaf <- merge(MUT_df, vaf_select, by.x="sample_id", by.y = "RNA_Sample")
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id"] <- "RNA_Sample"
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id.y"] <- "sample_id"

col_order <- c("Gene", "MutationKey_Hg38", "Prediction","splice_junction_status","splice_junction_position","sample_id", "case_id","RNA_Sample","DNA_Sample","NormalizedExpression","ECDF", "Pvalue", "VAF_RNA","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf")
MUT_vaf <- MUT_vaf[, col_order]
```

Download the vaf, inferred percentiles and pvalues of the mutated samples:
```{r echo=FALSE}
datatable(MUT_vaf,
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp',
              scrollX = TRUE,
              scrollY =TRUE
))
```


```{r echo=FALSE}
summary_res <- rbind(summary_res,MUT_vaf)
```




```{r echo=FALSE}
## Generate Mut_df
MUT_df <- GeneSJ[GeneSJ$GROUP == "MUT",c("sample_id","case_id")]
```


```{r echo=FALSE}
## Add not found effects information: Donor Gain 1
MUT_df <- MUT_df[,c("sample_id", "case_id")]
MUT_df$NormalizedExpression <- NA
MUT_df$Prediction <- "Aceptor Gain"
MUT_df$splice_junction_status <- "AlternativeSJ not found in MUT samples"
MUT_df$splice_junction_position <- "chr17:7675234-7676033"
MUT_df$Pvalue <- NA
MUT_df$ECDF <- NA

vaf_select <- annot_vaf[,c("Gene","MutationKey_Hg38","sample_id","RNA_Sample","DNA_Sample","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf", "VAF_RNA")]
MUT_vaf <- merge(MUT_df, vaf_select, by.x="sample_id", by.y = "RNA_Sample")
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id"] <- "RNA_Sample"
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id.y"] <- "sample_id"

col_order <- c("Gene", "MutationKey_Hg38", "Prediction","splice_junction_status","splice_junction_position","sample_id", "case_id","RNA_Sample","DNA_Sample","NormalizedExpression", "ECDF","Pvalue", "VAF_RNA","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf")
MUT_vaf <- MUT_vaf[, col_order]

## Save
summary_res <- rbind(summary_res, MUT_vaf)
```



```{r echo=FALSE}
## Add not found effects information:
MUT_df <- MUT_df[,c("sample_id", "case_id")]
MUT_df$NormalizedExpression <- NA
MUT_df$Prediction <- "Aceptor Gain"
MUT_df$splice_junction_status <- "AlternativeSJ not found in MUT samples"
MUT_df$splice_junction_position <- "chr1:114713979-114716076"
MUT_df$Pvalue <- NA
MUT_df$ECDF <- NA

vaf_select <- annot_vaf[,c("Gene","MutationKey_Hg38","sample_id","RNA_Sample","DNA_Sample","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf", "VAF_RNA")]
MUT_vaf <- merge(MUT_df, vaf_select, by.x="sample_id", by.y = "RNA_Sample")
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id"] <- "RNA_Sample"
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id.y"] <- "sample_id"

col_order <- c("Gene", "MutationKey_Hg38", "Prediction","splice_junction_status","splice_junction_position","sample_id", "case_id","RNA_Sample","DNA_Sample","NormalizedExpression","ECDF","Pvalue", "VAF_RNA","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf")
MUT_vaf <- MUT_vaf[, col_order]

## Save
summary_res <- rbind(summary_res, MUT_vaf)
```

```{r echo=FALSE}
rm(GeneSJ, normalization_df, GeneSJ_subset, annot_vaf, plot_df, samples_df,table, plot_boxplot, plot_dotplot, vaf_select, SJCounts, MUT_df, MUT_vaf)
```


# TP53 chr17,7675217,T,C {.tabset}
Variant found in **4 patients** of the BeatAML (**4 samples**)

* Patients with TP53 chr17,7675217,T,C variant: **4 patients (4 samples)**
* Patients with the variant and RNASeq for validation: **4 patients (4 samples)**

The splicing alterations being assessed are: 

* **Aceptor Loss**: chr17:7675237-7675993; aceptor splice site chr17:7675237.
* **Aceptor Gain**: chr17:7675216-7675993, found in the mutated samples.

Variant information:
```{r echo=FALSE}
datatable(to_be_validated[to_be_validated$MutationKey_Hg38 == "chr17,7675217,T,C",],
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp', 
              scrollX = TRUE,  
              scrollY = TRUE,
              columnDefs = list(list(targets = '_all', className = 'dt-center')))
)
```

Load the extracted splice junctions of the gene harboring the mutation.
```{r}
extractedSJ_path <- paste0(extractedSJ_dir_in,"TP53_UM_annotSJ.tsv")
GeneSJ <- read.delim(extractedSJ_path, sep ="\t")
```

Set the sample's group: Mutated (MUT) or No Mutated (WT)
```{r}
samples_df <- found_variants[found_variants$Gene=="TP53" & found_variants$MutationKey_Hg38 == "chr17,7675217,T,C",]

cases <- samples_df$RNA_Sample[samples_df$Validable == "Validable"]
GeneSJ$GROUP <- ifelse(GeneSJ$sample_id %in% cases , "MUT", "WT")
```


## SJ Lookup {.tabset}

Search for the splice junctions of interest in the extracted splice junctions of the gene by position (chr_SJstart_SJend).

### Aceptor Gain
Search: chr17:7675216-7675993

Show all the splice junctions containing the position chr17:7675216-7675993
```{r}
colnames(GeneSJ)[grep("7675216_7675993",colnames(GeneSJ))]
```
Found: chr17_7675216_7675993

Reads of all the AML samples (mutated and no mutated) for the splice junction:
```{r}
GeneSJ$chr17_7675216_7675993
```

Samples with the SJ of interest:
```{r}
table(GeneSJ$chr17_7675216_7675993>0) 
```

Groups of the samples having the alternative splice junction:
```{r}
table(GeneSJ$GROUP[GeneSJ$chr17_7675216_7675993 > 0])
```
Alternative SJ found in the mutated samples.

### Canonical SJ
Canonical DE: chr17:7674972-7675052

Canonical UE: chr17:7675237-7675993; aceptor splice site chr17:7675237

Reads of all the AML samples (mutated and no mutated) for the splice junction:
```{r}
GeneSJ$chr17_7674972_7675052
```

Reads of all the AML samples (mutated and no mutated) for the splice junction:
```{r}
GeneSJ$chr17_7675237_7675993
```


## Normalization
Count the reads of all the splice junctions of the gene harboring the variant:
```{r}
GeneSJ$rowSum_SJtotal <- rowSums(GeneSJ[,grep("chr", names(GeneSJ))])
```

Normalization of the expression by the total read counts of all the splice junctions of the gene:
```{r}
GeneSJ$Normalized_CanonDE <- (GeneSJ$chr17_7674972_7675052)/GeneSJ$rowSum_SJtotal*100
GeneSJ$Normalized_CanonUE <- (GeneSJ$chr17_7675237_7675993)/GeneSJ$rowSum_SJtotal*100

GeneSJ$Normalized_AG <- (GeneSJ$chr17_7675216_7675993)/GeneSJ$rowSum_SJtotal*100
```

Download the normalized values for the assessed splice junctions of all the AML samples:
```{r echo=FALSE}
normalization_df <- GeneSJ[,c("sample_id","case_id", "GROUP","Normalized_CanonUE", "Normalized_CanonDE","Normalized_AG","chr17_7675237_7675993", "chr17_7674972_7675052", "chr17_7675216_7675993", "rowSum_SJtotal","file_id.BAM","file_name.BAM","file_id.STAR.SJCounts","file_name.STAR.SJCounts" )]

colnames(normalization_df)[grep("chr17_7674972_7675052", names(normalization_df))] <- "chr17_7674972_7675052_CanonDE"
colnames(normalization_df)[grep("chr17_7675237_7675993", names(normalization_df))] <- "chr17_7675237_7675993_CanonUE"
colnames(normalization_df)[grep("chr17_7675216_7675993", names(normalization_df))] <- "chr17_7675216_7675993_ExAG"

write.xlsx(normalization_df, paste0(normalizedEx_dir_out, "BeatAML.TP53.chr17-7675217-T-C.xlsx"), rowNames = FALSE)
```

```{r echo=FALSE}
datatable(normalization_df,
extensions = 'Buttons',
rownames = FALSE, 
filter = 'top',
options = list(paging = TRUE,
              pageLength = 4,
              autoWidth = FALSE,
              dom = 'lifrtBp',
              buttons = c('excel'), 
              scrollX = TRUE,  
              scrollY = TRUE,
              columnDefs = list(list(targets = '_all', className = 'dt-center'))
))
```

## VAF
```{r echo=FALSE}
GeneSJ_subset <- GeneSJ[GeneSJ$sample_id %in% cases,]
GeneSJ_subset$Gene <- "TP53"
GeneSJ_subset$MutationKey_Hg38 <- "chr17,7675217,T,C"

annot_vaf <- merge(found_variants[,c("MutationKey_Hg38","Gene","sample_id","case_id",                                               "RNA_Sample","DNA_Sample","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf",                         "MuTect2_filter","VarScan2_filter","SomaticSniper_filter","MuSE_filter","MUTReads","WTReads","VAF")], 
                   GeneSJ_subset[,c("Gene","sample_id", "MutationKey_Hg38","case_id",  "Normalized_CanonUE", "Normalized_CanonDE", "Normalized_AG")], 
                   by.x=c("RNA_Sample", "case_id","Gene", "MutationKey_Hg38"), 
                   by.y=c("sample_id", "case_id","Gene", "MutationKey_Hg38"))


annot_vaf <- annot_vaf %>% select(Gene,MutationKey_Hg38,
                                  sample_id,
                                  case_id, RNA_Sample,DNA_Sample,
                                  MuTect2_vaf,VarScan2_vaf,SomaticSniper_vaf,MuSE_vaf,
                                  VAF,
                                  Normalized_CanonUE, Normalized_CanonDE,
                                  Normalized_AG,
                                  everything()
                                  )

colnames(annot_vaf)[grep("VAF", names(annot_vaf))] <- "VAF_RNA"
colnames(annot_vaf)[grep("MUTReads", names(annot_vaf))] <- "MUTReads_RNA"
colnames(annot_vaf)[grep("WTReads", names(annot_vaf))] <- "WTReads_RNA"

write.xlsx(annot_vaf, paste0(normalizedEx_dir_out, "BeatAML.TP53.chr17-7675217-T-C.MUT.xlsx"), rowNames = FALSE)
```

Mutated samples vaf:
```{r echo=FALSE}
datatable(annot_vaf,
extensions = 'Buttons',
rownames = FALSE, 
filter = 'top',
options = list(paging = TRUE,
              pageLength = 4,
              autoWidth = TRUE,
              dom = 'lifrtBp',
              buttons = c('excel'), 
              scrollX = TRUE,  
              scrollY = TRUE
              )
)
```

## Plots {.tabset}

```{r echo=FALSE}
GeneSJ$GROUP <- ifelse(GeneSJ$GROUP == "MUT", "MUT", "WT")
GeneSJ$GROUP <- as.factor(GeneSJ$GROUP)
```

### Static Dot Plots 
Canonical splice junction chr17:7675237-7675993, aceptor splice site chr17:7675237
```{r echo=FALSE}

ggplot(GeneSJ, aes(sample_id,Normalized_CanonUE,color=GROUP)) + 
  geom_point(size=.8) +
  ggtitle("Normalized expression of TP53 Upstream Exon Splice Junction") +
  labs(color='GROUP')
```

Splicing alterations:

```{r echo=FALSE}
ggplot(GeneSJ, aes(sample_id,Normalized_AG,color=GROUP)) +
  geom_point(size=.8) +
  ggtitle("Normalized expression of TP53 Aceptor Gain") +
  labs(color='GROUP') 
```


### Interactive Dot Plots

```{r echo=FALSE}
plot_df <- merge(GeneSJ,annot_vaf[c("RNA_Sample", "VAF_RNA", 
                                    "MuTect2_vaf", "VarScan2_vaf", "SomaticSniper_vaf", "MuSE_vaf")],
                 by.x="sample_id", by.y="RNA_Sample", all.x = TRUE)
```

Canonical splice junction chr17:7675237-7675993, aceptor splice site chr17:7675237
```{r echo=FALSE}
ggplotly(ggplot(plot_df, aes(sample_id,Normalized_CanonUE, color=GROUP)) + 
  geom_point(aes(text=paste0("VAF RNA: ",VAF_RNA,"\nVAF MuTect: ",MuTect2_vaf,
                             "\nVAF VarScan2: ",VarScan2_vaf,
                             "\nVAF SomaticSniper: ",SomaticSniper_vaf,
                             "\nVAF MuSE: ",MuSE_vaf)),size= 0.8)  +
  ggtitle("Normalized expression of TP53 Upstream Exon Splice Junction") +
  labs(color='GROUP') 
)
```

Splicing alteration:
```{r echo=FALSE}
ggplotly(ggplot(plot_df, aes(sample_id,Normalized_AG,color=GROUP)) + 
  geom_point(aes(text=paste0("VAF RNA: ",VAF_RNA,"\nVAF MuTect: ",MuTect2_vaf,
                             "\nVAF VarScan2: ",VarScan2_vaf,
                             "\nVAF SomaticSniper: ",SomaticSniper_vaf,
                             "\nVAF MuSE: ",MuSE_vaf)),size= 0.8) +
  ggtitle("Normalized expression of TP53 aceptor Gain") +
  labs(color='GROUP') 
)
```


### Violin Plots
Violin Plots for the alternative splice junctions interrogated:
```{r echo=FALSE}
plot_boxplot <- plot_df[plot_df$GROUP %in% c("WT"),]
plot_dotplot <- plot_df[plot_df$GROUP %in% c("MUT"),]

GEN = "TP53"
SJ = "chr17:7675237-7675993"

ggplot(plot_df, aes(y=Normalized_CanonUE, x=factor(GROUP, level=c("WT", "MUT")))) + 
       geom_violin(scale="area") +
       geom_point(data = plot_dotplot, 
                  aes(y=Normalized_CanonUE, x=GROUP),
                  color="black",
                  size=2
                  ) +
        
       ylab("Normalized SJ expression") +
       xlab(paste(GEN, SJ))
```


```{r echo=FALSE}
plot_boxplot <- plot_df[plot_df$GROUP %in% c("WT"),]
plot_dotplot <- plot_df[plot_df$GROUP %in% c("MUT"),]

GEN = "TP53"
SJ = "chr17:7675216-7675993"

ggplot(plot_df, aes(y=Normalized_AG, x=factor(GROUP, level=c("WT", "MUT")))) + 
       geom_violin(scale="area") +
       geom_point(data = plot_dotplot, 
                  aes(y=Normalized_AG, x=GROUP),
                  color="black",
                  size=2
                  ) +
        
       ylab("Normalized SJ expression") +
       xlab(paste(GEN, SJ))
```

## Statistical Analysis {.tabset}

```{r}
SJCounts <- GeneSJ
```

### Aceptor Loss {.tabset}

#### Expression Difference 

Normality Test:
```{r}
shapiro.test(SJCounts$Normalized_CanonUE[SJCounts$GROUP == "WT"])
```

Value of Mean Normalized Expression of the Alternative SJ in WT samples:
```{r}
mean_WT_SJi <- mean(SJCounts$Normalized_CanonUE[SJCounts$GROUP == "WT"], na.rm=TRUE)
mean_WT_SJi
```

Normalized Expression Value of the Alternative SJ in the MUT sample:
```{r}
MUT_SJi <- SJCounts$Normalized_CanonUE[SJCounts$GROUP == "MUT"]
MUT_SJi
```
Deviation from the mean normalized expression:
```{r}
SJCounts$Difference <-  SJCounts$Normalized_CanonUE - mean_WT_SJi
```

```{r echo=FALSE}
table <- SJCounts[SJCounts$GROUP == "WT", c("sample_id", "Normalized_CanonUE", "Difference")]
table <- table[order(table$Difference, decreasing = TRUE),]
```

```{r echo=FALSE}
datatable(table,
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp'
))
```

Difference in the MUT sample: deviation of the Normalized expression of the MUT patient from the mean normalized WT expression
```{r}
SJCounts$Difference[SJCounts$GROUP == "MUT"]
```

```{r echo=FALSE}
ggplot(SJCounts, aes(sample_id,Difference)) + geom_point(aes(colour=factor(GROUP))) +
  ylab("Differential Expression") +
  labs(color = "GROUP",
       title = "Differential Expression Distribution")
```


```{r echo=FALSE}
ggplot(SJCounts, aes(y=Difference, x=factor(GROUP, level=c("WT", "MUT")))) + 
  geom_boxplot() +
  ylab("Differential Expression") +
  xlab("GROUP")
```

#### ECDF - Pvalue Inference 

```{r}
v_ecdf <- ecdf(SJCounts$Difference[SJCounts$GROUP == "WT"])
v_ecdf
```

```{r}
plot(ecdf(SJCounts$Difference[SJCounts$GROUP == "WT"]))
```

```{r}
v_ecdf(SJCounts$Difference[SJCounts$GROUP == "MUT"])
```


```{r}
MUT_df <- SJCounts[SJCounts$GROUP == "MUT",c("sample_id","case_id", "Normalized_CanonUE")] 
colnames(MUT_df) <- c("sample_id", "case_id", "NormalizedExpression")

MUT_df$ECDF <- v_ecdf(SJCounts$Difference[SJCounts$GROUP == "MUT"])
MUT_df$Pvalue <- MUT_df$ECDF

MUT_df$Prediction <- "Aceptor Loss"
MUT_df$splice_junction_status <- "CanonicalSJ"
MUT_df$splice_junction_position <- "chr17:7675237-7675993"

MUT_df$ECDF <- ifelse(MUT_df$NormalizedExpression == 0, NA,MUT_df$ECDF)
MUT_df$Pvalue <- ifelse(MUT_df$NormalizedExpression == 0, NA,MUT_df$Pvalue)

```

```{r echo=FALSE}
vaf_select <- annot_vaf[,c("Gene","MutationKey_Hg38","sample_id","RNA_Sample","DNA_Sample","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf", "VAF_RNA")]
MUT_vaf <- merge(MUT_df, vaf_select, by.x="sample_id", by.y = "RNA_Sample")
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id"] <- "RNA_Sample"
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id.y"] <- "sample_id"

col_order <- c("Gene", "MutationKey_Hg38", "Prediction","splice_junction_status","splice_junction_position","sample_id", "case_id","RNA_Sample","DNA_Sample","NormalizedExpression","ECDF", "Pvalue", "VAF_RNA","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf")
MUT_vaf <- MUT_vaf[, col_order]
```

Download the vaf, inferred percentiles and pvalues of the mutated samples:
```{r echo=FALSE}
datatable(MUT_vaf,
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp',
              scrollX=TRUE
))
```

```{r echo=FALSE}
summary_res <- rbind(summary_res,MUT_vaf)
```


### Aceptor Gain {.tabset}

#### Expression Difference 

Normality Test:
```{r}
shapiro.test(SJCounts$Normalized_AG[SJCounts$GROUP == "WT"])
```

Value of Mean Normalized Expression of the Alternative SJ in WT samples:
```{r}
mean_WT_SJi <- mean(SJCounts$Normalized_AG[SJCounts$GROUP == "WT"], na.rm=TRUE)
mean_WT_SJi
```

Normalized Expression Value of the Alternative SJ in the MUT sample:
```{r}
MUT_SJi <- SJCounts$Normalized_AG[SJCounts$GROUP == "MUT"]
MUT_SJi
```
Deviation from the mean normalized expression:
```{r}
SJCounts$Difference <-  SJCounts$Normalized_AG - mean_WT_SJi
```

```{r echo=FALSE}
table <- SJCounts[SJCounts$GROUP == "WT", c("sample_id", "Normalized_AG", "Difference")]
table <- table[order(table$Difference, decreasing = TRUE),]
```

```{r echo=FALSE}
datatable(table,
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp'
))
```

Difference in the MUT sample: deviation of the Normalized expression of the MUT patient from the mean normalized WT expression
```{r}
SJCounts$Difference[SJCounts$GROUP == "MUT"]
```

```{r echo=FALSE}
ggplot(SJCounts, aes(sample_id,Difference)) + geom_point(aes(colour=factor(GROUP))) +
  ylab("Differential Expression") +
  labs(color = "GROUP",
       title = "Differential Expression Distribution")
```


```{r echo=FALSE}
ggplot(SJCounts, aes(y=Difference, x=factor(GROUP, level=c("WT", "MUT")))) + 
  geom_boxplot() +
  ylab("Differential Expression") +
  xlab("GROUP")
```

#### ECDF - Pvalue Inference 

```{r}
v_ecdf <- ecdf(SJCounts$Difference[SJCounts$GROUP == "WT"])
v_ecdf
```

```{r}
plot(ecdf(SJCounts$Difference[SJCounts$GROUP == "WT"]))
```

```{r}
v_ecdf(SJCounts$Difference[SJCounts$GROUP == "MUT"])
```


```{r}
MUT_df <- SJCounts[SJCounts$GROUP == "MUT",c("sample_id","case_id", "Normalized_AG")] 
colnames(MUT_df) <- c("sample_id", "case_id", "NormalizedExpression")

MUT_df$ECDF <- v_ecdf(SJCounts$Difference[SJCounts$GROUP == "MUT"])
MUT_df$Pvalue <- 1- MUT_df$ECDF

MUT_df$Prediction <- "Aceptor Gain"
MUT_df$splice_junction_status <- "AlternativeSJ found in MUT samples"
MUT_df$splice_junction_position <- "chr17:7675216-7675993"

MUT_df$ECDF <- ifelse(MUT_df$NormalizedExpression == 0, NA,MUT_df$ECDF)
MUT_df$Pvalue <- ifelse(MUT_df$NormalizedExpression == 0, NA,MUT_df$Pvalue)
```

```{r echo=FALSE}
vaf_select <- annot_vaf[,c("Gene","MutationKey_Hg38","sample_id","RNA_Sample","DNA_Sample","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf", "VAF_RNA")]
MUT_vaf <- merge(MUT_df, vaf_select, by.x="sample_id", by.y = "RNA_Sample")
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id"] <- "RNA_Sample"
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id.y"] <- "sample_id"

col_order <- c("Gene", "MutationKey_Hg38", "Prediction","splice_junction_status","splice_junction_position","sample_id", "case_id","RNA_Sample","DNA_Sample","NormalizedExpression","ECDF", "Pvalue", "VAF_RNA","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf")
MUT_vaf <- MUT_vaf[, col_order]
```

Download the vaf, inferred percentiles and pvalues of the mutated samples:
```{r echo=FALSE}
datatable(MUT_vaf,
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp',
              scrollX = TRUE,
              scrollY =TRUE
))
```


```{r echo=FALSE}
summary_res <- rbind(summary_res,MUT_vaf)
```

```{r echo=FALSE}
rm(GeneSJ, normalization_df, GeneSJ_subset, annot_vaf, plot_df, samples_df,table, plot_boxplot, plot_dotplot, vaf_select, SJCounts, MUT_df, MUT_vaf)
```


# TP53 chr17,7674872,T,C {.tabset}
Variant found in **6 patients** of the BeatAML (**7 samples**).

* Patients with TP53 chr17,7674872,T,C variant: **6 patients (7 samples)**
* Patients with the variant and RNASeq for validation: **5 patient (5 samples)**

The splicing alterations being assessed are: 

* **Donor Gain**: predicted at 1bp from the variant, chr17,7674872, not found in the splice junction collection.
* **Exon Skipping**: chr17:7674291-7675052, found in the mutated samples.

Variant information:
```{r echo=FALSE}
datatable(to_be_validated[to_be_validated$MutationKey_Hg38 == "chr17,7674872,T,C",],
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp', 
              scrollX = TRUE,  
              scrollY = TRUE,
              columnDefs = list(list(targets = '_all', className = 'dt-center')))
)
```

Load the extracted splice junctions of the gene harboring the mutation.
```{r}
extractedSJ_path <- paste0(extractedSJ_dir_in,"TP53_UM_annotSJ.tsv")
GeneSJ <- read.delim(extractedSJ_path, sep ="\t")
```

Set the sample's group: Mutated (MUT) or No Mutated (WT)
```{r}
samples_df <- found_variants[found_variants$Gene=="TP53" & found_variants$MutationKey_Hg38 == "chr17,7674872,T,C",]

cases <- samples_df$RNA_Sample[samples_df$Validable == "Validable"]
GeneSJ$GROUP <- ifelse(GeneSJ$sample_id %in% cases , "MUT", "WT")
```


## SJ Lookup {.tabset}

Search for the splice junctions of interest in the extracted splice junctions of the gene by position (chr_SJstart_SJend).

### Donor Gain
Search: predicted at 1bp from the variant position, chr17,7674872

Show all the splice junctions containing the positions between 7674870 - 7674879
```{r}
colnames(GeneSJ)[grep("767487",colnames(GeneSJ))]
```
Alternative SJ not found in the splice junction collection.

### Exon Skipping
Search: chr17:7674291-7675052

Show all the splice junctions containing the position chr17:7674291-7675052
```{r}
colnames(GeneSJ)[grep("7674291_7675052",colnames(GeneSJ))]
```
Found: chr17_7674291_7675052

Reads of all the AML samples (mutated and no mutated) for the splice junction:
```{r}
GeneSJ$chr17_7674291_7675052
```

Samples with the SJ of interest:
```{r}
table(GeneSJ$chr17_7674291_7675052>0) 
```

Groups of the samples having the alternative splice junction:
```{r}
table(GeneSJ$GROUP[GeneSJ$chr17_7674291_7675052 > 0])
```
Alternative SJ found in the mutated samples.

### Canonical SJ
Exon upstream (UE): chr17:7674972-7675052

Exon downstream (DE): chr17:7674291-7674858

Reads of all the AML samples (mutated and no mutated) for the splice junction:
```{r}
GeneSJ$chr17_7674972_7675052
```

Reads of all the AML samples (mutated and no mutated) for the splice junction:
```{r}
GeneSJ$chr17_7674291_7674858
```


## Normalization
Count the reads of all the splice junctions of the gene harboring the variant:
```{r}
GeneSJ$rowSum_SJtotal <- rowSums(GeneSJ[,grep("chr", names(GeneSJ))])
```

Normalization of the expression by the total read counts of all the splice junctions of the gene:
```{r}
GeneSJ$Normalized_CanonUE <- (GeneSJ$chr17_7674972_7675052)/GeneSJ$rowSum_SJtotal*100
GeneSJ$Normalized_CanonDE <- (GeneSJ$chr17_7674291_7674858)/GeneSJ$rowSum_SJtotal*100
GeneSJ$Normalized_ES <- (GeneSJ$chr17_7674291_7675052)/GeneSJ$rowSum_SJtotal*100
```

Download the normalized values for the assessed splice junctions of all the AML samples:
```{r echo=FALSE}
normalization_df <- GeneSJ[,c("sample_id","case_id", "GROUP","Normalized_CanonDE", "Normalized_CanonUE", "Normalized_ES", "chr17_7674291_7674858",  "chr17_7674972_7675052","chr17_7674291_7675052", "rowSum_SJtotal","file_id.BAM","file_name.BAM","file_id.STAR.SJCounts","file_name.STAR.SJCounts" )]

colnames(normalization_df)[grep("chr17_7674972_767505", names(normalization_df))] <- "chr17_7674972_767505_CanonUE"
colnames(normalization_df)[grep("chr17_7674291_7674858", names(normalization_df))] <- "chr17_7674291_7674858_CanonDE"
colnames(normalization_df)[grep("chr17_7674291_7675052", names(normalization_df))] <- "chr17_7674291_7675052_ES"

write.xlsx(normalization_df, paste0(normalizedEx_dir_out, "BeatAML.TP53.chr17-7674872-T-C.xlsx"), rowNames = FALSE)
```

```{r echo=FALSE}
datatable(normalization_df,
extensions = 'Buttons',
rownames = FALSE, 
filter = 'top',
options = list(paging = TRUE,
              pageLength = 4,
              autoWidth = FALSE,
              dom = 'lifrtBp',
              buttons = c('excel'), 
              scrollX = TRUE,  
              scrollY = TRUE,
              columnDefs = list(list(targets = '_all', className = 'dt-center'))
))
```

## VAF
```{r echo=FALSE}
GeneSJ_subset <- GeneSJ[GeneSJ$sample_id %in% cases,]
GeneSJ_subset$Gene <- "TP53"
GeneSJ_subset$MutationKey_Hg38 <- "chr17,7674872,T,C"

annot_vaf <- merge(found_variants[,c("MutationKey_Hg38","Gene","sample_id","case_id",                                               "RNA_Sample","DNA_Sample","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf",                         "MuTect2_filter","VarScan2_filter","SomaticSniper_filter","MuSE_filter","MUTReads","WTReads","VAF")], 
                   GeneSJ_subset[,c("Gene","sample_id", "MutationKey_Hg38","case_id",  "Normalized_CanonDE", "Normalized_CanonUE", "Normalized_ES")], 
                   by.x=c("RNA_Sample", "case_id","Gene", "MutationKey_Hg38"), 
                   by.y=c("sample_id", "case_id","Gene", "MutationKey_Hg38"))


annot_vaf <- annot_vaf %>% select(Gene,MutationKey_Hg38,
                                  sample_id,
                                  case_id, RNA_Sample,DNA_Sample,
                                  MuTect2_vaf,VarScan2_vaf,SomaticSniper_vaf,MuSE_vaf,
                                  VAF,
                                  Normalized_CanonDE, Normalized_CanonUE,Normalized_ES,
                                  everything()
                                  )

colnames(annot_vaf)[grep("VAF", names(annot_vaf))] <- "VAF_RNA"
colnames(annot_vaf)[grep("MUTReads", names(annot_vaf))] <- "MUTReads_RNA"
colnames(annot_vaf)[grep("WTReads", names(annot_vaf))] <- "WTReads_RNA"

write.xlsx(annot_vaf, paste0(normalizedEx_dir_out, "BeatAML.TP53.chr17-7674872-T-C.MUT.xlsx"), rowNames = FALSE)
```

Mutated samples vaf:
```{r echo=FALSE}
datatable(annot_vaf,
extensions = 'Buttons',
rownames = FALSE, 
filter = 'top',
options = list(paging = TRUE,
              pageLength = 4,
              autoWidth = TRUE,
              dom = 'lifrtBp',
              buttons = c('excel'), 
              scrollX = TRUE,  
              scrollY = TRUE
              )
)
```

## Plots {.tabset}

```{r echo=FALSE}
GeneSJ$GROUP <- ifelse(GeneSJ$GROUP == "MUT", "MUT", "WT")
GeneSJ$GROUP <- as.factor(GeneSJ$GROUP)
```

### Static Dot Plots 

Splicing alterations:
```{r echo=FALSE}
ggplot(GeneSJ, aes(sample_id,Normalized_ES,color=GROUP)) +
  geom_point(size=.8) +
  ggtitle("Normalized expression of TP53 Exon Skipping") +
  labs(color='GROUP') 
```


### Interactive Dot Plots

```{r echo=FALSE}
plot_df <- merge(GeneSJ,annot_vaf[c("RNA_Sample", "VAF_RNA", 
                                    "MuTect2_vaf", "VarScan2_vaf", "SomaticSniper_vaf", "MuSE_vaf")],
                 by.x="sample_id", by.y="RNA_Sample", all.x = TRUE)
```

Splicing alteration:
```{r echo=FALSE}
ggplotly(ggplot(plot_df, aes(sample_id,Normalized_ES,color=GROUP)) + 
  geom_point(aes(text=paste0("VAF RNA: ",VAF_RNA,"\nVAF MuTect: ",MuTect2_vaf,
                             "\nVAF VarScan2: ",VarScan2_vaf,
                             "\nVAF SomaticSniper: ",SomaticSniper_vaf,
                             "\nVAF MuSE: ",MuSE_vaf)),size= 0.8) +
  ggtitle("Normalized expression of TP53 Exon Skipping") +
  labs(color='GROUP') 
)
```


### Violin Plots
Violin Plots for the alternative splice junctions interrogated:


```{r echo=FALSE}
plot_boxplot <- plot_df[plot_df$GROUP %in% c("WT"),]
plot_dotplot <- plot_df[plot_df$GROUP %in% c("MUT"),]

GEN = "TP53"
SJ = "chr17:7674291-7675052"

ggplot(plot_df, aes(y=Normalized_ES, x=factor(GROUP, level=c("WT", "MUT")))) + 
       geom_violin(scale="area") +
       geom_point(data = plot_dotplot, 
                  aes(y=Normalized_ES, x=GROUP),
                  color="black",
                  size=2
                  ) +
        
       ylab("Normalized SJ expression") +
       xlab(paste(GEN, SJ))
```

## Statistical Analysis {.tabset}

```{r}
SJCounts <- GeneSJ
```

### Exon Skipping {.tabset}

#### Expression Difference 

Normality Test:
```{r}
shapiro.test(SJCounts$Normalized_ES[SJCounts$GROUP == "WT"])
```

Value of Mean Normalized Expression of the Alternative SJ in WT samples:
```{r}
mean_WT_SJi <- mean(SJCounts$Normalized_ES[SJCounts$GROUP == "WT"], na.rm=TRUE)
mean_WT_SJi
```

Normalized Expression Value of the Alternative SJ in the MUT sample:
```{r}
MUT_SJi <- SJCounts$Normalized_ES[SJCounts$GROUP == "MUT"]
MUT_SJi
```
Deviation from the mean normalized expression:
```{r}
SJCounts$Difference <-  SJCounts$Normalized_ES - mean_WT_SJi
```

```{r echo=FALSE}
table <- SJCounts[SJCounts$GROUP == "WT", c("sample_id", "Normalized_ES", "Difference")]
table <- table[order(table$Difference, decreasing = TRUE),]
```

```{r echo=FALSE}
datatable(table,
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp'
))
```

Difference in the MUT sample: deviation of the Normalized expression of the MUT patient from the mean normalized WT expression
```{r}
SJCounts$Difference[SJCounts$GROUP == "MUT"]
```

```{r echo=FALSE}
ggplot(SJCounts, aes(sample_id,Difference)) + geom_point(aes(colour=factor(GROUP))) +
  ylab("Differential Expression") +
  labs(color = "GROUP",
       title = "Differential Expression Distribution")
```


```{r echo=FALSE}
ggplot(SJCounts, aes(y=Difference, x=factor(GROUP, level=c("WT", "MUT")))) + 
  geom_boxplot() +
  ylab("Differential Expression") +
  xlab("GROUP")
```

#### ECDF

```{r}
v_ecdf <- ecdf(SJCounts$Difference[SJCounts$GROUP == "WT"])
v_ecdf
```

```{r}
plot(ecdf(SJCounts$Difference[SJCounts$GROUP == "WT"]))
```

```{r}
v_ecdf(SJCounts$Difference[SJCounts$GROUP == "MUT"])
```


```{r}
MUT_df <- SJCounts[SJCounts$GROUP == "MUT",c("sample_id","case_id", "Normalized_ES")] 
colnames(MUT_df) <- c("sample_id", "case_id", "NormalizedExpression")

MUT_df$ECDF <- v_ecdf(SJCounts$Difference[SJCounts$GROUP == "MUT"])
MUT_df$Pvalue <- NA

MUT_df$Prediction <- "Exon Skipping"
MUT_df$splice_junction_status <- "AlternativeSJ found in MUT samples"
MUT_df$splice_junction_position <- "chr17:7674291-7675052"

MUT_df$ECDF <- ifelse(MUT_df$NormalizedExpression == 0, NA,MUT_df$ECDF)
```

```{r echo=FALSE}
vaf_select <- annot_vaf[,c("Gene","MutationKey_Hg38","sample_id","RNA_Sample","DNA_Sample","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf", "VAF_RNA")]
MUT_vaf <- merge(MUT_df, vaf_select, by.x="sample_id", by.y = "RNA_Sample")
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id"] <- "RNA_Sample"
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id.y"] <- "sample_id"

col_order <- c("Gene", "MutationKey_Hg38", "Prediction","splice_junction_status","splice_junction_position","sample_id", "case_id","RNA_Sample","DNA_Sample","NormalizedExpression","ECDF", "Pvalue", "VAF_RNA","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf")
MUT_vaf <- MUT_vaf[, col_order]
```

Download the vaf, inferred percentiles and pvalues of the mutated samples:
```{r echo=FALSE}
datatable(MUT_vaf,
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp',
              scrollX=TRUE
))
```

#### Mann Whitney U
Normality:
```{r}
shapiro.test(SJCounts$Normalized_ES[SJCounts$GROUP== "WT"])
```

```{r}
shapiro.test(SJCounts$Normalized_ES[SJCounts$GROUP== "MUT"])
```

Mann-Whitney:
```{r}
wt <- wilcox.test(x=SJCounts$Normalized_ES[SJCounts$GROUP== "MUT"], 
                  y=SJCounts$Normalized_ES[SJCounts$GROUP== "WT"],
                  alternative = "two.sided", 
                  paired = FALSE, 
                  conf.int = 0.95)

wt
```

```{r echo=FALSE}

stat_df <- data.frame(matrix(ncol = 17, nrow = 1))
colnames(stat_df) <- c("Gene","MutationKey_Hg38","Prediction","splice_junction_status","splice_junction_position", "sample_id","case_id","RNA_Sample","DNA_Sample","NormalizedExpression","ECDF","Pvalue","VAF_RNA","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf")

stat_df$Gene <- unique(MUT_vaf$Gene)
stat_df$MutationKey_Hg38 <- unique(MUT_vaf$MutationKey_Hg38)
stat_df$sample_id <- paste(MUT_vaf$sample_id, collapse = ';\n') 
stat_df$case_id <- paste(MUT_vaf$case_id, collapse = ';\n') 
stat_df$RNA_Sample <- paste(MUT_vaf$RNA_Sample, collapse = ';\n') 
stat_df$DNA_Sample <- paste(MUT_vaf$DNA_Sample, collapse = ';\n')
stat_df$NormalizedExpression <- paste0(MUT_vaf$NormalizedExpression, collapse= ";\n")
stat_df$ECDF <- paste0(MUT_vaf$ECDF, collapse = ";\n")
stat_df$VAF_RNA <-  paste(MUT_vaf$VAF_RNA, collapse = ';\n')
stat_df$MuTect2_vaf <-  paste(MUT_vaf$MuTect2_vaf, collapse = ';\n')
stat_df$VarScan2_vaf <-  paste(MUT_vaf$VarScan2_vaf, collapse = ';\n')
stat_df$SomaticSniper_vaf <-  paste(MUT_vaf$SomaticSniper_vaf, collapse = ';\n')
stat_df$MuSE_vaf <-  paste(MUT_vaf$MuSE_vaf, collapse = ';\n')
stat_df$Prediction <- "Exon Skipping"
stat_df$splice_junction_status <- "AlternativeSJ found in MUT samples"
stat_df$splice_junction_position <- "chr17:7674291-7675052"

stat_df$Pvalue <- wt$p.value
```

```{r echo=FALSE}
summary_res<- rbind(summary_res,stat_df)
```


```{r echo=FALSE}
## Add not found effects information:
MUT_df <- MUT_df[,c("sample_id", "case_id")]
MUT_df$NormalizedExpression <- NA
MUT_df$Prediction <- "Donor Gain"
MUT_df$splice_junction_status <- "AlternativeSJ not found in SJ collection"
MUT_df$splice_junction_position <- "chr17:7674872"
MUT_df$Pvalue <- NA

vaf_select <- annot_vaf[,c("Gene","MutationKey_Hg38","sample_id","RNA_Sample","DNA_Sample","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf", "VAF_RNA")]
MUT_vaf <- merge(MUT_df, vaf_select, by.x="sample_id", by.y = "RNA_Sample")
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id"] <- "RNA_Sample"
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id.y"] <- "sample_id"

col_order <- c("Gene", "MutationKey_Hg38", "Prediction","splice_junction_status","splice_junction_position","sample_id", "case_id","RNA_Sample","DNA_Sample","NormalizedExpression", "Pvalue", "VAF_RNA","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf")
MUT_vaf <- MUT_vaf[, col_order]

stat_df <- data.frame(matrix(ncol = 17, nrow = 1))
colnames(stat_df) <- c("Gene","MutationKey_Hg38","Prediction","splice_junction_status","splice_junction_position", "sample_id","case_id","RNA_Sample","DNA_Sample","NormalizedExpression","ECDF","Pvalue","VAF_RNA","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf")

stat_df$Gene <- unique(MUT_vaf$Gene)
stat_df$MutationKey_Hg38 <- unique(MUT_vaf$MutationKey_Hg38)
stat_df$sample_id <- paste(MUT_vaf$sample_id, collapse = ';\n') 
stat_df$case_id <- paste(MUT_vaf$case_id, collapse = ';\n') 
stat_df$RNA_Sample <- paste(MUT_vaf$RNA_Sample, collapse = ';\n') 
stat_df$DNA_Sample <- paste(MUT_vaf$DNA_Sample, collapse = ';\n')
stat_df$NormalizedExpression <- NA
stat_df$ECDF <- NA
stat_df$VAF_RNA <-  paste(MUT_vaf$VAF_RNA, collapse = ';\n')
stat_df$MuTect2_vaf <-  paste(MUT_vaf$MuTect2_vaf, collapse = ';\n')
stat_df$VarScan2_vaf <-  paste(MUT_vaf$VarScan2_vaf, collapse = ';\n')
stat_df$SomaticSniper_vaf <-  paste(MUT_vaf$SomaticSniper_vaf, collapse = ';\n')
stat_df$MuSE_vaf <-  paste(MUT_vaf$MuSE_vaf, collapse = ';\n')
stat_df$Prediction <- unique(MUT_vaf$Prediction)
stat_df$splice_junction_status <- unique(MUT_vaf$splice_junction_status)
stat_df$splice_junction_position <- unique(MUT_vaf$splice_junction_position)
## Save
summary_res <- rbind(summary_res, stat_df)
```


```{r echo=FALSE}
rm(GeneSJ, normalization_df, GeneSJ_subset, annot_vaf, plot_df, samples_df,table, plot_boxplot, plot_dotplot, vaf_select, SJCounts, MUT_df, MUT_vaf)
```


# TP53 chr17,7674894,G,A {.tabset}
Variant found in **2 patients** of the BeatAML (**2 samples**)

* Patients with TP53 chr17,7674894,G,A variant: **2 patients (2 samples)**
* Patients with the variant and RNASeq for validation: **2 patients (2 samples)**

The splicing alterations being assessed are: 

* **Donor Loss**: chr17:7674291-7674858; donor splice site chr17:7674858.
* **Exon Skipping**: chr17:7674291-7675052, found in the mutated samples.

Variant information:
```{r echo=FALSE}
datatable(to_be_validated[to_be_validated$MutationKey_Hg38 == "chr17,7674894,G,A",],
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp', 
              scrollX = TRUE,  
              scrollY = TRUE,
              columnDefs = list(list(targets = '_all', className = 'dt-center')))
)
```

Load the extracted splice junctions of the gene harboring the mutation.
```{r}
extractedSJ_path <- paste0(extractedSJ_dir_in,"TP53_UM_annotSJ.tsv")
GeneSJ <- read.delim(extractedSJ_path, sep ="\t")
```

Set the sample's group: Mutated (MUT) or No Mutated (WT)
```{r}
samples_df <- found_variants[found_variants$Gene=="TP53" & found_variants$MutationKey_Hg38 == "chr17,7674894,G,A",]

cases <- samples_df$RNA_Sample[samples_df$Validable == "Validable"]
GeneSJ$GROUP <- ifelse(GeneSJ$sample_id %in% cases , "MUT", "WT")
```


## SJ Lookup {.tabset}

Search for the splice junctions of interest in the extracted splice junctions of the gene by position (chr_SJstart_SJend).

### Exon Skipping
Search: chr17:7674291-7675052

Show all the splice junctions containing the position chr17:7674291-7675052
```{r}
colnames(GeneSJ)[grep("7674291_7675052",colnames(GeneSJ))]
```
Found: chr17_7674291_7675052

Reads of all the AML samples (mutated and no mutated) for the splice junction:
```{r}
GeneSJ$chr17_7674291_7675052
```

Samples with the SJ of interest:
```{r}
table(GeneSJ$chr17_7674291_7675052>0) 
```

Groups of the samples having the alternative splice junction:
```{r}
table(GeneSJ$GROUP[GeneSJ$chr17_7674291_7675052 > 0])
```
Alternative SJ found in the mutated samples.

### Canonical SJ
Exon upstream (UE): chr17:7674972-7675052

Exon downstream (DE): chr17:7674291-7674858, splice site chr17:7674858

Reads of all the AML samples (mutated and no mutated) for the splice junction:
```{r}
GeneSJ$chr17_7674972_7675052
```

Reads of all the AML samples (mutated and no mutated) for the splice junction:
```{r}
GeneSJ$chr17_7674291_7674858
```


## Normalization
Count the reads of all the splice junctions of the gene harboring the variant:
```{r}
GeneSJ$rowSum_SJtotal <- rowSums(GeneSJ[,grep("chr", names(GeneSJ))])
```

Normalization of the expression by the total read counts of all the splice junctions of the gene:
```{r}
GeneSJ$Normalized_CanonDE <- (GeneSJ$chr17_7674291_7674858)/GeneSJ$rowSum_SJtotal*100
GeneSJ$Normalized_ES <- (GeneSJ$chr17_7674291_7675052)/GeneSJ$rowSum_SJtotal*100
```

Download the normalized values for the assessed splice junctions of all the AML samples:
```{r echo=FALSE}
normalization_df <- GeneSJ[,c("sample_id","case_id", "GROUP","Normalized_CanonDE", "Normalized_ES", "chr17_7674291_7674858",  "chr17_7674291_7675052", "rowSum_SJtotal","file_id.BAM","file_name.BAM","file_id.STAR.SJCounts","file_name.STAR.SJCounts" )]

colnames(normalization_df)[grep("chr17_7674291_7674858", names(normalization_df))] <- "chr17_7674291_7674858_CanonDE"
colnames(normalization_df)[grep("chr17_7674291_7675052", names(normalization_df))] <- "chr17_7674291_7675052_ES"

write.xlsx(normalization_df, paste0(normalizedEx_dir_out, "BeatAML.TP53.chr17-7674894-G-A.xlsx"), rowNames = FALSE)
```

```{r echo=FALSE}
datatable(normalization_df,
extensions = 'Buttons',
rownames = FALSE, 
filter = 'top',
options = list(paging = TRUE,
              pageLength = 4,
              autoWidth = FALSE,
              dom = 'lifrtBp',
              buttons = c('excel'), 
              scrollX = TRUE,  
              scrollY = TRUE,
              columnDefs = list(list(targets = '_all', className = 'dt-center'))
))
```

## VAF
```{r echo=FALSE}
GeneSJ_subset <- GeneSJ[GeneSJ$sample_id %in% cases,]
GeneSJ_subset$Gene <- "TP53"
GeneSJ_subset$MutationKey_Hg38 <- "chr17,7674894,G,A"

annot_vaf <- merge(found_variants[,c("MutationKey_Hg38","Gene","sample_id","case_id",                                               "RNA_Sample","DNA_Sample","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf",                         "MuTect2_filter","VarScan2_filter","SomaticSniper_filter","MuSE_filter","MUTReads","WTReads","VAF")], 
                   GeneSJ_subset[,c("Gene","sample_id", "MutationKey_Hg38","case_id",  "Normalized_CanonDE", "Normalized_ES")], 
                   by.x=c("RNA_Sample", "case_id","Gene", "MutationKey_Hg38"), 
                   by.y=c("sample_id", "case_id","Gene", "MutationKey_Hg38"))


annot_vaf <- annot_vaf %>% select(Gene,MutationKey_Hg38,
                                  sample_id,
                                  case_id, RNA_Sample,DNA_Sample,
                                  MuTect2_vaf,VarScan2_vaf,SomaticSniper_vaf,MuSE_vaf,
                                  VAF,
                                  Normalized_CanonDE,Normalized_ES,
                                  everything()
                                  )

colnames(annot_vaf)[grep("VAF", names(annot_vaf))] <- "VAF_RNA"
colnames(annot_vaf)[grep("MUTReads", names(annot_vaf))] <- "MUTReads_RNA"
colnames(annot_vaf)[grep("WTReads", names(annot_vaf))] <- "WTReads_RNA"

write.xlsx(annot_vaf, paste0(normalizedEx_dir_out, "BeatAML.TP53.chr17-7674894-G-A.MUT.xlsx"), rowNames = FALSE)
```

Mutated samples vaf:
```{r echo=FALSE}
datatable(annot_vaf,
extensions = 'Buttons',
rownames = FALSE, 
filter = 'top',
options = list(paging = TRUE,
              pageLength = 4,
              autoWidth = TRUE,
              dom = 'lifrtBp',
              buttons = c('excel'), 
              scrollX = TRUE,  
              scrollY = TRUE
              )
)
```

## Plots {.tabset}

```{r echo=FALSE}
GeneSJ$GROUP <- ifelse(GeneSJ$GROUP == "MUT", "MUT", "WT")
GeneSJ$GROUP <- as.factor(GeneSJ$GROUP)
```

### Static Dot Plots 

Canonical splice junction Exon downstream (DE): chr17:7674291-7674858, donor splice site chr17:7674858
```{r}
ggplot(GeneSJ, aes(sample_id,Normalized_CanonDE,color=GROUP)) +
  geom_point(size=.8) +
  labs(color='GROUP', 
       title = "Normalized Expression of TP53 chr17:7674291-7674858", 
       subtitle="Potential Donor Loss Effect on Canonical SJ",
       y = "Normalized Expression")
```

Splicing alterations:
```{r echo=FALSE}
ggplot(GeneSJ, aes(sample_id,Normalized_ES,color=GROUP)) +
  geom_point(size=.8) +
  ggtitle("Normalized expression of TP53 Exon Skipping") +
  labs(color='GROUP') 
```


### Interactive Dot Plots

```{r echo=FALSE}
plot_df <- merge(GeneSJ,annot_vaf[c("RNA_Sample", "VAF_RNA", 
                                    "MuTect2_vaf", "VarScan2_vaf", "SomaticSniper_vaf", "MuSE_vaf")],
                 by.x="sample_id", by.y="RNA_Sample", all.x = TRUE)
```


Canonical splice junction Exon downstream (DE): chr17:7674291-7674858, donor splice site chr17:7674858
```{r echo=FALSE}
ggplotly(ggplot(plot_df, aes(sample_id,Normalized_CanonDE,color=GROUP)) + 
  geom_point(aes(text=paste0("VAF RNA: ",VAF_RNA,"\nVAF MuTect: ",MuTect2_vaf,
                             "\nVAF VarScan2: ",VarScan2_vaf,
                             "\nVAF SomaticSniper: ",SomaticSniper_vaf,
                             "\nVAF MuSE: ",MuSE_vaf)),size= 0.8) +
  ggtitle("Normalized expression of TP53 Downstream Splice Junction") +
  labs(color='GROUP') 
)
```

Splicing alteration:
```{r echo=FALSE}
ggplotly(ggplot(plot_df, aes(sample_id,Normalized_ES,color=GROUP)) + 
  geom_point(aes(text=paste0("VAF RNA: ",VAF_RNA,"\nVAF MuTect: ",MuTect2_vaf,
                             "\nVAF VarScan2: ",VarScan2_vaf,
                             "\nVAF SomaticSniper: ",SomaticSniper_vaf,
                             "\nVAF MuSE: ",MuSE_vaf)),size= 0.8) +
  ggtitle("Normalized expression of TP53 Exon Skipping") +
  labs(color='GROUP') 
)
```


### Violin Plots
Violin Plots for the alternative splice junctions interrogated:

Donor Loss:
```{r echo=FALSE}
plot_boxplot <- plot_df[plot_df$GROUP %in% c("WT"),]
plot_dotplot <- plot_df[plot_df$GROUP %in% c("MUT"),]

GEN = "TP53"
SJ = "chr17:7674291-7674858"

ggplot(plot_df, aes(y=Normalized_CanonDE, x=factor(GROUP, level=c("WT", "MUT")))) + 
       geom_violin(scale="area") +
       geom_point(data = plot_dotplot, 
                  aes(y=Normalized_CanonDE, x=GROUP),
                  color="black",
                  size=2
                  ) +
        
       ylab("Normalized SJ expression") +
       xlab(paste(GEN, SJ))
```

Exon Skipping:
```{r echo=FALSE}
plot_boxplot <- plot_df[plot_df$GROUP %in% c("WT"),]
plot_dotplot <- plot_df[plot_df$GROUP %in% c("MUT"),]

GEN = "TP53"
SJ = "chr17:7674291-7675052"

ggplot(plot_df, aes(y=Normalized_ES, x=factor(GROUP, level=c("WT", "MUT")))) + 
       geom_violin(scale="area") +
       geom_point(data = plot_dotplot, 
                  aes(y=Normalized_ES, x=GROUP),
                  color="black",
                  size=2
                  ) +
        
       ylab("Normalized SJ expression") +
       xlab(paste(GEN, SJ))
```

## Statistical Analysis {.tabset}

```{r}
SJCounts <- GeneSJ
```


### Donor Loss {.tabset}

#### Expression Difference 

Normality Test:
```{r}
shapiro.test(SJCounts$Normalized_CanonDE[SJCounts$GROUP == "WT"])
```

Value of Mean Normalized Expression of the Alternative SJ in WT samples:
```{r}
mean_WT_SJi <- mean(SJCounts$Normalized_CanonDE[SJCounts$GROUP == "WT"], na.rm=TRUE)
mean_WT_SJi
```

Normalized Expression Value of the Alternative SJ in the MUT sample:
```{r}
MUT_SJi <- SJCounts$Normalized_CanonDE[SJCounts$GROUP == "MUT"]
MUT_SJi
```
Deviation from the mean normalized expression:
```{r}
SJCounts$Difference <-  SJCounts$Normalized_CanonDE - mean_WT_SJi
```

```{r echo=FALSE}
table <- SJCounts[SJCounts$GROUP == "WT", c("sample_id", "Normalized_CanonDE", "Difference")]
table <- table[order(table$Difference, decreasing = TRUE),]
```

```{r echo=FALSE}
datatable(table,
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp'
))
```

Difference in the MUT sample: deviation of the Normalized expression of the MUT patient from the mean normalized WT expression
```{r}
SJCounts$Difference[SJCounts$GROUP == "MUT"]
```

```{r echo=FALSE}
ggplot(SJCounts, aes(sample_id,Difference)) + geom_point(aes(colour=factor(GROUP))) +
  ylab("Differential Expression") +
  labs(color = "GROUP",
       title = "Differential Expression Distribution")
```


```{r echo=FALSE}
ggplot(SJCounts, aes(y=Difference, x=factor(GROUP, level=c("WT", "MUT")))) + 
  geom_boxplot() +
  ylab("Differential Expression") +
  xlab("GROUP")
```

#### ECDF - Pvalue Inference 

```{r}
v_ecdf <- ecdf(SJCounts$Difference[SJCounts$GROUP == "WT"])
v_ecdf
```

```{r}
plot(ecdf(SJCounts$Difference[SJCounts$GROUP == "WT"]))
```

```{r}
v_ecdf(SJCounts$Difference[SJCounts$GROUP == "MUT"])
```


```{r}
MUT_df <- SJCounts[SJCounts$GROUP == "MUT",c("sample_id","case_id", "Normalized_CanonDE")] 
colnames(MUT_df) <- c("sample_id", "case_id", "NormalizedExpression")

MUT_df$ECDF <- v_ecdf(SJCounts$Difference[SJCounts$GROUP == "MUT"])
MUT_df$Pvalue <- MUT_df$ECDF

MUT_df$Prediction <- "Donor Loss"
MUT_df$splice_junction_status <- "CanonicalSJ"
MUT_df$splice_junction_position <- "chr17:7674291-7674858"

MUT_df$ECDF <- ifelse(MUT_df$NormalizedExpression == 0, NA,MUT_df$ECDF)
MUT_df$Pvalue <- ifelse(MUT_df$NormalizedExpression == 0, NA,MUT_df$Pvalue)
```

```{r echo=FALSE}
vaf_select <- annot_vaf[,c("Gene","MutationKey_Hg38","sample_id","RNA_Sample","DNA_Sample","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf", "VAF_RNA")]
MUT_vaf <- merge(MUT_df, vaf_select, by.x="sample_id", by.y = "RNA_Sample")
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id"] <- "RNA_Sample"
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id.y"] <- "sample_id"

col_order <- c("Gene", "MutationKey_Hg38", "Prediction","splice_junction_status","splice_junction_position","sample_id", "case_id","RNA_Sample","DNA_Sample","NormalizedExpression","ECDF", "Pvalue", "VAF_RNA","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf")
MUT_vaf <- MUT_vaf[, col_order]
```

Download the vaf, inferred percentiles and pvalues of the mutated samples:
```{r echo=FALSE}
datatable(MUT_vaf,
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp',
              scrollX=TRUE
))
```

```{r echo=FALSE}
summary_res <- rbind(summary_res,MUT_vaf)
```


### Exon Skipping {.tabset}

#### Expression Difference 

Normality Test:
```{r}
shapiro.test(SJCounts$Normalized_ES[SJCounts$GROUP == "WT"])
```

Value of Mean Normalized Expression of the Alternative SJ in WT samples:
```{r}
mean_WT_SJi <- mean(SJCounts$Normalized_ES[SJCounts$GROUP == "WT"], na.rm=TRUE)
mean_WT_SJi
```

Normalized Expression Value of the Alternative SJ in the MUT sample:
```{r}
MUT_SJi <- SJCounts$Normalized_ES[SJCounts$GROUP == "MUT"]
MUT_SJi
```
Deviation from the mean normalized expression:
```{r}
SJCounts$Difference <-  SJCounts$Normalized_ES - mean_WT_SJi
```

```{r echo=FALSE}
table <- SJCounts[SJCounts$GROUP == "WT", c("sample_id", "Normalized_ES", "Difference")]
table <- table[order(table$Difference, decreasing = TRUE),]
```

```{r echo=FALSE}
datatable(table,
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp'
))
```

Difference in the MUT sample: deviation of the Normalized expression of the MUT patient from the mean normalized WT expression
```{r}
SJCounts$Difference[SJCounts$GROUP == "MUT"]
```

```{r echo=FALSE}
ggplot(SJCounts, aes(sample_id,Difference)) + geom_point(aes(colour=factor(GROUP))) +
  ylab("Differential Expression") +
  labs(color = "GROUP",
       title = "Differential Expression Distribution")
```


```{r echo=FALSE}
ggplot(SJCounts, aes(y=Difference, x=factor(GROUP, level=c("WT", "MUT")))) + 
  geom_boxplot() +
  ylab("Differential Expression") +
  xlab("GROUP")
```

#### ECDF - Pvalue Inference 

```{r}
v_ecdf <- ecdf(SJCounts$Difference[SJCounts$GROUP == "WT"])
v_ecdf
```

```{r}
plot(ecdf(SJCounts$Difference[SJCounts$GROUP == "WT"]))
```

```{r}
v_ecdf(SJCounts$Difference[SJCounts$GROUP == "MUT"])
```


```{r}
MUT_df <- SJCounts[SJCounts$GROUP == "MUT",c("sample_id","case_id", "Normalized_ES")] 
colnames(MUT_df) <- c("sample_id", "case_id", "NormalizedExpression")

MUT_df$ECDF <- v_ecdf(SJCounts$Difference[SJCounts$GROUP == "MUT"])
MUT_df$Pvalue <- 1 - MUT_df$ECDF

MUT_df$Prediction <- "Exon Skipping"
MUT_df$splice_junction_status <- "AlternativeSJ found in MUT samples"
MUT_df$splice_junction_position <- "chr17:7674291-7675052"

MUT_df$ECDF <- ifelse(MUT_df$NormalizedExpression == 0, NA,MUT_df$ECDF)
MUT_df$Pvalue <- ifelse(MUT_df$NormalizedExpression == 0, NA,MUT_df$Pvalue)
```

```{r echo=FALSE}
vaf_select <- annot_vaf[,c("Gene","MutationKey_Hg38","sample_id","RNA_Sample","DNA_Sample","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf", "VAF_RNA")]
MUT_vaf <- merge(MUT_df, vaf_select, by.x="sample_id", by.y = "RNA_Sample")
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id"] <- "RNA_Sample"
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id.y"] <- "sample_id"

col_order <- c("Gene", "MutationKey_Hg38", "Prediction","splice_junction_status","splice_junction_position","sample_id", "case_id","RNA_Sample","DNA_Sample","NormalizedExpression","ECDF", "Pvalue", "VAF_RNA","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf")
MUT_vaf <- MUT_vaf[, col_order]
```

Download the vaf, inferred percentiles and pvalues of the mutated samples:
```{r echo=FALSE}
datatable(MUT_vaf,
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp',
              scrollX=TRUE
))
```

```{r echo=FALSE}
summary_res <- rbind(summary_res,MUT_vaf)
```

```{r echo=FALSE}
rm(GeneSJ, normalization_df, GeneSJ_subset, annot_vaf, plot_df, samples_df,table, plot_boxplot, plot_dotplot, vaf_select, SJCounts, MUT_df, MUT_vaf)
```


# ASXL1 chr20,32433787,C,T {.tabset}
Variant found in **1 patient** of the BeatAML (**1 sample**)

* Patients with ASXL1 chr20,32433787,C,T: **1 patient (1 sample)**
* Patients with the variant and RNASeq for validation: **1 patient (1 sample)**

The splicing alterations being assessed are: 

* **Donor Gain**: predicted at 2bp from the variant,chr20:32433786, not found in the splice junction collection.

Variant information:
```{r echo=FALSE}
datatable(to_be_validated[to_be_validated$MutationKey_Hg38 == "chr20,32433787,C,T",],
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp', 
              scrollX = TRUE,  
              scrollY = TRUE,
              columnDefs = list(list(targets = '_all', className = 'dt-center')))
)
```

Load the extracted splice junctions of the gene harboring the mutation.
```{r}
extractedSJ_path <- paste0(extractedSJ_dir_in,"ASXL1_UM_annotSJ.tsv")
GeneSJ <- read.delim(extractedSJ_path, sep ="\t")
```

Set the sample's group: Mutated (MUT) or No Mutated (WT)
```{r}
samples_df <- found_variants[found_variants$Gene=="ASXL1" & found_variants$MutationKey_Hg38 == "chr20,32433787,C,T",]

cases <- samples_df$RNA_Sample[samples_df$Validable == "Validable"]
GeneSJ$GROUP <- ifelse(GeneSJ$sample_id %in% cases , "MUT", "WT")
```


## SJ Lookup {.tabset}

Search for the splice junctions of interest in the extracted splice junctions of the gene by position (chr_SJstart_SJend).

### Donor Gain
Search: predicted at 2bp from the variant, chr20:32433786

Show all the splice junctions containing the positions between 32433780-32433789
```{r}
colnames(GeneSJ)[grep("3243378",colnames(GeneSJ))]
```
Alternative SJ not found in the splice junction collection.

## VAF
```{r echo=FALSE}
GeneSJ_subset <- GeneSJ[GeneSJ$sample_id %in% cases,]
GeneSJ_subset$Gene <- "ASXL1"
GeneSJ_subset$MutationKey_Hg38 <- "chr20,32433787,C,T"

annot_vaf <- merge(found_variants[,c("MutationKey_Hg38","Gene","sample_id","case_id",                                               "RNA_Sample","DNA_Sample","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf",                         "MuTect2_filter","VarScan2_filter","SomaticSniper_filter","MuSE_filter","MUTReads","WTReads","VAF")], 
                   GeneSJ_subset[,c("Gene","MutationKey_Hg38","sample_id", "case_id")], 
                   by.x=c("RNA_Sample", "case_id","Gene", "MutationKey_Hg38"), 
                   by.y=c("sample_id", "case_id","Gene", "MutationKey_Hg38"))


annot_vaf <- annot_vaf %>% select(Gene,MutationKey_Hg38,
                                  sample_id,
                                  case_id, RNA_Sample,DNA_Sample,
                                  MuTect2_vaf,VarScan2_vaf,SomaticSniper_vaf,MuSE_vaf,
                                  VAF,
                                  everything()
                                  )
colnames(annot_vaf)[grep("VAF", names(annot_vaf))] <- "VAF_RNA"
colnames(annot_vaf)[grep("MUTReads", names(annot_vaf))] <- "MUTReads_RNA"
colnames(annot_vaf)[grep("WTReads", names(annot_vaf))] <- "WTReads_RNA"

write.xlsx(annot_vaf, paste0(normalizedEx_dir_out, "BeatAML.ASXL1.chr20-32433787-C-T.MUT.xlsx"), rowNames = FALSE)
```

Mutated samples vaf:
```{r echo=FALSE}
datatable(annot_vaf,
extensions = 'Buttons',
rownames = FALSE, 
filter = 'top',
options = list(paging = TRUE,
              pageLength = 4,
              autoWidth = TRUE,
              dom = 'lifrtBp',
              buttons = c('excel'), 
              scrollX = TRUE,  
              scrollY = TRUE
              )
)
```


```{r echo=FALSE}
## Generate Mut_df
MUT_df <- GeneSJ[GeneSJ$GROUP == "MUT",c("sample_id","case_id")]
```

```{r echo=FALSE}
## Add not found effects information: Donor Gain 1
MUT_df <- MUT_df[,c("sample_id", "case_id")]
MUT_df$NormalizedExpression <- NA
MUT_df$Prediction <- "Donor Gain"
MUT_df$splice_junction_status <- "AlternativeSJ not found in SJ collection"
MUT_df$splice_junction_position <- "chr20:32433786"
MUT_df$Pvalue <- NA
MUT_df$ECDF <- NA

vaf_select <- annot_vaf[,c("Gene","MutationKey_Hg38","sample_id","RNA_Sample","DNA_Sample","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf", "VAF_RNA")]
MUT_vaf <- merge(MUT_df, vaf_select, by.x="sample_id", by.y = "RNA_Sample")
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id"] <- "RNA_Sample"
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id.y"] <- "sample_id"

col_order <- c("Gene", "MutationKey_Hg38", "Prediction","splice_junction_status","splice_junction_position","sample_id", "case_id","RNA_Sample","DNA_Sample","NormalizedExpression", "ECDF","Pvalue", "VAF_RNA","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf")
MUT_vaf <- MUT_vaf[, col_order]

## Save
summary_res <- rbind(summary_res, MUT_vaf)
```



# EP300 chr22,41160652,T,C {.tabset}
Variant found in **2 patients** of the BeatAML (**2 samples**)

* Patients with EP300 chr22,41160652,T,C: **2 patients (2 samples)**
* Patients with the variant and RNASeq for validation: **2 patients (2 samples)**

The splicing alterations being assessed are: 

* **Donor Gain**: predicted at the variant position, chr22:41160653, not found in the splice junction collection.

Variant information:
```{r echo=FALSE}
datatable(to_be_validated[to_be_validated$MutationKey_Hg38 == "chr22,41160652,T,C",],
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp', 
              scrollX = TRUE,  
              scrollY = TRUE,
              columnDefs = list(list(targets = '_all', className = 'dt-center')))
)
```

Load the extracted splice junctions of the gene harboring the mutation.
```{r}
extractedSJ_path <- paste0(extractedSJ_dir_in,"EP300_UM_annotSJ.tsv")
GeneSJ <- read.delim(extractedSJ_path, sep ="\t")
```

Set the sample's group: Mutated (MUT) or No Mutated (WT)
```{r}
samples_df <- found_variants[found_variants$Gene=="EP300" & found_variants$MutationKey_Hg38 == "chr22,41160652,T,C",]

cases <- samples_df$RNA_Sample[samples_df$Validable == "Validable"]
GeneSJ$GROUP <- ifelse(GeneSJ$sample_id %in% cases , "MUT", "WT")
```


## SJ Lookup {.tabset}

Search for the splice junctions of interest in the extracted splice junctions of the gene by position (chr_SJstart_SJend).

### Donor Gain
Search: predicted at the variant position, chr22:41160653

Show all the splice junctions containing the positions between 41160650-41160659
```{r}
colnames(GeneSJ)[grep("4116065",colnames(GeneSJ))]
```
Alternative SJ not found in the splice junction collection.


## VAF
```{r echo=FALSE}
GeneSJ_subset <- GeneSJ[GeneSJ$sample_id %in% cases,]
GeneSJ_subset$Gene <- "EP300"
GeneSJ_subset$MutationKey_Hg38 <- "chr22,41160652,T,C"

annot_vaf <- merge(found_variants[,c("MutationKey_Hg38","Gene","sample_id","case_id",                                               "RNA_Sample","DNA_Sample","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf",                         "MuTect2_filter","VarScan2_filter","SomaticSniper_filter","MuSE_filter","MUTReads","WTReads","VAF")], 
                   GeneSJ_subset[,c("Gene","MutationKey_Hg38","sample_id", "case_id")], 
                   by.x=c("RNA_Sample", "case_id","Gene", "MutationKey_Hg38"), 
                   by.y=c("sample_id", "case_id","Gene", "MutationKey_Hg38"))


annot_vaf <- annot_vaf %>% select(Gene,MutationKey_Hg38,
                                  sample_id,
                                  case_id, RNA_Sample,DNA_Sample,
                                  MuTect2_vaf,VarScan2_vaf,SomaticSniper_vaf,MuSE_vaf,
                                  VAF,
                                  everything()
                                  )
colnames(annot_vaf)[grep("VAF", names(annot_vaf))] <- "VAF_RNA"
colnames(annot_vaf)[grep("MUTReads", names(annot_vaf))] <- "MUTReads_RNA"
colnames(annot_vaf)[grep("WTReads", names(annot_vaf))] <- "WTReads_RNA"

write.xlsx(annot_vaf, paste0(normalizedEx_dir_out, "BeatAML.EP300.chr22-41160652-T-C.MUT.xlsx"), rowNames = FALSE)
```

Mutated samples vaf:
```{r echo=FALSE}
datatable(annot_vaf,
extensions = 'Buttons',
rownames = FALSE, 
filter = 'top',
options = list(paging = TRUE,
              pageLength = 4,
              autoWidth = TRUE,
              dom = 'lifrtBp',
              buttons = c('excel'), 
              scrollX = TRUE,  
              scrollY = TRUE
              )
)
```


```{r echo=FALSE}
## Generate Mut_df
MUT_df <- GeneSJ[GeneSJ$GROUP == "MUT",c("sample_id","case_id")]
```

```{r echo=FALSE}
## Add not found effects information: Donor Gain 1
MUT_df <- MUT_df[,c("sample_id", "case_id")]
MUT_df$NormalizedExpression <- NA
MUT_df$Prediction <- "Donor Gain"
MUT_df$splice_junction_status <- "AlternativeSJ not found in SJ collection"
MUT_df$splice_junction_position <- "chr22:41160653"
MUT_df$Pvalue <- NA
MUT_df$ECDF <- NA

vaf_select <- annot_vaf[,c("Gene","MutationKey_Hg38","sample_id","RNA_Sample","DNA_Sample","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf", "VAF_RNA")]
MUT_vaf <- merge(MUT_df, vaf_select, by.x="sample_id", by.y = "RNA_Sample")
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id"] <- "RNA_Sample"
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id.y"] <- "sample_id"

col_order <- c("Gene", "MutationKey_Hg38", "Prediction","splice_junction_status","splice_junction_position","sample_id", "case_id","RNA_Sample","DNA_Sample","NormalizedExpression", "ECDF" ,"Pvalue", "VAF_RNA","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf")
MUT_vaf <- MUT_vaf[, col_order]

## Save
summary_res <- rbind(summary_res, MUT_vaf)
```



# DNMT3A chr2,25239130,C,T {.tabset}
Variant found in **1 patient** of the BeatAML (**1 sample**)

* Patients with DNMT3A chr2,25239130,C,T variant: **1 patient (1 sample)**
* Patients with the variant and RNASeq for validation: **1 patient (1 sample)**

The splicing alterations being assessed are: 

* **Donor Loss**: chr2:25237006-25239129; donor splice site chr2:25239129.
* **Donor Gain**: chr2:25237006-25239139, found in the mutated samples.


Variant information:
```{r echo=FALSE}
datatable(to_be_validated[to_be_validated$MutationKey_Hg38 == "chr2,25239130,C,T",],
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp', 
              scrollX = TRUE,  
              scrollY = TRUE,
              columnDefs = list(list(targets = '_all', className = 'dt-center')))
)
```

Load the extracted splice junctions of the gene harboring the mutation.
```{r}
extractedSJ_path <- paste0(extractedSJ_dir_in,"DNMT3A_UM_annotSJ.tsv")
GeneSJ <- read.delim(extractedSJ_path, sep ="\t")
```

Set the sample's group: Mutated (MUT) or No Mutated (WT)
```{r}
samples_df <- found_variants[found_variants$Gene=="DNMT3A" & found_variants$MutationKey_Hg38 == "chr2,25239130,C,T",]

cases <- samples_df$RNA_Sample[samples_df$Validable == "Validable"]
GeneSJ$GROUP <- ifelse(GeneSJ$sample_id %in% cases , "MUT", "WT")
```


## SJ Lookup {.tabset}

Search for the splice junctions of interest in the extracted splice junctions of the gene by position (chr_SJstart_SJend).

### Donor Gain
Search: chr2:25237006-25239139

Show all the splice junctions containing the position chr2:25237006-25239139
```{r}
colnames(GeneSJ)[grep("25237006_25239139",colnames(GeneSJ))]
```
Found: chr2_25237006_25239139

Reads of all the AML samples (mutated and no mutated) for the splice junction:
```{r}
GeneSJ$chr2_25237006_25239139
```

Samples with the SJ of interest:
```{r}
table(GeneSJ$chr2_25237006_25239139>0) 
```

Groups of the samples having the alternative splice junction:
```{r}
table(GeneSJ$GROUP[GeneSJ$chr2_25237006_25239139 > 0])
```
Alternative SJ found in the mutated samples.

### Canonical SJ
Exon upstream (UE): chr2:25239216-25240301

Exon downstream (DE): chr2:25237006-25239129; donor splice site chr2:25239129 

Reads of all the AML samples (mutated and no mutated) for the splice junction:
```{r}
GeneSJ$chr2_25239216_25240301
```

Reads of all the AML samples (mutated and no mutated) for the splice junction:
```{r}
GeneSJ$chr2_25237006_25239129
```


## Normalization
Count the reads of all the splice junctions of the gene harboring the variant:
```{r}
GeneSJ$rowSum_SJtotal <- rowSums(GeneSJ[,grep("chr", names(GeneSJ))])
```

Normalization of the expression by the total read counts of all the splice junctions of the gene:
```{r}
GeneSJ$Normalized_CanonDE <- (GeneSJ$chr2_25237006_25239129)/GeneSJ$rowSum_SJtotal*100
GeneSJ$Normalized_DG <- (GeneSJ$chr2_25237006_25239139)/GeneSJ$rowSum_SJtotal*100
```

Download the normalized values for the assessed splice junctions of all the AML samples:
```{r echo=FALSE}
normalization_df <- GeneSJ[,c("sample_id","case_id", "GROUP","Normalized_CanonDE", "Normalized_DG", "chr2_25237006_25239129",  "chr2_25237006_25239139", "rowSum_SJtotal","file_id.BAM","file_name.BAM","file_id.STAR.SJCounts","file_name.STAR.SJCounts" )]

colnames(normalization_df)[grep("chr2_25237006_25239129", names(normalization_df))] <- "chr2_25237006_25239129_CanonDE"
colnames(normalization_df)[grep("chr2_25237006_25239139", names(normalization_df))] <- "chr2_25237006_25239139_DG"


write.xlsx(normalization_df, paste0(normalizedEx_dir_out, "BeatAML.DNMT3A.chr2-25239130-C-T.xlsx"), rowNames = FALSE)
```

```{r echo=FALSE}
datatable(normalization_df,
extensions = 'Buttons',
rownames = FALSE, 
filter = 'top',
options = list(paging = TRUE,
              pageLength = 4,
              autoWidth = FALSE,
              dom = 'lifrtBp',
              buttons = c('excel'), 
              scrollX = TRUE,  
              scrollY = TRUE,
              columnDefs = list(list(targets = '_all', className = 'dt-center'))
))
```

## VAF
```{r echo=FALSE}
GeneSJ_subset <- GeneSJ[GeneSJ$sample_id %in% cases,]
GeneSJ_subset$Gene <- "DNMT3A"
GeneSJ_subset$MutationKey_Hg38 <- "chr2,25239130,C,T"

annot_vaf <- merge(found_variants[,c("MutationKey_Hg38","Gene","sample_id","case_id",                                               "RNA_Sample","DNA_Sample","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf",                         "MuTect2_filter","VarScan2_filter","SomaticSniper_filter","MuSE_filter","MUTReads","WTReads","VAF")], 
                   GeneSJ_subset[,c("Gene","sample_id", "MutationKey_Hg38","case_id",  "Normalized_CanonDE", "Normalized_DG")], 
                   by.x=c("RNA_Sample", "case_id","Gene", "MutationKey_Hg38"), 
                   by.y=c("sample_id", "case_id","Gene", "MutationKey_Hg38"))


annot_vaf <- annot_vaf %>% select(Gene,MutationKey_Hg38,
                                  sample_id,
                                  case_id, RNA_Sample,DNA_Sample,
                                  MuTect2_vaf,VarScan2_vaf,SomaticSniper_vaf,MuSE_vaf,
                                  VAF,
                                  Normalized_CanonDE,Normalized_DG,
                                  everything()
                                  )

colnames(annot_vaf)[grep("VAF", names(annot_vaf))] <- "VAF_RNA"
colnames(annot_vaf)[grep("MUTReads", names(annot_vaf))] <- "MUTReads_RNA"
colnames(annot_vaf)[grep("WTReads", names(annot_vaf))] <- "WTReads_RNA"

write.xlsx(annot_vaf, paste0(normalizedEx_dir_out, "BeatAML.DNMT3A.chr2-25239130-C-T.MUT.xlsx"), rowNames = FALSE)
```

Mutated samples vaf:
```{r echo=FALSE}
datatable(annot_vaf,
extensions = 'Buttons',
rownames = FALSE, 
filter = 'top',
options = list(paging = TRUE,
              pageLength = 4,
              autoWidth = TRUE,
              dom = 'lifrtBp',
              buttons = c('excel'), 
              scrollX = TRUE,  
              scrollY = TRUE
              )
)
```

## Plots {.tabset}

```{r echo=FALSE}
GeneSJ$GROUP <- ifelse(GeneSJ$GROUP == "MUT", "MUT", "WT")
GeneSJ$GROUP <- as.factor(GeneSJ$GROUP)
```

### Static Dot Plots 

Canonical splice junction:
```{r echo=FALSE}
ggplot(GeneSJ, aes(sample_id,Normalized_CanonDE,color=GROUP)) +
  geom_point(size=.8) +
  labs(color='GROUP', 
       title = "Normalized Expression of DNMT3A chr2:25237006-25239129", 
       subtitle="Potential Donor Loss Effect on Canonical SJ",
       y = "Normalized Expression")
```

Splicing alterations:
```{r echo=FALSE}
ggplot(GeneSJ, aes(sample_id,Normalized_DG,color=GROUP)) +
  geom_point(size=.8) +
  labs(color='GROUP', 
       title = "Normalized Expression of DNMT3A chr2:25237006-25239139", 
       subtitle="Potential Donor Gain Effect",
       y = "Normalized Expression") 
```


### Interactive Dot Plots

```{r echo=FALSE}
plot_df <- merge(GeneSJ,annot_vaf[c("RNA_Sample", "VAF_RNA", 
                                    "MuTect2_vaf", "VarScan2_vaf", "SomaticSniper_vaf", "MuSE_vaf")],
                 by.x="sample_id", by.y="RNA_Sample", all.x = TRUE)
```


Canonical splice junction:
```{r echo=FALSE}
ggplotly(ggplot(plot_df, aes(sample_id,Normalized_CanonDE,color=GROUP)) + 
  geom_point(aes(text=paste0("VAF RNA: ",VAF_RNA,"\nVAF MuTect: ",MuTect2_vaf,
                             "\nVAF VarScan2: ",VarScan2_vaf,
                             "\nVAF SomaticSniper: ",SomaticSniper_vaf,
                             "\nVAF MuSE: ",MuSE_vaf)),size= 0.8) +
  labs(color='GROUP', 
       title = "Normalized Expression of DNMT3A chr2:25237006-25239129", 
       subtitle="Potential Donor Loss Effect on Canonical SJ",
       y = "Normalized Expression")
)
```

Splicing alteration:
```{r echo=FALSE}
ggplotly(ggplot(plot_df, aes(sample_id,Normalized_DG,color=GROUP)) + 
  geom_point(aes(text=paste0("VAF RNA: ",VAF_RNA,"\nVAF MuTect: ",MuTect2_vaf,
                             "\nVAF VarScan2: ",VarScan2_vaf,
                             "\nVAF SomaticSniper: ",SomaticSniper_vaf,
                             "\nVAF MuSE: ",MuSE_vaf)),size= 0.8) +
  labs(color='GROUP', 
       title = "Normalized Expression of DNMT3A chr2:25237006-25239139", 
       subtitle="Potential Donor Gain Effect",
       y = "Normalized Expression")
)
```


### Violin Plots
Violin Plots for the alternative splice junctions interrogated:

Donor Loss:
```{r echo=FALSE}
plot_boxplot <- plot_df[plot_df$GROUP %in% c("WT"),]
plot_dotplot <- plot_df[plot_df$GROUP %in% c("MUT"),]

GEN = "DNMT3A"
SJ = " chr2:25237006-25239129"

ggplot(plot_df, aes(y=Normalized_CanonDE, x=factor(GROUP, level=c("WT", "MUT")))) + 
       geom_violin(scale="area") +
       geom_point(data = plot_dotplot, 
                  aes(y=Normalized_CanonDE, x=GROUP),
                  color="black",
                  size=2
                  ) +
        
       ylab("Normalized SJ expression") +
       xlab(paste(GEN, SJ))
```

Exon Skipping:
```{r echo=FALSE}
plot_boxplot <- plot_df[plot_df$GROUP %in% c("WT"),]
plot_dotplot <- plot_df[plot_df$GROUP %in% c("MUT"),]

GEN = "TP53"
SJ = "chr17:7674291-7675052"

ggplot(plot_df, aes(y=Normalized_DG, x=factor(GROUP, level=c("WT", "MUT")))) + 
       geom_violin(scale="area") +
       geom_point(data = plot_dotplot, 
                  aes(y=Normalized_DG, x=GROUP),
                  color="black",
                  size=2
                  ) +
        
       ylab("Normalized SJ expression") +
       xlab(paste(GEN, SJ))
```

## Statistical Analysis {.tabset}

```{r}
SJCounts <- GeneSJ
```


### Donor Loss {.tabset}

#### Expression Difference 

Normality Test:
```{r}
shapiro.test(SJCounts$Normalized_CanonDE[SJCounts$GROUP == "WT"])
```

Value of Mean Normalized Expression of the Alternative SJ in WT samples:
```{r}
mean_WT_SJi <- mean(SJCounts$Normalized_CanonDE[SJCounts$GROUP == "WT"], na.rm=TRUE)
mean_WT_SJi
```

Normalized Expression Value of the Alternative SJ in the MUT sample:
```{r}
MUT_SJi <- SJCounts$Normalized_CanonDE[SJCounts$GROUP == "MUT"]
MUT_SJi
```
Deviation from the mean normalized expression:
```{r}
SJCounts$Difference <-  SJCounts$Normalized_CanonDE - mean_WT_SJi
```

```{r echo=FALSE}
table <- SJCounts[SJCounts$GROUP == "WT", c("sample_id", "Normalized_CanonDE", "Difference")]
table <- table[order(table$Difference, decreasing = TRUE),]
```

```{r echo=FALSE}
datatable(table,
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp'
))
```

Difference in the MUT sample: deviation of the Normalized expression of the MUT patient from the mean normalized WT expression
```{r}
SJCounts$Difference[SJCounts$GROUP == "MUT"]
```

```{r echo=FALSE}
ggplot(SJCounts, aes(sample_id,Difference)) + geom_point(aes(colour=factor(GROUP))) +
  ylab("Differential Expression") +
  labs(color = "GROUP",
       title = "Differential Expression Distribution")
```


```{r echo=FALSE}
ggplot(SJCounts, aes(y=Difference, x=factor(GROUP, level=c("WT", "MUT")))) + 
  geom_boxplot() +
  ylab("Differential Expression") +
  xlab("GROUP")
```

#### ECDF - Pvalue Inference 

```{r}
v_ecdf <- ecdf(SJCounts$Difference[SJCounts$GROUP == "WT"])
v_ecdf
```

```{r}
plot(ecdf(SJCounts$Difference[SJCounts$GROUP == "WT"]))
```

```{r}
v_ecdf(SJCounts$Difference[SJCounts$GROUP == "MUT"])
```


```{r}
MUT_df <- SJCounts[SJCounts$GROUP == "MUT",c("sample_id","case_id", "Normalized_CanonDE")]
colnames(MUT_df) <- c("sample_id", "case_id", "NormalizedExpression")

MUT_df$ECDF <- v_ecdf(SJCounts$Difference[SJCounts$GROUP == "MUT"])
MUT_df$Pvalue <- MUT_df$ECDF

MUT_df$Prediction <- "Donor Loss"
MUT_df$splice_junction_status <- "CanonicalSJ"
MUT_df$splice_junction_position <- "chr2:25237006-25239129"

MUT_df$ECDF <- ifelse(MUT_df$NormalizedExpression == 0, NA,MUT_df$ECDF)
MUT_df$Pvalue <- ifelse(MUT_df$NormalizedExpression == 0, NA,MUT_df$Pvalue)
```

```{r echo=FALSE}
vaf_select <- annot_vaf[,c("Gene","MutationKey_Hg38","sample_id","RNA_Sample","DNA_Sample","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf", "VAF_RNA")]
MUT_vaf <- merge(MUT_df, vaf_select, by.x="sample_id", by.y = "RNA_Sample")
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id"] <- "RNA_Sample"
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id.y"] <- "sample_id"

col_order <- c("Gene", "MutationKey_Hg38", "Prediction","splice_junction_status","splice_junction_position","sample_id", "case_id","RNA_Sample","DNA_Sample","NormalizedExpression","ECDF", "Pvalue", "VAF_RNA","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf")
MUT_vaf <- MUT_vaf[, col_order]
```

Download the vaf, inferred percentiles and pvalues of the mutated samples:
```{r echo=FALSE}
datatable(MUT_vaf,
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp',
              scrollX=TRUE
))
```

```{r echo=FALSE}
summary_res <- rbind(summary_res,MUT_vaf)
```


### Donor Gain {.tabset}

#### Expression Difference 

Normality Test:
```{r}
shapiro.test(SJCounts$Normalized_DG[SJCounts$GROUP == "WT"])
```

Value of Mean Normalized Expression of the Alternative SJ in WT samples:
```{r}
mean_WT_SJi <- mean(SJCounts$Normalized_DG[SJCounts$GROUP == "WT"], na.rm=TRUE)
mean_WT_SJi
```

Normalized Expression Value of the Alternative SJ in the MUT sample:
```{r}
MUT_SJi <- SJCounts$Normalized_DG[SJCounts$GROUP == "MUT"]
MUT_SJi
```
Deviation from the mean normalized expression:
```{r}
SJCounts$Difference <-  SJCounts$Normalized_DG - mean_WT_SJi
```

```{r echo=FALSE}
table <- SJCounts[SJCounts$GROUP == "WT", c("sample_id", "Normalized_DG", "Difference")]
table <- table[order(table$Difference, decreasing = TRUE),]
```

```{r echo=FALSE}
datatable(table,
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp'
))
```

Difference in the MUT sample: deviation of the Normalized expression of the MUT patient from the mean normalized WT expression
```{r}
SJCounts$Difference[SJCounts$GROUP == "MUT"]
```

```{r echo=FALSE}
ggplot(SJCounts, aes(sample_id,Difference)) + geom_point(aes(colour=factor(GROUP))) +
  ylab("Differential Expression") +
  labs(color = "GROUP",
       title = "Differential Expression Distribution")
```


```{r echo=FALSE}
ggplot(SJCounts, aes(y=Difference, x=factor(GROUP, level=c("WT", "MUT")))) + 
  geom_boxplot() +
  ylab("Differential Expression") +
  xlab("GROUP")
```

#### ECDF - Pvalue Inference 

```{r}
v_ecdf <- ecdf(SJCounts$Difference[SJCounts$GROUP == "WT"])
v_ecdf
```

```{r}
plot(ecdf(SJCounts$Difference[SJCounts$GROUP == "WT"]))
```

```{r}
v_ecdf(SJCounts$Difference[SJCounts$GROUP == "MUT"])
```


```{r}
MUT_df <- SJCounts[SJCounts$GROUP == "MUT",c("sample_id","case_id", "Normalized_DG")]
colnames(MUT_df) <- c("sample_id", "case_id", "NormalizedExpression")

MUT_df$ECDF <- v_ecdf(SJCounts$Difference[SJCounts$GROUP == "MUT"])
MUT_df$Pvalue <- 1 - MUT_df$ECDF

MUT_df$Prediction <- "Donor Gain"
MUT_df$splice_junction_status <- "AlternativeSJ found in MUT samples"
MUT_df$splice_junction_position <- "chr2:25237006-25239139"

MUT_df$ECDF <- ifelse(MUT_df$NormalizedExpression == 0, NA,MUT_df$ECDF)
MUT_df$Pvalue <- ifelse(MUT_df$NormalizedExpression == 0, NA,MUT_df$Pvalue)
```

```{r echo=FALSE}
vaf_select <- annot_vaf[,c("Gene","MutationKey_Hg38","sample_id","RNA_Sample","DNA_Sample","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf", "VAF_RNA")]
MUT_vaf <- merge(MUT_df, vaf_select, by.x="sample_id", by.y = "RNA_Sample")
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id"] <- "RNA_Sample"
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id.y"] <- "sample_id"

col_order <- c("Gene", "MutationKey_Hg38", "Prediction","splice_junction_status","splice_junction_position","sample_id", "case_id","RNA_Sample","DNA_Sample","NormalizedExpression","ECDF", "Pvalue", "VAF_RNA","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf")
MUT_vaf <- MUT_vaf[, col_order]
```

Download the vaf, inferred percentiles and pvalues of the mutated samples:
```{r echo=FALSE}
datatable(MUT_vaf,
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp',
              scrollX=TRUE
))
```

```{r echo=FALSE}
summary_res <- rbind(summary_res,MUT_vaf)
```

```{r echo=FALSE}
rm(GeneSJ, normalization_df, GeneSJ_subset, annot_vaf, plot_df, samples_df,table, plot_boxplot, plot_dotplot, vaf_select, SJCounts, MUT_df, MUT_vaf)
```


# DNMT3A chr2,25235792,T,C {.tabset}
Variant found in **1 patient** of the BeatAML (**1 sample**)

* Patients with DNMT3A chr2,25235792,T,C variant: **1 patient (1 sample)**
* Patients with the variant and RNASeq for validation: **1 patient (1 sample)**

The splicing alterations being assessed are: 

* **Donor Gain**: predicted at 5bp from the variant, chr2:25235796, not found in the splice junction collection.
* **Aceptor Gain**: chr2:25235821-25236935, found in the mutated samples.
* **Aceptor Loss**: chr2:25235826-25236935, aceptor splice site chr2:25238256.

Variant information:
```{r echo=FALSE}
datatable(to_be_validated[to_be_validated$MutationKey_Hg38 == "chr2,25235792,T,C",],
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp', 
              scrollX = TRUE,  
              scrollY = TRUE,
              columnDefs = list(list(targets = '_all', className = 'dt-center')))
)
```

Load the extracted splice junctions of the gene harboring the mutation.
```{r}
extractedSJ_path <- paste0(extractedSJ_dir_in,"DNMT3A_UM_annotSJ.tsv")
GeneSJ <- read.delim(extractedSJ_path, sep ="\t")
```

Set the sample's group: Mutated (MUT) or No Mutated (WT)
```{r}
samples_df <- found_variants[found_variants$Gene=="DNMT3A" & found_variants$MutationKey_Hg38 == "chr2,25235792,T,C",]

cases <- samples_df$RNA_Sample[samples_df$Validable == "Validable"]
GeneSJ$GROUP <- ifelse(GeneSJ$sample_id %in% cases , "MUT", "WT")
```


## SJ Lookup {.tabset}

Search for the splice junctions of interest in the extracted splice junctions of the gene by position (chr_SJstart_SJend).

### Donor Gain
Search: predicted at 5bp from the variant, chr2:25235796

Show all the splice junctions containing the positions between 25235790-25235799
```{r}
colnames(GeneSJ)[grep("2523579",colnames(GeneSJ))]
```
Alternative SJ not found in the splice junction collection.

### Aceptor Gain
Search: chr2:25235821-25236935

Show all the splice junctions containing the position chr2:25235821-25236935
```{r}
colnames(GeneSJ)[grep("25235821_25236935",colnames(GeneSJ))]
```
Found: chr2_25235821_25236935

Reads of all the AML samples (mutated and no mutated) for the splice junction:
```{r}
GeneSJ$chr2_25235821_25236935
```

Samples with the SJ of interest:
```{r}
table(GeneSJ$chr2_25235821_25236935>0) 
```

Groups of the samples having the alternative splice junction:
```{r}
table(GeneSJ$GROUP[GeneSJ$chr2_25235821_25236935 > 0])
```
Alternative SJ found in the mutated samples.

### Canonical SJ
Exon upstream (UE): chr2:25235826-25236935, aceptor splice site chr2:25238256

Exon downstream (DE): chr2:25234421-25235706, donor splice site chr2:25235706

Reads of all the AML samples (mutated and no mutated) for the splice junction:
```{r}
GeneSJ$chr2_25235826_25236935
```

Reads of all the AML samples (mutated and no mutated) for the splice junction:
```{r}
GeneSJ$chr2_25234421_25235706
```


## Normalization
Count the reads of all the splice junctions of the gene harboring the variant:
```{r}
GeneSJ$rowSum_SJtotal <- rowSums(GeneSJ[,grep("chr", names(GeneSJ))])
```

Normalization of the expression by the total read counts of all the splice junctions of the gene:
```{r}
GeneSJ$Normalized_CanonUE <- (GeneSJ$chr2_25235826_25236935)/GeneSJ$rowSum_SJtotal*100
GeneSJ$Normalized_CanonDE <- (GeneSJ$chr2_25234421_25235706)/GeneSJ$rowSum_SJtotal*100
GeneSJ$Normalized_AG <- (GeneSJ$chr2_25235821_25236935)/GeneSJ$rowSum_SJtotal*100
```

Download the normalized values for the assessed splice junctions of all the AML samples:
```{r echo=FALSE}
normalization_df <- GeneSJ[,c("sample_id","case_id", "GROUP","Normalized_CanonDE", "Normalized_CanonUE", "Normalized_AG","chr2_25235826_25236935", "chr2_25234421_25235706",  "chr2_25235821_25236935", "rowSum_SJtotal","file_id.BAM","file_name.BAM","file_id.STAR.SJCounts","file_name.STAR.SJCounts" )]

colnames(normalization_df)[grep("chr2_25235826_25236935", names(normalization_df))] <- "chr2_25235826_25236935_CanonUE"
colnames(normalization_df)[grep("chr2_25234421_25235706", names(normalization_df))] <- "chr2_25234421_25235706_CanonDE"
colnames(normalization_df)[grep("chr2_25235821_25236935", names(normalization_df))] <- "chr2_25235821_25236935_AG"


write.xlsx(normalization_df, paste0(normalizedEx_dir_out, "BeatAML.DNMT3A.chr2-25235792-T-C.xlsx"), rowNames = FALSE)
```

```{r echo=FALSE}
datatable(normalization_df,
extensions = 'Buttons',
rownames = FALSE, 
filter = 'top',
options = list(paging = TRUE,
              pageLength = 4,
              autoWidth = FALSE,
              dom = 'lifrtBp',
              buttons = c('excel'), 
              scrollX = TRUE,  
              scrollY = TRUE,
              columnDefs = list(list(targets = '_all', className = 'dt-center'))
))
```

## VAF
```{r echo=FALSE}
GeneSJ_subset <- GeneSJ[GeneSJ$sample_id %in% cases,]
GeneSJ_subset$Gene <- "DNMT3A"
GeneSJ_subset$MutationKey_Hg38 <- "chr2,25235792,T,C"

annot_vaf <- merge(found_variants[,c("MutationKey_Hg38","Gene","sample_id","case_id",                                               "RNA_Sample","DNA_Sample","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf",                         "MuTect2_filter","VarScan2_filter","SomaticSniper_filter","MuSE_filter","MUTReads","WTReads","VAF")], 
                   GeneSJ_subset[,c("Gene","sample_id", "MutationKey_Hg38","case_id",  "Normalized_CanonDE", "Normalized_CanonUE", "Normalized_AG")], 
                   by.x=c("RNA_Sample", "case_id","Gene", "MutationKey_Hg38"), 
                   by.y=c("sample_id", "case_id","Gene", "MutationKey_Hg38"))


annot_vaf <- annot_vaf %>% select(Gene,MutationKey_Hg38,
                                  sample_id,
                                  case_id, RNA_Sample,DNA_Sample,
                                  MuTect2_vaf,VarScan2_vaf,SomaticSniper_vaf,MuSE_vaf,
                                  VAF,
                                  Normalized_CanonDE, Normalized_CanonUE,Normalized_AG,
                                  everything()
                                  )

colnames(annot_vaf)[grep("VAF", names(annot_vaf))] <- "VAF_RNA"
colnames(annot_vaf)[grep("MUTReads", names(annot_vaf))] <- "MUTReads_RNA"
colnames(annot_vaf)[grep("WTReads", names(annot_vaf))] <- "WTReads_RNA"

write.xlsx(annot_vaf, paste0(normalizedEx_dir_out, "BeatAML.DNMT3A.chr2-25235792-T-C.MUT.xlsx"), rowNames = FALSE)
```

Mutated samples vaf:
```{r echo=FALSE}
datatable(annot_vaf,
extensions = 'Buttons',
rownames = FALSE, 
filter = 'top',
options = list(paging = TRUE,
              pageLength = 4,
              autoWidth = TRUE,
              dom = 'lifrtBp',
              buttons = c('excel'), 
              scrollX = TRUE,  
              scrollY = TRUE
              )
)
```

## Plots {.tabset}

```{r echo=FALSE}
GeneSJ$GROUP <- ifelse(GeneSJ$GROUP == "MUT", "MUT", "WT")
GeneSJ$GROUP <- as.factor(GeneSJ$GROUP)
```

### Static Dot Plots 

Canonical splice junction:
```{r echo=FALSE}
ggplot(GeneSJ, aes(sample_id,Normalized_CanonUE,color=GROUP)) +
  geom_point(size=.8) +
  labs(color='GROUP', 
       title = "Normalized Expression of DNMT3A chr2:25235826-25236935", 
       subtitle="Potential Aceptor Loss Effect on Canonical SJ",
       y = "Normalized Expression")
```

Splicing alterations:
```{r echo=FALSE}
ggplot(GeneSJ, aes(sample_id,Normalized_AG,color=GROUP)) +
  geom_point(size=.8) +
  labs(color='GROUP', 
       title = "Normalized Expression of DNMT3A chr2:25235821-25236935", 
       subtitle="Potential Aceptor Gain Effect",
       y = "Normalized Expression") 
```


### Interactive Dot Plots

```{r echo=FALSE}
plot_df <- merge(GeneSJ,annot_vaf[c("RNA_Sample", "VAF_RNA", 
                                    "MuTect2_vaf", "VarScan2_vaf", "SomaticSniper_vaf", "MuSE_vaf")],
                 by.x="sample_id", by.y="RNA_Sample", all.x = TRUE)
```


Canonical splice junction:
```{r echo=FALSE}
ggplotly(ggplot(plot_df, aes(sample_id,Normalized_CanonUE,color=GROUP)) + 
  geom_point(aes(text=paste0("VAF RNA: ",VAF_RNA,"\nVAF MuTect: ",MuTect2_vaf,
                             "\nVAF VarScan2: ",VarScan2_vaf,
                             "\nVAF SomaticSniper: ",SomaticSniper_vaf,
                             "\nVAF MuSE: ",MuSE_vaf)),size= 0.8) +
  labs(color='GROUP', 
       title = "Normalized Expression of DNMT3A chr2:25235826-25236935", 
       subtitle="Potential Aceptor Loss Effect on Canonical SJ",
       y = "Normalized Expression")
)
```

Splicing alteration:
```{r echo=FALSE}
ggplotly(ggplot(plot_df, aes(sample_id,Normalized_AG,color=GROUP)) + 
  geom_point(aes(text=paste0("VAF RNA: ",VAF_RNA,"\nVAF MuTect: ",MuTect2_vaf,
                             "\nVAF VarScan2: ",VarScan2_vaf,
                             "\nVAF SomaticSniper: ",SomaticSniper_vaf,
                             "\nVAF MuSE: ",MuSE_vaf)),size= 0.8) +
  labs(color='GROUP', 
       title = "Normalized Expression of DNMT3A chr2:25235821-25236935", 
       subtitle="Potential Donor Gain Effect",
       y = "Normalized Expression")
)
```


### Violin Plots
Violin Plots for the alternative splice junctions interrogated:

Aceptor Loss:
```{r echo=FALSE}
plot_boxplot <- plot_df[plot_df$GROUP %in% c("WT"),]
plot_dotplot <- plot_df[plot_df$GROUP %in% c("MUT"),]

GEN = "DNMT3A"
SJ = "chr2:25235826-25236935"

ggplot(plot_df, aes(y=Normalized_CanonUE, x=factor(GROUP, level=c("WT", "MUT")))) + 
       geom_violin(scale="area") +
       geom_point(data = plot_dotplot, 
                  aes(y=Normalized_CanonUE, x=GROUP),
                  color="black",
                  size=2
                  ) +
        
       ylab("Normalized SJ expression") +
       xlab(paste(GEN, SJ))
```

Aceptor Gain:
```{r echo=FALSE}
plot_boxplot <- plot_df[plot_df$GROUP %in% c("WT"),]
plot_dotplot <- plot_df[plot_df$GROUP %in% c("MUT"),]

GEN = "DNMT3A"
SJ = "chr2:25235821-25236935"

ggplot(plot_df, aes(y=Normalized_AG, x=factor(GROUP, level=c("WT", "MUT")))) + 
       geom_violin(scale="area") +
       geom_point(data = plot_dotplot, 
                  aes(y=Normalized_AG, x=GROUP),
                  color="black",
                  size=2
                  ) +
        
       ylab("Normalized SJ expression") +
       xlab(paste(GEN, SJ))
```

## Statistical Analysis {.tabset}

```{r}
SJCounts <- GeneSJ
```

### Aceptor Gain {.tabset}

#### Expression Difference 

Normality Test:
```{r}
shapiro.test(SJCounts$Normalized_AG[SJCounts$GROUP == "WT"])
```

Value of Mean Normalized Expression of the Alternative SJ in WT samples:
```{r}
mean_WT_SJi <- mean(SJCounts$Normalized_AG[SJCounts$GROUP == "WT"], na.rm=TRUE)
mean_WT_SJi
```

Normalized Expression Value of the Alternative SJ in the MUT sample:
```{r}
MUT_SJi <- SJCounts$Normalized_AG[SJCounts$GROUP == "MUT"]
MUT_SJi
```
Deviation from the mean normalized expression:
```{r}
SJCounts$Difference <-  SJCounts$Normalized_AG - mean_WT_SJi
```

```{r echo=FALSE}
table <- SJCounts[SJCounts$GROUP == "WT", c("sample_id", "Normalized_AG", "Difference")]
table <- table[order(table$Difference, decreasing = TRUE),]
```

```{r echo=FALSE}
datatable(table,
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp'
))
```

Difference in the MUT sample: deviation of the Normalized expression of the MUT patient from the mean normalized WT expression
```{r}
SJCounts$Difference[SJCounts$GROUP == "MUT"]
```

```{r echo=FALSE}
ggplot(SJCounts, aes(sample_id,Difference)) + geom_point(aes(colour=factor(GROUP))) +
  ylab("Differential Expression") +
  labs(color = "GROUP",
       title = "Differential Expression Distribution")
```


```{r echo=FALSE}
ggplot(SJCounts, aes(y=Difference, x=factor(GROUP, level=c("WT", "MUT")))) + 
  geom_boxplot() +
  ylab("Differential Expression") +
  xlab("GROUP")
```

#### ECDF - Pvalue Inference 

```{r}
v_ecdf <- ecdf(SJCounts$Difference[SJCounts$GROUP == "WT"])
v_ecdf
```

```{r}
plot(ecdf(SJCounts$Difference[SJCounts$GROUP == "WT"]))
```

```{r}
v_ecdf(SJCounts$Difference[SJCounts$GROUP == "MUT"])
```


```{r}
MUT_df <- SJCounts[SJCounts$GROUP == "MUT",c("sample_id","case_id", "Normalized_AG")]
colnames(MUT_df) <- c("sample_id", "case_id", "NormalizedExpression")

MUT_df$ECDF <- v_ecdf(SJCounts$Difference[SJCounts$GROUP == "MUT"])
MUT_df$Pvalue <- 1 - MUT_df$ECDF

MUT_df$Prediction <- "Aceptor Gain"
MUT_df$splice_junction_status <- "AlternativeSJ found in MUT samples"
MUT_df$splice_junction_position <- "chr2:25235821-25236935"

MUT_df$ECDF <- ifelse(MUT_df$NormalizedExpression == 0, NA,MUT_df$ECDF)
MUT_df$Pvalue <- ifelse(MUT_df$NormalizedExpression == 0, NA,MUT_df$Pvalue)
```

```{r echo=FALSE}
vaf_select <- annot_vaf[,c("Gene","MutationKey_Hg38","sample_id","RNA_Sample","DNA_Sample","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf", "VAF_RNA")]
MUT_vaf <- merge(MUT_df, vaf_select, by.x="sample_id", by.y = "RNA_Sample")
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id"] <- "RNA_Sample"
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id.y"] <- "sample_id"

col_order <- c("Gene", "MutationKey_Hg38", "Prediction","splice_junction_status","splice_junction_position","sample_id", "case_id","RNA_Sample","DNA_Sample","NormalizedExpression","ECDF", "Pvalue", "VAF_RNA","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf")
MUT_vaf <- MUT_vaf[, col_order]
```

Download the vaf, inferred percentiles and pvalues of the mutated samples:
```{r echo=FALSE}
datatable(MUT_vaf,
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp',
              scrollX=TRUE
))
```

```{r echo=FALSE}
summary_res <- rbind(summary_res,MUT_vaf)
```


### Aceptor Loss {.tabset}

#### Expression Difference 

Normality Test:
```{r}
shapiro.test(SJCounts$Normalized_CanonUE[SJCounts$GROUP == "WT"])
```

Value of Mean Normalized Expression of the Alternative SJ in WT samples:
```{r}
mean_WT_SJi <- mean(SJCounts$Normalized_CanonUE[SJCounts$GROUP == "WT"], na.rm=TRUE)
mean_WT_SJi
```

Normalized Expression Value of the Alternative SJ in the MUT sample:
```{r}
MUT_SJi <- SJCounts$Normalized_CanonUE[SJCounts$GROUP == "MUT"]
MUT_SJi
```
Deviation from the mean normalized expression:
```{r}
SJCounts$Difference <-  SJCounts$Normalized_CanonUE - mean_WT_SJi
```

```{r echo=FALSE}
table <- SJCounts[SJCounts$GROUP == "WT", c("sample_id", "Normalized_CanonUE", "Difference")]
table <- table[order(table$Difference, decreasing = TRUE),]
```

```{r echo=FALSE}
datatable(table,
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp'
))
```

Difference in the MUT sample: deviation of the Normalized expression of the MUT patient from the mean normalized WT expression
```{r}
SJCounts$Difference[SJCounts$GROUP == "MUT"]
```

```{r echo=FALSE}
ggplot(SJCounts, aes(sample_id,Difference)) + geom_point(aes(colour=factor(GROUP))) +
  ylab("Differential Expression") +
  labs(color = "GROUP",
       title = "Differential Expression Distribution")
```


```{r echo=FALSE}
ggplot(SJCounts, aes(y=Difference, x=factor(GROUP, level=c("WT", "MUT")))) + 
  geom_boxplot() +
  ylab("Differential Expression") +
  xlab("GROUP")
```

#### ECDF - Pvalue Inference 

```{r}
v_ecdf <- ecdf(SJCounts$Difference[SJCounts$GROUP == "WT"])
v_ecdf
```

```{r}
plot(ecdf(SJCounts$Difference[SJCounts$GROUP == "WT"]))
```

```{r}
v_ecdf(SJCounts$Difference[SJCounts$GROUP == "MUT"])
```


```{r}
MUT_df <- SJCounts[SJCounts$GROUP == "MUT",c("sample_id","case_id", "Normalized_CanonUE")]
colnames(MUT_df) <- c("sample_id", "case_id", "NormalizedExpression")

MUT_df$ECDF <- v_ecdf(SJCounts$Difference[SJCounts$GROUP == "MUT"])
MUT_df$Pvalue <- MUT_df$ECDF

MUT_df$Prediction <- "Aceptor Loss"
MUT_df$splice_junction_status <- "CanonicalSJ"
MUT_df$splice_junction_position <- "chr2:25235826-25236935"

MUT_df$ECDF <- ifelse(MUT_df$NormalizedExpression == 0, NA,MUT_df$ECDF)
MUT_df$Pvalue <- ifelse(MUT_df$NormalizedExpression == 0, NA,MUT_df$Pvalue)
```

```{r echo=FALSE}
vaf_select <- annot_vaf[,c("Gene","MutationKey_Hg38","sample_id","RNA_Sample","DNA_Sample","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf", "VAF_RNA")]
MUT_vaf <- merge(MUT_df, vaf_select, by.x="sample_id", by.y = "RNA_Sample")
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id"] <- "RNA_Sample"
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id.y"] <- "sample_id"

col_order <- c("Gene", "MutationKey_Hg38", "Prediction","splice_junction_status","splice_junction_position","sample_id", "case_id","RNA_Sample","DNA_Sample","NormalizedExpression","ECDF", "Pvalue", "VAF_RNA","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf")
MUT_vaf <- MUT_vaf[, col_order]
```

Download the vaf, inferred percentiles and pvalues of the mutated samples:
```{r echo=FALSE}
datatable(MUT_vaf,
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp',
              scrollX=TRUE
))
```

```{r echo=FALSE}
summary_res <- rbind(summary_res,MUT_vaf)
```

### Donor Loss {.tabset}

#### Expression Difference 

Normality Test:
```{r}
shapiro.test(SJCounts$Normalized_CanonDE[SJCounts$GROUP == "WT"])
```

Value of Mean Normalized Expression of the Alternative SJ in WT samples:
```{r}
mean_WT_SJi <- mean(SJCounts$Normalized_CanonDE[SJCounts$GROUP == "WT"], na.rm=TRUE)
mean_WT_SJi
```

Normalized Expression Value of the Alternative SJ in the MUT sample:
```{r}
MUT_SJi <- SJCounts$Normalized_CanonDE[SJCounts$GROUP == "MUT"]
MUT_SJi
```
Deviation from the mean normalized expression:
```{r}
SJCounts$Difference <-  SJCounts$Normalized_CanonDE - mean_WT_SJi
```

```{r echo=FALSE}
table <- SJCounts[SJCounts$GROUP == "WT", c("sample_id", "Normalized_CanonDE", "Difference")]
table <- table[order(table$Difference, decreasing = TRUE),]
```

```{r echo=FALSE}
datatable(table,
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp'
))
```

Difference in the MUT sample: deviation of the Normalized expression of the MUT patient from the mean normalized WT expression
```{r}
SJCounts$Difference[SJCounts$GROUP == "MUT"]
```

```{r echo=FALSE}
ggplot(SJCounts, aes(sample_id,Difference)) + geom_point(aes(colour=factor(GROUP))) +
  ylab("Differential Expression") +
  labs(color = "GROUP",
       title = "Differential Expression Distribution")
```


```{r echo=FALSE}
ggplot(SJCounts, aes(y=Difference, x=factor(GROUP, level=c("WT", "MUT")))) + 
  geom_boxplot() +
  ylab("Differential Expression") +
  xlab("GROUP")
```

#### ECDF - Pvalue Inference 

```{r}
v_ecdf <- ecdf(SJCounts$Difference[SJCounts$GROUP == "WT"])
v_ecdf
```

```{r}
plot(ecdf(SJCounts$Difference[SJCounts$GROUP == "WT"]))
```

```{r}
v_ecdf(SJCounts$Difference[SJCounts$GROUP == "MUT"])
```


```{r}
MUT_df <- SJCounts[SJCounts$GROUP == "MUT",c("sample_id","case_id", "Normalized_CanonDE")]
colnames(MUT_df) <- c("sample_id", "case_id", "NormalizedExpression")

MUT_df$ECDF <- v_ecdf(SJCounts$Difference[SJCounts$GROUP == "MUT"])
MUT_df$Pvalue <- MUT_df$ECDF

MUT_df$Prediction <- "Donor Loss"
MUT_df$splice_junction_status <- "CanonicalSJ"
MUT_df$splice_junction_position <- "chr2:25234421-25235706"

MUT_df$ECDF <- ifelse(MUT_df$NormalizedExpression == 0, NA,MUT_df$ECDF)
MUT_df$Pvalue <- ifelse(MUT_df$NormalizedExpression == 0, NA,MUT_df$Pvalue)
```

```{r echo=FALSE}
vaf_select <- annot_vaf[,c("Gene","MutationKey_Hg38","sample_id","RNA_Sample","DNA_Sample","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf", "VAF_RNA")]
MUT_vaf <- merge(MUT_df, vaf_select, by.x="sample_id", by.y = "RNA_Sample")
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id"] <- "RNA_Sample"
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id.y"] <- "sample_id"

col_order <- c("Gene", "MutationKey_Hg38", "Prediction","splice_junction_status","splice_junction_position","sample_id", "case_id","RNA_Sample","DNA_Sample","NormalizedExpression","ECDF", "Pvalue", "VAF_RNA","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf")
MUT_vaf <- MUT_vaf[, col_order]
```

Download the vaf, inferred percentiles and pvalues of the mutated samples:
```{r echo=FALSE}
datatable(MUT_vaf,
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp',
              scrollX=TRUE
))
```

```{r echo=FALSE}
summary_res <- rbind(summary_res,MUT_vaf)
```


```{r echo=FALSE}
## Add not found effects information:
MUT_df <- MUT_df[,c("sample_id", "case_id")]
MUT_df$NormalizedExpression <- NA
MUT_df$Prediction <- "Donor Gain"
MUT_df$splice_junction_status <- "AlternativeSJ not found in SJ collection"
MUT_df$splice_junction_position <- "chr2:25235796"
MUT_df$Pvalue <- NA
MUT_df$ECDF <- NA

vaf_select <- annot_vaf[,c("Gene","MutationKey_Hg38","sample_id","RNA_Sample","DNA_Sample","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf", "VAF_RNA")]
MUT_vaf <- merge(MUT_df, vaf_select, by.x="sample_id", by.y = "RNA_Sample")
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id"] <- "RNA_Sample"
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id.y"] <- "sample_id"

col_order <- c("Gene", "MutationKey_Hg38", "Prediction","splice_junction_status","splice_junction_position","sample_id", "case_id","RNA_Sample","DNA_Sample","NormalizedExpression","ECDF", "Pvalue", "VAF_RNA","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf")
MUT_vaf <- MUT_vaf[, col_order]

## Save
summary_res <- rbind(summary_res, MUT_vaf)
```

```{r echo=FALSE}
rm(GeneSJ, normalization_df, GeneSJ_subset, annot_vaf, plot_df, samples_df,table, plot_boxplot, plot_dotplot, vaf_select, SJCounts, MUT_df, MUT_vaf)
```


# DNMT3A chr2,25244214,G,A {.tabset}
Variant found in **1 patient** of the BeatAML (**1 sample**)

* Patients with DNMT3A chr2,25244214,G,A: **1 patient (1 sample)**
* Patients with the variant and RNASeq for validation: **1 patient (1 sample)**

The splicing alterations being assessed are: 

* **Donor Gain**: predicted at 2bp from the variant, chr2:25244215, not found in the splice junction collection.

Variant information:
```{r echo=FALSE}
datatable(to_be_validated[to_be_validated$MutationKey_Hg38 == "chr2,25244214,G,A",],
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp', 
              scrollX = TRUE,  
              scrollY = TRUE,
              columnDefs = list(list(targets = '_all', className = 'dt-center')))
)
```

Load the extracted splice junctions of the gene harboring the mutation.
```{r}
extractedSJ_path <- paste0(extractedSJ_dir_in,"DNMT3A_UM_annotSJ.tsv")
GeneSJ <- read.delim(extractedSJ_path, sep ="\t")
```

Set the sample's group: Mutated (MUT) or No Mutated (WT)
```{r}
samples_df <- found_variants[found_variants$Gene=="DNMT3A" & found_variants$MutationKey_Hg38 == "chr2,25244214,G,A",]

cases <- samples_df$RNA_Sample[samples_df$Validable == "Validable"]
GeneSJ$GROUP <- ifelse(GeneSJ$sample_id %in% cases , "MUT", "WT")
```


## SJ Lookup {.tabset}

Search for the splice junctions of interest in the extracted splice junctions of the gene by position (chr_SJstart_SJend).

### Donor Gain
Search: predicted at 2bp from the variant, chr2:25244215

Show all the splice junctions containing the positions between 25244210-25244219
```{r}
colnames(GeneSJ)[grep("2524421",colnames(GeneSJ))]
```
Alternative SJ not found in the splice junction collection.


## VAF
```{r echo=FALSE}
GeneSJ_subset <- GeneSJ[GeneSJ$sample_id %in% cases,]
GeneSJ_subset$Gene <- "DNMT3A"
GeneSJ_subset$MutationKey_Hg38 <- "chr2,25244214,G,A"

annot_vaf <- merge(found_variants[,c("MutationKey_Hg38","Gene","sample_id","case_id",                                               "RNA_Sample","DNA_Sample","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf",                         "MuTect2_filter","VarScan2_filter","SomaticSniper_filter","MuSE_filter","MUTReads","WTReads","VAF")], 
                   GeneSJ_subset[,c("Gene","MutationKey_Hg38","sample_id", "case_id")], 
                   by.x=c("RNA_Sample", "case_id","Gene", "MutationKey_Hg38"), 
                   by.y=c("sample_id", "case_id","Gene", "MutationKey_Hg38"))


annot_vaf <- annot_vaf %>% select(Gene,MutationKey_Hg38,
                                  sample_id,
                                  case_id, RNA_Sample,DNA_Sample,
                                  MuTect2_vaf,VarScan2_vaf,SomaticSniper_vaf,MuSE_vaf,
                                  VAF,
                                  everything()
                                  )
colnames(annot_vaf)[grep("VAF", names(annot_vaf))] <- "VAF_RNA"
colnames(annot_vaf)[grep("MUTReads", names(annot_vaf))] <- "MUTReads_RNA"
colnames(annot_vaf)[grep("WTReads", names(annot_vaf))] <- "WTReads_RNA"

write.xlsx(annot_vaf, paste0(normalizedEx_dir_out, "BeatAML.DNMT3A.chr2-25244214-G-A.MUT.xlsx"), rowNames = FALSE)
```

Mutated samples vaf:
```{r echo=FALSE}
datatable(annot_vaf,
extensions = 'Buttons',
rownames = FALSE, 
filter = 'top',
options = list(paging = TRUE,
              pageLength = 4,
              autoWidth = TRUE,
              dom = 'lifrtBp',
              buttons = c('excel'), 
              scrollX = TRUE,  
              scrollY = TRUE
              )
)
```


```{r echo=FALSE}
## Generate Mut_df
MUT_df <- GeneSJ[GeneSJ$GROUP == "MUT",c("sample_id","case_id")]
```

```{r echo=FALSE}
## Add not found effects information: Donor Gain 1
MUT_df <- MUT_df[,c("sample_id", "case_id")]
MUT_df$NormalizedExpression <- NA
MUT_df$Prediction <- "Donor Gain"
MUT_df$splice_junction_status <- "AlternativeSJ not found in SJ collection"
MUT_df$splice_junction_position <- "chr2:25244215"
MUT_df$Pvalue <- NA
MUT_df$ECDF <- NA

vaf_select <- annot_vaf[,c("Gene","MutationKey_Hg38","sample_id","RNA_Sample","DNA_Sample","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf", "VAF_RNA")]
MUT_vaf <- merge(MUT_df, vaf_select, by.x="sample_id", by.y = "RNA_Sample")
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id"] <- "RNA_Sample"
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id.y"] <- "sample_id"

col_order <- c("Gene", "MutationKey_Hg38", "Prediction","splice_junction_status","splice_junction_position","sample_id", "case_id","RNA_Sample","DNA_Sample","NormalizedExpression", "ECDF", "Pvalue", "VAF_RNA","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf")
MUT_vaf <- MUT_vaf[, col_order]

## Save
summary_res <- rbind(summary_res, MUT_vaf)
```

```{r echo=FALSE}
rm(GeneSJ, normalization_df, GeneSJ_subset, annot_vaf, plot_df, samples_df,table, plot_boxplot, plot_dotplot, vaf_select, SJCounts, MUT_df, MUT_vaf)
```


# DNMT3A chr2,25240693,C,A {.tabset}
Variant found in **1 patient** of the BeatAML (**1 sample**)

* Patients with DNMT3A chr2,25240693,C,A variant: **1 patient (1 sample)**
* Patients with the variant and RNASeq for validation: **1 patient (1 sample)**

The splicing alterations being assessed are: 

* **Donor Gain**: predicted at 2bp from the variant, chr2:25240694, not found in the splice junction collection.

Variant information:
```{r echo=FALSE}
datatable(to_be_validated[to_be_validated$MutationKey_Hg38 == "chr2,25240693,C,A",],
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp', 
              scrollX = TRUE,  
              scrollY = TRUE,
              columnDefs = list(list(targets = '_all', className = 'dt-center')))
)
```

Load the extracted splice junctions of the gene harboring the mutation.
```{r}
extractedSJ_path <- paste0(extractedSJ_dir_in,"DNMT3A_UM_annotSJ.tsv")
GeneSJ <- read.delim(extractedSJ_path, sep ="\t")
```

Set the sample's group: Mutated (MUT) or No Mutated (WT)
```{r}
samples_df <- found_variants[found_variants$Gene=="DNMT3A" & found_variants$MutationKey_Hg38 == "chr2,25240693,C,A",]

cases <- samples_df$RNA_Sample[samples_df$Validable == "Validable"]
GeneSJ$GROUP <- ifelse(GeneSJ$sample_id %in% cases , "MUT", "WT")
```


## SJ Lookup {.tabset}

Search for the splice junctions of interest in the extracted splice junctions of the gene by position (chr_SJstart_SJend).

### Donor Gain
Search: predicted at 2bp from the variant, chr2:25240694

Show all the splice junctions containing the positions between 25240690-25240699
```{r}
colnames(GeneSJ)[grep("2524069",colnames(GeneSJ))]
```
Alternative SJ not found in the splice junction collection.

## VAF
```{r echo=FALSE}
GeneSJ_subset <- GeneSJ[GeneSJ$sample_id %in% cases,]
GeneSJ_subset$Gene <- "DNMT3A"
GeneSJ_subset$MutationKey_Hg38 <- "chr2,25240693,C,A"

annot_vaf <- merge(found_variants[,c("MutationKey_Hg38","Gene","sample_id","case_id",                                               "RNA_Sample","DNA_Sample","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf",                         "MuTect2_filter","VarScan2_filter","SomaticSniper_filter","MuSE_filter","MUTReads","WTReads","VAF")], 
                   GeneSJ_subset[,c("Gene","MutationKey_Hg38","sample_id", "case_id")], 
                   by.x=c("RNA_Sample", "case_id","Gene", "MutationKey_Hg38"), 
                   by.y=c("sample_id", "case_id","Gene", "MutationKey_Hg38"))


annot_vaf <- annot_vaf %>% select(Gene,MutationKey_Hg38,
                                  sample_id,
                                  case_id, RNA_Sample,DNA_Sample,
                                  MuTect2_vaf,VarScan2_vaf,SomaticSniper_vaf,MuSE_vaf,
                                  VAF,
                                  everything()
                                  )
colnames(annot_vaf)[grep("VAF", names(annot_vaf))] <- "VAF_RNA"
colnames(annot_vaf)[grep("MUTReads", names(annot_vaf))] <- "MUTReads_RNA"
colnames(annot_vaf)[grep("WTReads", names(annot_vaf))] <- "WTReads_RNA"

write.xlsx(annot_vaf, paste0(normalizedEx_dir_out, "BeatAML.DNMT3A.chr2-25240693-C-A.MUT.xlsx"), rowNames = FALSE)
```

Mutated samples vaf:
```{r echo=FALSE}
datatable(annot_vaf,
extensions = 'Buttons',
rownames = FALSE, 
filter = 'top',
options = list(paging = TRUE,
              pageLength = 4,
              autoWidth = TRUE,
              dom = 'lifrtBp',
              buttons = c('excel'), 
              scrollX = TRUE,  
              scrollY = TRUE
              )
)
```


```{r echo=FALSE}
## Generate Mut_df
MUT_df <- GeneSJ[GeneSJ$GROUP == "MUT",c("sample_id","case_id")]
```

```{r echo=FALSE}
## Add not found effects information: Donor Gain 1
MUT_df <- MUT_df[,c("sample_id", "case_id")]
MUT_df$NormalizedExpression <- NA
MUT_df$Prediction <- "Donor Gain"
MUT_df$splice_junction_status <- "AlternativeSJ not found in SJ collection"
MUT_df$splice_junction_position <- "chr2:25240694"
MUT_df$Pvalue <- NA
MUT_df$ECDF <- NA

vaf_select <- annot_vaf[,c("Gene","MutationKey_Hg38","sample_id","RNA_Sample","DNA_Sample","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf", "VAF_RNA")]
MUT_vaf <- merge(MUT_df, vaf_select, by.x="sample_id", by.y = "RNA_Sample")
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id"] <- "RNA_Sample"
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id.y"] <- "sample_id"

col_order <- c("Gene", "MutationKey_Hg38", "Prediction","splice_junction_status","splice_junction_position","sample_id", "case_id","RNA_Sample","DNA_Sample","NormalizedExpression","ECDF","Pvalue", "VAF_RNA","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf")
MUT_vaf <- MUT_vaf[, col_order]

## Save
summary_res <- rbind(summary_res, MUT_vaf)
```

```{r echo=FALSE}
rm(GeneSJ, normalization_df, GeneSJ_subset, annot_vaf, plot_df, samples_df,table, plot_boxplot, plot_dotplot, vaf_select, SJCounts, MUT_df, MUT_vaf)
```


# KMT2D	chr12,49050056,G,A {.tabset}
Variant found in **1 patient** of the BeatAML (**1 sample**)

* Patients with KMT2D	chr12,49050056,G,A: **1 patient (1 sample)**
* Patients with the variant and RNASeq for validation: **1 patient (1 sample)**

The splicing alterations being assessed are: 

* **Donor Gain**: predicted at 2bp from the variant, chr12:49050057, not found in the splice junction collection.

Variant information:
```{r echo=FALSE}
datatable(to_be_validated[to_be_validated$MutationKey_Hg38 == "chr12,49050056,G,A",],
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp', 
              scrollX = TRUE,  
              scrollY = TRUE,
              columnDefs = list(list(targets = '_all', className = 'dt-center')))
)
```

Load the extracted splice junctions of the gene harboring the mutation.
```{r}
extractedSJ_path <- paste0(extractedSJ_dir_in,"KMT2D_UM_annotSJ.tsv")
GeneSJ <- read.delim(extractedSJ_path, sep ="\t")
```

Set the sample's group: Mutated (MUT) or No Mutated (WT)
```{r}
samples_df <- found_variants[found_variants$Gene=="KMT2D" & found_variants$MutationKey_Hg38 == "chr12,49050056,G,A",]

cases <- samples_df$RNA_Sample[samples_df$Validable == "Validable"]
GeneSJ$GROUP <- ifelse(GeneSJ$sample_id %in% cases , "MUT", "WT")
```


## SJ Lookup {.tabset}

Search for the splice junctions of interest in the extracted splice junctions of the gene by position (chr_SJstart_SJend).

### Donor Gain
Search: predicted at 2bp from the variant, chr12:49050057

Show all the splice junctions containing the positions between 49050050-49050059
```{r}
colnames(GeneSJ)[grep("49050056",colnames(GeneSJ))]
```
Alternative SJ not found in the splice junction collection.



## VAF
```{r echo=FALSE}
GeneSJ_subset <- GeneSJ[GeneSJ$sample_id %in% cases,]
GeneSJ_subset$Gene <- "KMT2D"
GeneSJ_subset$MutationKey_Hg38 <- "chr12,49050056,G,A"

annot_vaf <- merge(found_variants[,c("MutationKey_Hg38","Gene","sample_id","case_id",                                               "RNA_Sample","DNA_Sample","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf",                         "MuTect2_filter","VarScan2_filter","SomaticSniper_filter","MuSE_filter","MUTReads","WTReads","VAF")], 
                   GeneSJ_subset[,c("Gene","MutationKey_Hg38","sample_id", "case_id")], 
                   by.x=c("RNA_Sample", "case_id","Gene", "MutationKey_Hg38"), 
                   by.y=c("sample_id", "case_id","Gene", "MutationKey_Hg38"))


annot_vaf <- annot_vaf %>% select(Gene,MutationKey_Hg38,
                                  sample_id,
                                  case_id, RNA_Sample,DNA_Sample,
                                  MuTect2_vaf,VarScan2_vaf,SomaticSniper_vaf,MuSE_vaf,
                                  VAF,
                                  everything()
                                  )
colnames(annot_vaf)[grep("VAF", names(annot_vaf))] <- "VAF_RNA"
colnames(annot_vaf)[grep("MUTReads", names(annot_vaf))] <- "MUTReads_RNA"
colnames(annot_vaf)[grep("WTReads", names(annot_vaf))] <- "WTReads_RNA"

write.xlsx(annot_vaf, paste0(normalizedEx_dir_out, "BeatAML.KMT2D.chr12-49050056-G-A.MUT.xlsx"), rowNames = FALSE)
```

Mutated samples vaf:
```{r echo=FALSE}
datatable(annot_vaf,
extensions = 'Buttons',
rownames = FALSE, 
filter = 'top',
options = list(paging = TRUE,
              pageLength = 4,
              autoWidth = TRUE,
              dom = 'lifrtBp',
              buttons = c('excel'), 
              scrollX = TRUE,  
              scrollY = TRUE
              )
)
```


```{r echo=FALSE}
## Generate Mut_df
MUT_df <- GeneSJ[GeneSJ$GROUP == "MUT",c("sample_id","case_id")]
```

```{r echo=FALSE}
## Add not found effects information: Donor Gain 1
MUT_df <- MUT_df[,c("sample_id", "case_id")]
MUT_df$NormalizedExpression <- NA
MUT_df$Prediction <- "Donor Gain"
MUT_df$splice_junction_status <- "AlternativeSJ not found in SJ collection"
MUT_df$splice_junction_position <- "chr12:49050057"
MUT_df$Pvalue <- NA
MUT_df$ECDF <- NA

vaf_select <- annot_vaf[,c("Gene","MutationKey_Hg38","sample_id","RNA_Sample","DNA_Sample","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf", "VAF_RNA")]
MUT_vaf <- merge(MUT_df, vaf_select, by.x="sample_id", by.y = "RNA_Sample")
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id"] <- "RNA_Sample"
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id.y"] <- "sample_id"

col_order <- c("Gene", "MutationKey_Hg38", "Prediction","splice_junction_status","splice_junction_position","sample_id", "case_id","RNA_Sample","DNA_Sample","NormalizedExpression", "ECDF","Pvalue", "VAF_RNA","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf")
MUT_vaf <- MUT_vaf[, col_order]

## Save
summary_res <- rbind(summary_res, MUT_vaf)
```

```{r echo=FALSE}
rm(GeneSJ, normalization_df, GeneSJ_subset, annot_vaf, plot_df, samples_df,table, plot_boxplot, plot_dotplot, vaf_select, SJCounts, MUT_df, MUT_vaf)
```


# KMT2D chr12,49031255,G,A {.tabset}
Variant found in **3 patients** of the BeatAML (**3 samples**)

* Patients with KMT2D chr12,49031255,G,A variant: **3 patients (3 samples)**
* Patients with the variant and RNASeq for validation: **3 patients (3 samples)**

The splicing alterations being assessed are: 

* **Donor Gain**: predicted at 2bp from the variant, chr12:49031256, not found in the splice junction collection.
* **Donor Loss**: chr12:49031034-49031174; donor splice site chr12:49031174.
* **Aceptor Gain**: chr12:49031313-49032512, found in the mutated samples.
* **Aceptor Loss**: chr12:49033965-49034066; aceptor splice site chr12:49033965.


Variant information:
```{r echo=FALSE}
datatable(to_be_validated[to_be_validated$MutationKey_Hg38 == "chr12,49031255,G,A",],
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp', 
              scrollX = TRUE,  
              scrollY = TRUE,
              columnDefs = list(list(targets = '_all', className = 'dt-center')))
)
```

Load the extracted splice junctions of the gene harboring the mutation.
```{r}
extractedSJ_path <- paste0(extractedSJ_dir_in,"KMT2D_UM_annotSJ.tsv")
GeneSJ <- read.delim(extractedSJ_path, sep ="\t")
```

Set the sample's group: Mutated (MUT) or No Mutated (WT)
```{r}
samples_df <- found_variants[found_variants$Gene=="KMT2D" & found_variants$MutationKey_Hg38 == "chr12,49031255,G,A",]

cases <- samples_df$RNA_Sample[samples_df$Validable == "Validable"]
GeneSJ$GROUP <- ifelse(GeneSJ$sample_id %in% cases , "MUT", "WT")
```


## SJ Lookup {.tabset}

Search for the splice junctions of interest in the extracted splice junctions of the gene by position (chr_SJstart_SJend).

### Donor Gain
Search: predicted at 2bp from the variant, chr12:49031256

Show all the splice junctions containing the positions between 49031250-49031259
```{r}
colnames(GeneSJ)[grep("4903125",colnames(GeneSJ))]
```
Alternative SJ not found in the splice junction collection.

### Aceptor Gain
Search: chr12:49031313


```{r}
colnames(GeneSJ)[grep("4903131",colnames(GeneSJ))]
```

Show all the splice junctions containing the position chr12:49031313-49034067
```{r}
colnames(GeneSJ)[grep("49031313_49034",colnames(GeneSJ))]
```
```{r}
t(GeneSJ[GeneSJ$GROUP == "MUT",c("sample_id",colnames(GeneSJ)[grep("49031313",colnames(GeneSJ))])])
```

Show all the splice junctions containing the position chr12:49031313-49032512
```{r}
colnames(GeneSJ)[grep("49031313_49032512",colnames(GeneSJ))]
```

Found: chr12_49031313_49032512

Reads of all the AML samples (mutated and no mutated) for the splice junction:
```{r}
GeneSJ$chr12_49031313_49032512
```

Samples with the SJ of interest:
```{r}
table(GeneSJ$chr12_49031313_49032512>0) 
```

Groups of the samples having the alternative splice junction:
```{r}
table(GeneSJ$GROUP[GeneSJ$chr12_49031313_49032512 > 0])
```
Alternative SJ found in the mutated samples.

### Canonical SJ
Exon upstream (UE): chr12:49033965-49034066; aceptor splice site chr12:49033965

Exon downstream (DE): chr12:49031034-49031174; donor splice site chr12:49031174


Reads of all the AML samples (mutated and no mutated) for the splice junction:
```{r}
GeneSJ$chr12_49033965_49034066
```

Reads of all the AML samples (mutated and no mutated) for the splice junction:
```{r}
GeneSJ$chr12_49031034_49031174
```


## Normalization
Count the reads of all the splice junctions of the gene harboring the variant:
```{r}
GeneSJ$rowSum_SJtotal <- rowSums(GeneSJ[,grep("chr", names(GeneSJ))])
```

Normalization of the expression by the total read counts of all the splice junctions of the gene:
```{r}
GeneSJ$Normalized_CanonDE <- (GeneSJ$chr12_49031034_49031174)/GeneSJ$rowSum_SJtotal*100
GeneSJ$Normalized_CanonUE <- (GeneSJ$chr12_49033965_49034066)/GeneSJ$rowSum_SJtotal*100
GeneSJ$Normalized_AG <- (GeneSJ$chr12_49031313_49032512)/GeneSJ$rowSum_SJtotal*100
```

Download the normalized values for the assessed splice junctions of all the AML samples:
```{r echo=FALSE}
normalization_df <- GeneSJ[,c("sample_id","case_id", "GROUP","Normalized_CanonDE", "Normalized_CanonUE","Normalized_AG", "chr12_49031034_49031174", "chr12_49033965_49034066","chr12_49031313_49032512", "rowSum_SJtotal","file_id.BAM","file_name.BAM","file_id.STAR.SJCounts","file_name.STAR.SJCounts" )]

colnames(normalization_df)[grep("chr12_49031034_49031174", names(normalization_df))] <- "chr12_49031034_49031174_CanonDE"
colnames(normalization_df)[grep("chr12_49033965_49034066", names(normalization_df))] <- "chr12_49033965_49034066_CanonUE"
colnames(normalization_df)[grep("chr12_49031313_49032512", names(normalization_df))] <- "chr12_49031313_49032512_CanonAG"


write.xlsx(normalization_df, paste0(normalizedEx_dir_out, "BeatAML.KMT2D.chr12-49031255-G-A.xlsx"), rowNames = FALSE)
```

```{r echo=FALSE}
datatable(normalization_df,
extensions = 'Buttons',
rownames = FALSE, 
filter = 'top',
options = list(paging = TRUE,
              pageLength = 4,
              autoWidth = FALSE,
              dom = 'lifrtBp',
              buttons = c('excel'), 
              scrollX = TRUE,  
              scrollY = TRUE,
              columnDefs = list(list(targets = '_all', className = 'dt-center'))
))
```

## VAF
```{r echo=FALSE}
GeneSJ_subset <- GeneSJ[GeneSJ$sample_id %in% cases,]
GeneSJ_subset$Gene <- "KMT2D"
GeneSJ_subset$MutationKey_Hg38 <- "chr12,49031255,G,A"

annot_vaf <- merge(found_variants[,c("MutationKey_Hg38","Gene","sample_id","case_id",                                               "RNA_Sample","DNA_Sample","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf",                         "MuTect2_filter","VarScan2_filter","SomaticSniper_filter","MuSE_filter","MUTReads","WTReads","VAF")], 
                   GeneSJ_subset[,c("Gene","sample_id", "MutationKey_Hg38","case_id",  "Normalized_CanonDE", "Normalized_CanonUE", "Normalized_AG")], 
                   by.x=c("RNA_Sample", "case_id","Gene", "MutationKey_Hg38"), 
                   by.y=c("sample_id", "case_id","Gene", "MutationKey_Hg38"))


annot_vaf <- annot_vaf %>% select(Gene,MutationKey_Hg38,
                                  sample_id,
                                  case_id, RNA_Sample,DNA_Sample,
                                  MuTect2_vaf,VarScan2_vaf,SomaticSniper_vaf,MuSE_vaf,
                                  VAF,
                                  Normalized_CanonDE, Normalized_CanonUE,Normalized_AG,
                                  everything()
                                  )

colnames(annot_vaf)[grep("VAF", names(annot_vaf))] <- "VAF_RNA"
colnames(annot_vaf)[grep("MUTReads", names(annot_vaf))] <- "MUTReads_RNA"
colnames(annot_vaf)[grep("WTReads", names(annot_vaf))] <- "WTReads_RNA"

write.xlsx(annot_vaf, paste0(normalizedEx_dir_out, "BeatAML.KMT2D.chr12-49031255-G-A.MUT.xlsx"), rowNames = FALSE)
```

Mutated samples vaf:
```{r echo=FALSE}
datatable(annot_vaf,
extensions = 'Buttons',
rownames = FALSE, 
filter = 'top',
options = list(paging = TRUE,
              pageLength = 4,
              autoWidth = TRUE,
              dom = 'lifrtBp',
              buttons = c('excel'), 
              scrollX = TRUE,  
              scrollY = TRUE
              )
)
```

## Plots {.tabset}

```{r echo=FALSE}
GeneSJ$GROUP <- ifelse(GeneSJ$GROUP == "MUT", "MUT", "WT")
GeneSJ$GROUP <- as.factor(GeneSJ$GROUP)
```

### Static Dot Plots 

Canonical splice junction:
```{r echo=FALSE}
ggplot(GeneSJ, aes(sample_id,Normalized_CanonUE,color=GROUP)) +
  geom_point(size=.8) +
  labs(color='GROUP', 
       title = "Normalized Expression of KMT2D chr12:49033965-49034066", 
       subtitle="Potential Aceptor Loss Effect on Canonical SJ",
       y = "Normalized Expression")
```

```{r echo=FALSE}
ggplot(GeneSJ, aes(sample_id,Normalized_CanonDE,color=GROUP)) +
  geom_point(size=.8) +
  labs(color='GROUP', 
       title = "Normalized Expression of KMT2D chr12:49031034-49031174", 
       subtitle="Potential Donor Loss Effect on Canonical SJ",
       y = "Normalized Expression")
```

Splicing alterations:
```{r echo=FALSE}
ggplot(GeneSJ, aes(sample_id,Normalized_AG,color=GROUP)) +
  geom_point(size=.8) +
  labs(color='GROUP', 
       title = "Normalized Expression of KMT2D chr12:49031313-49032512", 
       subtitle="Potential Aceptor Gain Effect",
       y = "Normalized Expression") 
```


### Interactive Dot Plots

```{r echo=FALSE}
plot_df <- merge(GeneSJ,annot_vaf[c("RNA_Sample", "VAF_RNA", 
                                    "MuTect2_vaf", "VarScan2_vaf", "SomaticSniper_vaf", "MuSE_vaf")],
                 by.x="sample_id", by.y="RNA_Sample", all.x = TRUE)
```


Canonical splice junction:
```{r echo=FALSE}
ggplotly(ggplot(plot_df, aes(sample_id,Normalized_CanonUE,color=GROUP)) + 
  geom_point(aes(text=paste0("VAF RNA: ",VAF_RNA,"\nVAF MuTect: ",MuTect2_vaf,
                             "\nVAF VarScan2: ",VarScan2_vaf,
                             "\nVAF SomaticSniper: ",SomaticSniper_vaf,
                             "\nVAF MuSE: ",MuSE_vaf)),size= 0.8) +
  labs(color='GROUP', 
       title = "Normalized Expression of KMT2D chr12:49033965-49034066", 
       subtitle="Potential Aceptor Loss Effect on Canonical SJ",
       y = "Normalized Expression")
)
```

```{r echo=FALSE}
ggplotly(ggplot(plot_df, aes(sample_id,Normalized_CanonDE,color=GROUP)) + 
  geom_point(aes(text=paste0("VAF RNA: ",VAF_RNA,"\nVAF MuTect: ",MuTect2_vaf,
                             "\nVAF VarScan2: ",VarScan2_vaf,
                             "\nVAF SomaticSniper: ",SomaticSniper_vaf,
                             "\nVAF MuSE: ",MuSE_vaf)),size= 0.8) +
  labs(color='GROUP', 
       title = "Normalized Expression of KMT2D chr12:49031034-49031174", 
       subtitle="Potential Donor Loss Effect on Canonical SJ",
       y = "Normalized Expression")
)
```


Splicing alteration:
```{r echo=FALSE}
ggplotly(ggplot(plot_df, aes(sample_id,Normalized_AG,color=GROUP)) + 
  geom_point(aes(text=paste0("VAF RNA: ",VAF_RNA,"\nVAF MuTect: ",MuTect2_vaf,
                             "\nVAF VarScan2: ",VarScan2_vaf,
                             "\nVAF SomaticSniper: ",SomaticSniper_vaf,
                             "\nVAF MuSE: ",MuSE_vaf)),size= 0.8) +
  labs(color='GROUP', 
       title = "Normalized Expression of KMT2D chr12:49031313-49032512", 
       subtitle="Potential Donor Gain Effect",
       y = "Normalized Expression")
)
```


### Violin Plots
Violin Plots for the alternative splice junctions interrogated:

Aceptor Loss:
```{r echo=FALSE}
plot_boxplot <- plot_df[plot_df$GROUP %in% c("WT"),]
plot_dotplot <- plot_df[plot_df$GROUP %in% c("MUT"),]

GEN = "KMT2D"
SJ = "chr12:49033965-49034066"

ggplot(plot_df, aes(y=Normalized_CanonUE, x=factor(GROUP, level=c("WT", "MUT")))) + 
       geom_violin(scale="area") +
       geom_point(data = plot_dotplot, 
                  aes(y=Normalized_CanonUE, x=GROUP),
                  color="black",
                  size=2
                  ) +
        
       ylab("Normalized SJ expression") +
       xlab(paste(GEN, SJ))
```

Donor Loss:
```{r echo=FALSE}
plot_boxplot <- plot_df[plot_df$GROUP %in% c("WT"),]
plot_dotplot <- plot_df[plot_df$GROUP %in% c("MUT"),]

GEN = "KMT2D"
SJ = "chr12:49031034-49031174"

ggplot(plot_df, aes(y=Normalized_CanonDE, x=factor(GROUP, level=c("WT", "MUT")))) + 
       geom_violin(scale="area") +
       geom_point(data = plot_dotplot, 
                  aes(y=Normalized_CanonDE, x=GROUP),
                  color="black",
                  size=2
                  ) +
        
       ylab("Normalized SJ expression") +
       xlab(paste(GEN, SJ))
```


Aceptor Gain:
```{r echo=FALSE}
plot_boxplot <- plot_df[plot_df$GROUP %in% c("WT"),]
plot_dotplot <- plot_df[plot_df$GROUP %in% c("MUT"),]

GEN = "KMT2D"
SJ = "chr12:49031313-49032512"

ggplot(plot_df, aes(y=Normalized_AG, x=factor(GROUP, level=c("WT", "MUT")))) + 
       geom_violin(scale="area") +
       geom_point(data = plot_dotplot, 
                  aes(y=Normalized_AG, x=GROUP),
                  color="black",
                  size=2
                  ) +
        
       ylab("Normalized SJ expression") +
       xlab(paste(GEN, SJ))
```

## Statistical Analysis {.tabset}

```{r}
SJCounts <- GeneSJ
```


### Donor Loss {.tabset}

#### Expression Difference 

Normality Test:
```{r}
shapiro.test(SJCounts$Normalized_CanonDE[SJCounts$GROUP == "WT"])
```

Value of Mean Normalized Expression of the Alternative SJ in WT samples:
```{r}
mean_WT_SJi <- mean(SJCounts$Normalized_CanonDE[SJCounts$GROUP == "WT"], na.rm=TRUE)
mean_WT_SJi
```

Normalized Expression Value of the Alternative SJ in the MUT sample:
```{r}
MUT_SJi <- SJCounts$Normalized_CanonDE[SJCounts$GROUP == "MUT"]
MUT_SJi
```
Deviation from the mean normalized expression:
```{r}
SJCounts$Difference <-  SJCounts$Normalized_CanonDE - mean_WT_SJi
```

```{r echo=FALSE}
table <- SJCounts[SJCounts$GROUP == "WT", c("sample_id", "Normalized_CanonDE", "Difference")]
table <- table[order(table$Difference, decreasing = TRUE),]
```

```{r echo=FALSE}
datatable(table,
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp'
))
```

Difference in the MUT sample: deviation of the Normalized expression of the MUT patient from the mean normalized WT expression
```{r}
SJCounts$Difference[SJCounts$GROUP == "MUT"]
```

```{r echo=FALSE}
ggplot(SJCounts, aes(sample_id,Difference)) + geom_point(aes(colour=factor(GROUP))) +
  ylab("Differential Expression") +
  labs(color = "GROUP",
       title = "Differential Expression Distribution")
```


```{r echo=FALSE}
ggplot(SJCounts, aes(y=Difference, x=factor(GROUP, level=c("WT", "MUT")))) + 
  geom_boxplot() +
  ylab("Differential Expression") +
  xlab("GROUP")
```

#### ECDF - Pvalue Inference 

```{r}
v_ecdf <- ecdf(SJCounts$Difference[SJCounts$GROUP == "WT"])
v_ecdf
```

```{r}
plot(ecdf(SJCounts$Difference[SJCounts$GROUP == "WT"]))
```

```{r}
v_ecdf(SJCounts$Difference[SJCounts$GROUP == "MUT"])
```


```{r}
MUT_df <- SJCounts[SJCounts$GROUP == "MUT",c("sample_id","case_id", "Normalized_CanonDE")]
colnames(MUT_df) <- c("sample_id", "case_id", "NormalizedExpression")

MUT_df$ECDF <- v_ecdf(SJCounts$Difference[SJCounts$GROUP == "MUT"])
MUT_df$Pvalue <- MUT_df$ECDF

MUT_df$Prediction <- "Donor Loss"
MUT_df$splice_junction_status <- "CanonicalSJ"
MUT_df$splice_junction_position <- "chr12:49031034-49031174"

MUT_df$ECDF <- ifelse(MUT_df$NormalizedExpression == 0, NA,MUT_df$ECDF)
MUT_df$Pvalue <- ifelse(MUT_df$NormalizedExpression == 0, NA,MUT_df$Pvalue)
```

```{r echo=FALSE}
vaf_select <- annot_vaf[,c("Gene","MutationKey_Hg38","sample_id","RNA_Sample","DNA_Sample","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf", "VAF_RNA")]
MUT_vaf <- merge(MUT_df, vaf_select, by.x="sample_id", by.y = "RNA_Sample")
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id"] <- "RNA_Sample"
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id.y"] <- "sample_id"

col_order <- c("Gene", "MutationKey_Hg38", "Prediction","splice_junction_status","splice_junction_position","sample_id", "case_id","RNA_Sample","DNA_Sample","NormalizedExpression","ECDF", "Pvalue", "VAF_RNA","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf")
MUT_vaf <- MUT_vaf[, col_order]
```

Download the vaf, inferred percentiles and pvalues of the mutated samples:
```{r echo=FALSE}
datatable(MUT_vaf,
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp',
              scrollX=TRUE
))
```

```{r echo=FALSE}
summary_res <- rbind(summary_res,MUT_vaf)
```


### Aceptor Loss {.tabset}

#### Expression Difference 

Normality Test:
```{r}
shapiro.test(SJCounts$Normalized_CanonUE[SJCounts$GROUP == "WT"])
```

Value of Mean Normalized Expression of the Alternative SJ in WT samples:
```{r}
mean_WT_SJi <- mean(SJCounts$Normalized_CanonUE[SJCounts$GROUP == "WT"], na.rm=TRUE)
mean_WT_SJi
```

Normalized Expression Value of the Alternative SJ in the MUT sample:
```{r}
MUT_SJi <- SJCounts$Normalized_CanonUE[SJCounts$GROUP == "MUT"]
MUT_SJi
```
Deviation from the mean normalized expression:
```{r}
SJCounts$Difference <-  SJCounts$Normalized_CanonUE - mean_WT_SJi
```

```{r echo=FALSE}
table <- SJCounts[SJCounts$GROUP == "WT", c("sample_id", "Normalized_CanonUE", "Difference")]
table <- table[order(table$Difference, decreasing = TRUE),]
```

```{r echo=FALSE}
datatable(table,
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp'
))
```

Difference in the MUT sample: deviation of the Normalized expression of the MUT patient from the mean normalized WT expression
```{r}
SJCounts$Difference[SJCounts$GROUP == "MUT"]
```

```{r echo=FALSE}
ggplot(SJCounts, aes(sample_id,Difference)) + geom_point(aes(colour=factor(GROUP))) +
  ylab("Differential Expression") +
  labs(color = "GROUP",
       title = "Differential Expression Distribution")
```


```{r echo=FALSE}
ggplot(SJCounts, aes(y=Difference, x=factor(GROUP, level=c("WT", "MUT")))) + 
  geom_boxplot() +
  ylab("Differential Expression") +
  xlab("GROUP")
```

#### ECDF - Pvalue Inference 

```{r}
v_ecdf <- ecdf(SJCounts$Difference[SJCounts$GROUP == "WT"])
v_ecdf
```

```{r}
plot(ecdf(SJCounts$Difference[SJCounts$GROUP == "WT"]))
```

```{r}
v_ecdf(SJCounts$Difference[SJCounts$GROUP == "MUT"])
```


```{r}
MUT_df <- SJCounts[SJCounts$GROUP == "MUT",c("sample_id","case_id", "Normalized_CanonUE")]
colnames(MUT_df) <- c("sample_id", "case_id", "NormalizedExpression")

MUT_df$ECDF <- v_ecdf(SJCounts$Difference[SJCounts$GROUP == "MUT"])
MUT_df$Pvalue <- MUT_df$ECDF

MUT_df$Prediction <- "Aceptor Loss"
MUT_df$splice_junction_status <- "CanonicalSJ"
MUT_df$splice_junction_position <- "chr12:49033965-49034066"

MUT_df$ECDF <- ifelse(MUT_df$NormalizedExpression == 0, NA,MUT_df$ECDF)
MUT_df$Pvalue <- ifelse(MUT_df$NormalizedExpression == 0, NA,MUT_df$Pvalue)
```

```{r echo=FALSE}
vaf_select <- annot_vaf[,c("Gene","MutationKey_Hg38","sample_id","RNA_Sample","DNA_Sample","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf", "VAF_RNA")]
MUT_vaf <- merge(MUT_df, vaf_select, by.x="sample_id", by.y = "RNA_Sample")
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id"] <- "RNA_Sample"
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id.y"] <- "sample_id"

col_order <- c("Gene", "MutationKey_Hg38", "Prediction","splice_junction_status","splice_junction_position","sample_id", "case_id","RNA_Sample","DNA_Sample","NormalizedExpression","ECDF", "Pvalue", "VAF_RNA","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf")
MUT_vaf <- MUT_vaf[, col_order]
```

Download the vaf, inferred percentiles and pvalues of the mutated samples:
```{r echo=FALSE}
datatable(MUT_vaf,
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp',
              scrollX=TRUE
))
```

```{r echo=FALSE}
summary_res <- rbind(summary_res,MUT_vaf)
```


### Aceptor Gain {.tabset}

#### Expression Difference 

Normality Test:
```{r}
shapiro.test(SJCounts$Normalized_AG[SJCounts$GROUP == "WT"])
```

Value of Mean Normalized Expression of the Alternative SJ in WT samples:
```{r}
mean_WT_SJi <- mean(SJCounts$Normalized_AG[SJCounts$GROUP == "WT"], na.rm=TRUE)
mean_WT_SJi
```

Normalized Expression Value of the Alternative SJ in the MUT sample:
```{r}
MUT_SJi <- SJCounts$Normalized_AG[SJCounts$GROUP == "MUT"]
MUT_SJi
```
Deviation from the mean normalized expression:
```{r}
SJCounts$Difference <-  SJCounts$Normalized_AG - mean_WT_SJi
```

```{r echo=FALSE}
table <- SJCounts[SJCounts$GROUP == "WT", c("sample_id", "Normalized_AG", "Difference")]
table <- table[order(table$Difference, decreasing = TRUE),]
```

```{r echo=FALSE}
datatable(table,
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp'
))
```

Difference in the MUT sample: deviation of the Normalized expression of the MUT patient from the mean normalized WT expression
```{r}
SJCounts$Difference[SJCounts$GROUP == "MUT"]
```

```{r echo=FALSE}
ggplot(SJCounts, aes(sample_id,Difference)) + geom_point(aes(colour=factor(GROUP))) +
  ylab("Differential Expression") +
  labs(color = "GROUP",
       title = "Differential Expression Distribution")
```


```{r echo=FALSE}
ggplot(SJCounts, aes(y=Difference, x=factor(GROUP, level=c("WT", "MUT")))) + 
  geom_boxplot() +
  ylab("Differential Expression") +
  xlab("GROUP")
```

#### ECDF - Pvalue Inference 

```{r}
v_ecdf <- ecdf(SJCounts$Difference[SJCounts$GROUP == "WT"])
v_ecdf
```

```{r}
plot(ecdf(SJCounts$Difference[SJCounts$GROUP == "WT"]))
```

```{r}
v_ecdf(SJCounts$Difference[SJCounts$GROUP == "MUT"])
```


```{r}
MUT_df <- SJCounts[SJCounts$GROUP == "MUT",c("sample_id","case_id", "Normalized_AG")]
colnames(MUT_df) <- c("sample_id", "case_id", "NormalizedExpression")

MUT_df$ECDF <- v_ecdf(SJCounts$Difference[SJCounts$GROUP == "MUT"])
MUT_df$Pvalue <- 1 - MUT_df$ECDF

MUT_df$Prediction <- "Aceptor Gain"
MUT_df$splice_junction_status <- "AlternativeSJ found in MUT samples"
MUT_df$splice_junction_position <- "chr12:49031313-49032512"

MUT_df$ECDF <- ifelse(MUT_df$NormalizedExpression == 0, NA,MUT_df$ECDF)
MUT_df$Pvalue <- ifelse(MUT_df$NormalizedExpression == 0, NA,MUT_df$Pvalue)
```

```{r echo=FALSE}
vaf_select <- annot_vaf[,c("Gene","MutationKey_Hg38","sample_id","RNA_Sample","DNA_Sample","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf", "VAF_RNA")]
MUT_vaf <- merge(MUT_df, vaf_select, by.x="sample_id", by.y = "RNA_Sample")
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id"] <- "RNA_Sample"
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id.y"] <- "sample_id"

col_order <- c("Gene", "MutationKey_Hg38", "Prediction","splice_junction_status","splice_junction_position","sample_id", "case_id","RNA_Sample","DNA_Sample","NormalizedExpression","ECDF", "Pvalue", "VAF_RNA","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf")
MUT_vaf <- MUT_vaf[, col_order]
```

Download the vaf, inferred percentiles and pvalues of the mutated samples:
```{r echo=FALSE}
datatable(MUT_vaf,
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp',
              scrollX=TRUE
))
```

```{r echo=FALSE}
summary_res <- rbind(summary_res,MUT_vaf)
```



```{r echo=FALSE}
## Add not found effects information:
MUT_df <- MUT_df[,c("sample_id", "case_id")]
MUT_df$NormalizedExpression <- NA
MUT_df$Prediction <- "Donor Gain"
MUT_df$splice_junction_status <- "AlternativeSJ not found in SJ collection"
MUT_df$splice_junction_position <- "chr12:49031256"
MUT_df$Pvalue <- NA
MUT_df$ECDF <- NA

vaf_select <- annot_vaf[,c("Gene","MutationKey_Hg38","sample_id","RNA_Sample","DNA_Sample","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf", "VAF_RNA")]
MUT_vaf <- merge(MUT_df, vaf_select, by.x="sample_id", by.y = "RNA_Sample")
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id"] <- "RNA_Sample"
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id.y"] <- "sample_id"

col_order <- c("Gene", "MutationKey_Hg38", "Prediction","splice_junction_status","splice_junction_position","sample_id", "case_id","RNA_Sample","DNA_Sample","NormalizedExpression","ECDF","Pvalue",  "VAF_RNA","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf")
MUT_vaf <- MUT_vaf[, col_order]

## Save
summary_res <- rbind(summary_res, MUT_vaf)
```

```{r echo=FALSE}
rm(GeneSJ, normalization_df, GeneSJ_subset, annot_vaf, plot_df, samples_df,table, plot_boxplot, plot_dotplot, vaf_select, SJCounts, MUT_df, MUT_vaf)
```

# TET2 chr4,105259774,G,A {.tabset}
Variant found in **1 patient** of the BeatAML (**1 sample**)

* Patients with TET2 chr4,105259774,G,A variant: **1 patient (1 sample)**
* Patients with the variant and RNASeq for validation: **1 patient (1 sample)**

The splicing alterations being assessed are: 

* **Aceptor Gain**: predicted at 38bp from the variant, chr4:105259811, not found in the splice junction collection.
* **Donor Loss**: chr4:105259770-105261758; donor splice site chr4:105259770.
* **Donor Gain**: chr4:105259708-105261758, found in the mutated samples.
* **Exon Skipping**: chr4:105243779-105261758, not found in the splice junction collection.


```{r echo=FALSE}
datatable(to_be_validated[to_be_validated$MutationKey_Hg38 == "chr4,105259774,G,A",],
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp', 
              scrollX = TRUE,  
              scrollY = TRUE,
              columnDefs = list(list(targets = '_all', className = 'dt-center')))
)
```

Load the extracted splice junctions of the gene harboring the mutation.
```{r}
extractedSJ_path <- paste0(extractedSJ_dir_in,"TET2_UM_annotSJ.tsv")
GeneSJ <- read.delim(extractedSJ_path, sep ="\t")
```

Set the sample's group: Mutated (MUT) or No Mutated (WT)
```{r}
samples_df <- found_variants[found_variants$Gene=="TET2" & found_variants$MutationKey_Hg38 == "chr4,105259774,G,A",]

cases <- samples_df$RNA_Sample[samples_df$Validable == "Validable"]
GeneSJ$GROUP <- ifelse(GeneSJ$sample_id %in% cases , "MUT", "WT")
```


## SJ Lookup {.tabset}

Search for the splice junctions of interest in the extracted splice junctions of the gene by position (chr_SJstart_SJend).

### Aceptor Gain
Search: predicted at 38bp from the variant, chr4:105259811

Show all the splice junctions containing the positions between 105259810-105259819
```{r}
colnames(GeneSJ)[grep("10525981",colnames(GeneSJ))]
```
Alternative SJ not found in the splice junction collection.

### Donor Gain
Search: chr4:105259708-105261758

Show all the splice junctions containing the position chr4:105259708-105261758
```{r}
colnames(GeneSJ)[grep("105259708_105261758",colnames(GeneSJ))]
```

Found: chr4_105259708_105261758

Reads of all the AML samples (mutated and no mutated) for the splice junction:
```{r}
GeneSJ$chr4_105259708_105261758
```

Samples with the SJ of interest:
```{r}
table(GeneSJ$chr4_105259708_105261758>0) 
```

Groups of the samples having the alternative splice junction:
```{r}
table(GeneSJ$GROUP[GeneSJ$chr4_105259708_105261758 > 0])
```
Alternative SJ found in the mutated samples.

### Exon Skipping
Search: chr4:105243779-105261758 

Show all the splice junctions containing the position chr4:105243779-105261758
```{r}
colnames(GeneSJ)[grep("105243779_105261758",colnames(GeneSJ))]
```
Alternative SJ not found in the splice junction collection.


### Canonical SJ
Exon 6-7 chr4:105243779-105259618

Exon 7-8 chr4:105259770-105261758; donor splice site chr4:105259770


Reads of all the AML samples (mutated and no mutated) for the splice junction:
```{r}
GeneSJ$chr4_105243779_105259618
```

Reads of all the AML samples (mutated and no mutated) for the splice junction:
```{r}
GeneSJ$chr4_105259770_105261758
```


## Normalization
Count the reads of all the splice junctions of the gene harboring the variant:
```{r}
GeneSJ$rowSum_SJtotal <- rowSums(GeneSJ[,grep("chr", names(GeneSJ))])
```

Normalization of the expression by the total read counts of all the splice junctions of the gene:
```{r}
GeneSJ$Normalized_CanonEx7_8 <- (GeneSJ$chr4_105259770_105261758)/GeneSJ$rowSum_SJtotal*100
GeneSJ$Normalized_Ex7DG <- (GeneSJ$chr4_105259708_105261758)/GeneSJ$rowSum_SJtotal*100
```

Download the normalized values for the assessed splice junctions of all the AML samples:
```{r echo=FALSE}
normalization_df <- GeneSJ[,c("sample_id","case_id", "GROUP","Normalized_CanonEx7_8","Normalized_Ex7DG", "chr4_105259770_105261758", "chr4_105259708_105261758", "rowSum_SJtotal","file_id.BAM","file_name.BAM","file_id.STAR.SJCounts","file_name.STAR.SJCounts" )]

colnames(normalization_df)[grep("chr12_49031034_49031174", names(normalization_df))] <- "chr4_105259770_105261758_CanonEx7_8"
colnames(normalization_df)[grep("chr12_49031034_49031174", names(normalization_df))] <- "chr4_105259708_105261758_Ex7DG"

write.xlsx(normalization_df, paste0(normalizedEx_dir_out, "BeatAML.TET2.chr4-105259774-G-A.xlsx"), rowNames = FALSE)
```

```{r echo=FALSE}
datatable(normalization_df,
extensions = 'Buttons',
rownames = FALSE, 
filter = 'top',
options = list(paging = TRUE,
              pageLength = 4,
              autoWidth = FALSE,
              dom = 'lifrtBp',
              buttons = c('excel'), 
              scrollX = TRUE,  
              scrollY = TRUE,
              columnDefs = list(list(targets = '_all', className = 'dt-center'))
))
```

## VAF
```{r echo=FALSE}
GeneSJ_subset <- GeneSJ[GeneSJ$sample_id %in% cases,]
GeneSJ_subset$Gene <- "TET2"
GeneSJ_subset$MutationKey_Hg38 <- "chr4,105259774,G,A"

annot_vaf <- merge(found_variants[,c("MutationKey_Hg38","Gene","sample_id","case_id",                                               "RNA_Sample","DNA_Sample","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf",                         "MuTect2_filter","VarScan2_filter","SomaticSniper_filter","MuSE_filter","MUTReads","WTReads","VAF")], 
                   GeneSJ_subset[,c("Gene","sample_id", "MutationKey_Hg38","case_id", "Normalized_CanonEx7_8", "Normalized_Ex7DG")], 
                   by.x=c("RNA_Sample", "case_id","Gene", "MutationKey_Hg38"), 
                   by.y=c("sample_id", "case_id","Gene", "MutationKey_Hg38"))


annot_vaf <- annot_vaf %>% select(Gene,MutationKey_Hg38,
                                  sample_id,
                                  case_id, RNA_Sample,DNA_Sample,
                                  MuTect2_vaf,VarScan2_vaf,SomaticSniper_vaf,MuSE_vaf,
                                  VAF,
                                  Normalized_CanonEx7_8,
                                  Normalized_Ex7DG,
                                  everything()
                                  )

colnames(annot_vaf)[grep("VAF", names(annot_vaf))] <- "VAF_RNA"
colnames(annot_vaf)[grep("MUTReads", names(annot_vaf))] <- "MUTReads_RNA"
colnames(annot_vaf)[grep("WTReads", names(annot_vaf))] <- "WTReads_RNA"

write.xlsx(annot_vaf, paste0(normalizedEx_dir_out, "BeatAML.TET2.chr4-105259774-G-A.MUT.xlsx"), rowNames = FALSE)
```

Mutated samples vaf:
```{r echo=FALSE}
datatable(annot_vaf,
extensions = 'Buttons',
rownames = FALSE, 
filter = 'top',
options = list(paging = TRUE,
              pageLength = 4,
              autoWidth = TRUE,
              dom = 'lifrtBp',
              buttons = c('excel'), 
              scrollX = TRUE,  
              scrollY = TRUE
              )
)
```

## Plots {.tabset}

```{r echo=FALSE}
GeneSJ$GROUP <- ifelse(GeneSJ$GROUP == "MUT", "MUT", "WT")
GeneSJ$GROUP <- as.factor(GeneSJ$GROUP)
```

### Static Dot Plots 

Canonical splice junction:
```{r echo=FALSE}
ggplot(GeneSJ, aes(sample_id,Normalized_CanonEx7_8,color=GROUP)) +
  geom_point(size=.8) +
  labs(color='GROUP', 
       title = "Normalized Expression of TET2 chr4:105259770-105261758", 
       subtitle="Potential Donor Loss Effect",
       y = "Normalized Expression")
```

Splicing alterations:
```{r echo=FALSE}
ggplot(GeneSJ, aes(sample_id,Normalized_Ex7DG,color=GROUP)) +
  geom_point(size=.8) +
  ggtitle("Normalized expression of TET2 Exon 7 Donor Gain") +
  labs(color='GROUP')  
```


### Interactive Dot Plots

```{r echo=FALSE}
plot_df <- merge(GeneSJ,annot_vaf[c("RNA_Sample", "VAF_RNA", 
                                    "MuTect2_vaf", "VarScan2_vaf", "SomaticSniper_vaf", "MuSE_vaf")],
                 by.x="sample_id", by.y="RNA_Sample", all.x = TRUE)
```


Canonical splice junction:
```{r echo=FALSE}
ggplotly(ggplot(plot_df, aes(sample_id,Normalized_CanonEx7_8,color=GROUP)) + 
  geom_point(aes(text=paste0("VAF RNA: ",VAF_RNA,"\nVAF MuTect: ",MuTect2_vaf,
                             "\nVAF VarScan2: ",VarScan2_vaf,
                             "\nVAF SomaticSniper: ",SomaticSniper_vaf,
                             "\nVAF MuSE: ",MuSE_vaf)),size= 0.8) +
  ggtitle("Normalized expression of TET2 Exon 7-8 Splice Junction") +
  labs(color='GROUP') 
)
```


Splicing alteration:
```{r echo=FALSE}
ggplotly(ggplot(plot_df, aes(sample_id,Normalized_Ex7DG,color=GROUP)) + 
  geom_point(aes(text=paste0("VAF RNA: ",VAF_RNA,"\nVAF MuTect: ",MuTect2_vaf,
                             "\nVAF VarScan2: ",VarScan2_vaf,
                             "\nVAF SomaticSniper: ",SomaticSniper_vaf,
                             "\nVAF MuSE: ",MuSE_vaf)),size= 0.8) +
  ggtitle("Normalized expression of TET2 Exon 7 Donor Gain") +
  labs(color='GROUP') 
)
```


### Violin Plots
Violin Plots for the alternative splice junctions interrogated:

Donor Loss:
```{r echo=FALSE}
plot_boxplot <- plot_df[plot_df$GROUP %in% c("WT"),]
plot_dotplot <- plot_df[plot_df$GROUP %in% c("MUT"),]

GEN = "TET2"
SJ = "chr4:105259770-105261758"

ggplot(plot_df, aes(y=Normalized_CanonEx7_8, x=factor(GROUP, level=c("WT", "MUT")))) + 
       geom_violin(scale="area") +
       geom_point(data = plot_dotplot, 
                  aes(y=Normalized_CanonEx7_8, x=GROUP),
                  color="black",
                  size=2
                  ) +
        
       ylab("Normalized SJ expression") +
       xlab(paste(GEN, SJ))
```


Donor Gain:
```{r echo=FALSE}
plot_boxplot <- plot_df[plot_df$GROUP %in% c("WT"),]
plot_dotplot <- plot_df[plot_df$GROUP %in% c("MUT"),]

GEN = "TET2"
SJ = "chr4:105259708-105261758"

ggplot(plot_df, aes(y=Normalized_Ex7DG, x=factor(GROUP, level=c("WT", "MUT")))) + 
       geom_violin(scale="area") +
       geom_point(data = plot_dotplot, 
                  aes(y=Normalized_Ex7DG, x=GROUP),
                  color="black",
                  size=2
                  ) +
        
       ylab("Normalized SJ expression") +
       xlab(paste(GEN, SJ))
```

## Statistical Analysis {.tabset}

```{r}
SJCounts <- GeneSJ
```

### Donor Loss {.tabset}

#### Expression Difference 

Normality Test:
```{r}
shapiro.test(SJCounts$Normalized_CanonEx7_8[SJCounts$GROUP == "WT"])
```

Value of Mean Normalized Expression of the Alternative SJ in WT samples:
```{r}
mean_WT_SJi <- mean(SJCounts$Normalized_CanonEx7_8[SJCounts$GROUP == "WT"], na.rm=TRUE)
mean_WT_SJi
```

Normalized Expression Value of the Alternative SJ in the MUT sample:
```{r}
MUT_SJi <- SJCounts$Normalized_CanonEx7_8[SJCounts$GROUP == "MUT"]
MUT_SJi
```
Deviation from the mean normalized expression:
```{r}
SJCounts$Difference <-  SJCounts$Normalized_CanonEx7_8 - mean_WT_SJi
```

```{r echo=FALSE}
table <- SJCounts[SJCounts$GROUP == "WT", c("sample_id", "Normalized_CanonEx7_8", "Difference")]
table <- table[order(table$Difference, decreasing = TRUE),]
```

```{r echo=FALSE}
datatable(table,
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp'
))
```

Difference in the MUT sample: deviation of the Normalized expression of the MUT patient from the mean normalized WT expression
```{r}
SJCounts$Difference[SJCounts$GROUP == "MUT"]
```

```{r echo=FALSE}
ggplot(SJCounts, aes(sample_id,Difference)) + geom_point(aes(colour=factor(GROUP))) +
  ylab("Differential Expression") +
  labs(color = "GROUP",
       title = "Differential Expression Distribution")
```


```{r echo=FALSE}
ggplot(SJCounts, aes(y=Difference, x=factor(GROUP, level=c("WT", "MUT")))) + 
  geom_boxplot() +
  ylab("Differential Expression") +
  xlab("GROUP")
```

#### ECDF - Pvalue Inference 

```{r}
v_ecdf <- ecdf(SJCounts$Difference[SJCounts$GROUP == "WT"])
v_ecdf
```

```{r}
plot(ecdf(SJCounts$Difference[SJCounts$GROUP == "WT"]))
```

```{r}
v_ecdf(SJCounts$Difference[SJCounts$GROUP == "MUT"])
```


```{r}
MUT_df <- SJCounts[SJCounts$GROUP == "MUT",c("sample_id","case_id", "Normalized_CanonEx7_8")]
colnames(MUT_df) <- c("sample_id", "case_id", "NormalizedExpression")

MUT_df$ECDF <- v_ecdf(SJCounts$Difference[SJCounts$GROUP == "MUT"])
MUT_df$Pvalue <- MUT_df$ECDF

MUT_df$Prediction <- "Donor Loss"
MUT_df$splice_junction_status <- "CanonicalSJ"
MUT_df$splice_junction_position <- "chr4:105259770-105261758"

MUT_df$ECDF <- ifelse(MUT_df$NormalizedExpression == 0, NA,MUT_df$ECDF)
MUT_df$Pvalue <- ifelse(MUT_df$NormalizedExpression == 0, NA,MUT_df$Pvalue)
```

```{r echo=FALSE}
vaf_select <- annot_vaf[,c("Gene","MutationKey_Hg38","sample_id","RNA_Sample","DNA_Sample","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf", "VAF_RNA")]
MUT_vaf <- merge(MUT_df, vaf_select, by.x="sample_id", by.y = "RNA_Sample")
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id"] <- "RNA_Sample"
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id.y"] <- "sample_id"

col_order <- c("Gene", "MutationKey_Hg38", "Prediction","splice_junction_status","splice_junction_position","sample_id", "case_id","RNA_Sample","DNA_Sample","NormalizedExpression","ECDF", "Pvalue", "VAF_RNA","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf")
MUT_vaf <- MUT_vaf[, col_order]
```

Download the vaf, inferred percentiles and pvalues of the mutated samples:
```{r echo=FALSE}
datatable(MUT_vaf,
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp',
              scrollX=TRUE
))
```

```{r echo=FALSE}
summary_res <- rbind(summary_res,MUT_vaf)
```


### Donor Gain {.tabset}

#### Expression Difference 

Normality Test:
```{r}
shapiro.test(SJCounts$Normalized_Ex7DG[SJCounts$GROUP == "WT"])
```

Value of Mean Normalized Expression of the Alternative SJ in WT samples:
```{r}
mean_WT_SJi <- mean(SJCounts$Normalized_Ex7DG[SJCounts$GROUP == "WT"], na.rm=TRUE)
mean_WT_SJi
```

Normalized Expression Value of the Alternative SJ in the MUT sample:
```{r}
MUT_SJi <- SJCounts$Normalized_Ex7DG[SJCounts$GROUP == "MUT"]
MUT_SJi
```
Deviation from the mean normalized expression:
```{r}
SJCounts$Difference <-  SJCounts$Normalized_Ex7DG - mean_WT_SJi
```

```{r echo=FALSE}
table <- SJCounts[SJCounts$GROUP == "WT", c("sample_id", "Normalized_Ex7DG", "Difference")]
table <- table[order(table$Difference, decreasing = TRUE),]
```

```{r echo=FALSE}
datatable(table,
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp'
))
```

Difference in the MUT sample: deviation of the Normalized expression of the MUT patient from the mean normalized WT expression
```{r}
SJCounts$Difference[SJCounts$GROUP == "MUT"]
```

```{r echo=FALSE}
ggplot(SJCounts, aes(sample_id,Difference)) + geom_point(aes(colour=factor(GROUP))) +
  ylab("Differential Expression") +
  labs(color = "GROUP",
       title = "Differential Expression Distribution")
```


```{r echo=FALSE}
ggplot(SJCounts, aes(y=Difference, x=factor(GROUP, level=c("WT", "MUT")))) + 
  geom_boxplot() +
  ylab("Differential Expression") +
  xlab("GROUP")
```

#### ECDF - Pvalue Inference 

```{r}
v_ecdf <- ecdf(SJCounts$Difference[SJCounts$GROUP == "WT"])
v_ecdf
```

```{r}
plot(ecdf(SJCounts$Difference[SJCounts$GROUP == "WT"]))
```

```{r}
v_ecdf(SJCounts$Difference[SJCounts$GROUP == "MUT"])
```


```{r}
MUT_df <- SJCounts[SJCounts$GROUP == "MUT",c("sample_id","case_id", "Normalized_Ex7DG")]
colnames(MUT_df) <- c("sample_id", "case_id", "NormalizedExpression")

MUT_df$ECDF <- v_ecdf(SJCounts$Difference[SJCounts$GROUP == "MUT"])
MUT_df$Pvalue <- 1 - MUT_df$ECDF

MUT_df$Prediction <- "Donor Gain"
MUT_df$splice_junction_status <- "AlternativeSJ found in MUT samples"
MUT_df$splice_junction_position <- "chr4:105259708-105261758"

MUT_df$ECDF <- ifelse(MUT_df$NormalizedExpression == 0, NA,MUT_df$ECDF)
MUT_df$Pvalue <- ifelse(MUT_df$NormalizedExpression == 0, NA,MUT_df$Pvalue)

```

```{r echo=FALSE}
vaf_select <- annot_vaf[,c("Gene","MutationKey_Hg38","sample_id","RNA_Sample","DNA_Sample","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf", "VAF_RNA")]
MUT_vaf <- merge(MUT_df, vaf_select, by.x="sample_id", by.y = "RNA_Sample")
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id"] <- "RNA_Sample"
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id.y"] <- "sample_id"

col_order <- c("Gene", "MutationKey_Hg38", "Prediction","splice_junction_status","splice_junction_position","sample_id", "case_id","RNA_Sample","DNA_Sample","NormalizedExpression","ECDF", "Pvalue", "VAF_RNA","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf")
MUT_vaf <- MUT_vaf[, col_order]
```

Download the vaf, inferred percentiles and pvalues of the mutated samples:
```{r echo=FALSE}
datatable(MUT_vaf,
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp',
              scrollX=TRUE
))
```

```{r echo=FALSE}
summary_res <- rbind(summary_res,MUT_vaf)
```

```{r echo=FALSE}
## Add not found effects information:
MUT_df <- MUT_df[,c("sample_id", "case_id")]
MUT_df$NormalizedExpression <- NA
MUT_df$Prediction <- "Aceptor Gain"
MUT_df$splice_junction_status <- "AlternativeSJ not found in SJ collection"
MUT_df$splice_junction_position <- "chr4:105259811"
MUT_df$Pvalue <- NA
MUT_df$ECDF <- NA

vaf_select <- annot_vaf[,c("Gene","MutationKey_Hg38","sample_id","RNA_Sample","DNA_Sample","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf", "VAF_RNA")]
MUT_vaf <- merge(MUT_df, vaf_select, by.x="sample_id", by.y = "RNA_Sample")
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id"] <- "RNA_Sample"
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id.y"] <- "sample_id"

col_order <- c("Gene", "MutationKey_Hg38", "Prediction","splice_junction_status","splice_junction_position","sample_id", "case_id","RNA_Sample","DNA_Sample","NormalizedExpression", "ECDF","Pvalue", "VAF_RNA","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf")
MUT_vaf <- MUT_vaf[, col_order]

## Save
summary_res <- rbind(summary_res, MUT_vaf)
```

```{r echo=FALSE}
## Add not found effects information:
MUT_df <- MUT_df[,c("sample_id", "case_id")]
MUT_df$NormalizedExpression <- NA
MUT_df$Prediction <- "Exon Skipping"
MUT_df$splice_junction_status <- "AlternativeSJ not found in MUT samples"
MUT_df$splice_junction_position <- "chr4:105243779-105261758"
MUT_df$Pvalue <- NA
MUT_df$ECDF <- NA

vaf_select <- annot_vaf[,c("Gene","MutationKey_Hg38","sample_id","RNA_Sample","DNA_Sample","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf", "VAF_RNA")]
MUT_vaf <- merge(MUT_df, vaf_select, by.x="sample_id", by.y = "RNA_Sample")
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id"] <- "RNA_Sample"
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id.y"] <- "sample_id"

col_order <- c("Gene", "MutationKey_Hg38", "Prediction","splice_junction_status","splice_junction_position","sample_id", "case_id","RNA_Sample","DNA_Sample","NormalizedExpression","ECDF", "Pvalue", "VAF_RNA","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf")
MUT_vaf <- MUT_vaf[, col_order]

## Save
summary_res <- rbind(summary_res, MUT_vaf)
```
```{r echo=FALSE}
rm(GeneSJ, normalization_df, GeneSJ_subset, annot_vaf, plot_df, samples_df,table, plot_boxplot, plot_dotplot, vaf_select, SJCounts, MUT_df, MUT_vaf)
```


# PTPN11 chr12,112489084,G,T {.tabset}
Variant found in **3 patients** of the BeatAML (**3 samples**)

* Patients with PTPN11 chr12,112489084,G,T variant: **3 patients (3 samples)**
* Patients with the variant and RNASeq for validation: **2 patients (2 samples)**

The splicing alterations being assessed are: 

* **Donor Gain**: chr12:112489079-112502143, found in the mutated samples.
* **Donor Loss**: chr12:112489176-112502143; donor splice site chr12:112489176

Variant information:
```{r echo=FALSE}
datatable(to_be_validated[to_be_validated$MutationKey_Hg38 == "chr12,112489084,G,T",],
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp', 
              scrollX = TRUE,  
              scrollY = TRUE,
              columnDefs = list(list(targets = '_all', className = 'dt-center')))
)
```

Load the extracted splice junctions of the gene harboring the mutation.
```{r}
extractedSJ_path <- paste0(extractedSJ_dir_in,"PTPN11_UM_annotSJ.tsv")
GeneSJ <- read.delim(extractedSJ_path, sep ="\t")
```

Set the sample's group: Mutated (MUT) or No Mutated (WT)
```{r}
samples_df <- found_variants[found_variants$Gene=="PTPN11" & found_variants$MutationKey_Hg38 == "chr12,112489084,G,T",]

cases <- samples_df$RNA_Sample[samples_df$Validable == "Validable"]
GeneSJ$GROUP <- ifelse(GeneSJ$sample_id %in% cases , "MUT", "WT")
```


## SJ Lookup {.tabset}

Search for the splice junctions of interest in the extracted splice junctions of the gene by position (chr_SJstart_SJend).

### Donor Gain
Search: chr12:112489079-112502143

```{r}
colnames(GeneSJ)[grep("112489079",colnames(GeneSJ))]
```

Show all the splice junctions containing the position chr12:112489079-112502143
```{r}
colnames(GeneSJ)[grep("112489079_112502143",colnames(GeneSJ))]
```

Found: chr12_112489079_112502143

Reads of all the AML samples (mutated and no mutated) for the splice junction:
```{r}
GeneSJ$chr12_112489079_112502143
```

Samples with the SJ of interest:
```{r}
table(GeneSJ$chr12_112489079_112502143>0) 
```

Groups of the samples having the alternative splice junction:
```{r}
table(GeneSJ$GROUP[GeneSJ$chr12_112489079_112502143 > 0])
```
Alternative SJ found in the mutated samples.

### Canonical SJ
Exon 12-13: chr12:112488511-112489023

Exon 13-14: chr12:112489176-112502143; splice site chr12:112489176

Reads of all the AML samples (mutated and no mutated) for the splice junction:
```{r}
GeneSJ$chr12_112488511_112489023
```

Reads of all the AML samples (mutated and no mutated) for the splice junction:
```{r}
GeneSJ$chr12_112489176_112502143
```


## Normalization
Count the reads of all the splice junctions of the gene harboring the variant:
```{r}
GeneSJ$rowSum_SJtotal <- rowSums(GeneSJ[,grep("chr", names(GeneSJ))])
```

Normalization of the expression by the total read counts of all the splice junctions of the gene:
```{r}
GeneSJ$Normalized_CanonEx13_14 <- (GeneSJ$chr12_112489176_112502143)/GeneSJ$rowSum_SJtotal*100

GeneSJ$Normalized_Ex13DG <- (GeneSJ$chr12_112489079_112502143)/GeneSJ$rowSum_SJtotal*100
```

Download the normalized values for the assessed splice junctions of all the AML samples:
```{r echo=FALSE}
normalization_df <- GeneSJ[,c("sample_id","case_id", "GROUP","Normalized_CanonEx13_14","Normalized_Ex13DG", "chr12_112489176_112502143", "chr12_112489079_112502143", "rowSum_SJtotal","file_id.BAM","file_name.BAM","file_id.STAR.SJCounts","file_name.STAR.SJCounts" )]

colnames(normalization_df)[grep("chr12_112489176_112502143", names(normalization_df))] <- "chr12_112489176_112502143_CanonEx13_14"
colnames(normalization_df)[grep("chr12_112489079_112502143", names(normalization_df))] <- "chr12_112489079_112502143_Ex13DG"

write.xlsx(normalization_df, paste0(normalizedEx_dir_out, "BeatAML.PTPN11.chr12-112489084-G-T.xlsx"), rowNames = FALSE)
```

```{r echo=FALSE}
datatable(normalization_df,
extensions = 'Buttons',
rownames = FALSE, 
filter = 'top',
options = list(paging = TRUE,
              pageLength = 4,
              autoWidth = FALSE,
              dom = 'lifrtBp',
              buttons = c('excel'), 
              scrollX = TRUE,  
              scrollY = TRUE,
              columnDefs = list(list(targets = '_all', className = 'dt-center'))
))
```

## VAF
```{r echo=FALSE}
GeneSJ_subset <- GeneSJ[GeneSJ$sample_id %in% cases,]
GeneSJ_subset$Gene <- "PTPN11"
GeneSJ_subset$MutationKey_Hg38 <- "chr12,112489084,G,T"

annot_vaf <- merge(found_variants[,c("MutationKey_Hg38","Gene","sample_id","case_id",                                               "RNA_Sample","DNA_Sample","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf",                         "MuTect2_filter","VarScan2_filter","SomaticSniper_filter","MuSE_filter","MUTReads","WTReads","VAF")], 
                   GeneSJ_subset[,c("Gene","sample_id", "MutationKey_Hg38","case_id", "Normalized_CanonEx13_14", "Normalized_Ex13DG")], 
                   by.x=c("RNA_Sample", "case_id","Gene", "MutationKey_Hg38"), 
                   by.y=c("sample_id", "case_id","Gene", "MutationKey_Hg38"))


annot_vaf <- annot_vaf %>% select(Gene,MutationKey_Hg38,
                                  sample_id,
                                  case_id, RNA_Sample,DNA_Sample,
                                  MuTect2_vaf,VarScan2_vaf,SomaticSniper_vaf,MuSE_vaf,
                                  VAF,
                                  Normalized_CanonEx13_14,Normalized_Ex13DG,
                                  everything()
                                  )

colnames(annot_vaf)[grep("VAF", names(annot_vaf))] <- "VAF_RNA"
colnames(annot_vaf)[grep("MUTReads", names(annot_vaf))] <- "MUTReads_RNA"
colnames(annot_vaf)[grep("WTReads", names(annot_vaf))] <- "WTReads_RNA"

write.xlsx(annot_vaf, paste0(normalizedEx_dir_out, "BeatAML.PTPN11.chr12-112489084-G-T.MUT.xlsx"), rowNames = FALSE)
```

Mutated samples vaf:
```{r echo=FALSE}
datatable(annot_vaf,
extensions = 'Buttons',
rownames = FALSE, 
filter = 'top',
options = list(paging = TRUE,
              pageLength = 4,
              autoWidth = TRUE,
              dom = 'lifrtBp',
              buttons = c('excel'), 
              scrollX = TRUE,  
              scrollY = TRUE
              )
)
```

## Plots {.tabset}

```{r echo=FALSE}
GeneSJ$GROUP <- ifelse(GeneSJ$GROUP == "MUT", "MUT", "WT")
GeneSJ$GROUP <- as.factor(GeneSJ$GROUP)
```

### Static Dot Plots 

Canonical splice junction:
```{r echo=FALSE}
ggplot(GeneSJ, aes(sample_id,Normalized_CanonEx13_14,color=GROUP)) +
  geom_point(size=.8) +
  labs(color='GROUP', 
       title = "Normalized Expression of PTPN11 chr12:112489079-112502143", 
       subtitle="Potential Donor Gain Effect",
       y = "Normalized Expression")
```

Splicing alterations:
```{r echo=FALSE}
ggplot(GeneSJ, aes(sample_id,Normalized_Ex13DG,color=GROUP)) +
  geom_point(size=.8) +
  labs(color='GROUP', 
       title = "Normalized Expression of PTPN11 chr12:112489176-112502143", 
       subtitle="Potential Donor Loss Effect",
       y = "Normalized Expression")
```


### Interactive Dot Plots

```{r echo=FALSE}
plot_df <- merge(GeneSJ,annot_vaf[c("RNA_Sample", "VAF_RNA", 
                                    "MuTect2_vaf", "VarScan2_vaf", "SomaticSniper_vaf", "MuSE_vaf")],
                 by.x="sample_id", by.y="RNA_Sample", all.x = TRUE)
```


Canonical splice junction:
```{r echo=FALSE}
ggplotly(ggplot(plot_df, aes(sample_id,Normalized_CanonEx13_14,color=GROUP)) + 
  geom_point(aes(text=paste0("VAF RNA: ",VAF_RNA,"\nVAF MuTect: ",MuTect2_vaf,
                             "\nVAF VarScan2: ",VarScan2_vaf,
                             "\nVAF SomaticSniper: ",SomaticSniper_vaf,
                             "\nVAF MuSE: ",MuSE_vaf)),size= 0.8) +
  labs(color='GROUP', 
       title = "Normalized Expression of PTPN11 chr12:112489176-112502143", 
       subtitle="Potential Donor Loss Effect",
       y = "Normalized Expression")
)
```


Splicing alteration:
```{r echo=FALSE}
ggplotly(ggplot(plot_df, aes(sample_id,Normalized_Ex13DG,color=GROUP)) + 
  geom_point(aes(text=paste0("VAF RNA: ",VAF_RNA,"\nVAF MuTect: ",MuTect2_vaf,
                             "\nVAF VarScan2: ",VarScan2_vaf,
                             "\nVAF SomaticSniper: ",SomaticSniper_vaf,
                             "\nVAF MuSE: ",MuSE_vaf)),size= 0.8) +
  labs(color='GROUP', 
       title = "Normalized Expression of PTPN11 chr12:112489079-112502143", 
       subtitle="Potential Donor Gain Effect",
       y = "Normalized Expression")
)
```


### Violin Plots
Violin Plots for the alternative splice junctions interrogated:

Donor Loss:
```{r echo=FALSE}
plot_boxplot <- plot_df[plot_df$GROUP %in% c("WT"),]
plot_dotplot <- plot_df[plot_df$GROUP %in% c("MUT"),]

GEN = "PTPN11"
SJ = "chr12:112489176-112502143"

ggplot(plot_df, aes(y=Normalized_CanonEx13_14, x=factor(GROUP, level=c("WT", "MUT")))) + 
       geom_violin(scale="area") +
       geom_point(data = plot_dotplot, 
                  aes(y=Normalized_CanonEx13_14, x=GROUP),
                  color="black",
                  size=2
                  ) +
        
       ylab("Normalized SJ expression") +
       xlab(paste(GEN, SJ))
```


Donor Gain:
```{r echo=FALSE}
plot_boxplot <- plot_df[plot_df$GROUP %in% c("WT"),]
plot_dotplot <- plot_df[plot_df$GROUP %in% c("MUT"),]

GEN = "PTPN11"
SJ = "chr12:112489079-112502143"

ggplot(plot_df, aes(y=Normalized_Ex13DG, x=factor(GROUP, level=c("WT", "MUT")))) + 
       geom_violin(scale="area") +
       geom_point(data = plot_dotplot, 
                  aes(y=Normalized_Ex13DG, x=GROUP),
                  color="black",
                  size=2
                  ) +
        
       ylab("Normalized SJ expression") +
       xlab(paste(GEN, SJ))
```

## Statistical Analysis {.tabset}

```{r}
SJCounts <- GeneSJ
```

### Donor Loss {.tabset}

#### Expression Difference 

Normality Test:
```{r}
shapiro.test(SJCounts$Normalized_CanonEx13_14[SJCounts$GROUP == "WT"])
```

Value of Mean Normalized Expression of the Alternative SJ in WT samples:
```{r}
mean_WT_SJi <- mean(SJCounts$Normalized_CanonEx13_14[SJCounts$GROUP == "WT"], na.rm=TRUE)
mean_WT_SJi
```

Normalized Expression Value of the Alternative SJ in the MUT sample:
```{r}
MUT_SJi <- SJCounts$Normalized_CanonEx13_14[SJCounts$GROUP == "MUT"]
MUT_SJi
```
Deviation from the mean normalized expression:
```{r}
SJCounts$Difference <-  SJCounts$Normalized_CanonEx13_14 - mean_WT_SJi
```

```{r echo=FALSE}
table <- SJCounts[SJCounts$GROUP == "WT", c("sample_id", "Normalized_CanonEx13_14", "Difference")]
table <- table[order(table$Difference, decreasing = TRUE),]
```

```{r echo=FALSE}
datatable(table,
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp'
))
```

Difference in the MUT sample: deviation of the Normalized expression of the MUT patient from the mean normalized WT expression
```{r}
SJCounts$Difference[SJCounts$GROUP == "MUT"]
```

```{r echo=FALSE}
ggplot(SJCounts, aes(sample_id,Difference)) + geom_point(aes(colour=factor(GROUP))) +
  ylab("Differential Expression") +
  labs(color = "GROUP",
       title = "Differential Expression Distribution")
```


```{r echo=FALSE}
ggplot(SJCounts, aes(y=Difference, x=factor(GROUP, level=c("WT", "MUT")))) + 
  geom_boxplot() +
  ylab("Differential Expression") +
  xlab("GROUP")
```

#### ECDF - Pvalue Inference 

```{r}
v_ecdf <- ecdf(SJCounts$Difference[SJCounts$GROUP == "WT"])
v_ecdf
```

```{r}
plot(ecdf(SJCounts$Difference[SJCounts$GROUP == "WT"]))
```

```{r}
v_ecdf(SJCounts$Difference[SJCounts$GROUP == "MUT"])
```


```{r}
MUT_df <- SJCounts[SJCounts$GROUP == "MUT",c("sample_id","case_id", "Normalized_CanonEx13_14")] 
colnames(MUT_df) <- c("sample_id", "case_id", "NormalizedExpression")

MUT_df$ECDF <- v_ecdf(SJCounts$Difference[SJCounts$GROUP == "MUT"])
MUT_df$Pvalue <- MUT_df$ECDF

MUT_df$Prediction <- "Donor Loss"
MUT_df$splice_junction_status <- "CanonicalSJ"
MUT_df$splice_junction_position <- "chr12:112489176-112502143"

MUT_df$ECDF <- ifelse(MUT_df$NormalizedExpression == 0, NA,MUT_df$ECDF)
MUT_df$Pvalue <- ifelse(MUT_df$NormalizedExpression == 0, NA,MUT_df$Pvalue)
```

```{r echo=FALSE}
vaf_select <- annot_vaf[,c("Gene","MutationKey_Hg38","sample_id","RNA_Sample","DNA_Sample","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf", "VAF_RNA")]
MUT_vaf <- merge(MUT_df, vaf_select, by.x="sample_id", by.y = "RNA_Sample")
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id"] <- "RNA_Sample"
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id.y"] <- "sample_id"

col_order <- c("Gene", "MutationKey_Hg38", "Prediction","splice_junction_status","splice_junction_position","sample_id", "case_id","RNA_Sample","DNA_Sample","NormalizedExpression","ECDF", "Pvalue", "VAF_RNA","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf")
MUT_vaf <- MUT_vaf[, col_order]
```

Download the vaf, inferred percentiles and pvalues of the mutated samples:
```{r echo=FALSE}
datatable(MUT_vaf,
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp',
              scrollX=TRUE
))
```

```{r echo=FALSE}
summary_res <- rbind(summary_res,MUT_vaf)
```


### Donor Gain {.tabset}

#### Expression Difference 

Normality Test:
```{r}
shapiro.test(SJCounts$Normalized_Ex13DG[SJCounts$GROUP == "WT"])
```

Value of Mean Normalized Expression of the Alternative SJ in WT samples:
```{r}
mean_WT_SJi <- mean(SJCounts$Normalized_Ex13DG[SJCounts$GROUP == "WT"], na.rm=TRUE)
mean_WT_SJi
```

Normalized Expression Value of the Alternative SJ in the MUT sample:
```{r}
MUT_SJi <- SJCounts$Normalized_Ex13DG[SJCounts$GROUP == "MUT"]
MUT_SJi
```
Deviation from the mean normalized expression:
```{r}
SJCounts$Difference <-  SJCounts$Normalized_Ex13DG - mean_WT_SJi
```

```{r echo=FALSE}
table <- SJCounts[SJCounts$GROUP == "WT", c("sample_id", "Normalized_Ex13DG", "Difference")]
table <- table[order(table$Difference, decreasing = TRUE),]
```

```{r echo=FALSE}
datatable(table,
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp'
))
```

Difference in the MUT sample: deviation of the Normalized expression of the MUT patient from the mean normalized WT expression
```{r}
SJCounts$Difference[SJCounts$GROUP == "MUT"]
```

```{r echo=FALSE}
ggplot(SJCounts, aes(sample_id,Difference)) + geom_point(aes(colour=factor(GROUP))) +
  ylab("Differential Expression") +
  labs(color = "GROUP",
       title = "Differential Expression Distribution")
```


```{r echo=FALSE}
ggplot(SJCounts, aes(y=Difference, x=factor(GROUP, level=c("WT", "MUT")))) + 
  geom_boxplot() +
  ylab("Differential Expression") +
  xlab("GROUP")
```

#### ECDF - Pvalue Inference 

```{r}
v_ecdf <- ecdf(SJCounts$Difference[SJCounts$GROUP == "WT"])
v_ecdf
```

```{r}
plot(ecdf(SJCounts$Difference[SJCounts$GROUP == "WT"]))
```

```{r}
v_ecdf(SJCounts$Difference[SJCounts$GROUP == "MUT"])
```


```{r}
MUT_df <- SJCounts[SJCounts$GROUP == "MUT",c("sample_id","case_id", "Normalized_Ex13DG")]
colnames(MUT_df) <- c("sample_id", "case_id", "NormalizedExpression")

MUT_df$ECDF <- v_ecdf(SJCounts$Difference[SJCounts$GROUP == "MUT"])
MUT_df$Pvalue <- 1 - MUT_df$ECDF

MUT_df$Prediction <- "Donor Gain"
MUT_df$splice_junction_status <- "AlternativeSJ found in MUT samples"
MUT_df$splice_junction_position <- "chr12:112489079-112502143"

MUT_df$ECDF <- ifelse(MUT_df$NormalizedExpression == 0, NA,MUT_df$ECDF)
MUT_df$Pvalue <- ifelse(MUT_df$NormalizedExpression == 0, NA,MUT_df$Pvalue)
```

```{r echo=FALSE}
vaf_select <- annot_vaf[,c("Gene","MutationKey_Hg38","sample_id","RNA_Sample","DNA_Sample","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf", "VAF_RNA")]
MUT_vaf <- merge(MUT_df, vaf_select, by.x="sample_id", by.y = "RNA_Sample")
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id"] <- "RNA_Sample"
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id.y"] <- "sample_id"

col_order <- c("Gene", "MutationKey_Hg38", "Prediction","splice_junction_status","splice_junction_position","sample_id", "case_id","RNA_Sample","DNA_Sample","NormalizedExpression","ECDF", "Pvalue", "VAF_RNA","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf")
MUT_vaf <- MUT_vaf[, col_order]
```

Download the vaf, inferred percentiles and pvalues of the mutated samples:
```{r echo=FALSE}
datatable(MUT_vaf,
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp',
              scrollX=TRUE
))
```

```{r echo=FALSE}
summary_res <- rbind(summary_res,MUT_vaf)
```

```{r echo=FALSE}
rm(GeneSJ, normalization_df, GeneSJ_subset, annot_vaf, plot_df, samples_df,table, plot_boxplot, plot_dotplot, vaf_select, SJCounts, MUT_df, MUT_vaf)
```

# KDM6A chrX,45062737,C,T {.tabset}
Variant found in **2 patients** of the BeatAML (**2 samples**)

* Patients with KDM6A chrX,45062737,C,T variant: **2 patients (2 samples)**
* Patients with the variant and RNASeq for validation: **2 patients (2 samples)**

The splicing alterations being assessed are: 

* **Donor Gain**: predicted a 2bp from the variant, chrX:45062740, not found in the splice junction collection.
* **Donor Gain**: chrX:45062720-45063421, found in the mutated samples.
* **Donor Loss**: chrX:45062749-45063421; donor splice site chrX:45062749.



Variant information:
```{r echo=FALSE}
datatable(to_be_validated[to_be_validated$MutationKey_Hg38 == "chrX,45062737,C,T",],
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp', 
              scrollX = TRUE,  
              scrollY = TRUE,
              columnDefs = list(list(targets = '_all', className = 'dt-center')))
)
```

Load the extracted splice junctions of the gene harboring the mutation.
```{r}
extractedSJ_path <- paste0(extractedSJ_dir_in,"KDM6A_UM_annotSJ.tsv")
GeneSJ <- read.delim(extractedSJ_path, sep ="\t")
```

Set the sample's group: Mutated (MUT) or No Mutated (WT)
```{r}
samples_df <- found_variants[found_variants$Gene=="KDM6A" & found_variants$MutationKey_Hg38 == "chrX,45062737,C,T",]

cases <- samples_df$RNA_Sample[samples_df$Validable == "Validable"]
GeneSJ$GROUP <- ifelse(GeneSJ$sample_id %in% cases , "MUT", "WT")
```


## SJ Lookup {.tabset}

Search for the splice junctions of interest in the extracted splice junctions of the gene by position (chr_SJstart_SJend).


### Donor Gain
Search: predicted a 2bp from the variant, chrX:45062740

Show all the splice junctions containing the positions between 45062740-45062749. Found canonical donor: chrX:45062749
```{r}
colnames(GeneSJ)[grep("4506274",colnames(GeneSJ))]
```
Show all the splice junctions containing the positions between 45062730-45062739. 
```{r}
colnames(GeneSJ)[grep("4506273",colnames(GeneSJ))]
```

Alternative SJ not found in the splice junction collection.


### Donor Gain
Search: chrX:45062720-45063421 

```{r}
colnames(GeneSJ)[grep("4506272",colnames(GeneSJ))]
```

Show all the splice junctions containing the position chrX:45062720-45063421 
```{r}
colnames(GeneSJ)[grep("45062720_45063421",colnames(GeneSJ))]
```

Found: chrX_45062720_45063421

Reads of all the AML samples (mutated and no mutated) for the splice junction:
```{r}
GeneSJ$chrX_45062720_45063421
```

Samples with the SJ of interest:
```{r}
table(GeneSJ$chrX_45062720_45063421>0) 
```

Groups of the samples having the alternative splice junction:
```{r}
table(GeneSJ$GROUP[GeneSJ$chrX_45062720_45063421 > 0])
```
Alternative SJ found in the mutated samples.

### Canonical SJ
Exon upstream (UE): chrX:45061420-45062646

Exon downstream (DE): chrX:45062749-45063421; donor splice site: chrX:45062749 

Reads of all the AML samples (mutated and no mutated) for the splice junction:
```{r}
GeneSJ$chrX_45061420_45062646
```

Reads of all the AML samples (mutated and no mutated) for the splice junction:
```{r}
GeneSJ$chrX_45062749_45063421
```


## Normalization
Count the reads of all the splice junctions of the gene harboring the variant:
```{r}
GeneSJ$rowSum_SJtotal <- rowSums(GeneSJ[,grep("chr", names(GeneSJ))])
```

Normalization of the expression by the total read counts of all the splice junctions of the gene:
```{r}
GeneSJ$Normalized_CanonDE <- (GeneSJ$chrX_45062749_45063421)/GeneSJ$rowSum_SJtotal*100
GeneSJ$Normalized_DG <- (GeneSJ$chrX_45062720_45063421)/GeneSJ$rowSum_SJtotal*100
```

Download the normalized values for the assessed splice junctions of all the AML samples:
```{r echo=FALSE}
normalization_df <- GeneSJ[,c("sample_id","case_id", "GROUP","Normalized_CanonDE", "Normalized_DG", "chrX_45062749_45063421",  "chrX_45062720_45063421", "rowSum_SJtotal","file_id.BAM","file_name.BAM","file_id.STAR.SJCounts","file_name.STAR.SJCounts" )]

colnames(normalization_df)[grep("chrX_45062749_45063421", names(normalization_df))] <- "chrX_45062749_45063421_CanonUE"
colnames(normalization_df)[grep("chrX_45062720_45063421", names(normalization_df))] <- "chrX_45062720_45063421_DG"

write.xlsx(normalization_df, paste0(normalizedEx_dir_out, "BeatAML.KDM6A.chrX-45062737-C-T.xlsx"), rowNames = FALSE)
```

```{r echo=FALSE}
datatable(normalization_df,
extensions = 'Buttons',
rownames = FALSE, 
filter = 'top',
options = list(paging = TRUE,
              pageLength = 4,
              autoWidth = FALSE,
              dom = 'lifrtBp',
              buttons = c('excel'), 
              scrollX = TRUE,  
              scrollY = TRUE,
              columnDefs = list(list(targets = '_all', className = 'dt-center'))
))
```

## VAF
```{r echo=FALSE}
GeneSJ_subset <- GeneSJ[GeneSJ$sample_id %in% cases,]
GeneSJ_subset$Gene <- "KDM6A"
GeneSJ_subset$MutationKey_Hg38 <- "chrX,45062737,C,T"

annot_vaf <- merge(found_variants[,c("MutationKey_Hg38","Gene","sample_id","case_id",                                               "RNA_Sample","DNA_Sample","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf",                         "MuTect2_filter","VarScan2_filter","SomaticSniper_filter","MuSE_filter","MUTReads","WTReads","VAF")], 
                   GeneSJ_subset[,c("Gene","sample_id", "MutationKey_Hg38","case_id",  "Normalized_CanonDE", "Normalized_DG")], 
                   by.x=c("RNA_Sample", "case_id","Gene", "MutationKey_Hg38"), 
                   by.y=c("sample_id", "case_id","Gene", "MutationKey_Hg38"))


annot_vaf <- annot_vaf %>% select(Gene,MutationKey_Hg38,
                                  sample_id,
                                  case_id, RNA_Sample,DNA_Sample,
                                  MuTect2_vaf,VarScan2_vaf,SomaticSniper_vaf,MuSE_vaf,
                                  VAF,
                                  Normalized_CanonDE,Normalized_DG,
                                  everything()
                                  )

colnames(annot_vaf)[grep("VAF", names(annot_vaf))] <- "VAF_RNA"
colnames(annot_vaf)[grep("MUTReads", names(annot_vaf))] <- "MUTReads_RNA"
colnames(annot_vaf)[grep("WTReads", names(annot_vaf))] <- "WTReads_RNA"

write.xlsx(annot_vaf, paste0(normalizedEx_dir_out, "BeatAML.KDM6A.chrX-45062737-C-T.MUT.xlsx"), rowNames = FALSE)
```

Mutated samples vaf:
```{r echo=FALSE}
datatable(annot_vaf,
extensions = 'Buttons',
rownames = FALSE, 
filter = 'top',
options = list(paging = TRUE,
              pageLength = 4,
              autoWidth = TRUE,
              dom = 'lifrtBp',
              buttons = c('excel'), 
              scrollX = TRUE,  
              scrollY = TRUE
              )
)
```

## Plots {.tabset}

```{r echo=FALSE}
GeneSJ$GROUP <- ifelse(GeneSJ$GROUP == "MUT", "MUT", "WT")
GeneSJ$GROUP <- as.factor(GeneSJ$GROUP)
```

### Static Dot Plots 

Canonical splice junction:
```{r echo=FALSE}
ggplot(GeneSJ, aes(sample_id,Normalized_CanonDE,color=GROUP)) +
  geom_point(size=.8) +
  labs(color='GROUP', 
       title = "Normalized Expression of KDM6A chrX:45062749-45063421", 
       subtitle="Potential Donor Loss Effect on Canonical SJ",
       y = "Normalized Expression")
```

Splicing alterations:
```{r echo=FALSE}
ggplot(GeneSJ, aes(sample_id,Normalized_DG,color=GROUP)) +
  geom_point(size=.8) +
  labs(color='GROUP', 
       title = "Normalized Expression of KDM6A chrX:45062720-45063421", 
       subtitle="Potential Donor Gain",
       y = "Normalized Expression")
```


### Interactive Dot Plots

```{r echo=FALSE}
plot_df <- merge(GeneSJ,annot_vaf[c("RNA_Sample", "VAF_RNA", 
                                    "MuTect2_vaf", "VarScan2_vaf", "SomaticSniper_vaf", "MuSE_vaf")],
                 by.x="sample_id", by.y="RNA_Sample", all.x = TRUE)
```


Canonical splice junction:
```{r echo=FALSE}
ggplotly(ggplot(plot_df, aes(sample_id,Normalized_CanonDE,color=GROUP)) + 
  geom_point(aes(text=paste0("VAF RNA: ",VAF_RNA,"\nVAF MuTect: ",MuTect2_vaf,
                             "\nVAF VarScan2: ",VarScan2_vaf,
                             "\nVAF SomaticSniper: ",SomaticSniper_vaf,
                             "\nVAF MuSE: ",MuSE_vaf)),size= 0.8) +
  labs(color='GROUP', 
       title = "Normalized Expression of KDM6A chrX:45062749-45063421", 
       subtitle="Potential Donor Loss Effect",
       y = "Normalized Expression")
)
```


Splicing alteration:
```{r echo=FALSE}
ggplotly(ggplot(plot_df, aes(sample_id,Normalized_DG,color=GROUP)) + 
  geom_point(aes(text=paste0("VAF RNA: ",VAF_RNA,"\nVAF MuTect: ",MuTect2_vaf,
                             "\nVAF VarScan2: ",VarScan2_vaf,
                             "\nVAF SomaticSniper: ",SomaticSniper_vaf,
                             "\nVAF MuSE: ",MuSE_vaf)),size= 0.8) +
  labs(color='GROUP', 
       title = "Normalized Expression of KDM6A chrX:45062720-45063421", 
       subtitle="Potential Donor Gain Effect",
       y = "Normalized Expression")
)
```


### Violin Plots
Violin Plots for the alternative splice junctions interrogated:

Donor Loss:
```{r echo=FALSE}
plot_boxplot <- plot_df[plot_df$GROUP %in% c("WT"),]
plot_dotplot <- plot_df[plot_df$GROUP %in% c("MUT"),]

GEN = "KDM6A"
SJ = "chrX:45062749-45063421"

ggplot(plot_df, aes(y=Normalized_CanonDE, x=factor(GROUP, level=c("WT", "MUT")))) + 
       geom_violin(scale="area") +
       geom_point(data = plot_dotplot, 
                  aes(y=Normalized_CanonDE, x=GROUP),
                  color="black",
                  size=2
                  ) +
        
       ylab("Normalized SJ expression") +
       xlab(paste(GEN, SJ))
```


Donor Gain:
```{r echo=FALSE}
plot_boxplot <- plot_df[plot_df$GROUP %in% c("WT"),]
plot_dotplot <- plot_df[plot_df$GROUP %in% c("MUT"),]

GEN = "KDM6A"
SJ = "chrX:45062720-45063421"

ggplot(plot_df, aes(y=Normalized_DG, x=factor(GROUP, level=c("WT", "MUT")))) + 
       geom_violin(scale="area") +
       geom_point(data = plot_dotplot, 
                  aes(y=Normalized_DG, x=GROUP),
                  color="black",
                  size=2
                  ) +
        
       ylab("Normalized SJ expression") +
       xlab(paste(GEN, SJ))
```

## Statistical Analysis {.tabset}

```{r}
SJCounts <- GeneSJ
```

### Donor Loss {.tabset}

#### Expression Difference 

Normality Test:
```{r}
shapiro.test(SJCounts$Normalized_CanonDE[SJCounts$GROUP == "WT"])
```

Value of Mean Normalized Expression of the Alternative SJ in WT samples:
```{r}
mean_WT_SJi <- mean(SJCounts$Normalized_CanonDE[SJCounts$GROUP == "WT"], na.rm=TRUE)
mean_WT_SJi
```

Normalized Expression Value of the Alternative SJ in the MUT sample:
```{r}
MUT_SJi <- SJCounts$Normalized_CanonDE[SJCounts$GROUP == "MUT"]
MUT_SJi
```
Deviation from the mean normalized expression:
```{r}
SJCounts$Difference <-  SJCounts$Normalized_CanonDE - mean_WT_SJi
```

```{r echo=FALSE}
table <- SJCounts[SJCounts$GROUP == "WT", c("sample_id", "Normalized_CanonDE", "Difference")]
table <- table[order(table$Difference, decreasing = TRUE),]
```

```{r echo=FALSE}
datatable(table,
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp'
))
```

Difference in the MUT sample: deviation of the Normalized expression of the MUT patient from the mean normalized WT expression
```{r}
SJCounts$Difference[SJCounts$GROUP == "MUT"]
```

```{r echo=FALSE}
ggplot(SJCounts, aes(sample_id,Difference)) + geom_point(aes(colour=factor(GROUP))) +
  ylab("Differential Expression") +
  labs(color = "GROUP",
       title = "Differential Expression Distribution")
```


```{r echo=FALSE}
ggplot(SJCounts, aes(y=Difference, x=factor(GROUP, level=c("WT", "MUT")))) + 
  geom_boxplot() +
  ylab("Differential Expression") +
  xlab("GROUP")
```

#### ECDF - Pvalue Inference 

```{r}
v_ecdf <- ecdf(SJCounts$Difference[SJCounts$GROUP == "WT"])
v_ecdf
```

```{r}
plot(ecdf(SJCounts$Difference[SJCounts$GROUP == "WT"]))
```

```{r}
v_ecdf(SJCounts$Difference[SJCounts$GROUP == "MUT"])
```


```{r}
MUT_df <- SJCounts[SJCounts$GROUP == "MUT",c("sample_id","case_id", "Normalized_CanonDE")]
colnames(MUT_df) <- c("sample_id", "case_id", "NormalizedExpression")

MUT_df$ECDF <- v_ecdf(SJCounts$Difference[SJCounts$GROUP == "MUT"])
MUT_df$Pvalue <- MUT_df$ECDF

MUT_df$Prediction <- "Donor Loss"
MUT_df$splice_junction_status <- "CanonicalSJ"
MUT_df$splice_junction_position <- "chrX:45062749-45063421"

MUT_df$ECDF <- ifelse(MUT_df$NormalizedExpression == 0, NA,MUT_df$ECDF)
MUT_df$Pvalue <- ifelse(MUT_df$NormalizedExpression == 0, NA,MUT_df$Pvalue)
```

```{r echo=FALSE}
vaf_select <- annot_vaf[,c("Gene","MutationKey_Hg38","sample_id","RNA_Sample","DNA_Sample","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf", "VAF_RNA")]
MUT_vaf <- merge(MUT_df, vaf_select, by.x="sample_id", by.y = "RNA_Sample")
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id"] <- "RNA_Sample"
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id.y"] <- "sample_id"

col_order <- c("Gene", "MutationKey_Hg38", "Prediction","splice_junction_status","splice_junction_position","sample_id", "case_id","RNA_Sample","DNA_Sample","NormalizedExpression","ECDF", "Pvalue", "VAF_RNA","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf")
MUT_vaf <- MUT_vaf[, col_order]
```

Download the vaf, inferred percentiles and pvalues of the mutated samples:
```{r echo=FALSE}
datatable(MUT_vaf,
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp',
              scrollX=TRUE
))
```

```{r echo=FALSE}
summary_res <- rbind(summary_res,MUT_vaf)
```


### Donor Gain {.tabset}

#### Expression Difference 

Normality Test:
```{r}
shapiro.test(SJCounts$Normalized_DG[SJCounts$GROUP == "WT"])
```

Value of Mean Normalized Expression of the Alternative SJ in WT samples:
```{r}
mean_WT_SJi <- mean(SJCounts$Normalized_DG[SJCounts$GROUP == "WT"], na.rm=TRUE)
mean_WT_SJi
```

Normalized Expression Value of the Alternative SJ in the MUT sample:
```{r}
MUT_SJi <- SJCounts$Normalized_DG[SJCounts$GROUP == "MUT"]
MUT_SJi
```
Deviation from the mean normalized expression:
```{r}
SJCounts$Difference <-  SJCounts$Normalized_DG - mean_WT_SJi
```

```{r echo=FALSE}
table <- SJCounts[SJCounts$GROUP == "WT", c("sample_id", "Normalized_DG", "Difference")]
table <- table[order(table$Difference, decreasing = TRUE),]
```

```{r echo=FALSE}
datatable(table,
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp'
))
```

Difference in the MUT sample: deviation of the Normalized expression of the MUT patient from the mean normalized WT expression
```{r}
SJCounts$Difference[SJCounts$GROUP == "MUT"]
```

```{r echo=FALSE}
ggplot(SJCounts, aes(sample_id,Difference)) + geom_point(aes(colour=factor(GROUP))) +
  ylab("Differential Expression") +
  labs(color = "GROUP",
       title = "Differential Expression Distribution")
```


```{r echo=FALSE}
ggplot(SJCounts, aes(y=Difference, x=factor(GROUP, level=c("WT", "MUT")))) + 
  geom_boxplot() +
  ylab("Differential Expression") +
  xlab("GROUP")
```

#### ECDF - Pvalue Inference 

```{r}
v_ecdf <- ecdf(SJCounts$Difference[SJCounts$GROUP == "WT"])
v_ecdf
```

```{r}
plot(ecdf(SJCounts$Difference[SJCounts$GROUP == "WT"]))
```

```{r}
v_ecdf(SJCounts$Difference[SJCounts$GROUP == "MUT"])
```


```{r}
MUT_df <- SJCounts[SJCounts$GROUP == "MUT",c("sample_id","case_id", "Normalized_DG")] 
colnames(MUT_df) <- c("sample_id", "case_id", "NormalizedExpression")

MUT_df$ECDF <- v_ecdf(SJCounts$Difference[SJCounts$GROUP == "MUT"])
MUT_df$Pvalue <- 1 - MUT_df$ECDF

MUT_df$Prediction <- "Donor Gain"
MUT_df$splice_junction_status <- "AlternativeSJ found in MUT samples"
MUT_df$splice_junction_position <- "chrX:45062720-45063421"

MUT_df$ECDF <- ifelse(MUT_df$NormalizedExpression == 0, NA,MUT_df$ECDF)
MUT_df$Pvalue <- ifelse(MUT_df$NormalizedExpression == 0, NA,MUT_df$Pvalue)
```

```{r echo=FALSE}
vaf_select <- annot_vaf[,c("Gene","MutationKey_Hg38","sample_id","RNA_Sample","DNA_Sample","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf", "VAF_RNA")]
MUT_vaf <- merge(MUT_df, vaf_select, by.x="sample_id", by.y = "RNA_Sample")
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id"] <- "RNA_Sample"
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id.y"] <- "sample_id"

col_order <- c("Gene", "MutationKey_Hg38", "Prediction","splice_junction_status","splice_junction_position","sample_id", "case_id","RNA_Sample","DNA_Sample","NormalizedExpression","ECDF", "Pvalue", "VAF_RNA","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf")
MUT_vaf <- MUT_vaf[, col_order]
```

Download the vaf, inferred percentiles and pvalues of the mutated samples:
```{r echo=FALSE}
datatable(MUT_vaf,
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp',
              scrollX=TRUE
))
```

```{r echo=FALSE}
summary_res <- rbind(summary_res,MUT_vaf)
```



```{r echo=FALSE}
## Add not found effects information:
MUT_df <- MUT_df[,c("sample_id", "case_id")]
MUT_df$NormalizedExpression <- NA
MUT_df$Prediction <- "Donor Gain"
MUT_df$splice_junction_status <- "AlternativeSJ not found in SJ collection"
MUT_df$splice_junction_position <- "chrX:45062740"
MUT_df$Pvalue <- NA
MUT_df$ECDF <- NA

vaf_select <- annot_vaf[,c("Gene","MutationKey_Hg38","sample_id","RNA_Sample","DNA_Sample","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf", "VAF_RNA")]
MUT_vaf <- merge(MUT_df, vaf_select, by.x="sample_id", by.y = "RNA_Sample")
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id"] <- "RNA_Sample"
colnames(MUT_vaf)[colnames(MUT_vaf) == "sample_id.y"] <- "sample_id"

col_order <- c("Gene", "MutationKey_Hg38", "Prediction","splice_junction_status","splice_junction_position","sample_id", "case_id","RNA_Sample","DNA_Sample","NormalizedExpression", "ECDF","Pvalue", "VAF_RNA","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf")
MUT_vaf <- MUT_vaf[, col_order]

## Save
summary_res <- rbind(summary_res, MUT_vaf)
```

```{r echo=FALSE}
rm(GeneSJ, normalization_df, GeneSJ_subset, annot_vaf, plot_df, samples_df,table, plot_boxplot, plot_dotplot, vaf_select, SJCounts, MUT_df, MUT_vaf)
```


# Results Summary
```{r echo=FALSE}
## Add non validable variants 
# STAG2, TET2 and NF1
MUT <- data.frame(matrix(ncol = 8, nrow = 3))
colnames(MUT) <- c("Gene", "MutationKey_Hg38", "Prediction", "splice_junction_status", "splice_junction_position","NormalizedExpression", "ECDF", "Pvalue")
MUT$Gene <- c("STAG2", "TET2", "NF1")
MUT$MutationKey_Hg38 <- c("chrX,124056127,G,A", "chr4,105237352,G,A","chr17,31338092,C,T")
MUT$Prediction <- rep("No validable", times=3)
MUT$splice_junction_status <- rep("No validable", times=3)
MUT$splice_junction_position <- rep("No validable", times=3)
MUT$Pvalue <- rep(NA, times=3)
MUT$ECDF <- rep(NA, times=3)

annot_vaf <- merge(found_variants[,c("MutationKey_Hg38","Gene","sample_id","case_id",                                               "RNA_Sample","DNA_Sample","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf","VAF")], 
                   MUT, 
                   by.x=c("Gene", "MutationKey_Hg38"), 
                   by.y=c("Gene", "MutationKey_Hg38"))

colnames(annot_vaf)[grepl("VAF", colnames(annot_vaf))] <- "VAF_RNA"

col_order <- c("Gene", "MutationKey_Hg38", "Prediction","splice_junction_status","splice_junction_position","sample_id", "case_id","RNA_Sample","DNA_Sample","NormalizedExpression", "ECDF", "Pvalue", "VAF_RNA","MuTect2_vaf","VarScan2_vaf","SomaticSniper_vaf","MuSE_vaf")
MUT_vaf <- annot_vaf[, col_order]

summary_res <- rbind(summary_res, MUT_vaf)

```

Download the vaf, inferred percentiles and pvalues of all the splicing alterations evaluated:
```{r echo=FALSE}
datatable(summary_res,
extensions = "Buttons",
rownames = FALSE, 
options = list(paging = TRUE,
              pageLength = 4,
              buttons = c('excel'), 
              autoWidth = TRUE,
              dom = 'lifrtBp',
              scrollX=TRUE,
              scrollY=TRUE
))
```

```{r echo=FALSE}
summary_res$Cohort <- "BeatAML"

summary_res$Order <- seq(nrow(summary_res))

summary_res <- merge(summary_res, to_be_validated[,c("MutationKey_Hg38", "HGVSc", "HGVSp")], by = "MutationKey_Hg38", all.x=TRUE)

summary_res <- summary_res[order(summary_res$Order),]

col_order <- c("Cohort", "Gene",     "MutationKey_Hg38", "HGVSc", "HGVSp", "Prediction", "splice_junction_status","splice_junction_position","sample_id","case_id", 
"RNA_Sample", "DNA_Sample","NormalizedExpression", "ECDF",  "Pvalue", "VAF_RNA",  "MuTect2_vaf","VarScan2_vaf", "SomaticSniper_vaf", 
"MuSE_vaf")

summary_res <- summary_res[,col_order]

writexl::write_xlsx(summary_res, paste0(summary_dir_out, "/BeatAML.report.xlsx"))
```
